{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract _C. floridanus_ and _C. pseudogracilis_ haplotypes from metabarcoding data."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before extracting the _Crangonyx spp._ haplotypes we need to import the functions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import metaBEAT_global_misc_functions as mb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Crangonyx_haplotypes’: File exists\r\n"
     ]
    }
   ],
   "source": [
    "mkdir Crangonyx_haplotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/C_floridanus/3-extract_haplotypes/Crangonyx_haplotypes\n"
     ]
    }
   ],
   "source": [
    "cd Crangonyx_haplotypes/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now extract the OTU ids for the relevant species that we filtered above 2%."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Specified BIOM input format 'json' - ok!\n",
      "SEARCH TERM: 'Crangonyx_floridanus'\n",
      "Found taxonomy metadata with OTUs - ok!\n",
      "Screening at taxonomic level: 'all'\n",
      "\n",
      "\n",
      "Identified 1 OTU(s) assigned to 'Crangonyx_floridanus'.\n",
      "SEARCH TERM: 'Crangonyx_pseudogracilis'\n",
      "Found taxonomy metadata with OTUs - ok!\n",
      "Screening at taxonomic level: 'all'\n",
      "\n",
      "\n",
      "Identified 1 OTU(s) assigned to 'Crangonyx_pseudogracilis'.\n",
      "\n",
      "{'Crangonyx_floridanus': [u'CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex'], 'Crangonyx_pseudogracilis': [u'CH105-1-4-May-nc|1_2105_9715_15085_1_ex']}\n",
      "\n",
      "OTU list:\n",
      "[u'CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex', u'CH105-1-4-May-nc|1_2105_9715_15085_1_ex']\n"
     ]
    }
   ],
   "source": [
    "OTU_table = mb.load_BIOM('../../2-metaBEAT/filtered.biom')\n",
    "\n",
    "dictionary = mb.find_target_OTUs_by_taxonomy(OTU_table, target='Crangonyx_floridanus', level='all')\n",
    "\n",
    "OTUs = {}\n",
    "OTUs['Crangonyx_floridanus'] = dictionary.keys()\n",
    "\n",
    "dictionary = mb.find_target_OTUs_by_taxonomy(OTU_table, target='Crangonyx_pseudogracilis', level='all')\n",
    "\n",
    "OTUs['Crangonyx_pseudogracilis'] = dictionary.keys()\n",
    "\n",
    "print \"\"\n",
    "print OTUs\n",
    "\n",
    "\n",
    "OTUs_as_list = []\n",
    "for sp in OTUs:\n",
    "    OTUs_as_list.extend(OTUs[sp])\n",
    "\n",
    "print \"\\nOTU list:\"\n",
    "print OTUs_as_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Identify samples contributing to each of the OTUs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OTU centroid ID - # of samples\n",
      "CH105-1-4-May-nc|1_2105_9715_15085_1_ex - 75\n",
      "CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex - 77\n"
     ]
    }
   ],
   "source": [
    "per_OTU_samples={}\n",
    "\n",
    "samples = OTU_table.ids(axis='sample')\n",
    "\n",
    "for OTU in OTUs_as_list:\n",
    "#    print OTU\n",
    "    per_OTU_samples[OTU] = []\n",
    "    obs = OTU_table.data(OTU, axis='observation')\n",
    "    for i in range(len(obs)):\n",
    "        if int(obs[i]) > 0:\n",
    "#            print \"\\t%s\" %samples[i]\n",
    "            per_OTU_samples[OTU].append(\".\".join(samples[i].split(\".\")[:-1]))\n",
    "\n",
    "    \n",
    "print \"OTU centroid ID - # of samples\"\n",
    "for OTU in per_OTU_samples:\n",
    "    print \"%s - %s\" %(OTU,len(per_OTU_samples[OTU]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Identify centroid IDS for the relevant OTUs, including query target alignments ('H')."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OTU ids to process:\n",
      "CH105-1-4-May-nc|1_2105_9715_15085_1_ex 75\n",
      "CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex 77\n",
      "#######\n",
      "\n",
      "CH105-1-4-May-nc|1_2105_9715_15085_1_ex 190\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1101_24330_23874_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1102_10220_11006_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1102_27565_22423_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1102_7800_20792_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1102_8729_18418_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1103_11450_9537_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1103_12563_12372_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1103_17548_18310_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1103_28869_13963_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1104_22429_18099_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1104_29122_17093_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1104_6632_7010_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1105_21661_12374_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1105_25153_15980_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1106_11101_13741_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1106_22165_8519_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1106_4461_12263_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1107_8534_10537_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1108_12105_6796_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1108_15756_20165_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1109_16945_20936_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1110_15360_27795_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1111_8704_12191_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1112_10564_20603_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1112_9067_12181_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1113_16637_21489_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1113_18798_14438_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1113_6543_25167_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1114_17423_22807_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_1114_5811_12712_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2101_13349_10607_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2102_12847_12016_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2102_8242_26920_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2103_13183_23191_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2103_23670_15957_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2103_7276_18079_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2103_8282_17074_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2104_12159_15457_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2104_13350_13191_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2104_16146_20056_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2105_19984_26450_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2105_4263_10599_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2105_6704_17543_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2105_8872_15343_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2106_14015_28538_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2106_29369_17802_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2107_16462_6850_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2107_24186_20654_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2107_28718_16879_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2107_7225_12330_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2108_17060_9755_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2108_9477_20327_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2110_17692_21574_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2110_8440_14312_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2111_13493_24589_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2111_2331_13823_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2111_24870_18232_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2112_10485_24664_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2113_18081_4955_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2113_7436_14015_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2114_13502_14830_1_ex\n",
      "\tfound centroid: CH103-1-6-May-nc|1_2114_8381_3963_1_ex\n",
      "\tfound centroid: CH105-1-4-May-nc|1_2105_9715_15085_1_ex\n",
      "\tfound centroid: CH106-1-3-May-nc|1_1107_8315_18054_1_ex\n",
      "\tfound centroid: CH106-1-3-May-nc|1_1110_21439_22183_1_ex\n",
      "\tfound centroid: CH106-1-3-May-nc|1_1111_5630_16700_1_ex\n",
      "\tfound centroid: CH109-pl1-2-1-Oct-nc|1_1111_17771_13255_1_ex\n",
      "\tfound centroid: CH110-pl1-2-2-Oct-nc|1_1102_9949_5557_1_ex\n",
      "\tfound centroid: CH110-pl1-2-2-Oct-nc|1_1103_25080_14554_1_ex\n",
      "\tfound centroid: CH111-pl1-2-3-Oct-nc|1_1111_8862_24940_1_ex\n",
      "\tfound centroid: CH112-pl1-2-4-Oct-nc|1_1102_12605_4879_1_ex\n",
      "\tfound centroid: CH113-pl1-2-5-Oct-nc|1_1102_10926_16218_1_ex\n",
      "\tfound centroid: CH114-pl1-2-6-Oct-nc|1_1102_10014_10337_1_ex\n",
      "\tfound centroid: CH115-pl1-2-7-Oct-nc|1_1101_15281_24572_1_ex\n",
      "\tfound centroid: CH116-pl1-2-8-Oct-nc|1_1103_14526_14749_1_ex\n",
      "\tfound centroid: CH117-pl1-3-1-Oct-nc|1_1101_7880_21159_1_ex\n",
      "\tfound centroid: CH118-pl1-3-2-Oct-nc|1_1107_19428_27614_1_ex\n",
      "\tfound centroid: CH119-pl1-3-3-Oct-nc|1_1112_7975_13869_1_ex\n",
      "\tfound centroid: CH201-2-6-May-nc|1_2105_16108_25950_1_ex\n",
      "\tfound centroid: CH203-2-4-May-nc|1_1101_9790_13250_1_ex\n",
      "\tfound centroid: CH204-2-3-May-nc|1_1113_25654_15317_1_ex\n",
      "\tfound centroid: CH205-2-2-May-nc|1_1102_9440_27809_1_ex\n",
      "\tfound centroid: CH206-2-1-May-nc|1_1102_9367_9576_1_ex\n",
      "\tfound centroid: CH209-3-6-May-nc|1_1101_23088_22908_1_ex\n",
      "\tfound centroid: CH301-pl1-3-4-Oct-nc|1_1101_21762_24316_1_ex\n",
      "\tfound centroid: CH302-pl1-3-5-Oct-nc|1_1103_18851_7119_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_1102_15265_21686_1_ex\n",
      "\tfound centroid: CH304-pl1-3-7-Oct-nc|1_1106_17586_25029_1_ex\n",
      "\tfound centroid: CH305-pl1-3-8-Oct-nc|1_2110_10509_22653_1_ex\n",
      "\tfound centroid: CH306-pl1-4-1-Oct-nc|1_1104_14965_4318_1_ex\n",
      "\tfound centroid: CH307-pl1-4-2-Oct-nc|1_2102_5028_14498_1_ex\n",
      "\tfound centroid: CH308-3-3-May-nc|1_1110_22091_26391_1_ex\n",
      "\tfound centroid: CH308-pl1-4-3-Oct-nc|1_1101_13768_18790_1_ex\n",
      "\tfound centroid: CH308-pl1-4-3-Oct-nc|1_2110_9229_10986_1_ex\n",
      "\tfound centroid: CH309-3-2-May-nc|1_1107_21219_21838_1_ex\n",
      "\tfound centroid: CH309-pl1-4-4-Oct-nc|1_1101_20773_4368_1_ex\n",
      "\tfound centroid: CH310-3-1-May-nc|1_1101_18604_17140_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_1102_9343_19183_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_1105_12388_26376_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_1106_16155_9070_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_1108_21815_6755_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_1109_3862_19237_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_1110_7811_17532_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_1113_22501_11697_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_2102_17322_23679_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_2105_11367_13744_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_2110_3817_15988_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_2114_15974_20148_1_ex\n",
      "\tfound centroid: CH311-4-8-May-nc|1_1103_25211_6822_1_ex\n",
      "\tfound centroid: CH311-4-8-May-nc|1_1106_15509_26959_1_ex\n",
      "\tfound centroid: CH311-4-8-May-nc|1_1106_9801_25151_1_ex\n",
      "\tfound centroid: CH311-4-8-May-nc|1_1109_24932_7432_1_ex\n",
      "\tfound centroid: CH311-4-8-May-nc|1_1111_12946_17679_1_ex\n",
      "\tfound centroid: CH311-4-8-May-nc|1_1111_5648_19871_1_ex\n",
      "\tfound centroid: CH311-4-8-May-nc|1_2105_11929_22986_1_ex\n",
      "\tfound centroid: CH311-4-8-May-nc|1_2111_16656_25801_1_ex\n",
      "\tfound centroid: CH312-4-7-May-nc|1_1101_24633_10184_1_ex\n",
      "\tfound centroid: CH312-4-7-May-nc|1_1101_26865_8994_1_ex\n",
      "\tfound centroid: CH312-4-7-May-nc|1_1111_15185_17021_1_ex\n",
      "\tfound centroid: CH312-4-7-May-nc|1_1111_25351_20883_1_ex\n",
      "\tfound centroid: CH312-4-7-May-nc|1_2111_11547_5110_1_ex\n",
      "\tfound centroid: CH312-4-7-May-nc|1_2111_20310_26284_1_ex\n",
      "\tfound centroid: CH312-4-7-May-nc|1_2112_4207_14922_1_ex\n",
      "\tfound centroid: CH312-4-7-May-nc|1_2114_10808_12998_1_ex\n",
      "\tfound centroid: CH313-4-6-May-nc|1_1101_17223_3813_1_ex\n",
      "\tfound centroid: CH314-4-5-May-nc|1_2113_14373_16842_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_1101_10662_24904_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_1103_22687_12430_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_1104_15681_4019_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_1104_9004_10630_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_1106_6401_16448_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_1109_11432_26032_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_1114_20947_26158_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_2104_20609_4106_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_2105_26332_17343_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_2107_23036_6643_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_2108_7435_24691_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_2109_10549_12380_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_2112_12159_9393_1_ex\n",
      "\tfound centroid: CH315-4-4-May-nc|1_2113_14536_22073_1_ex\n",
      "\tfound centroid: CH316-4-3-May-nc|1_1101_17804_4020_1_ex\n",
      "\tfound centroid: CH317-4-2-May-nc|1_1113_11527_10580_1_ex\n",
      "\tfound centroid: CH401-4-1-May-nc|1_2102_8546_20040_1_ex\n",
      "\tfound centroid: CH401-pl1-4-6-Oct-nc|1_1110_12109_12328_1_ex\n",
      "\tfound centroid: CH402-5-8-May-nc|1_2114_5820_10860_1_ex\n",
      "\tfound centroid: CH403-5-7-May-nc|1_2110_22767_17537_1_ex\n",
      "\tfound centroid: CH404-5-6-May-nc|1_1109_23362_14780_1_ex\n",
      "\tfound centroid: CH405-5-5-May-nc|1_2114_7356_13964_1_ex\n",
      "\tfound centroid: CH406-5-4-May-nc|1_1101_6058_21446_1_ex\n",
      "\tfound centroid: CH406-5-4-May-nc|1_2102_19675_24297_1_ex\n",
      "\tfound centroid: CH407-5-3-May-nc|1_1106_2335_17338_1_ex\n",
      "\tfound centroid: CH407-5-3-May-nc|1_1114_6447_21242_1_ex\n",
      "\tfound centroid: CH407-5-3-May-nc|1_2111_17745_21913_1_ex\n",
      "\tfound centroid: CH408-5-2-May-nc|1_1104_18113_23871_1_ex\n",
      "\tfound centroid: CH409-5-1-May-nc|1_2113_12901_9622_1_ex\n",
      "\tfound centroid: CH501-6-7-May-nc|1_2104_16742_20287_1_ex\n",
      "\tfound centroid: CH501-pl1-4-7-Oct-nc|1_1114_21367_11968_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_1102_28237_13381_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_1104_8157_16617_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_1107_21070_14741_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_2101_26438_18573_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_2102_19390_13080_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_2103_11685_26793_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_2108_17273_27632_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_2109_21966_16778_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_2111_10293_27692_1_ex\n",
      "\tfound centroid: CH503-6-6-May-nc|1_2112_13408_22985_1_ex\n",
      "\tfound centroid: CH503-pl1-5-1-Oct-nc|1_2102_12578_9443_1_ex\n",
      "\tfound centroid: CH504-pl1-5-2-Oct-nc|1_1109_13608_27725_1_ex\n",
      "\tfound centroid: CH505-6-4-May-nc|1_2105_8375_19090_1_ex\n",
      "\tfound centroid: CH505-pl1-5-3-Oct-nc|1_1101_15663_9824_1_ex\n",
      "\tfound centroid: CH506-6-3-May-nc|1_2105_16158_22211_1_ex\n",
      "\tfound centroid: CH506-pl1-5-4-Oct-nc|1_2108_19105_19739_1_ex\n",
      "\tfound centroid: CH507-pl1-5-5-Oct-nc|1_1110_17381_9508_1_ex\n",
      "\tfound centroid: CH507-pl1-5-5-Oct-nc|1_1114_18451_14159_1_ex\n",
      "\tfound centroid: CH508-pl1-5-6-Oct-nc|1_2105_9297_23643_1_ex\n",
      "\tfound centroid: CH509-pl1-5-7-Oct-nc|1_1107_20609_7175_1_ex\n",
      "\tfound centroid: CH510-7-7-May-nc|1_1104_8944_8704_1_ex\n",
      "\tfound centroid: CH510-pl1-5-8-Oct-nc|1_1107_18552_12641_1_ex\n",
      "\tfound centroid: CH601-pl1-6-1-Oct-nc|1_1102_14770_27980_1_ex\n",
      "\tfound centroid: CH602-7-6-May-nc|1_1103_11222_3470_1_ex\n",
      "\tfound centroid: CH602-pl1-6-2-Oct-nc|1_1103_14601_27439_1_ex\n",
      "\tfound centroid: CH603-pl1-6-3-Oct-nc|1_2106_19344_17133_1_ex\n",
      "\tfound centroid: CH604-pl1-6-4-Oct-nc|1_1113_14747_10301_1_ex\n",
      "\tfound centroid: CH605-pl1-6-5-Oct-nc|1_1102_13824_27047_1_ex\n",
      "\tfound centroid: CH605-pl1-6-5-Oct-nc|1_1110_16237_24188_1_ex\n",
      "\tfound centroid: CH606-pl1-6-6-Oct-nc|1_2109_10040_14462_1_ex\n",
      "\tfound centroid: CH607-pl1-6-7-Oct-nc|1_1101_17852_23911_1_ex\n",
      "\tfound centroid: CH608-pl1-6-8-Oct-nc|1_1107_5331_10619_1_ex\n",
      "\tfound centroid: neg-7-5-May-nc|1_2104_14460_24998_1_ex\n",
      "CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex 135\n",
      "\tfound centroid: CH-BL1-pl1-7-3-Oct-nc|1_2111_14114_15025_1_ex\n",
      "\tfound centroid: CH-BL2-pl1-7-4-Oct-nc|1_1102_10818_23263_1_ex\n",
      "\tfound centroid: CH101-1-8-May-nc|1_2114_22385_22080_1_ex\n",
      "\tfound centroid: CH101-pl1-1-1-Oct-nc|1_1101_19163_12271_1_ex\n",
      "\tfound centroid: CH101-pl1-1-1-Oct-nc|1_1105_9774_10777_1_ex\n",
      "\tfound centroid: CH101-pl1-1-1-Oct-nc|1_2105_17136_11806_1_ex\n",
      "\tfound centroid: CH102-1-7-May-nc|1_1102_17379_27773_1_ex\n",
      "\tfound centroid: CH102-pl1-1-2-Oct-nc|1_2112_22478_8174_1_ex\n",
      "\tfound centroid: CH103-pl1-1-3-Oct-nc|1_1102_9913_11787_1_ex\n",
      "\tfound centroid: CH104-1-5-May-nc|1_1103_23331_12069_1_ex\n",
      "\tfound centroid: CH104-1-5-May-nc|1_1104_4936_17123_1_ex\n",
      "\tfound centroid: CH104-1-5-May-nc|1_2105_16181_17403_1_ex\n",
      "\tfound centroid: CH104-pl1-1-4-Oct-nc|1_1101_18512_16301_1_ex\n",
      "\tfound centroid: CH104-pl1-1-4-Oct-nc|1_1107_12247_23331_1_ex\n",
      "\tfound centroid: CH105-pl1-1-5-Oct-nc|1_1111_17590_9870_1_ex\n",
      "\tfound centroid: CH106-pl1-1-6-Oct-nc|1_2110_15659_23451_1_ex\n",
      "\tfound centroid: CH107-1-2-May-nc|1_2107_7483_23780_1_ex\n",
      "\tfound centroid: CH107-pl1-1-7-Oct-nc|1_1103_19227_9435_1_ex\n",
      "\tfound centroid: CH108-1-1-May-nc|1_1105_22538_22905_1_ex\n",
      "\tfound centroid: CH108-pl1-1-8-Oct-nc|1_1103_20873_20660_1_ex\n",
      "\tfound centroid: CH109-2-8-May-nc|1_1104_4149_17971_1_ex\n",
      "\tfound centroid: CH109-2-8-May-nc|1_1105_9084_11993_1_ex\n",
      "\tfound centroid: CH109-2-8-May-nc|1_1111_17882_15316_1_ex\n",
      "\tfound centroid: CH109-2-8-May-nc|1_2103_19217_16098_1_ex\n",
      "\tfound centroid: CH109-2-8-May-nc|1_2107_12255_24001_1_ex\n",
      "\tfound centroid: CH109-pl1-2-1-Oct-nc|1_1103_21976_14388_1_ex\n",
      "\tfound centroid: CH109-pl1-2-1-Oct-nc|1_1109_9384_21255_1_ex\n",
      "\tfound centroid: CH110-2-7-May-nc|1_1102_14199_27658_1_ex\n",
      "\tfound centroid: CH110-pl1-2-2-Oct-nc|1_1109_19936_9823_1_ex\n",
      "\tfound centroid: CH110-pl1-2-2-Oct-nc|1_1110_10713_27863_1_ex\n",
      "\tfound centroid: CH110-pl1-2-2-Oct-nc|1_1111_11765_16896_1_ex\n",
      "\tfound centroid: CH110-pl1-2-2-Oct-nc|1_2102_18477_7604_1_ex\n",
      "\tfound centroid: CH110-pl1-2-2-Oct-nc|1_2106_16297_7752_1_ex\n",
      "\tfound centroid: CH110-pl1-2-2-Oct-nc|1_2112_17573_21662_1_ex\n",
      "\tfound centroid: CH111-pl1-2-3-Oct-nc|1_1104_20289_25468_1_ex\n",
      "\tfound centroid: CH111-pl1-2-3-Oct-nc|1_1108_10303_8437_1_ex\n",
      "\tfound centroid: CH112-pl1-2-4-Oct-nc|1_2109_10322_12814_1_ex\n",
      "\tfound centroid: CH113-pl1-2-5-Oct-nc|1_2101_27873_17936_1_ex\n",
      "\tfound centroid: CH114-pl1-2-6-Oct-nc|1_1105_15737_20636_1_ex\n",
      "\tfound centroid: CH115-pl1-2-7-Oct-nc|1_2110_24793_6820_1_ex\n",
      "\tfound centroid: CH116-pl1-2-8-Oct-nc|1_1101_15577_17803_1_ex\n",
      "\tfound centroid: CH117-pl1-3-1-Oct-nc|1_1102_22697_21915_1_ex\n",
      "\tfound centroid: CH117-pl1-3-1-Oct-nc|1_1102_4395_10505_1_ex\n",
      "\tfound centroid: CH117-pl1-3-1-Oct-nc|1_1112_21052_10874_1_ex\n",
      "\tfound centroid: CH118-pl1-3-2-Oct-nc|1_2106_15982_25320_1_ex\n",
      "\tfound centroid: CH118-pl1-3-2-Oct-nc|1_2108_16428_5700_1_ex\n",
      "\tfound centroid: CH118-pl1-3-2-Oct-nc|1_2110_24095_23599_1_ex\n",
      "\tfound centroid: CH119-pl1-3-3-Oct-nc|1_1114_11909_28155_1_ex\n",
      "\tfound centroid: CH119-pl1-3-3-Oct-nc|1_1114_17361_12540_1_ex\n",
      "\tfound centroid: CH202-2-5-May-nc|1_1111_14115_28583_1_ex\n",
      "\tfound centroid: CH202-2-5-May-nc|1_2101_8683_13805_1_ex\n",
      "\tfound centroid: CH202-2-5-May-nc|1_2102_7674_11967_1_ex\n",
      "\tfound centroid: CH202-2-5-May-nc|1_2111_13976_21014_1_ex\n",
      "\tfound centroid: CH203-2-4-May-nc|1_1107_13081_11271_1_ex\n",
      "\tfound centroid: CH204-2-3-May-nc|1_1101_15249_8454_1_ex\n",
      "\tfound centroid: CH206-2-1-May-nc|1_1114_27579_13621_1_ex\n",
      "\tfound centroid: CH207-3-8-May-nc|1_1108_13776_14705_1_ex\n",
      "\tfound centroid: CH208-3-7-May-nc|1_1102_25191_15965_1_ex\n",
      "\tfound centroid: CH208-3-7-May-nc|1_1107_24978_13102_1_ex\n",
      "\tfound centroid: CH208-3-7-May-nc|1_2107_13823_23739_1_ex\n",
      "\tfound centroid: CH208-3-7-May-nc|1_2111_10616_10934_1_ex\n",
      "\tfound centroid: CH208-3-7-May-nc|1_2113_13133_17253_1_ex\n",
      "\tfound centroid: CH209-3-6-May-nc|1_1101_11049_12097_1_ex\n",
      "\tfound centroid: CH210-3-5-May-nc|1_2102_7803_8624_1_ex\n",
      "\tfound centroid: CH301-pl1-3-4-Oct-nc|1_1111_12453_27617_1_ex\n",
      "\tfound centroid: CH302-pl1-3-5-Oct-nc|1_1102_20186_15022_1_ex\n",
      "\tfound centroid: CH302-pl1-3-5-Oct-nc|1_1108_3424_17463_1_ex\n",
      "\tfound centroid: CH302-pl1-3-5-Oct-nc|1_1109_12193_6319_1_ex\n",
      "\tfound centroid: CH302-pl1-3-5-Oct-nc|1_2112_23354_19511_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_1101_21365_26861_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_1101_27561_20028_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_1102_23591_13453_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_1104_21351_17798_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_1108_3489_19275_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_1109_21051_24535_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_1111_22301_13305_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_2101_15444_21743_1_ex\n",
      "\tfound centroid: CH303-pl1-3-6-Oct-nc|1_2101_2773_19044_1_ex\n",
      "\tfound centroid: CH304-pl1-3-7-Oct-nc|1_1105_25326_8473_1_ex\n",
      "\tfound centroid: CH305-pl1-3-8-Oct-nc|1_1103_15287_26025_1_ex\n",
      "\tfound centroid: CH306-pl1-4-1-Oct-nc|1_2113_14905_26849_1_ex\n",
      "\tfound centroid: CH307-pl1-4-2-Oct-nc|1_2111_26938_16630_1_ex\n",
      "\tfound centroid: CH308-pl1-4-3-Oct-nc|1_1103_14822_12234_1_ex\n",
      "\tfound centroid: CH308-pl1-4-3-Oct-nc|1_2101_24262_20530_1_ex\n",
      "\tfound centroid: CH309-3-2-May-nc|1_1102_27653_12510_1_ex\n",
      "\tfound centroid: CH309-pl1-4-4-Oct-nc|1_2111_19020_11638_1_ex\n",
      "\tfound centroid: CH310-3-1-May-nc|1_2107_9200_6395_1_ex\n",
      "\tfound centroid: CH310-pl1-4-5-Oct-nc|1_1103_25241_20637_1_ex\n",
      "\tfound centroid: CH401-pl1-4-6-Oct-nc|1_1103_18623_9709_1_ex\n",
      "\tfound centroid: CH401-pl1-4-6-Oct-nc|1_1113_16057_10251_1_ex\n",
      "\tfound centroid: CH401-pl1-4-6-Oct-nc|1_2113_25554_12926_1_ex\n",
      "\tfound centroid: CH501-pl1-4-7-Oct-nc|1_2110_14536_5793_1_ex\n",
      "\tfound centroid: CH502-pl1-4-8-Oct-nc|1_1101_24418_25110_1_ex\n",
      "\tfound centroid: CH503-pl1-5-1-Oct-nc|1_2111_21188_16601_1_ex\n",
      "\tfound centroid: CH504-6-5-May-nc|1_2101_8721_18397_1_ex\n",
      "\tfound centroid: CH504-pl1-5-2-Oct-nc|1_1103_6377_16927_1_ex\n",
      "\tfound centroid: CH505-pl1-5-3-Oct-nc|1_1111_27423_13701_1_ex\n",
      "\tfound centroid: CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex\n",
      "\tfound centroid: CH507-6-2-May-nc|1_1103_16590_12766_1_ex\n",
      "\tfound centroid: CH507-pl1-5-5-Oct-nc|1_1105_7417_23503_1_ex\n",
      "\tfound centroid: CH508-6-1-May-nc|1_1101_8646_6236_1_ex\n",
      "\tfound centroid: CH508-pl1-5-6-Oct-nc|1_1101_18482_26952_1_ex\n",
      "\tfound centroid: CH508-pl1-5-6-Oct-nc|1_1107_20724_7454_1_ex\n",
      "\tfound centroid: CH508-pl1-5-6-Oct-nc|1_1113_12072_10057_1_ex\n",
      "\tfound centroid: CH509-7-8-May-nc|1_1105_24118_22790_1_ex\n",
      "\tfound centroid: CH509-7-8-May-nc|1_1114_5815_14050_1_ex\n",
      "\tfound centroid: CH509-7-8-May-nc|1_2101_5882_5905_1_ex\n",
      "\tfound centroid: CH509-pl1-5-7-Oct-nc|1_1103_14803_12121_1_ex\n",
      "\tfound centroid: CH509-pl1-5-7-Oct-nc|1_1105_12208_14070_1_ex\n",
      "\tfound centroid: CH510-pl1-5-8-Oct-nc|1_1101_17841_4428_1_ex\n",
      "\tfound centroid: CH510-pl1-5-8-Oct-nc|1_1111_8731_20032_1_ex\n",
      "\tfound centroid: CH510-pl1-5-8-Oct-nc|1_1114_12008_15110_1_ex\n",
      "\tfound centroid: CH510-pl1-5-8-Oct-nc|1_2104_27519_9634_1_ex\n",
      "\tfound centroid: CH601-pl1-6-1-Oct-nc|1_1102_12629_8189_1_ex\n",
      "\tfound centroid: CH601-pl1-6-1-Oct-nc|1_2103_18237_7942_1_ex\n",
      "\tfound centroid: CH602-pl1-6-2-Oct-nc|1_2112_17474_4212_1_ex\n",
      "\tfound centroid: CH603-pl1-6-3-Oct-nc|1_1102_3745_17711_1_ex\n",
      "\tfound centroid: CH603-pl1-6-3-Oct-nc|1_2103_11992_17643_1_ex\n",
      "\tfound centroid: CH603-pl1-6-3-Oct-nc|1_2111_14816_26851_1_ex\n",
      "\tfound centroid: CH603-pl1-6-3-Oct-nc|1_2111_22687_15876_1_ex\n",
      "\tfound centroid: CH604-pl1-6-4-Oct-nc|1_1101_9513_5482_1_ex\n",
      "\tfound centroid: CH604-pl1-6-4-Oct-nc|1_1104_11193_6308_1_ex\n",
      "\tfound centroid: CH604-pl1-6-4-Oct-nc|1_2103_19820_7570_1_ex\n",
      "\tfound centroid: CH605-pl1-6-5-Oct-nc|1_1102_7714_23811_1_ex\n",
      "\tfound centroid: CH605-pl1-6-5-Oct-nc|1_2103_23579_23692_1_ex\n",
      "\tfound centroid: CH606-pl1-6-6-Oct-nc|1_2114_18585_9443_1_ex\n",
      "\tfound centroid: CH607-pl1-6-7-Oct-nc|1_1110_22079_4171_1_ex\n",
      "\tfound centroid: CH608-pl1-6-8-Oct-nc|1_2102_22468_24274_1_ex\n",
      "\tfound centroid: CH609-pl1-7-1-Oct-nc|1_1102_6617_14688_1_ex\n",
      "\tfound centroid: CH609-pl1-7-1-Oct-nc|1_1112_11242_24062_1_ex\n",
      "\tfound centroid: CH610-pl1-7-2-Oct-nc|1_1105_18104_6500_1_ex\n",
      "\tfound centroid: neg-7-5-May-nc|1_1105_18185_18390_1_ex\n",
      "\tfound centroid: pcr-NEG-pl1-7-5-Oct-nc|1_2110_26480_22422_1_ex\n",
      "\tfound centroid: pcr-NEG-pl1-7-7-Oct-nc|1_1101_9829_3153_1_ex\n",
      "\tfound centroid: pcr-POS-pl1-7-6-Oct-nc|1_1108_6151_24160_1_ex\n"
     ]
    }
   ],
   "source": [
    "per_OTU_centroids={}\n",
    "\n",
    "print \"OTU ids to process:\"\n",
    "for OTU in per_OTU_samples:\n",
    "    print OTU,len(per_OTU_samples[OTU])\n",
    "print \"#######\\n\"\n",
    "\n",
    "uc=open('../../2-metaBEAT/GLOBAL/global.uc', 'r')\n",
    "\n",
    "for line in uc:\n",
    "    if line.startswith('H'):\n",
    "        if line.strip().split(\"\\t\")[9] in OTUs_as_list:\n",
    "#            print \"hit: %s\\t%s\" %(line.strip().split(\"\\t\")[9],line.strip().split(\"\\t\")[8])\n",
    "            if not per_OTU_centroids.has_key(line.strip().split(\"\\t\")[9]):\n",
    "                per_OTU_centroids[line.strip().split(\"\\t\")[9]]=[line.strip().split(\"\\t\")[9]]\n",
    "\n",
    "                \n",
    "            if line.strip().split(\"\\t\")[8].split(\"|\")[0] in per_OTU_samples[line.strip().split(\"\\t\")[9]]:\n",
    "                per_OTU_centroids[line.strip().split(\"\\t\")[9]].append(line.strip().split(\"\\t\")[8])\n",
    "#                print \"found sample: %s\" %line.strip().split(\"\\t\")[8].split(\"|\")[0]\n",
    "                        \n",
    "uc.close()\n",
    "\n",
    "\n",
    "for OTU in per_OTU_centroids:\n",
    "    print OTU,str(len(per_OTU_centroids[OTU]))\n",
    "    for c in sorted(per_OTU_centroids[OTU]):\n",
    "        print \"\\tfound centroid: \"+c\n",
    "#print per_OTU_centroids\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Specify a unique id for each OTU."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[u'CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex', u'CH105-1-4-May-nc|1_2105_9715_15085_1_ex']\n",
      "['Crangonyx_floridanus_0_OTU', 'Crangonyx_pseudogracilis_0_OTU']\n"
     ]
    }
   ],
   "source": [
    "OTUs_synonyms = []\n",
    "for sp in OTUs:\n",
    "    count=0\n",
    "#    OTUs_as_list.extend(OTUs[sp])\n",
    "    \n",
    "    for otu in OTUs[sp]:\n",
    "        OTUs_synonyms.append(sp+'_'+str(count)+'_OTU')\n",
    "        count+=1\n",
    "    \n",
    "print OTUs_as_list\n",
    "print OTUs_synonyms\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Identify samples contributing to each of the OTUs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "CH105-1-4-May-nc|1_2105_9715_15085_1_ex 75 ['CH103-1-6-May-nc', 'CH105-1-4-May-nc', 'CH106-1-3-May-nc', 'CH109-pl1-2-1-Oct-nc', 'CH110-pl1-2-2-Oct-nc', 'CH111-pl1-2-3-Oct-nc', 'CH112-pl1-2-4-Oct-nc', 'CH113-pl1-2-5-Oct-nc', 'CH114-pl1-2-6-Oct-nc', 'CH115-pl1-2-7-Oct-nc', 'CH116-pl1-2-8-Oct-nc', 'CH117-pl1-3-1-Oct-nc', 'CH118-pl1-3-2-Oct-nc', 'CH119-pl1-3-3-Oct-nc', 'CH201-2-6-May-nc', 'CH203-2-4-May-nc', 'CH204-2-3-May-nc', 'CH205-2-2-May-nc', 'CH206-2-1-May-nc', 'CH209-3-6-May-nc', 'CH301-pl1-3-4-Oct-nc', 'CH302-pl1-3-5-Oct-nc', 'CH303-pl1-3-6-Oct-nc', 'CH304-pl1-3-7-Oct-nc', 'CH305-pl1-3-8-Oct-nc', 'CH306-pl1-4-1-Oct-nc', 'CH307-pl1-4-2-Oct-nc', 'CH308-3-3-May-nc', 'CH308-pl1-4-3-Oct-nc', 'CH309-3-2-May-nc', 'CH309-pl1-4-4-Oct-nc', 'CH310-3-1-May-nc', 'CH310-pl1-4-5-Oct-nc', 'CH311-4-8-May-nc', 'CH312-4-7-May-nc', 'CH313-4-6-May-nc', 'CH314-4-5-May-nc', 'CH315-4-4-May-nc', 'CH316-4-3-May-nc', 'CH317-4-2-May-nc', 'CH401-4-1-May-nc', 'CH401-pl1-4-6-Oct-nc', 'CH402-5-8-May-nc', 'CH403-5-7-May-nc', 'CH404-5-6-May-nc', 'CH405-5-5-May-nc', 'CH406-5-4-May-nc', 'CH407-5-3-May-nc', 'CH408-5-2-May-nc', 'CH409-5-1-May-nc', 'CH501-6-7-May-nc', 'CH501-pl1-4-7-Oct-nc', 'CH502-pl1-4-8-Oct-nc', 'CH503-6-6-May-nc', 'CH503-pl1-5-1-Oct-nc', 'CH504-pl1-5-2-Oct-nc', 'CH505-6-4-May-nc', 'CH505-pl1-5-3-Oct-nc', 'CH506-6-3-May-nc', 'CH506-pl1-5-4-Oct-nc', 'CH507-pl1-5-5-Oct-nc', 'CH508-pl1-5-6-Oct-nc', 'CH509-pl1-5-7-Oct-nc', 'CH510-7-7-May-nc', 'CH510-pl1-5-8-Oct-nc', 'CH601-pl1-6-1-Oct-nc', 'CH602-7-6-May-nc', 'CH602-pl1-6-2-Oct-nc', 'CH603-pl1-6-3-Oct-nc', 'CH604-pl1-6-4-Oct-nc', 'CH605-pl1-6-5-Oct-nc', 'CH606-pl1-6-6-Oct-nc', 'CH607-pl1-6-7-Oct-nc', 'CH608-pl1-6-8-Oct-nc', 'neg-7-5-May-nc']\n",
      "\n",
      "CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex 77 ['CH-BL1-pl1-7-3-Oct-nc', 'CH-BL2-pl1-7-4-Oct-nc', 'CH101-1-8-May-nc', 'CH101-pl1-1-1-Oct-nc', 'CH102-1-7-May-nc', 'CH102-pl1-1-2-Oct-nc', 'CH103-pl1-1-3-Oct-nc', 'CH104-1-5-May-nc', 'CH104-pl1-1-4-Oct-nc', 'CH105-pl1-1-5-Oct-nc', 'CH106-pl1-1-6-Oct-nc', 'CH107-1-2-May-nc', 'CH107-pl1-1-7-Oct-nc', 'CH108-1-1-May-nc', 'CH108-pl1-1-8-Oct-nc', 'CH109-2-8-May-nc', 'CH109-pl1-2-1-Oct-nc', 'CH110-2-7-May-nc', 'CH110-pl1-2-2-Oct-nc', 'CH111-pl1-2-3-Oct-nc', 'CH112-pl1-2-4-Oct-nc', 'CH113-pl1-2-5-Oct-nc', 'CH114-pl1-2-6-Oct-nc', 'CH115-pl1-2-7-Oct-nc', 'CH116-pl1-2-8-Oct-nc', 'CH117-pl1-3-1-Oct-nc', 'CH118-pl1-3-2-Oct-nc', 'CH119-pl1-3-3-Oct-nc', 'CH202-2-5-May-nc', 'CH203-2-4-May-nc', 'CH204-2-3-May-nc', 'CH206-2-1-May-nc', 'CH207-3-8-May-nc', 'CH208-3-7-May-nc', 'CH209-3-6-May-nc', 'CH210-3-5-May-nc', 'CH301-pl1-3-4-Oct-nc', 'CH302-pl1-3-5-Oct-nc', 'CH303-pl1-3-6-Oct-nc', 'CH304-pl1-3-7-Oct-nc', 'CH305-pl1-3-8-Oct-nc', 'CH306-pl1-4-1-Oct-nc', 'CH307-pl1-4-2-Oct-nc', 'CH308-pl1-4-3-Oct-nc', 'CH309-3-2-May-nc', 'CH309-pl1-4-4-Oct-nc', 'CH310-3-1-May-nc', 'CH310-pl1-4-5-Oct-nc', 'CH401-pl1-4-6-Oct-nc', 'CH501-pl1-4-7-Oct-nc', 'CH502-pl1-4-8-Oct-nc', 'CH503-pl1-5-1-Oct-nc', 'CH504-6-5-May-nc', 'CH504-pl1-5-2-Oct-nc', 'CH505-pl1-5-3-Oct-nc', 'CH506-pl1-5-4-Oct-nc', 'CH507-6-2-May-nc', 'CH507-pl1-5-5-Oct-nc', 'CH508-6-1-May-nc', 'CH508-pl1-5-6-Oct-nc', 'CH509-7-8-May-nc', 'CH509-pl1-5-7-Oct-nc', 'CH510-pl1-5-8-Oct-nc', 'CH601-pl1-6-1-Oct-nc', 'CH602-pl1-6-2-Oct-nc', 'CH603-pl1-6-3-Oct-nc', 'CH604-pl1-6-4-Oct-nc', 'CH605-pl1-6-5-Oct-nc', 'CH606-pl1-6-6-Oct-nc', 'CH607-pl1-6-7-Oct-nc', 'CH608-pl1-6-8-Oct-nc', 'CH609-pl1-7-1-Oct-nc', 'CH610-pl1-7-2-Oct-nc', 'neg-7-5-May-nc', 'pcr-NEG-pl1-7-5-Oct-nc', 'pcr-NEG-pl1-7-7-Oct-nc', 'pcr-POS-pl1-7-6-Oct-nc']\n"
     ]
    }
   ],
   "source": [
    "per_OTU_samples={}\n",
    "            \n",
    "for otu in per_OTU_centroids:\n",
    "    per_OTU_samples[otu] = []\n",
    "    for centroid in per_OTU_centroids[otu]:\n",
    "        per_OTU_samples[otu].append(centroid.split(\"|\")[0])\n",
    "#        print centroid.split(\"|\")[1]\n",
    "\n",
    "    per_OTU_samples[otu]=list(set(per_OTU_samples[otu]))\n",
    "    print \"\\n\"+otu,len(per_OTU_samples[otu]),sorted(per_OTU_samples[otu])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a global fasta file containing all reads for each OTU."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OTU: CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex -> Crangonyx_floridanus_0_OTU.fa\n",
      "OTU: CH105-1-4-May-nc|1_2105_9715_15085_1_ex -> Crangonyx_pseudogracilis_0_OTU.fa\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "for i in range(len(OTUs_as_list)):\n",
    "    otu=OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    print \"OTU: %s -> %s.fa\" %(otu,syn)\n",
    "    read_ids_per_OTU=[]\n",
    "    seqs_per_OTU=[]\n",
    "    for sample in per_OTU_samples[otu]:\n",
    "        seqs_per_sample=[]\n",
    "#        print \"Sample: %s\" %sample\n",
    "        read_ids=[]\n",
    "#        collect relevant centroids\n",
    "        centroids=[]\n",
    "        for c in per_OTU_centroids[otu]:\n",
    "#            print \"centroid: %s\" %c\n",
    "            if c.startswith(sample+'|'):\n",
    "                for c_ind in c.split(\"|\")[1:]:\n",
    "                    centroids.append(c_ind)\n",
    "                \n",
    "#        print \"Centroids: %s\" %centroids\n",
    "        \n",
    "#        extract read ids from uc file\n",
    "        read_ids.extend(centroids[:])\n",
    "        uc=open('../../2-metaBEAT/'+sample+'/'+sample+'.uc', 'r')\n",
    "        for line in uc:\n",
    "            if line.startswith('H'):\n",
    "                if line.strip().split(\"\\t\")[9] in centroids:\n",
    "#                    print line\n",
    "                    read_ids.append(line.strip().split(\"\\t\")[8])\n",
    "        uc.close()            \n",
    "#        print \"READ IDS: %i\" %len(read_ids)\n",
    "        \n",
    "#        extract reads per sample\n",
    "        \n",
    "        fasta=open('../../2-metaBEAT/'+sample+'/'+sample+'_queries.fasta', 'r')     # trimmed.fasta', 'r')\n",
    "        for r in SeqIO.parse(fasta, 'fasta'):\n",
    "            if r.id in read_ids:\n",
    "                r.id = sample+'|'+r.id\n",
    "                r.description = r.id\n",
    "                seqs_per_OTU.append(r)\n",
    "                \n",
    "            \n",
    "        fasta.close()\n",
    "        \n",
    "        \n",
    "#    Write out global fasta per OTU, containing all reads across all samples\n",
    "    out=open(syn+'.fa', 'w')\n",
    "    SeqIO.write(seqs_per_OTU, out, 'fasta')\n",
    "    out.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dereplicate all sequences at 95% identity match."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading file Crangonyx_floridanus_0_OTU.fa 0%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 1%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 2%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 2%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 3%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 4%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 5%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 5%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 6%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 7%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 8%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 8%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 9%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 10%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 10%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 11%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 12%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 13%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 13%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 14%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 15%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 16%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 16%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 17%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 18%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 19%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 19%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 20%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 21%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 22%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 22%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 23%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 24%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 25%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 25%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 26%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 27%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 28%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 28%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 29%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 30%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 30%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 31%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 32%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 33%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 33%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 34%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 35%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 36%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 36%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 37%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 38%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 39%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 39%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 40%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 41%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 42%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 42%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 43%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 44%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 45%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 45%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 46%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 47%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 48%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 48%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 49%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 50%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 51%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 51%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 52%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 53%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 53%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 54%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 55%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 56%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 56%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 57%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 58%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 59%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 59%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 60%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 61%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 62%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 62%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 63%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 64%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 65%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 65%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 66%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 67%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 68%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 68%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 69%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 70%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 70%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 71%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 72%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 73%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 73%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 74%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 75%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 76%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 76%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 77%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 78%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 79%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 79%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 80%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 81%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 82%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 82%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 83%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 84%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 85%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 85%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 86%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 87%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 87%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 88%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 89%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 90%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 90%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 91%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 92%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 93%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 93%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 94%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 95%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 96%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 96%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 97%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 98%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 99%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 99%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 100%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.fa 100%\n",
      "42661 nt in 135 seqs, min 308, max 343, avg 316\n",
      "Indexing sequences 0%  \r",
      "Indexing sequences 0%  \r",
      "Indexing sequences 1%  \r",
      "Indexing sequences 1%  \r",
      "Indexing sequences 2%  \r",
      "Indexing sequences 3%  \r",
      "Indexing sequences 4%  \r",
      "Indexing sequences 4%  \r",
      "Indexing sequences 5%  \r",
      "Indexing sequences 6%  \r",
      "Indexing sequences 7%  \r",
      "Indexing sequences 7%  \r",
      "Indexing sequences 8%  \r",
      "Indexing sequences 9%  \r",
      "Indexing sequences 10%  \r",
      "Indexing sequences 10%  \r",
      "Indexing sequences 11%  \r",
      "Indexing sequences 12%  \r",
      "Indexing sequences 13%  \r",
      "Indexing sequences 13%  \r",
      "Indexing sequences 14%  \r",
      "Indexing sequences 15%  \r",
      "Indexing sequences 16%  \r",
      "Indexing sequences 16%  \r",
      "Indexing sequences 17%  \r",
      "Indexing sequences 18%  \r",
      "Indexing sequences 19%  \r",
      "Indexing sequences 19%  \r",
      "Indexing sequences 20%  \r",
      "Indexing sequences 21%  \r",
      "Indexing sequences 21%  \r",
      "Indexing sequences 22%  \r",
      "Indexing sequences 23%  \r",
      "Indexing sequences 24%  \r",
      "Indexing sequences 24%  \r",
      "Indexing sequences 25%  \r",
      "Indexing sequences 26%  \r",
      "Indexing sequences 27%  \r",
      "Indexing sequences 27%  \r",
      "Indexing sequences 28%  \r",
      "Indexing sequences 29%  \r",
      "Indexing sequences 30%  \r",
      "Indexing sequences 30%  \r",
      "Indexing sequences 31%  \r",
      "Indexing sequences 32%  \r",
      "Indexing sequences 33%  \r",
      "Indexing sequences 33%  \r",
      "Indexing sequences 34%  \r",
      "Indexing sequences 35%  \r",
      "Indexing sequences 36%  \r",
      "Indexing sequences 36%  \r",
      "Indexing sequences 37%  \r",
      "Indexing sequences 38%  \r",
      "Indexing sequences 39%  \r",
      "Indexing sequences 39%  \r",
      "Indexing sequences 40%  \r",
      "Indexing sequences 41%  \r",
      "Indexing sequences 41%  \r",
      "Indexing sequences 42%  \r",
      "Indexing sequences 43%  \r",
      "Indexing sequences 44%  \r",
      "Indexing sequences 44%  \r",
      "Indexing sequences 45%  \r",
      "Indexing sequences 46%  \r",
      "Indexing sequences 47%  \r",
      "Indexing sequences 47%  \r",
      "Indexing sequences 48%  \r",
      "Indexing sequences 49%  \r",
      "Indexing sequences 50%  \r",
      "Indexing sequences 50%  \r",
      "Indexing sequences 51%  \r",
      "Indexing sequences 52%  \r",
      "Indexing sequences 53%  \r",
      "Indexing sequences 53%  \r",
      "Indexing sequences 54%  \r",
      "Indexing sequences 55%  \r",
      "Indexing sequences 56%  \r",
      "Indexing sequences 56%  \r",
      "Indexing sequences 57%  \r",
      "Indexing sequences 58%  \r",
      "Indexing sequences 59%  \r",
      "Indexing sequences 59%  \r",
      "Indexing sequences 60%  \r",
      "Indexing sequences 61%  \r",
      "Indexing sequences 61%  \r",
      "Indexing sequences 62%  \r",
      "Indexing sequences 63%  \r",
      "Indexing sequences 64%  \r",
      "Indexing sequences 64%  \r",
      "Indexing sequences 65%  \r",
      "Indexing sequences 66%  \r",
      "Indexing sequences 67%  \r",
      "Indexing sequences 67%  \r",
      "Indexing sequences 68%  \r",
      "Indexing sequences 69%  \r",
      "Indexing sequences 70%  \r",
      "Indexing sequences 70%  \r",
      "Indexing sequences 71%  \r",
      "Indexing sequences 72%  \r",
      "Indexing sequences 73%  \r",
      "Indexing sequences 73%  \r",
      "Indexing sequences 74%  \r",
      "Indexing sequences 75%  \r",
      "Indexing sequences 76%  \r",
      "Indexing sequences 76%  \r",
      "Indexing sequences 77%  \r",
      "Indexing sequences 78%  \r",
      "Indexing sequences 79%  \r",
      "Indexing sequences 79%  \r",
      "Indexing sequences 80%  \r",
      "Indexing sequences 81%  \r",
      "Indexing sequences 81%  \r",
      "Indexing sequences 82%  \r",
      "Indexing sequences 83%  \r",
      "Indexing sequences 84%  \r",
      "Indexing sequences 84%  \r",
      "Indexing sequences 85%  \r",
      "Indexing sequences 86%  \r",
      "Indexing sequences 87%  \r",
      "Indexing sequences 87%  \r",
      "Indexing sequences 88%  \r",
      "Indexing sequences 89%  \r",
      "Indexing sequences 90%  \r",
      "Indexing sequences 90%  \r",
      "Indexing sequences 91%  \r",
      "Indexing sequences 92%  \r",
      "Indexing sequences 93%  \r",
      "Indexing sequences 93%  \r",
      "Indexing sequences 94%  \r",
      "Indexing sequences 95%  \r",
      "Indexing sequences 96%  \r",
      "Indexing sequences 96%  \r",
      "Indexing sequences 97%  \r",
      "Indexing sequences 98%  \r",
      "Indexing sequences 99%  \r",
      "Indexing sequences 99%  \r",
      "Indexing sequences 100%\n",
      "Dereplicating 0%  \r",
      "Dereplicating 0%  \r",
      "Dereplicating 1%  \r",
      "Dereplicating 1%  \r",
      "Dereplicating 2%  \r",
      "Dereplicating 3%  \r",
      "Dereplicating 4%  \r",
      "Dereplicating 4%  \r",
      "Dereplicating 5%  \r",
      "Dereplicating 6%  \r",
      "Dereplicating 7%  \r",
      "Dereplicating 7%  \r",
      "Dereplicating 8%  \r",
      "Dereplicating 9%  \r",
      "Dereplicating 10%  \r",
      "Dereplicating 10%  \r",
      "Dereplicating 11%  \r",
      "Dereplicating 12%  \r",
      "Dereplicating 13%  \r",
      "Dereplicating 13%  \r",
      "Dereplicating 14%  \r",
      "Dereplicating 15%  \r",
      "Dereplicating 16%  \r",
      "Dereplicating 16%  \r",
      "Dereplicating 17%  \r",
      "Dereplicating 18%  \r",
      "Dereplicating 19%  \r",
      "Dereplicating 19%  \r",
      "Dereplicating 20%  \r",
      "Dereplicating 21%  \r",
      "Dereplicating 21%  \r",
      "Dereplicating 22%  \r",
      "Dereplicating 23%  \r",
      "Dereplicating 24%  \r",
      "Dereplicating 24%  \r",
      "Dereplicating 25%  \r",
      "Dereplicating 26%  \r",
      "Dereplicating 27%  \r",
      "Dereplicating 27%  \r",
      "Dereplicating 28%  \r",
      "Dereplicating 29%  \r",
      "Dereplicating 30%  \r",
      "Dereplicating 30%  \r",
      "Dereplicating 31%  \r",
      "Dereplicating 32%  \r",
      "Dereplicating 33%  \r",
      "Dereplicating 33%  \r",
      "Dereplicating 34%  \r",
      "Dereplicating 35%  \r",
      "Dereplicating 36%  \r",
      "Dereplicating 36%  \r",
      "Dereplicating 37%  \r",
      "Dereplicating 38%  \r",
      "Dereplicating 39%  \r",
      "Dereplicating 39%  \r",
      "Dereplicating 40%  \r",
      "Dereplicating 41%  \r",
      "Dereplicating 41%  \r",
      "Dereplicating 42%  \r",
      "Dereplicating 43%  \r",
      "Dereplicating 44%  \r",
      "Dereplicating 44%  \r",
      "Dereplicating 45%  \r",
      "Dereplicating 46%  \r",
      "Dereplicating 47%  \r",
      "Dereplicating 47%  \r",
      "Dereplicating 48%  \r",
      "Dereplicating 49%  \r",
      "Dereplicating 50%  \r",
      "Dereplicating 50%  \r",
      "Dereplicating 51%  \r",
      "Dereplicating 52%  \r",
      "Dereplicating 53%  \r",
      "Dereplicating 53%  \r",
      "Dereplicating 54%  \r",
      "Dereplicating 55%  \r",
      "Dereplicating 56%  \r",
      "Dereplicating 56%  \r",
      "Dereplicating 57%  \r",
      "Dereplicating 58%  \r",
      "Dereplicating 59%  \r",
      "Dereplicating 59%  \r",
      "Dereplicating 60%  \r",
      "Dereplicating 61%  \r",
      "Dereplicating 61%  \r",
      "Dereplicating 62%  \r",
      "Dereplicating 63%  \r",
      "Dereplicating 64%  \r",
      "Dereplicating 64%  \r",
      "Dereplicating 65%  \r",
      "Dereplicating 66%  \r",
      "Dereplicating 67%  \r",
      "Dereplicating 67%  \r",
      "Dereplicating 68%  \r",
      "Dereplicating 69%  \r",
      "Dereplicating 70%  \r",
      "Dereplicating 70%  \r",
      "Dereplicating 71%  \r",
      "Dereplicating 72%  \r",
      "Dereplicating 73%  \r",
      "Dereplicating 73%  \r",
      "Dereplicating 74%  \r",
      "Dereplicating 75%  \r",
      "Dereplicating 76%  \r",
      "Dereplicating 76%  \r",
      "Dereplicating 77%  \r",
      "Dereplicating 78%  \r",
      "Dereplicating 79%  \r",
      "Dereplicating 79%  \r",
      "Dereplicating 80%  \r",
      "Dereplicating 81%  \r",
      "Dereplicating 81%  \r",
      "Dereplicating 82%  \r",
      "Dereplicating 83%  \r",
      "Dereplicating 84%  \r",
      "Dereplicating 84%  \r",
      "Dereplicating 85%  \r",
      "Dereplicating 86%  \r",
      "Dereplicating 87%  \r",
      "Dereplicating 87%  \r",
      "Dereplicating 88%  \r",
      "Dereplicating 89%  \r",
      "Dereplicating 90%  \r",
      "Dereplicating 90%  \r",
      "Dereplicating 91%  \r",
      "Dereplicating 92%  \r",
      "Dereplicating 93%  \r",
      "Dereplicating 93%  \r",
      "Dereplicating 94%  \r",
      "Dereplicating 95%  \r",
      "Dereplicating 96%  \r",
      "Dereplicating 96%  \r",
      "Dereplicating 97%  \r",
      "Dereplicating 98%  \r",
      "Dereplicating 99%  \r",
      "Dereplicating 99%  \r",
      "Dereplicating 100%\n",
      "Sorting 0%  \r",
      "Sorting 100%\n",
      "133 unique sequences, avg cluster 1.0, median 1, max 3\n",
      "Writing output file 0%  \r",
      "Writing output file 0%  \r",
      "Writing output file 1%  \r",
      "Writing output file 2%  \r",
      "Writing output file 2%  \r",
      "Writing output file 3%  \r",
      "Writing output file 4%  \r",
      "Writing output file 5%  \r",
      "Writing output file 5%  \r",
      "Writing output file 6%  \r",
      "Writing output file 7%  \r",
      "Writing output file 8%  \r",
      "Writing output file 8%  \r",
      "Writing output file 9%  \r",
      "Writing output file 10%  \r",
      "Writing output file 11%  \r",
      "Writing output file 11%  \r",
      "Writing output file 12%  \r",
      "Writing output file 13%  \r",
      "Writing output file 14%  \r",
      "Writing output file 14%  \r",
      "Writing output file 15%  \r",
      "Writing output file 16%  \r",
      "Writing output file 17%  \r",
      "Writing output file 17%  \r",
      "Writing output file 18%  \r",
      "Writing output file 19%  \r",
      "Writing output file 20%  \r",
      "Writing output file 20%  \r",
      "Writing output file 21%  \r",
      "Writing output file 22%  \r",
      "Writing output file 23%  \r",
      "Writing output file 23%  \r",
      "Writing output file 24%  \r",
      "Writing output file 25%  \r",
      "Writing output file 26%  \r",
      "Writing output file 26%  \r",
      "Writing output file 27%  \r",
      "Writing output file 28%  \r",
      "Writing output file 29%  \r",
      "Writing output file 29%  \r",
      "Writing output file 30%  \r",
      "Writing output file 31%  \r",
      "Writing output file 32%  \r",
      "Writing output file 32%  \r",
      "Writing output file 33%  \r",
      "Writing output file 34%  \r",
      "Writing output file 35%  \r",
      "Writing output file 35%  \r",
      "Writing output file 36%  \r",
      "Writing output file 37%  \r",
      "Writing output file 38%  \r",
      "Writing output file 38%  \r",
      "Writing output file 39%  \r",
      "Writing output file 40%  \r",
      "Writing output file 41%  \r",
      "Writing output file 41%  \r",
      "Writing output file 42%  \r",
      "Writing output file 43%  \r",
      "Writing output file 44%  \r",
      "Writing output file 44%  \r",
      "Writing output file 45%  \r",
      "Writing output file 46%  \r",
      "Writing output file 47%  \r",
      "Writing output file 47%  \r",
      "Writing output file 48%  \r",
      "Writing output file 49%  \r",
      "Writing output file 50%  \r",
      "Writing output file 50%  \r",
      "Writing output file 51%  \r",
      "Writing output file 52%  \r",
      "Writing output file 53%  \r",
      "Writing output file 53%  \r",
      "Writing output file 54%  \r",
      "Writing output file 55%  \r",
      "Writing output file 56%  \r",
      "Writing output file 56%  \r",
      "Writing output file 57%  \r",
      "Writing output file 58%  \r",
      "Writing output file 59%  \r",
      "Writing output file 59%  \r",
      "Writing output file 60%  \r",
      "Writing output file 61%  \r",
      "Writing output file 62%  \r",
      "Writing output file 62%  \r",
      "Writing output file 63%  \r",
      "Writing output file 64%  \r",
      "Writing output file 65%  \r",
      "Writing output file 65%  \r",
      "Writing output file 66%  \r",
      "Writing output file 67%  \r",
      "Writing output file 68%  \r",
      "Writing output file 68%  \r",
      "Writing output file 69%  \r",
      "Writing output file 70%  \r",
      "Writing output file 71%  \r",
      "Writing output file 71%  \r",
      "Writing output file 72%  \r",
      "Writing output file 73%  \r",
      "Writing output file 74%  \r",
      "Writing output file 74%  \r",
      "Writing output file 75%  \r",
      "Writing output file 76%  \r",
      "Writing output file 77%  \r",
      "Writing output file 77%  \r",
      "Writing output file 78%  \r",
      "Writing output file 79%  \r",
      "Writing output file 80%  \r",
      "Writing output file 80%  \r",
      "Writing output file 81%  \r",
      "Writing output file 82%  \r",
      "Writing output file 83%  \r",
      "Writing output file 83%  \r",
      "Writing output file 84%  \r",
      "Writing output file 85%  \r",
      "Writing output file 86%  \r",
      "Writing output file 86%  \r",
      "Writing output file 87%  \r",
      "Writing output file 88%  \r",
      "Writing output file 89%  \r",
      "Writing output file 89%  \r",
      "Writing output file 90%  \r",
      "Writing output file 91%  \r",
      "Writing output file 92%  \r",
      "Writing output file 92%  \r",
      "Writing output file 93%  \r",
      "Writing output file 94%  \r",
      "Writing output file 95%  \r",
      "Writing output file 95%  \r",
      "Writing output file 96%  \r",
      "Writing output file 97%  \r",
      "Writing output file 98%  \r",
      "Writing output file 98%  \r",
      "Writing output file 99%  \r",
      "Writing output file 100%\n",
      "Writing uc file, first part 0%  \r",
      "Writing uc file, first part 0%  \r",
      "Writing uc file, first part 1%  \r",
      "Writing uc file, first part 2%  \r",
      "Writing uc file, first part 2%  \r",
      "Writing uc file, first part 3%  \r",
      "Writing uc file, first part 4%  \r",
      "Writing uc file, first part 5%  \r",
      "Writing uc file, first part 5%  \r",
      "Writing uc file, first part 6%  \r",
      "Writing uc file, first part 7%  \r",
      "Writing uc file, first part 8%  \r",
      "Writing uc file, first part 8%  \r",
      "Writing uc file, first part 9%  \r",
      "Writing uc file, first part 10%  \r",
      "Writing uc file, first part 11%  \r",
      "Writing uc file, first part 11%  \r",
      "Writing uc file, first part 12%  \r",
      "Writing uc file, first part 13%  \r",
      "Writing uc file, first part 14%  \r",
      "Writing uc file, first part 14%  \r",
      "Writing uc file, first part 15%  \r",
      "Writing uc file, first part 16%  \r",
      "Writing uc file, first part 17%  \r",
      "Writing uc file, first part 17%  \r",
      "Writing uc file, first part 18%  \r",
      "Writing uc file, first part 19%  \r",
      "Writing uc file, first part 20%  \r",
      "Writing uc file, first part 20%  \r",
      "Writing uc file, first part 21%  \r",
      "Writing uc file, first part 22%  \r",
      "Writing uc file, first part 23%  \r",
      "Writing uc file, first part 23%  \r",
      "Writing uc file, first part 24%  \r",
      "Writing uc file, first part 25%  \r",
      "Writing uc file, first part 26%  \r",
      "Writing uc file, first part 26%  \r",
      "Writing uc file, first part 27%  \r",
      "Writing uc file, first part 28%  \r",
      "Writing uc file, first part 29%  \r",
      "Writing uc file, first part 29%  \r",
      "Writing uc file, first part 30%  \r",
      "Writing uc file, first part 31%  \r",
      "Writing uc file, first part 32%  \r",
      "Writing uc file, first part 32%  \r",
      "Writing uc file, first part 33%  \r",
      "Writing uc file, first part 34%  \r",
      "Writing uc file, first part 35%  \r",
      "Writing uc file, first part 35%  \r",
      "Writing uc file, first part 36%  \r",
      "Writing uc file, first part 37%  \r",
      "Writing uc file, first part 38%  \r",
      "Writing uc file, first part 38%  \r",
      "Writing uc file, first part 39%  \r",
      "Writing uc file, first part 40%  \r",
      "Writing uc file, first part 41%  \r",
      "Writing uc file, first part 41%  \r",
      "Writing uc file, first part 42%  \r",
      "Writing uc file, first part 43%  \r",
      "Writing uc file, first part 44%  \r",
      "Writing uc file, first part 44%  \r",
      "Writing uc file, first part 45%  \r",
      "Writing uc file, first part 46%  \r",
      "Writing uc file, first part 47%  \r",
      "Writing uc file, first part 47%  \r",
      "Writing uc file, first part 48%  \r",
      "Writing uc file, first part 49%  \r",
      "Writing uc file, first part 50%  \r",
      "Writing uc file, first part 50%  \r",
      "Writing uc file, first part 51%  \r",
      "Writing uc file, first part 52%  \r",
      "Writing uc file, first part 53%  \r",
      "Writing uc file, first part 53%  \r",
      "Writing uc file, first part 54%  \r",
      "Writing uc file, first part 55%  \r",
      "Writing uc file, first part 56%  \r",
      "Writing uc file, first part 56%  \r",
      "Writing uc file, first part 57%  \r",
      "Writing uc file, first part 58%  \r",
      "Writing uc file, first part 59%  \r",
      "Writing uc file, first part 59%  \r",
      "Writing uc file, first part 60%  \r",
      "Writing uc file, first part 61%  \r",
      "Writing uc file, first part 62%  \r",
      "Writing uc file, first part 62%  \r",
      "Writing uc file, first part 63%  \r",
      "Writing uc file, first part 64%  \r",
      "Writing uc file, first part 65%  \r",
      "Writing uc file, first part 65%  \r",
      "Writing uc file, first part 66%  \r",
      "Writing uc file, first part 67%  \r",
      "Writing uc file, first part 68%  \r",
      "Writing uc file, first part 68%  \r",
      "Writing uc file, first part 69%  \r",
      "Writing uc file, first part 70%  \r",
      "Writing uc file, first part 71%  \r",
      "Writing uc file, first part 71%  \r",
      "Writing uc file, first part 72%  \r",
      "Writing uc file, first part 73%  \r",
      "Writing uc file, first part 74%  \r",
      "Writing uc file, first part 74%  \r",
      "Writing uc file, first part 75%  \r",
      "Writing uc file, first part 76%  \r",
      "Writing uc file, first part 77%  \r",
      "Writing uc file, first part 77%  \r",
      "Writing uc file, first part 78%  \r",
      "Writing uc file, first part 79%  \r",
      "Writing uc file, first part 80%  \r",
      "Writing uc file, first part 80%  \r",
      "Writing uc file, first part 81%  \r",
      "Writing uc file, first part 82%  \r",
      "Writing uc file, first part 83%  \r",
      "Writing uc file, first part 83%  \r",
      "Writing uc file, first part 84%  \r",
      "Writing uc file, first part 85%  \r",
      "Writing uc file, first part 86%  \r",
      "Writing uc file, first part 86%  \r",
      "Writing uc file, first part 87%  \r",
      "Writing uc file, first part 88%  \r",
      "Writing uc file, first part 89%  \r",
      "Writing uc file, first part 89%  \r",
      "Writing uc file, first part 90%  \r",
      "Writing uc file, first part 91%  \r",
      "Writing uc file, first part 92%  \r",
      "Writing uc file, first part 92%  \r",
      "Writing uc file, first part 93%  \r",
      "Writing uc file, first part 94%  \r",
      "Writing uc file, first part 95%  \r",
      "Writing uc file, first part 95%  \r",
      "Writing uc file, first part 96%  \r",
      "Writing uc file, first part 97%  \r",
      "Writing uc file, first part 98%  \r",
      "Writing uc file, first part 98%  \r",
      "Writing uc file, first part 99%  \r",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 0%  \r",
      "Writing uc file, second part 0%  \r",
      "Writing uc file, second part 1%  \r",
      "Writing uc file, second part 2%  \r",
      "Writing uc file, second part 2%  \r",
      "Writing uc file, second part 3%  \r",
      "Writing uc file, second part 4%  \r",
      "Writing uc file, second part 5%  \r",
      "Writing uc file, second part 5%  \r",
      "Writing uc file, second part 6%  \r",
      "Writing uc file, second part 7%  \r",
      "Writing uc file, second part 8%  \r",
      "Writing uc file, second part 8%  \r",
      "Writing uc file, second part 9%  \r",
      "Writing uc file, second part 10%  \r",
      "Writing uc file, second part 11%  \r",
      "Writing uc file, second part 11%  \r",
      "Writing uc file, second part 12%  \r",
      "Writing uc file, second part 13%  \r",
      "Writing uc file, second part 14%  \r",
      "Writing uc file, second part 14%  \r",
      "Writing uc file, second part 15%  \r",
      "Writing uc file, second part 16%  \r",
      "Writing uc file, second part 17%  \r",
      "Writing uc file, second part 17%  \r",
      "Writing uc file, second part 18%  \r",
      "Writing uc file, second part 19%  \r",
      "Writing uc file, second part 20%  \r",
      "Writing uc file, second part 20%  \r",
      "Writing uc file, second part 21%  \r",
      "Writing uc file, second part 22%  \r",
      "Writing uc file, second part 23%  \r",
      "Writing uc file, second part 23%  \r",
      "Writing uc file, second part 24%  \r",
      "Writing uc file, second part 25%  \r",
      "Writing uc file, second part 26%  \r",
      "Writing uc file, second part 26%  \r",
      "Writing uc file, second part 27%  \r",
      "Writing uc file, second part 28%  \r",
      "Writing uc file, second part 29%  \r",
      "Writing uc file, second part 29%  \r",
      "Writing uc file, second part 30%  \r",
      "Writing uc file, second part 31%  \r",
      "Writing uc file, second part 32%  \r",
      "Writing uc file, second part 32%  \r",
      "Writing uc file, second part 33%  \r",
      "Writing uc file, second part 34%  \r",
      "Writing uc file, second part 35%  \r",
      "Writing uc file, second part 35%  \r",
      "Writing uc file, second part 36%  \r",
      "Writing uc file, second part 37%  \r",
      "Writing uc file, second part 38%  \r",
      "Writing uc file, second part 38%  \r",
      "Writing uc file, second part 39%  \r",
      "Writing uc file, second part 40%  \r",
      "Writing uc file, second part 41%  \r",
      "Writing uc file, second part 41%  \r",
      "Writing uc file, second part 42%  \r",
      "Writing uc file, second part 43%  \r",
      "Writing uc file, second part 44%  \r",
      "Writing uc file, second part 44%  \r",
      "Writing uc file, second part 45%  \r",
      "Writing uc file, second part 46%  \r",
      "Writing uc file, second part 47%  \r",
      "Writing uc file, second part 47%  \r",
      "Writing uc file, second part 48%  \r",
      "Writing uc file, second part 49%  \r",
      "Writing uc file, second part 50%  \r",
      "Writing uc file, second part 50%  \r",
      "Writing uc file, second part 51%  \r",
      "Writing uc file, second part 52%  \r",
      "Writing uc file, second part 53%  \r",
      "Writing uc file, second part 53%  \r",
      "Writing uc file, second part 54%  \r",
      "Writing uc file, second part 55%  \r",
      "Writing uc file, second part 56%  \r",
      "Writing uc file, second part 56%  \r",
      "Writing uc file, second part 57%  \r",
      "Writing uc file, second part 58%  \r",
      "Writing uc file, second part 59%  \r",
      "Writing uc file, second part 59%  \r",
      "Writing uc file, second part 60%  \r",
      "Writing uc file, second part 61%  \r",
      "Writing uc file, second part 62%  \r",
      "Writing uc file, second part 62%  \r",
      "Writing uc file, second part 63%  \r",
      "Writing uc file, second part 64%  \r",
      "Writing uc file, second part 65%  \r",
      "Writing uc file, second part 65%  \r",
      "Writing uc file, second part 66%  \r",
      "Writing uc file, second part 67%  \r",
      "Writing uc file, second part 68%  \r",
      "Writing uc file, second part 68%  \r",
      "Writing uc file, second part 69%  \r",
      "Writing uc file, second part 70%  \r",
      "Writing uc file, second part 71%  \r",
      "Writing uc file, second part 71%  \r",
      "Writing uc file, second part 72%  \r",
      "Writing uc file, second part 73%  \r",
      "Writing uc file, second part 74%  \r",
      "Writing uc file, second part 74%  \r",
      "Writing uc file, second part 75%  \r",
      "Writing uc file, second part 76%  \r",
      "Writing uc file, second part 77%  \r",
      "Writing uc file, second part 77%  \r",
      "Writing uc file, second part 78%  \r",
      "Writing uc file, second part 79%  \r",
      "Writing uc file, second part 80%  \r",
      "Writing uc file, second part 80%  \r",
      "Writing uc file, second part 81%  \r",
      "Writing uc file, second part 82%  \r",
      "Writing uc file, second part 83%  \r",
      "Writing uc file, second part 83%  \r",
      "Writing uc file, second part 84%  \r",
      "Writing uc file, second part 85%  \r",
      "Writing uc file, second part 86%  \r",
      "Writing uc file, second part 86%  \r",
      "Writing uc file, second part 87%  \r",
      "Writing uc file, second part 88%  \r",
      "Writing uc file, second part 89%  \r",
      "Writing uc file, second part 89%  \r",
      "Writing uc file, second part 90%  \r",
      "Writing uc file, second part 91%  \r",
      "Writing uc file, second part 92%  \r",
      "Writing uc file, second part 92%  \r",
      "Writing uc file, second part 93%  \r",
      "Writing uc file, second part 94%  \r",
      "Writing uc file, second part 95%  \r",
      "Writing uc file, second part 95%  \r",
      "Writing uc file, second part 96%  \r",
      "Writing uc file, second part 97%  \r",
      "Writing uc file, second part 98%  \r",
      "Writing uc file, second part 98%  \r",
      "Writing uc file, second part 99%  \r",
      "Writing uc file, second part 100%\n",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 0%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 1%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 1%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 2%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 2%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 3%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 3%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 4%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 4%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 5%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 5%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 6%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 6%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 7%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 7%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 8%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 9%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 9%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 10%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 10%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 11%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 11%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 12%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 12%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 13%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 13%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 14%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 14%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 15%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 15%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 16%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 16%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 17%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 18%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 18%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 19%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 19%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 20%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 20%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 21%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 21%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 22%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 22%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 23%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 23%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 24%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 24%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 25%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 25%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 26%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 27%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 27%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 28%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 28%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 29%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 29%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 30%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 30%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 31%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 31%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 32%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 32%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 33%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 33%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 34%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 35%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 35%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 36%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 36%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 37%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 37%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 38%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 38%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 39%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 39%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 40%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 40%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 41%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 41%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 42%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 42%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 43%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 43%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 44%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 45%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 45%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 46%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 46%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 47%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 47%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 48%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 48%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 49%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 49%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 50%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 50%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 51%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 51%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 52%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 52%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 53%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 54%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 54%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 55%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 55%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 56%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 56%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 57%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 57%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 58%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 58%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 59%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 59%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 60%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 60%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 61%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 61%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 62%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 62%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 63%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 63%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 64%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 64%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 65%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 65%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 66%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 67%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 67%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 68%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 68%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 69%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 69%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 70%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 70%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 71%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 71%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 72%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 72%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 73%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 73%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 74%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 74%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 75%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 75%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 76%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 76%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 77%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 77%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 78%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 79%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 79%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 80%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 80%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 81%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 81%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 82%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 82%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 83%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 83%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 84%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 84%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 85%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 85%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 86%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 86%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 87%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 87%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 88%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 88%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 89%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 90%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 90%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 91%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 91%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 92%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 92%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 93%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 93%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 94%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 94%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 95%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 95%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 96%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 96%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 97%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 97%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 98%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 98%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 99%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 100%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.fa 100%\n",
      "59494 nt in 190 seqs, min 305, max 339, avg 313\n",
      "Indexing sequences 0%  \r",
      "Indexing sequences 0%  \r",
      "Indexing sequences 1%  \r",
      "Indexing sequences 1%  \r",
      "Indexing sequences 2%  \r",
      "Indexing sequences 2%  \r",
      "Indexing sequences 3%  \r",
      "Indexing sequences 3%  \r",
      "Indexing sequences 4%  \r",
      "Indexing sequences 4%  \r",
      "Indexing sequences 5%  \r",
      "Indexing sequences 5%  \r",
      "Indexing sequences 6%  \r",
      "Indexing sequences 6%  \r",
      "Indexing sequences 7%  \r",
      "Indexing sequences 7%  \r",
      "Indexing sequences 8%  \r",
      "Indexing sequences 8%  \r",
      "Indexing sequences 9%  \r",
      "Indexing sequences 9%  \r",
      "Indexing sequences 10%  \r",
      "Indexing sequences 11%  \r",
      "Indexing sequences 11%  \r",
      "Indexing sequences 12%  \r",
      "Indexing sequences 12%  \r",
      "Indexing sequences 13%  \r",
      "Indexing sequences 13%  \r",
      "Indexing sequences 14%  \r",
      "Indexing sequences 14%  \r",
      "Indexing sequences 15%  \r",
      "Indexing sequences 15%  \r",
      "Indexing sequences 16%  \r",
      "Indexing sequences 16%  \r",
      "Indexing sequences 17%  \r",
      "Indexing sequences 17%  \r",
      "Indexing sequences 18%  \r",
      "Indexing sequences 18%  \r",
      "Indexing sequences 19%  \r",
      "Indexing sequences 19%  \r",
      "Indexing sequences 20%  \r",
      "Indexing sequences 21%  \r",
      "Indexing sequences 21%  \r",
      "Indexing sequences 22%  \r",
      "Indexing sequences 22%  \r",
      "Indexing sequences 23%  \r",
      "Indexing sequences 23%  \r",
      "Indexing sequences 24%  \r",
      "Indexing sequences 24%  \r",
      "Indexing sequences 25%  \r",
      "Indexing sequences 25%  \r",
      "Indexing sequences 26%  \r",
      "Indexing sequences 26%  \r",
      "Indexing sequences 27%  \r",
      "Indexing sequences 27%  \r",
      "Indexing sequences 28%  \r",
      "Indexing sequences 28%  \r",
      "Indexing sequences 29%  \r",
      "Indexing sequences 29%  \r",
      "Indexing sequences 30%  \r",
      "Indexing sequences 31%  \r",
      "Indexing sequences 31%  \r",
      "Indexing sequences 32%  \r",
      "Indexing sequences 32%  \r",
      "Indexing sequences 33%  \r",
      "Indexing sequences 33%  \r",
      "Indexing sequences 34%  \r",
      "Indexing sequences 34%  \r",
      "Indexing sequences 35%  \r",
      "Indexing sequences 35%  \r",
      "Indexing sequences 36%  \r",
      "Indexing sequences 36%  \r",
      "Indexing sequences 37%  \r",
      "Indexing sequences 37%  \r",
      "Indexing sequences 38%  \r",
      "Indexing sequences 38%  \r",
      "Indexing sequences 39%  \r",
      "Indexing sequences 39%  \r",
      "Indexing sequences 40%  \r",
      "Indexing sequences 41%  \r",
      "Indexing sequences 41%  \r",
      "Indexing sequences 42%  \r",
      "Indexing sequences 42%  \r",
      "Indexing sequences 43%  \r",
      "Indexing sequences 43%  \r",
      "Indexing sequences 44%  \r",
      "Indexing sequences 44%  \r",
      "Indexing sequences 45%  \r",
      "Indexing sequences 45%  \r",
      "Indexing sequences 46%  \r",
      "Indexing sequences 46%  \r",
      "Indexing sequences 47%  \r",
      "Indexing sequences 47%  \r",
      "Indexing sequences 48%  \r",
      "Indexing sequences 48%  \r",
      "Indexing sequences 49%  \r",
      "Indexing sequences 49%  \r",
      "Indexing sequences 50%  \r",
      "Indexing sequences 51%  \r",
      "Indexing sequences 51%  \r",
      "Indexing sequences 52%  \r",
      "Indexing sequences 52%  \r",
      "Indexing sequences 53%  \r",
      "Indexing sequences 53%  \r",
      "Indexing sequences 54%  \r",
      "Indexing sequences 54%  \r",
      "Indexing sequences 55%  \r",
      "Indexing sequences 55%  \r",
      "Indexing sequences 56%  \r",
      "Indexing sequences 56%  \r",
      "Indexing sequences 57%  \r",
      "Indexing sequences 57%  \r",
      "Indexing sequences 58%  \r",
      "Indexing sequences 58%  \r",
      "Indexing sequences 59%  \r",
      "Indexing sequences 59%  \r",
      "Indexing sequences 60%  \r",
      "Indexing sequences 61%  \r",
      "Indexing sequences 61%  \r",
      "Indexing sequences 62%  \r",
      "Indexing sequences 62%  \r",
      "Indexing sequences 63%  \r",
      "Indexing sequences 63%  \r",
      "Indexing sequences 64%  \r",
      "Indexing sequences 64%  \r",
      "Indexing sequences 65%  \r",
      "Indexing sequences 65%  \r",
      "Indexing sequences 66%  \r",
      "Indexing sequences 66%  \r",
      "Indexing sequences 67%  \r",
      "Indexing sequences 67%  \r",
      "Indexing sequences 68%  \r",
      "Indexing sequences 68%  \r",
      "Indexing sequences 69%  \r",
      "Indexing sequences 69%  \r",
      "Indexing sequences 70%  \r",
      "Indexing sequences 71%  \r",
      "Indexing sequences 71%  \r",
      "Indexing sequences 72%  \r",
      "Indexing sequences 72%  \r",
      "Indexing sequences 73%  \r",
      "Indexing sequences 73%  \r",
      "Indexing sequences 74%  \r",
      "Indexing sequences 74%  \r",
      "Indexing sequences 75%  \r",
      "Indexing sequences 75%  \r",
      "Indexing sequences 76%  \r",
      "Indexing sequences 76%  \r",
      "Indexing sequences 77%  \r",
      "Indexing sequences 77%  \r",
      "Indexing sequences 78%  \r",
      "Indexing sequences 78%  \r",
      "Indexing sequences 79%  \r",
      "Indexing sequences 79%  \r",
      "Indexing sequences 80%  \r",
      "Indexing sequences 81%  \r",
      "Indexing sequences 81%  \r",
      "Indexing sequences 82%  \r",
      "Indexing sequences 82%  \r",
      "Indexing sequences 83%  \r",
      "Indexing sequences 83%  \r",
      "Indexing sequences 84%  \r",
      "Indexing sequences 84%  \r",
      "Indexing sequences 85%  \r",
      "Indexing sequences 85%  \r",
      "Indexing sequences 86%  \r",
      "Indexing sequences 86%  \r",
      "Indexing sequences 87%  \r",
      "Indexing sequences 87%  \r",
      "Indexing sequences 88%  \r",
      "Indexing sequences 88%  \r",
      "Indexing sequences 89%  \r",
      "Indexing sequences 89%  \r",
      "Indexing sequences 90%  \r",
      "Indexing sequences 91%  \r",
      "Indexing sequences 91%  \r",
      "Indexing sequences 92%  \r",
      "Indexing sequences 92%  \r",
      "Indexing sequences 93%  \r",
      "Indexing sequences 93%  \r",
      "Indexing sequences 94%  \r",
      "Indexing sequences 94%  \r",
      "Indexing sequences 95%  \r",
      "Indexing sequences 95%  \r",
      "Indexing sequences 96%  \r",
      "Indexing sequences 96%  \r",
      "Indexing sequences 97%  \r",
      "Indexing sequences 97%  \r",
      "Indexing sequences 98%  \r",
      "Indexing sequences 98%  \r",
      "Indexing sequences 99%  \r",
      "Indexing sequences 99%  \r",
      "Indexing sequences 100%\n",
      "Dereplicating 0%  \r",
      "Dereplicating 0%  \r",
      "Dereplicating 1%  \r",
      "Dereplicating 1%  \r",
      "Dereplicating 2%  \r",
      "Dereplicating 2%  \r",
      "Dereplicating 3%  \r",
      "Dereplicating 3%  \r",
      "Dereplicating 4%  \r",
      "Dereplicating 4%  \r",
      "Dereplicating 5%  \r",
      "Dereplicating 5%  \r",
      "Dereplicating 6%  \r",
      "Dereplicating 6%  \r",
      "Dereplicating 7%  \r",
      "Dereplicating 7%  \r",
      "Dereplicating 8%  \r",
      "Dereplicating 8%  \r",
      "Dereplicating 9%  \r",
      "Dereplicating 9%  \r",
      "Dereplicating 10%  \r",
      "Dereplicating 11%  \r",
      "Dereplicating 11%  \r",
      "Dereplicating 12%  \r",
      "Dereplicating 12%  \r",
      "Dereplicating 13%  \r",
      "Dereplicating 13%  \r",
      "Dereplicating 14%  \r",
      "Dereplicating 14%  \r",
      "Dereplicating 15%  \r",
      "Dereplicating 15%  \r",
      "Dereplicating 16%  \r",
      "Dereplicating 16%  \r",
      "Dereplicating 17%  \r",
      "Dereplicating 17%  \r",
      "Dereplicating 18%  \r",
      "Dereplicating 18%  \r",
      "Dereplicating 19%  \r",
      "Dereplicating 19%  \r",
      "Dereplicating 20%  \r",
      "Dereplicating 21%  \r",
      "Dereplicating 21%  \r",
      "Dereplicating 22%  \r",
      "Dereplicating 22%  \r",
      "Dereplicating 23%  \r",
      "Dereplicating 23%  \r",
      "Dereplicating 24%  \r",
      "Dereplicating 24%  \r",
      "Dereplicating 25%  \r",
      "Dereplicating 25%  \r",
      "Dereplicating 26%  \r",
      "Dereplicating 26%  \r",
      "Dereplicating 27%  \r",
      "Dereplicating 27%  \r",
      "Dereplicating 28%  \r",
      "Dereplicating 28%  \r",
      "Dereplicating 29%  \r",
      "Dereplicating 29%  \r",
      "Dereplicating 30%  \r",
      "Dereplicating 31%  \r",
      "Dereplicating 31%  \r",
      "Dereplicating 32%  \r",
      "Dereplicating 32%  \r",
      "Dereplicating 33%  \r",
      "Dereplicating 33%  \r",
      "Dereplicating 34%  \r",
      "Dereplicating 34%  \r",
      "Dereplicating 35%  \r",
      "Dereplicating 35%  \r",
      "Dereplicating 36%  \r",
      "Dereplicating 36%  \r",
      "Dereplicating 37%  \r",
      "Dereplicating 37%  \r",
      "Dereplicating 38%  \r",
      "Dereplicating 38%  \r",
      "Dereplicating 39%  \r",
      "Dereplicating 39%  \r",
      "Dereplicating 40%  \r",
      "Dereplicating 41%  \r",
      "Dereplicating 41%  \r",
      "Dereplicating 42%  \r",
      "Dereplicating 42%  \r",
      "Dereplicating 43%  \r",
      "Dereplicating 43%  \r",
      "Dereplicating 44%  \r",
      "Dereplicating 44%  \r",
      "Dereplicating 45%  \r",
      "Dereplicating 45%  \r",
      "Dereplicating 46%  \r",
      "Dereplicating 46%  \r",
      "Dereplicating 47%  \r",
      "Dereplicating 47%  \r",
      "Dereplicating 48%  \r",
      "Dereplicating 48%  \r",
      "Dereplicating 49%  \r",
      "Dereplicating 49%  \r",
      "Dereplicating 50%  \r",
      "Dereplicating 51%  \r",
      "Dereplicating 51%  \r",
      "Dereplicating 52%  \r",
      "Dereplicating 52%  \r",
      "Dereplicating 53%  \r",
      "Dereplicating 53%  \r",
      "Dereplicating 54%  \r",
      "Dereplicating 54%  \r",
      "Dereplicating 55%  \r",
      "Dereplicating 55%  \r",
      "Dereplicating 56%  \r",
      "Dereplicating 56%  \r",
      "Dereplicating 57%  \r",
      "Dereplicating 57%  \r",
      "Dereplicating 58%  \r",
      "Dereplicating 58%  \r",
      "Dereplicating 59%  \r",
      "Dereplicating 59%  \r",
      "Dereplicating 60%  \r",
      "Dereplicating 61%  \r",
      "Dereplicating 61%  \r",
      "Dereplicating 62%  \r",
      "Dereplicating 62%  \r",
      "Dereplicating 63%  \r",
      "Dereplicating 63%  \r",
      "Dereplicating 64%  \r",
      "Dereplicating 64%  \r",
      "Dereplicating 65%  \r",
      "Dereplicating 65%  \r",
      "Dereplicating 66%  \r",
      "Dereplicating 66%  \r",
      "Dereplicating 67%  \r",
      "Dereplicating 67%  \r",
      "Dereplicating 68%  \r",
      "Dereplicating 68%  \r",
      "Dereplicating 69%  \r",
      "Dereplicating 69%  \r",
      "Dereplicating 70%  \r",
      "Dereplicating 71%  \r",
      "Dereplicating 71%  \r",
      "Dereplicating 72%  \r",
      "Dereplicating 72%  \r",
      "Dereplicating 73%  \r",
      "Dereplicating 73%  \r",
      "Dereplicating 74%  \r",
      "Dereplicating 74%  \r",
      "Dereplicating 75%  \r",
      "Dereplicating 75%  \r",
      "Dereplicating 76%  \r",
      "Dereplicating 76%  \r",
      "Dereplicating 77%  \r",
      "Dereplicating 77%  \r",
      "Dereplicating 78%  \r",
      "Dereplicating 78%  \r",
      "Dereplicating 79%  \r",
      "Dereplicating 79%  \r",
      "Dereplicating 80%  \r",
      "Dereplicating 81%  \r",
      "Dereplicating 81%  \r",
      "Dereplicating 82%  \r",
      "Dereplicating 82%  \r",
      "Dereplicating 83%  \r",
      "Dereplicating 83%  \r",
      "Dereplicating 84%  \r",
      "Dereplicating 84%  \r",
      "Dereplicating 85%  \r",
      "Dereplicating 85%  \r",
      "Dereplicating 86%  \r",
      "Dereplicating 86%  \r",
      "Dereplicating 87%  \r",
      "Dereplicating 87%  \r",
      "Dereplicating 88%  \r",
      "Dereplicating 88%  \r",
      "Dereplicating 89%  \r",
      "Dereplicating 89%  \r",
      "Dereplicating 90%  \r",
      "Dereplicating 91%  \r",
      "Dereplicating 91%  \r",
      "Dereplicating 92%  \r",
      "Dereplicating 92%  \r",
      "Dereplicating 93%  \r",
      "Dereplicating 93%  \r",
      "Dereplicating 94%  \r",
      "Dereplicating 94%  \r",
      "Dereplicating 95%  \r",
      "Dereplicating 95%  \r",
      "Dereplicating 96%  \r",
      "Dereplicating 96%  \r",
      "Dereplicating 97%  \r",
      "Dereplicating 97%  \r",
      "Dereplicating 98%  \r",
      "Dereplicating 98%  \r",
      "Dereplicating 99%  \r",
      "Dereplicating 99%  \r",
      "Dereplicating 100%\n",
      "Sorting 0%  \r",
      "Sorting 100%\n",
      "185 unique sequences, avg cluster 1.0, median 1, max 3\n",
      "Writing output file 0%  \r",
      "Writing output file 0%  \r",
      "Writing output file 1%  \r",
      "Writing output file 1%  \r",
      "Writing output file 2%  \r",
      "Writing output file 2%  \r",
      "Writing output file 3%  \r",
      "Writing output file 3%  \r",
      "Writing output file 4%  \r",
      "Writing output file 4%  \r",
      "Writing output file 5%  \r",
      "Writing output file 5%  \r",
      "Writing output file 6%  \r",
      "Writing output file 6%  \r",
      "Writing output file 7%  \r",
      "Writing output file 8%  \r",
      "Writing output file 8%  \r",
      "Writing output file 9%  \r",
      "Writing output file 9%  \r",
      "Writing output file 10%  \r",
      "Writing output file 10%  \r",
      "Writing output file 11%  \r",
      "Writing output file 11%  \r",
      "Writing output file 12%  \r",
      "Writing output file 12%  \r",
      "Writing output file 13%  \r",
      "Writing output file 14%  \r",
      "Writing output file 14%  \r",
      "Writing output file 15%  \r",
      "Writing output file 15%  \r",
      "Writing output file 16%  \r",
      "Writing output file 16%  \r",
      "Writing output file 17%  \r",
      "Writing output file 17%  \r",
      "Writing output file 18%  \r",
      "Writing output file 18%  \r",
      "Writing output file 19%  \r",
      "Writing output file 19%  \r",
      "Writing output file 20%  \r",
      "Writing output file 21%  \r",
      "Writing output file 21%  \r",
      "Writing output file 22%  \r",
      "Writing output file 22%  \r",
      "Writing output file 23%  \r",
      "Writing output file 23%  \r",
      "Writing output file 24%  \r",
      "Writing output file 24%  \r",
      "Writing output file 25%  \r",
      "Writing output file 25%  \r",
      "Writing output file 26%  \r",
      "Writing output file 26%  \r",
      "Writing output file 27%  \r",
      "Writing output file 28%  \r",
      "Writing output file 28%  \r",
      "Writing output file 29%  \r",
      "Writing output file 29%  \r",
      "Writing output file 30%  \r",
      "Writing output file 30%  \r",
      "Writing output file 31%  \r",
      "Writing output file 31%  \r",
      "Writing output file 32%  \r",
      "Writing output file 32%  \r",
      "Writing output file 33%  \r",
      "Writing output file 34%  \r",
      "Writing output file 34%  \r",
      "Writing output file 35%  \r",
      "Writing output file 35%  \r",
      "Writing output file 36%  \r",
      "Writing output file 36%  \r",
      "Writing output file 37%  \r",
      "Writing output file 37%  \r",
      "Writing output file 38%  \r",
      "Writing output file 38%  \r",
      "Writing output file 39%  \r",
      "Writing output file 39%  \r",
      "Writing output file 40%  \r",
      "Writing output file 41%  \r",
      "Writing output file 41%  \r",
      "Writing output file 42%  \r",
      "Writing output file 42%  \r",
      "Writing output file 43%  \r",
      "Writing output file 43%  \r",
      "Writing output file 44%  \r",
      "Writing output file 44%  \r",
      "Writing output file 45%  \r",
      "Writing output file 45%  \r",
      "Writing output file 46%  \r",
      "Writing output file 46%  \r",
      "Writing output file 47%  \r",
      "Writing output file 48%  \r",
      "Writing output file 48%  \r",
      "Writing output file 49%  \r",
      "Writing output file 49%  \r",
      "Writing output file 50%  \r",
      "Writing output file 50%  \r",
      "Writing output file 51%  \r",
      "Writing output file 51%  \r",
      "Writing output file 52%  \r",
      "Writing output file 52%  \r",
      "Writing output file 53%  \r",
      "Writing output file 54%  \r",
      "Writing output file 54%  \r",
      "Writing output file 55%  \r",
      "Writing output file 55%  \r",
      "Writing output file 56%  \r",
      "Writing output file 56%  \r",
      "Writing output file 57%  \r",
      "Writing output file 57%  \r",
      "Writing output file 58%  \r",
      "Writing output file 58%  \r",
      "Writing output file 59%  \r",
      "Writing output file 59%  \r",
      "Writing output file 60%  \r",
      "Writing output file 61%  \r",
      "Writing output file 61%  \r",
      "Writing output file 62%  \r",
      "Writing output file 62%  \r",
      "Writing output file 63%  \r",
      "Writing output file 63%  \r",
      "Writing output file 64%  \r",
      "Writing output file 64%  \r",
      "Writing output file 65%  \r",
      "Writing output file 65%  \r",
      "Writing output file 66%  \r",
      "Writing output file 66%  \r",
      "Writing output file 67%  \r",
      "Writing output file 68%  \r",
      "Writing output file 68%  \r",
      "Writing output file 69%  \r",
      "Writing output file 69%  \r",
      "Writing output file 70%  \r",
      "Writing output file 70%  \r",
      "Writing output file 71%  \r",
      "Writing output file 71%  \r",
      "Writing output file 72%  \r",
      "Writing output file 72%  \r",
      "Writing output file 73%  \r",
      "Writing output file 74%  \r",
      "Writing output file 74%  \r",
      "Writing output file 75%  \r",
      "Writing output file 75%  \r",
      "Writing output file 76%  \r",
      "Writing output file 76%  \r",
      "Writing output file 77%  \r",
      "Writing output file 77%  \r",
      "Writing output file 78%  \r",
      "Writing output file 78%  \r",
      "Writing output file 79%  \r",
      "Writing output file 79%  \r",
      "Writing output file 80%  \r",
      "Writing output file 81%  \r",
      "Writing output file 81%  \r",
      "Writing output file 82%  \r",
      "Writing output file 82%  \r",
      "Writing output file 83%  \r",
      "Writing output file 83%  \r",
      "Writing output file 84%  \r",
      "Writing output file 84%  \r",
      "Writing output file 85%  \r",
      "Writing output file 85%  \r",
      "Writing output file 86%  \r",
      "Writing output file 86%  \r",
      "Writing output file 87%  \r",
      "Writing output file 88%  \r",
      "Writing output file 88%  \r",
      "Writing output file 89%  \r",
      "Writing output file 89%  \r",
      "Writing output file 90%  \r",
      "Writing output file 90%  \r",
      "Writing output file 91%  \r",
      "Writing output file 91%  \r",
      "Writing output file 92%  \r",
      "Writing output file 92%  \r",
      "Writing output file 93%  \r",
      "Writing output file 94%  \r",
      "Writing output file 94%  \r",
      "Writing output file 95%  \r",
      "Writing output file 95%  \r",
      "Writing output file 96%  \r",
      "Writing output file 96%  \r",
      "Writing output file 97%  \r",
      "Writing output file 97%  \r",
      "Writing output file 98%  \r",
      "Writing output file 98%  \r",
      "Writing output file 99%  \r",
      "Writing output file 99%  \r",
      "Writing output file 100%\n",
      "Writing uc file, first part 0%  \r",
      "Writing uc file, first part 0%  \r",
      "Writing uc file, first part 1%  \r",
      "Writing uc file, first part 1%  \r",
      "Writing uc file, first part 2%  \r",
      "Writing uc file, first part 2%  \r",
      "Writing uc file, first part 3%  \r",
      "Writing uc file, first part 3%  \r",
      "Writing uc file, first part 4%  \r",
      "Writing uc file, first part 4%  \r",
      "Writing uc file, first part 5%  \r",
      "Writing uc file, first part 5%  \r",
      "Writing uc file, first part 6%  \r",
      "Writing uc file, first part 6%  \r",
      "Writing uc file, first part 7%  \r",
      "Writing uc file, first part 8%  \r",
      "Writing uc file, first part 8%  \r",
      "Writing uc file, first part 9%  \r",
      "Writing uc file, first part 9%  \r",
      "Writing uc file, first part 10%  \r",
      "Writing uc file, first part 10%  \r",
      "Writing uc file, first part 11%  \r",
      "Writing uc file, first part 11%  \r",
      "Writing uc file, first part 12%  \r",
      "Writing uc file, first part 12%  \r",
      "Writing uc file, first part 13%  \r",
      "Writing uc file, first part 14%  \r",
      "Writing uc file, first part 14%  \r",
      "Writing uc file, first part 15%  \r",
      "Writing uc file, first part 15%  \r",
      "Writing uc file, first part 16%  \r",
      "Writing uc file, first part 16%  \r",
      "Writing uc file, first part 17%  \r",
      "Writing uc file, first part 17%  \r",
      "Writing uc file, first part 18%  \r",
      "Writing uc file, first part 18%  \r",
      "Writing uc file, first part 19%  \r",
      "Writing uc file, first part 19%  \r",
      "Writing uc file, first part 20%  \r",
      "Writing uc file, first part 21%  \r",
      "Writing uc file, first part 21%  \r",
      "Writing uc file, first part 22%  \r",
      "Writing uc file, first part 22%  \r",
      "Writing uc file, first part 23%  \r",
      "Writing uc file, first part 23%  \r",
      "Writing uc file, first part 24%  \r",
      "Writing uc file, first part 24%  \r",
      "Writing uc file, first part 25%  \r",
      "Writing uc file, first part 25%  \r",
      "Writing uc file, first part 26%  \r",
      "Writing uc file, first part 26%  \r",
      "Writing uc file, first part 27%  \r",
      "Writing uc file, first part 28%  \r",
      "Writing uc file, first part 28%  \r",
      "Writing uc file, first part 29%  \r",
      "Writing uc file, first part 29%  \r",
      "Writing uc file, first part 30%  \r",
      "Writing uc file, first part 30%  \r",
      "Writing uc file, first part 31%  \r",
      "Writing uc file, first part 31%  \r",
      "Writing uc file, first part 32%  \r",
      "Writing uc file, first part 32%  \r",
      "Writing uc file, first part 33%  \r",
      "Writing uc file, first part 34%  \r",
      "Writing uc file, first part 34%  \r",
      "Writing uc file, first part 35%  \r",
      "Writing uc file, first part 35%  \r",
      "Writing uc file, first part 36%  \r",
      "Writing uc file, first part 36%  \r",
      "Writing uc file, first part 37%  \r",
      "Writing uc file, first part 37%  \r",
      "Writing uc file, first part 38%  \r",
      "Writing uc file, first part 38%  \r",
      "Writing uc file, first part 39%  \r",
      "Writing uc file, first part 39%  \r",
      "Writing uc file, first part 40%  \r",
      "Writing uc file, first part 41%  \r",
      "Writing uc file, first part 41%  \r",
      "Writing uc file, first part 42%  \r",
      "Writing uc file, first part 42%  \r",
      "Writing uc file, first part 43%  \r",
      "Writing uc file, first part 43%  \r",
      "Writing uc file, first part 44%  \r",
      "Writing uc file, first part 44%  \r",
      "Writing uc file, first part 45%  \r",
      "Writing uc file, first part 45%  \r",
      "Writing uc file, first part 46%  \r",
      "Writing uc file, first part 46%  \r",
      "Writing uc file, first part 47%  \r",
      "Writing uc file, first part 48%  \r",
      "Writing uc file, first part 48%  \r",
      "Writing uc file, first part 49%  \r",
      "Writing uc file, first part 49%  \r",
      "Writing uc file, first part 50%  \r",
      "Writing uc file, first part 50%  \r",
      "Writing uc file, first part 51%  \r",
      "Writing uc file, first part 51%  \r",
      "Writing uc file, first part 52%  \r",
      "Writing uc file, first part 52%  \r",
      "Writing uc file, first part 53%  \r",
      "Writing uc file, first part 54%  \r",
      "Writing uc file, first part 54%  \r",
      "Writing uc file, first part 55%  \r",
      "Writing uc file, first part 55%  \r",
      "Writing uc file, first part 56%  \r",
      "Writing uc file, first part 56%  \r",
      "Writing uc file, first part 57%  \r",
      "Writing uc file, first part 57%  \r",
      "Writing uc file, first part 58%  \r",
      "Writing uc file, first part 58%  \r",
      "Writing uc file, first part 59%  \r",
      "Writing uc file, first part 59%  \r",
      "Writing uc file, first part 60%  \r",
      "Writing uc file, first part 61%  \r",
      "Writing uc file, first part 61%  \r",
      "Writing uc file, first part 62%  \r",
      "Writing uc file, first part 62%  \r",
      "Writing uc file, first part 63%  \r",
      "Writing uc file, first part 63%  \r",
      "Writing uc file, first part 64%  \r",
      "Writing uc file, first part 64%  \r",
      "Writing uc file, first part 65%  \r",
      "Writing uc file, first part 65%  \r",
      "Writing uc file, first part 66%  \r",
      "Writing uc file, first part 66%  \r",
      "Writing uc file, first part 67%  \r",
      "Writing uc file, first part 68%  \r",
      "Writing uc file, first part 68%  \r",
      "Writing uc file, first part 69%  \r",
      "Writing uc file, first part 69%  \r",
      "Writing uc file, first part 70%  \r",
      "Writing uc file, first part 70%  \r",
      "Writing uc file, first part 71%  \r",
      "Writing uc file, first part 71%  \r",
      "Writing uc file, first part 72%  \r",
      "Writing uc file, first part 72%  \r",
      "Writing uc file, first part 73%  \r",
      "Writing uc file, first part 74%  \r",
      "Writing uc file, first part 74%  \r",
      "Writing uc file, first part 75%  \r",
      "Writing uc file, first part 75%  \r",
      "Writing uc file, first part 76%  \r",
      "Writing uc file, first part 76%  \r",
      "Writing uc file, first part 77%  \r",
      "Writing uc file, first part 77%  \r",
      "Writing uc file, first part 78%  \r",
      "Writing uc file, first part 78%  \r",
      "Writing uc file, first part 79%  \r",
      "Writing uc file, first part 79%  \r",
      "Writing uc file, first part 80%  \r",
      "Writing uc file, first part 81%  \r",
      "Writing uc file, first part 81%  \r",
      "Writing uc file, first part 82%  \r",
      "Writing uc file, first part 82%  \r",
      "Writing uc file, first part 83%  \r",
      "Writing uc file, first part 83%  \r",
      "Writing uc file, first part 84%  \r",
      "Writing uc file, first part 84%  \r",
      "Writing uc file, first part 85%  \r",
      "Writing uc file, first part 85%  \r",
      "Writing uc file, first part 86%  \r",
      "Writing uc file, first part 86%  \r",
      "Writing uc file, first part 87%  \r",
      "Writing uc file, first part 88%  \r",
      "Writing uc file, first part 88%  \r",
      "Writing uc file, first part 89%  \r",
      "Writing uc file, first part 89%  \r",
      "Writing uc file, first part 90%  \r",
      "Writing uc file, first part 90%  \r",
      "Writing uc file, first part 91%  \r",
      "Writing uc file, first part 91%  \r",
      "Writing uc file, first part 92%  \r",
      "Writing uc file, first part 92%  \r",
      "Writing uc file, first part 93%  \r",
      "Writing uc file, first part 94%  \r",
      "Writing uc file, first part 94%  \r",
      "Writing uc file, first part 95%  \r",
      "Writing uc file, first part 95%  \r",
      "Writing uc file, first part 96%  \r",
      "Writing uc file, first part 96%  \r",
      "Writing uc file, first part 97%  \r",
      "Writing uc file, first part 97%  \r",
      "Writing uc file, first part 98%  \r",
      "Writing uc file, first part 98%  \r",
      "Writing uc file, first part 99%  \r",
      "Writing uc file, first part 99%  \r",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 0%  \r",
      "Writing uc file, second part 0%  \r",
      "Writing uc file, second part 1%  \r",
      "Writing uc file, second part 1%  \r",
      "Writing uc file, second part 2%  \r",
      "Writing uc file, second part 2%  \r",
      "Writing uc file, second part 3%  \r",
      "Writing uc file, second part 3%  \r",
      "Writing uc file, second part 4%  \r",
      "Writing uc file, second part 4%  \r",
      "Writing uc file, second part 5%  \r",
      "Writing uc file, second part 5%  \r",
      "Writing uc file, second part 6%  \r",
      "Writing uc file, second part 6%  \r",
      "Writing uc file, second part 7%  \r",
      "Writing uc file, second part 8%  \r",
      "Writing uc file, second part 8%  \r",
      "Writing uc file, second part 9%  \r",
      "Writing uc file, second part 9%  \r",
      "Writing uc file, second part 10%  \r",
      "Writing uc file, second part 10%  \r",
      "Writing uc file, second part 11%  \r",
      "Writing uc file, second part 11%  \r",
      "Writing uc file, second part 12%  \r",
      "Writing uc file, second part 12%  \r",
      "Writing uc file, second part 13%  \r",
      "Writing uc file, second part 14%  \r",
      "Writing uc file, second part 14%  \r",
      "Writing uc file, second part 15%  \r",
      "Writing uc file, second part 15%  \r",
      "Writing uc file, second part 16%  \r",
      "Writing uc file, second part 16%  \r",
      "Writing uc file, second part 17%  \r",
      "Writing uc file, second part 17%  \r",
      "Writing uc file, second part 18%  \r",
      "Writing uc file, second part 18%  \r",
      "Writing uc file, second part 19%  \r",
      "Writing uc file, second part 19%  \r",
      "Writing uc file, second part 20%  \r",
      "Writing uc file, second part 21%  \r",
      "Writing uc file, second part 21%  \r",
      "Writing uc file, second part 22%  \r",
      "Writing uc file, second part 22%  \r",
      "Writing uc file, second part 23%  \r",
      "Writing uc file, second part 23%  \r",
      "Writing uc file, second part 24%  \r",
      "Writing uc file, second part 24%  \r",
      "Writing uc file, second part 25%  \r",
      "Writing uc file, second part 25%  \r",
      "Writing uc file, second part 26%  \r",
      "Writing uc file, second part 26%  \r",
      "Writing uc file, second part 27%  \r",
      "Writing uc file, second part 28%  \r",
      "Writing uc file, second part 28%  \r",
      "Writing uc file, second part 29%  \r",
      "Writing uc file, second part 29%  \r",
      "Writing uc file, second part 30%  \r",
      "Writing uc file, second part 30%  \r",
      "Writing uc file, second part 31%  \r",
      "Writing uc file, second part 31%  \r",
      "Writing uc file, second part 32%  \r",
      "Writing uc file, second part 32%  \r",
      "Writing uc file, second part 33%  \r",
      "Writing uc file, second part 34%  \r",
      "Writing uc file, second part 34%  \r",
      "Writing uc file, second part 35%  \r",
      "Writing uc file, second part 35%  \r",
      "Writing uc file, second part 36%  \r",
      "Writing uc file, second part 36%  \r",
      "Writing uc file, second part 37%  \r",
      "Writing uc file, second part 37%  \r",
      "Writing uc file, second part 38%  \r",
      "Writing uc file, second part 38%  \r",
      "Writing uc file, second part 39%  \r",
      "Writing uc file, second part 39%  \r",
      "Writing uc file, second part 40%  \r",
      "Writing uc file, second part 41%  \r",
      "Writing uc file, second part 41%  \r",
      "Writing uc file, second part 42%  \r",
      "Writing uc file, second part 42%  \r",
      "Writing uc file, second part 43%  \r",
      "Writing uc file, second part 43%  \r",
      "Writing uc file, second part 44%  \r",
      "Writing uc file, second part 44%  \r",
      "Writing uc file, second part 45%  \r",
      "Writing uc file, second part 45%  \r",
      "Writing uc file, second part 46%  \r",
      "Writing uc file, second part 46%  \r",
      "Writing uc file, second part 47%  \r",
      "Writing uc file, second part 48%  \r",
      "Writing uc file, second part 48%  \r",
      "Writing uc file, second part 49%  \r",
      "Writing uc file, second part 49%  \r",
      "Writing uc file, second part 50%  \r",
      "Writing uc file, second part 50%  \r",
      "Writing uc file, second part 51%  \r",
      "Writing uc file, second part 51%  \r",
      "Writing uc file, second part 52%  \r",
      "Writing uc file, second part 52%  \r",
      "Writing uc file, second part 53%  \r",
      "Writing uc file, second part 54%  \r",
      "Writing uc file, second part 54%  \r",
      "Writing uc file, second part 55%  \r",
      "Writing uc file, second part 55%  \r",
      "Writing uc file, second part 56%  \r",
      "Writing uc file, second part 56%  \r",
      "Writing uc file, second part 57%  \r",
      "Writing uc file, second part 57%  \r",
      "Writing uc file, second part 58%  \r",
      "Writing uc file, second part 58%  \r",
      "Writing uc file, second part 59%  \r",
      "Writing uc file, second part 59%  \r",
      "Writing uc file, second part 60%  \r",
      "Writing uc file, second part 61%  \r",
      "Writing uc file, second part 61%  \r",
      "Writing uc file, second part 62%  \r",
      "Writing uc file, second part 62%  \r",
      "Writing uc file, second part 63%  \r",
      "Writing uc file, second part 63%  \r",
      "Writing uc file, second part 64%  \r",
      "Writing uc file, second part 64%  \r",
      "Writing uc file, second part 65%  \r",
      "Writing uc file, second part 65%  \r",
      "Writing uc file, second part 66%  \r",
      "Writing uc file, second part 66%  \r",
      "Writing uc file, second part 67%  \r",
      "Writing uc file, second part 68%  \r",
      "Writing uc file, second part 68%  \r",
      "Writing uc file, second part 69%  \r",
      "Writing uc file, second part 69%  \r",
      "Writing uc file, second part 70%  \r",
      "Writing uc file, second part 70%  \r",
      "Writing uc file, second part 71%  \r",
      "Writing uc file, second part 71%  \r",
      "Writing uc file, second part 72%  \r",
      "Writing uc file, second part 72%  \r",
      "Writing uc file, second part 73%  \r",
      "Writing uc file, second part 74%  \r",
      "Writing uc file, second part 74%  \r",
      "Writing uc file, second part 75%  \r",
      "Writing uc file, second part 75%  \r",
      "Writing uc file, second part 76%  \r",
      "Writing uc file, second part 76%  \r",
      "Writing uc file, second part 77%  \r",
      "Writing uc file, second part 77%  \r",
      "Writing uc file, second part 78%  \r",
      "Writing uc file, second part 78%  \r",
      "Writing uc file, second part 79%  \r",
      "Writing uc file, second part 79%  \r",
      "Writing uc file, second part 80%  \r",
      "Writing uc file, second part 81%  \r",
      "Writing uc file, second part 81%  \r",
      "Writing uc file, second part 82%  \r",
      "Writing uc file, second part 82%  \r",
      "Writing uc file, second part 83%  \r",
      "Writing uc file, second part 83%  \r",
      "Writing uc file, second part 84%  \r",
      "Writing uc file, second part 84%  \r",
      "Writing uc file, second part 85%  \r",
      "Writing uc file, second part 85%  \r",
      "Writing uc file, second part 86%  \r",
      "Writing uc file, second part 86%  \r",
      "Writing uc file, second part 87%  \r",
      "Writing uc file, second part 88%  \r",
      "Writing uc file, second part 88%  \r",
      "Writing uc file, second part 89%  \r",
      "Writing uc file, second part 89%  \r",
      "Writing uc file, second part 90%  \r",
      "Writing uc file, second part 90%  \r",
      "Writing uc file, second part 91%  \r",
      "Writing uc file, second part 91%  \r",
      "Writing uc file, second part 92%  \r",
      "Writing uc file, second part 92%  \r",
      "Writing uc file, second part 93%  \r",
      "Writing uc file, second part 94%  \r",
      "Writing uc file, second part 94%  \r",
      "Writing uc file, second part 95%  \r",
      "Writing uc file, second part 95%  \r",
      "Writing uc file, second part 96%  \r",
      "Writing uc file, second part 96%  \r",
      "Writing uc file, second part 97%  \r",
      "Writing uc file, second part 97%  \r",
      "Writing uc file, second part 98%  \r",
      "Writing uc file, second part 98%  \r",
      "Writing uc file, second part 99%  \r",
      "Writing uc file, second part 99%  \r",
      "Writing uc file, second part 100%\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for OTU in $(ls -1 *.fa)\n",
    "do\n",
    "    vsearch --derep_fulllength $OTU \\\n",
    "    --strand both --output derep_$OTU \\\n",
    "    --uc derep_$OTU.uc --id 0.95\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each OTU, extract the most abundant dereplicated sequence as reference. Compare all other sequences to this one via `usearch_global`. Parse output and reverse complement any sequences if necessary."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Extract top sequence and write to file. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU\n",
      "Crangonyx_pseudogracilis_0_OTU\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "for i in range(len(OTUs_as_list)):\n",
    "    otu=OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    print syn\n",
    "    seqs = SeqIO.parse(open('derep_'+syn+'.fa','r'),'fasta')\n",
    "\n",
    "    seq = seqs.next()\n",
    "\n",
    "#    seq.seq = seq.seq.reverse_complement()\n",
    "\n",
    "    out=open('derep_'+syn+'.ref.fasta','w')\n",
    "    SeqIO.write(seq, out, 'fasta')\n",
    "    out.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Compare all sequences against the most abundant dereplicated sequences with `usearch_global`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading file derep_Crangonyx_floridanus_0_OTU.ref.fasta 0%  \r",
      "Reading file derep_Crangonyx_floridanus_0_OTU.ref.fasta 100%  \r",
      "Reading file derep_Crangonyx_floridanus_0_OTU.ref.fasta 100%\n",
      "314 nt in 1 seqs, min 314, max 314, avg 314\n",
      "Indexing sequences 0%  \r",
      "Indexing sequences 0%  \r",
      "Indexing sequences 100%\n",
      "Masking 0%  \r",
      "Masking 100%\n",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 100%\n",
      "Creating index of unique k-mers 0%  \r",
      "Creating index of unique k-mers 0%  \r",
      "Creating index of unique k-mers 100%\n",
      "Searching 0%  \r",
      "Searching 1%  \r",
      "Searching 3%  \r",
      "Searching 4%  \r",
      "Searching 5%  \r",
      "Searching 7%  \r",
      "Searching 8%  \r",
      "Searching 10%  \r",
      "Searching 11%  \r",
      "Searching 14%  \r",
      "Searching 16%  \r",
      "Searching 17%  \r",
      "Searching 20%  \r",
      "Searching 20%  \r",
      "Searching 23%  \r",
      "Searching 23%  \r",
      "Searching 26%  \r",
      "Searching 27%  \r",
      "Searching 29%  \r",
      "Searching 30%  \r",
      "Searching 31%  \r",
      "Searching 32%  \r",
      "Searching 33%  \r",
      "Searching 34%  \r",
      "Searching 35%  \r",
      "Searching 36%  \r",
      "Searching 36%  \r",
      "Searching 39%  \r",
      "Searching 40%  \r",
      "Searching 44%  \r",
      "Searching 44%  \r",
      "Searching 46%  \r",
      "Searching 50%  \r",
      "Searching 51%  \r",
      "Searching 53%  \r",
      "Searching 53%  \r",
      "Searching 54%  \r",
      "Searching 55%  \r",
      "Searching 56%  \r",
      "Searching 59%  \r",
      "Searching 59%  \r",
      "Searching 61%  \r",
      "Searching 62%  \r",
      "Searching 63%  \r",
      "Searching 65%  \r",
      "Searching 65%  \r",
      "Searching 68%  \r",
      "Searching 69%  \r",
      "Searching 70%  \r",
      "Searching 71%  \r",
      "Searching 72%  \r",
      "Searching 75%  \r",
      "Searching 76%  \r",
      "Searching 77%  \r",
      "Searching 78%  \r",
      "Searching 78%  \r",
      "Searching 80%  \r",
      "Searching 82%  \r",
      "Searching 83%  \r",
      "Searching 85%  \r",
      "Searching 86%  \r",
      "Searching 87%  \r",
      "Searching 87%  \r",
      "Searching 89%  \r",
      "Searching 90%  \r",
      "Searching 93%  \r",
      "Searching 94%  \r",
      "Searching 95%  \r",
      "Searching 96%  \r",
      "Searching 97%  \r",
      "Searching 98%  \r",
      "Searching 99%  \r",
      "Searching 99%  \r",
      "Searching 100%  \r",
      "Searching 100%\n",
      "Matching query sequences: 133 of 133 (100.00%)\n",
      "Reading file derep_Crangonyx_pseudogracilis_0_OTU.ref.fasta 0%  \r",
      "Reading file derep_Crangonyx_pseudogracilis_0_OTU.ref.fasta 100%  \r",
      "Reading file derep_Crangonyx_pseudogracilis_0_OTU.ref.fasta 100%\n",
      "312 nt in 1 seqs, min 312, max 312, avg 312\n",
      "Indexing sequences 0%  \r",
      "Indexing sequences 0%  \r",
      "Indexing sequences 100%\n",
      "Masking 0%  \r",
      "Masking 100%\n",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 100%\n",
      "Creating index of unique k-mers 0%  \r",
      "Creating index of unique k-mers 0%  \r",
      "Creating index of unique k-mers 100%\n",
      "Searching 0%  \r",
      "Searching 1%  \r",
      "Searching 2%  \r",
      "Searching 2%  \r",
      "Searching 3%  \r",
      "Searching 3%  \r",
      "Searching 6%  \r",
      "Searching 7%  \r",
      "Searching 7%  \r",
      "Searching 8%  \r",
      "Searching 8%  \r",
      "Searching 9%  \r",
      "Searching 10%  \r",
      "Searching 12%  \r",
      "Searching 12%  \r",
      "Searching 13%  \r",
      "Searching 14%  \r",
      "Searching 15%  \r",
      "Searching 16%  \r",
      "Searching 17%  \r",
      "Searching 17%  \r",
      "Searching 18%  \r",
      "Searching 18%  \r",
      "Searching 19%  \r",
      "Searching 21%  \r",
      "Searching 22%  \r",
      "Searching 23%  \r",
      "Searching 24%  \r",
      "Searching 25%  \r",
      "Searching 25%  \r",
      "Searching 26%  \r",
      "Searching 27%  \r",
      "Searching 27%  \r",
      "Searching 28%  \r",
      "Searching 29%  \r",
      "Searching 29%  \r",
      "Searching 30%  \r",
      "Searching 31%  \r",
      "Searching 31%  \r",
      "Searching 32%  \r",
      "Searching 33%  \r",
      "Searching 33%  \r",
      "Searching 38%  \r",
      "Searching 38%  \r",
      "Searching 39%  \r",
      "Searching 40%  \r",
      "Searching 40%  \r",
      "Searching 41%  \r",
      "Searching 42%  \r",
      "Searching 44%  \r",
      "Searching 45%  \r",
      "Searching 46%  \r",
      "Searching 47%  \r",
      "Searching 47%  \r",
      "Searching 49%  \r",
      "Searching 49%  \r",
      "Searching 51%  \r",
      "Searching 51%  \r",
      "Searching 52%  \r",
      "Searching 52%  \r",
      "Searching 53%  \r",
      "Searching 54%  \r",
      "Searching 55%  \r",
      "Searching 55%  \r",
      "Searching 56%  \r",
      "Searching 56%  \r",
      "Searching 57%  \r",
      "Searching 58%  \r",
      "Searching 58%  \r",
      "Searching 59%  \r",
      "Searching 59%  \r",
      "Searching 60%  \r",
      "Searching 61%  \r",
      "Searching 62%  \r",
      "Searching 64%  \r",
      "Searching 65%  \r",
      "Searching 66%  \r",
      "Searching 66%  \r",
      "Searching 67%  \r",
      "Searching 67%  \r",
      "Searching 68%  \r",
      "Searching 69%  \r",
      "Searching 69%  \r",
      "Searching 70%  \r",
      "Searching 71%  \r",
      "Searching 72%  \r",
      "Searching 72%  \r",
      "Searching 73%  \r",
      "Searching 73%  \r",
      "Searching 74%  \r",
      "Searching 74%  \r",
      "Searching 76%  \r",
      "Searching 77%  \r",
      "Searching 79%  \r",
      "Searching 79%  \r",
      "Searching 80%  \r",
      "Searching 80%  \r",
      "Searching 82%  \r",
      "Searching 83%  \r",
      "Searching 85%  \r",
      "Searching 86%  \r",
      "Searching 86%  \r",
      "Searching 87%  \r",
      "Searching 88%  \r",
      "Searching 89%  \r",
      "Searching 89%  \r",
      "Searching 90%  \r",
      "Searching 90%  \r",
      "Searching 91%  \r",
      "Searching 91%  \r",
      "Searching 92%  \r",
      "Searching 94%  \r",
      "Searching 94%  \r",
      "Searching 95%  \r",
      "Searching 95%  \r",
      "Searching 96%  \r",
      "Searching 96%  \r",
      "Searching 97%  \r",
      "Searching 97%  \r",
      "Searching 98%  \r",
      "Searching 99%  \r",
      "Searching 100%  \r",
      "Searching 100%\n",
      "Matching query sequences: 185 of 185 (100.00%)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for ref in $(ls -1 derep* | grep \"fasta\")\n",
    "do\n",
    "    full=$(echo -e \"$ref\" | sed 's/\\.ref//' | sed 's/sta$//')\n",
    "    \n",
    "    vsearch --usearch_global $full \\\n",
    "    --strand both \\\n",
    "    --db $ref \\\n",
    "    --id 0.9 \\\n",
    "    --blast6out $full.blast.out \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Parse output and identify sequences to reverse complement."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU\n",
      "[]\n",
      "Crangonyx_pseudogracilis_0_OTU\n",
      "[]\n"
     ]
    }
   ],
   "source": [
    "to_reverse = {}\n",
    "\n",
    "for i in range(len(OTUs_as_list)):\n",
    "    otu=OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    print syn\n",
    "    to_reverse[syn] = []\n",
    "    blast = open('derep_'+syn+'.fa.blast.out','r')\n",
    "\n",
    "    for rec in blast:\n",
    "        cols = rec.strip().split(\"\\t\")\n",
    "        if cols[6] > cols[7]:\n",
    "            to_reverse[syn].append(cols[0])\n",
    "        \n",
    "    print to_reverse[syn]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reverse complement if necessary."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU\n",
      "nothing to reverse\n",
      "Crangonyx_pseudogracilis_0_OTU\n",
      "nothing to reverse\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "for i in range(len(OTUs_as_list)):\n",
    "    otu=OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    print syn\n",
    "    if to_reverse[syn]:\n",
    "        seqs = SeqIO.parse(open('derep_'+syn+'.fa','r'),'fasta')\n",
    "        outseqs = []\n",
    "        for s in seqs:\n",
    "            print \"search: %s\" %s.id\n",
    "            if s.id in to_reverse[syn]:\n",
    "                print \"#%s\\t%s\" %(to_reverse[syn].index(s.id), to_reverse[syn][to_reverse[syn].index(s.id)])\n",
    "                s.seq = s.seq.reverse_complement()\n",
    "                del(to_reverse[syn][to_reverse[syn].index(s.id)])\n",
    "                outseqs.append(s)\n",
    "        out = open('derep_'+syn+'.fa','w')\n",
    "        SeqIO.write(outseqs, out, 'fasta')\n",
    "        out.close()\n",
    "    else:\n",
    "        print \"nothing to reverse\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Write top ten sequences to file and align for manual inspection."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU\n",
      "0 pcr-NEG-pl1-7-5-Oct-nc|1_2110_26480_22422_1_ex\n",
      "1 CH-BL2-pl1-7-4-Oct-nc|1_1102_10818_23263_1_ex\n",
      "2 CH101-1-8-May-nc|1_2114_22385_22080_1_ex\n",
      "3 CH101-pl1-1-1-Oct-nc|1_1101_19163_12271_1_ex\n",
      "4 CH101-pl1-1-1-Oct-nc|1_1105_9774_10777_1_ex\n",
      "5 CH101-pl1-1-1-Oct-nc|1_2105_17136_11806_1_ex\n",
      "6 CH102-1-7-May-nc|1_1102_17379_27773_1_ex\n",
      "7 CH102-pl1-1-2-Oct-nc|1_2112_22478_8174_1_ex\n",
      "8 CH103-pl1-1-3-Oct-nc|1_1102_9913_11787_1_ex\n",
      "9 CH104-1-5-May-nc|1_1103_23331_12069_1_ex\n",
      "Crangonyx_pseudogracilis_0_OTU\n",
      "0 CH113-pl1-2-5-Oct-nc|1_1102_10926_16218_1_ex\n",
      "1 CH204-2-3-May-nc|1_1113_25654_15317_1_ex\n",
      "2 CH304-pl1-3-7-Oct-nc|1_1106_17586_25029_1_ex\n",
      "3 CH501-pl1-4-7-Oct-nc|1_1114_21367_11968_1_ex\n",
      "4 CH103-1-6-May-nc|1_1101_24330_23874_1_ex\n",
      "5 CH103-1-6-May-nc|1_1102_10220_11006_1_ex\n",
      "6 CH103-1-6-May-nc|1_1102_27565_22423_1_ex\n",
      "7 CH103-1-6-May-nc|1_1102_7800_20792_1_ex\n",
      "8 CH103-1-6-May-nc|1_1102_8729_18418_1_ex\n",
      "9 CH103-1-6-May-nc|1_1103_11450_9537_1_ex\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "for i in range(len(OTUs_as_list)):\n",
    "    otu=OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    print syn\n",
    "    seqs = SeqIO.parse(open('derep_'+syn+'.fa','r'),'fasta')\n",
    "    count=0\n",
    "    seqs_to_print = []\n",
    "    for s in seqs:\n",
    "        print count,s.id\n",
    "        count+=1\n",
    "        seqs_to_print.append(s)\n",
    "        if count > 9:\n",
    "            break\n",
    "    out=open('derep_'+syn+'.top10.fa','w')\n",
    "    SeqIO.write(seqs_to_print,out,'fasta')\n",
    "    out.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Align top 10 with mafft."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "nseq =  10\n",
      "distance =  local\n",
      "iterate =  16\n",
      "cycle =  1\n",
      "nthread = 0\n",
      "lastonce = 0\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "    0 / 10\r",
      "    1 / 10\r",
      "    2 / 10\r",
      "    3 / 10\r",
      "    4 / 10\r",
      "    5 / 10\r",
      "    6 / 10\r",
      "    7 / 10\r",
      "    8 / 10\r\n",
      "\n",
      "##### writing hat3\n",
      "pairlocalalign (nuc) Version 7.123b alg=L, model=DNA200 (2),  2.000 ( 6.000), -0.099 (-0.297)\n",
      "0 thread(s)\n",
      "nthread = 0\n",
      "blosum 62 / kimura 200\n",
      "Loading 'hat3' ... \n",
      "done.\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "Loading 'hat2' ... done.\n",
      "Constructing a UPGMA tree ... \n",
      "\r",
      "    0 / 10\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "\r",
      "STEP     1 /9 c\r",
      "STEP     2 /9 c\r",
      "STEP     3 /9 c\r",
      "STEP     4 /9 c\r",
      "STEP     5 /9 c\r",
      "STEP     6 /9 c\r",
      "STEP     7 /9 c\r",
      "STEP     8 /9 c\r",
      "STEP     9 /9 c\n",
      "done.\n",
      "tbfast (nuc) Version 7.123b alg=A, model=DNA200 (2),  1.530 ( 4.590), -0.000 (-0.000)\n",
      "0 thread(s)\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "\n",
      "\r",
      "    0 / 10\n",
      "Segment   1/  1    1- 354\n",
      "STEP 001-001-0  identical.   \r",
      "STEP 001-001-1  identical.   \r",
      "STEP 001-002-0  identical.   \r",
      "STEP 001-002-1  identical.   \r",
      "STEP 001-003-0  identical.   \r",
      "STEP 001-003-1  identical.   \r",
      "STEP 001-004-0  identical.   \r",
      "STEP 001-004-1  identical.   \r",
      "STEP 001-005-0  identical.   \r",
      "STEP 001-005-1  identical.   \r",
      "STEP 001-006-0  identical.   \r",
      "STEP 001-006-1  identical.   \r",
      "STEP 001-007-0  identical.   \r",
      "STEP 001-007-1  identical.   \r",
      "STEP 001-008-0  identical.   \r",
      "STEP 001-008-1  identical.   \r",
      "STEP 001-009-1  identical.   \r",
      "STEP 002-009-1  identical.   \r",
      "STEP 002-008-0  identical.   \r",
      "STEP 002-008-1  identical.   \r\n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.123b alg=A, model=DNA200 (2),  1.530 ( 4.590), -0.000 (-0.000)\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --legacygappenalty option.\n",
      "\n",
      "\n",
      "nseq =  10\n",
      "distance =  local\n",
      "iterate =  16\n",
      "cycle =  1\n",
      "nthread = 0\n",
      "lastonce = 0\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "    0 / 10\r",
      "    1 / 10\r",
      "    2 / 10\r",
      "    3 / 10\r",
      "    4 / 10\r",
      "    5 / 10\r",
      "    6 / 10\r",
      "    7 / 10\r",
      "    8 / 10\r\n",
      "\n",
      "##### writing hat3\n",
      "pairlocalalign (nuc) Version 7.123b alg=L, model=DNA200 (2),  2.000 ( 6.000), -0.099 (-0.297)\n",
      "0 thread(s)\n",
      "nthread = 0\n",
      "blosum 62 / kimura 200\n",
      "Loading 'hat3' ... \n",
      "done.\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "Loading 'hat2' ... done.\n",
      "Constructing a UPGMA tree ... \n",
      "\r",
      "    0 / 10\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "\r",
      "STEP     1 /9 c\r",
      "STEP     2 /9 c\r",
      "STEP     3 /9 c\r",
      "STEP     4 /9 c\r",
      "STEP     5 /9 c\r",
      "STEP     6 /9 c\r",
      "STEP     7 /9 c\r",
      "STEP     8 /9 c\r",
      "STEP     9 /9 c\n",
      "done.\n",
      "tbfast (nuc) Version 7.123b alg=A, model=DNA200 (2),  1.530 ( 4.590), -0.000 (-0.000)\n",
      "0 thread(s)\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "\n",
      "\r",
      "    0 / 10\n",
      "Segment   1/  1    1- 317\n",
      "STEP 001-001-0  identical.   \r",
      "STEP 001-001-1  identical.   \r",
      "STEP 001-002-0  identical.   \r",
      "STEP 001-002-1  identical.   \r",
      "STEP 001-003-0  identical.   \r",
      "STEP 001-003-1  identical.   \r",
      "STEP 001-004-0  identical.   \r",
      "STEP 001-004-1  identical.   \r",
      "STEP 001-005-0  identical.   \r",
      "STEP 001-005-1  identical.   \r",
      "STEP 001-006-0  identical.   \r",
      "STEP 001-006-1  identical.   \r",
      "STEP 001-007-0  identical.   \r",
      "STEP 001-007-1  identical.   \r",
      "STEP 001-008-0  identical.   \r",
      "STEP 001-008-1  identical.   \r",
      "STEP 001-009-1  identical.   \r",
      "STEP 002-009-1  identical.   \r",
      "STEP 002-008-0  identical.   \r",
      "STEP 002-008-1  identical.   \r\n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.123b alg=A, model=DNA200 (2),  1.530 ( 4.590), -0.000 (-0.000)\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --legacygappenalty option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for file in $(ls -1 *top10*)\n",
    "do\n",
    "    out=$(echo -e \"$file\" | sed 's/fa$/aln.fa/')\n",
    "    mafft --localpair --maxiterate 1000 $file > $out\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Representatives of observed OTUs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Curated minibarcode (Leray CO1 region) sequences were selected as representatives and saved to two separate files (one per taxa)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing Crangonyx_floridanus_OTU.minibc.ref.fasta\n"
     ]
    }
   ],
   "source": [
    "%%file Crangonyx_floridanus_OTU.minibc.ref.fasta\n",
    ">CH101-pl1-1-1-Oct-nc|1_2105_17136_11806_1_ex\n",
    "tttagcatctacagctgctcatagaggtgcttctgtagacttagctattttctctcttcacctagcaggtgcctcctctattttaggttcaattaactttatttccacagtaataaatatacgagtaaaaaatatattaatagaccaaatccctttatttgtttgagctattttcttcactactattcttcttcttcttctttctttacctgttctagcaggagctatcacaatacttttaacagaccgtaatctcaatacatcattctttgacccttctggggggggtgaccctatcttgtaccagcatctctt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing Crangonyx_pseudogracilis_OTU.minibc.ref.fasta\n"
     ]
    }
   ],
   "source": [
    "%%file Crangonyx_pseudogracilis_OTU.minibc.ref.fasta\n",
    ">CH304-pl1-3-7-Oct-nc|1_1106_17586_25029_1_ex\n",
    "ctctatcatcaataacagcccacagaggttcatcagtagacctggctattttttctctccacctagctggtgcatcctcaattttaggagctatcaattttctatccacaataataaatataaaagtaaaaaaccttcttatagaccaagttcctttatttgtttgagcaattttttttacaacaattcttctccttctgtctctacctgttttagccggagctatcactatactattgacagaccgcaatcttaatacatcattctttgatccatcaggaggtggagaccctattctatatcaacatctttt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Comparing all sequences against the minibc reference sequences."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create blast databases."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 07/24/2019 18:10:02\n",
      "New DB name:   Crangonyx_floridanus_minibc_ref\n",
      "New DB title:  Crangonyx_floridanus_OTU.minibc.ref.fasta\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.000191927 seconds.\n",
      "\n",
      "\n",
      "Building a new DB, current time: 07/24/2019 18:10:02\n",
      "New DB name:   Crangonyx_pseudogracilis_minibc_ref\n",
      "New DB title:  Crangonyx_pseudogracilis_OTU.minibc.ref.fasta\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 1 sequences in 0.000222921 seconds.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "makeblastdb -in Crangonyx_floridanus_OTU.minibc.ref.fasta -dbtype nucl -out Crangonyx_floridanus_minibc_ref\n",
    "makeblastdb -in Crangonyx_pseudogracilis_OTU.minibc.ref.fasta -dbtype nucl -out Crangonyx_pseudogracilis_minibc_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run Blast on all *_OTU.fa* files, excluding all the dereplicated files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for file in $(ls -1 *.fa | grep \"derep\" -v)\n",
    "do\n",
    "    prefix=$(echo -e \"$file\" | sed 's/\\.fa$//')\n",
    "    sp=$(echo -e \"$file\" | cut -d \"_\" -f 1,2)\n",
    "        \n",
    "    blastn -db $sp\\_minibc_ref -query $file -outfmt 6 -out $prefix.vs.minibc.blastn.out\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Parse outputs and identify clipping points for sequences longer than the reference."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU\n",
      "Crangonyx_pseudogracilis_0_OTU\n"
     ]
    }
   ],
   "source": [
    "full_length=313\n",
    "\n",
    "\n",
    "global_clips = {}\n",
    "\n",
    "for i in range(len(OTUs_as_list)):\n",
    "    otu=OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    print syn\n",
    "    global_clips[syn] = {}\n",
    "    blast = open(syn+'.vs.minibc.blastn.out','r')\n",
    "    \n",
    "    for OTU in blast:\n",
    "        cols=OTU.strip().split(\"\\t\")\n",
    "        temp = []\n",
    "    #    print \"#%s\" %cols[0]\n",
    "        if int(cols[8]) < int(cols[9]):\n",
    "    #        print \"\\torientation ok\"\n",
    "            temp.append(0)\n",
    "            #check the first end for overhang\n",
    "            if int(cols[6]) == 1: #sequence alignment starts at position 1 -> lower clipping point is 0, i.e. not needed\n",
    "    #            print \"\\t1 - first end starts with 1 - no clipping on this side\"\n",
    "                temp.append(0)\n",
    "            else: #Alignment does not start at position 1, then specify clippoint to clip to same length as ref\n",
    "                temp.append(int(cols[6])-int(cols[8]))\n",
    "    #            print \"\\t1 - Alignment starts at pos %s vs. %s in ref - clip at: %s\" %(cols[6],cols[8],temp[-1])\n",
    "            \n",
    "            #check second end for overhang\n",
    "            if int(cols[9]) == full_length:\n",
    "                temp.append(int(cols[7]))\n",
    "    #            print \"\\t2 - Full lenght alignment ending at pos %s in query\" %cols[7]\n",
    "            else:\n",
    "                temp.append((full_length-int(cols[9]))+int(cols[7])) \n",
    "    #            print \"\\t2 - incomplete alignment ends with query pos %s and ref pos %s - clip at: %s\" %(cols[7], cols[9], temp[-1])\n",
    "        \n",
    "        \n",
    "        else:\n",
    "    #        print \"\\treverse complement\"\n",
    "            temp.append(1)\n",
    "            if int(cols[6]) == 1: #sequence alignment starts at position 1 -> lower clipping point is 0, i.e. not needed\n",
    "    #            print \"\\t1 - first end alignment starts at base 1 - no clipping on this side\"\n",
    "                temp.append(0)\n",
    "            else:\n",
    "                temp.append(int(cols[6])-(full_length-int(cols[8]))-1)\n",
    "    #            print \"\\t1 - alignment starts with query pos %s and ref pos %s - clip at: %s\" %(cols[6], cols[8], temp[-1])\n",
    "            \n",
    "            if int(cols[9]) == 1:\n",
    "    #            print \"\\t2 - full length alignment ending in pos %s in query\" %cols[7]\n",
    "                temp.append(int(cols[7]))\n",
    "            else:\n",
    "                temp.append(int(cols[7])+int(cols[9]))\n",
    "    #            print \"\\t2 - incomplete alignment ends with query %s at ref pos %s - clip at: %s\" %(cols[7], cols[9], temp[-1])\n",
    "\n",
    "        for i in range(len(temp)):\n",
    "            if temp[i] < 0:\n",
    "                temp[i] = 0\n",
    "            \n",
    "#    print \"\\t\"+str(temp)\n",
    "        global_clips[syn][cols[0]] = temp[:]\n",
    "    \n",
    "#print global_clips"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Clip and reverese complement based on blast results if necessary."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU\n",
      "Crangonyx_pseudogracilis_0_OTU\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "for i in range(len(OTUs_as_list)):\n",
    "    otu=OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    print syn\n",
    "\n",
    "    outseqs = []\n",
    "    seqs = SeqIO.parse(open(syn+'.fa','r'),'fasta')\n",
    "\n",
    "    for s in seqs:\n",
    "        s.seq = s.seq[global_clips[syn][s.id][1]:global_clips[syn][s.id][2]]\n",
    "        if global_clips[syn][s.id][0] == 1:\n",
    "            s.seq = s.seq.reverse_complement()\n",
    "        \n",
    "        outseqs.append(s)\n",
    "    \n",
    "    out = open(syn+'.clipped.fasta','w')\n",
    "    SeqIO.write(outseqs,out,'fasta')\n",
    "    out.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each OTU bin reads per sample."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU\n",
      "\twriting data for sample: CH101-pl1-1-1-Oct-nc\n",
      "\twriting data for sample: CH113-pl1-2-5-Oct-nc\n",
      "\twriting data for sample: CH206-2-1-May-nc\n",
      "\twriting data for sample: CH304-pl1-3-7-Oct-nc\n",
      "\twriting data for sample: CH112-pl1-2-4-Oct-nc\n",
      "\twriting data for sample: CH602-pl1-6-2-Oct-nc\n",
      "\twriting data for sample: CH603-pl1-6-3-Oct-nc\n",
      "\twriting data for sample: CH103-pl1-1-3-Oct-nc\n",
      "\twriting data for sample: CH502-pl1-4-8-Oct-nc\n",
      "\twriting data for sample: CH110-2-7-May-nc\n",
      "\twriting data for sample: CH102-1-7-May-nc\n",
      "\twriting data for sample: CH203-2-4-May-nc\n",
      "\twriting data for sample: CH114-pl1-2-6-Oct-nc\n",
      "\twriting data for sample: CH310-pl1-4-5-Oct-nc\n",
      "\twriting data for sample: CH308-pl1-4-3-Oct-nc\n",
      "\twriting data for sample: CH111-pl1-2-3-Oct-nc\n",
      "\twriting data for sample: CH609-pl1-7-1-Oct-nc\n",
      "\twriting data for sample: CH607-pl1-6-7-Oct-nc\n",
      "\twriting data for sample: CH104-1-5-May-nc\n",
      "\twriting data for sample: CH508-6-1-May-nc\n",
      "\twriting data for sample: CH606-pl1-6-6-Oct-nc\n",
      "\twriting data for sample: CH107-1-2-May-nc\n",
      "\twriting data for sample: CH-BL2-pl1-7-4-Oct-nc\n",
      "\twriting data for sample: CH210-3-5-May-nc\n",
      "\twriting data for sample: CH302-pl1-3-5-Oct-nc\n",
      "\twriting data for sample: CH101-1-8-May-nc\n",
      "\twriting data for sample: CH601-pl1-6-1-Oct-nc\n",
      "\twriting data for sample: CH202-2-5-May-nc\n",
      "\twriting data for sample: pcr-POS-pl1-7-6-Oct-nc\n",
      "\twriting data for sample: CH504-pl1-5-2-Oct-nc\n",
      "\twriting data for sample: CH504-6-5-May-nc\n",
      "\twriting data for sample: CH102-pl1-1-2-Oct-nc\n",
      "\twriting data for sample: CH109-pl1-2-1-Oct-nc\n",
      "\twriting data for sample: CH303-pl1-3-6-Oct-nc\n",
      "\twriting data for sample: CH309-pl1-4-4-Oct-nc\n",
      "\twriting data for sample: CH209-3-6-May-nc\n",
      "\twriting data for sample: CH119-pl1-3-3-Oct-nc\n",
      "\twriting data for sample: CH401-pl1-4-6-Oct-nc\n",
      "\twriting data for sample: CH118-pl1-3-2-Oct-nc\n",
      "\twriting data for sample: CH604-pl1-6-4-Oct-nc\n",
      "\twriting data for sample: CH506-pl1-5-4-Oct-nc\n",
      "\twriting data for sample: CH204-2-3-May-nc\n",
      "\twriting data for sample: CH507-pl1-5-5-Oct-nc\n",
      "\twriting data for sample: CH207-3-8-May-nc\n",
      "\twriting data for sample: CH108-1-1-May-nc\n",
      "\twriting data for sample: pcr-NEG-pl1-7-7-Oct-nc\n",
      "\twriting data for sample: CH306-pl1-4-1-Oct-nc\n",
      "\twriting data for sample: CH510-pl1-5-8-Oct-nc\n",
      "\twriting data for sample: CH507-6-2-May-nc\n",
      "\twriting data for sample: CH104-pl1-1-4-Oct-nc\n",
      "\twriting data for sample: CH208-3-7-May-nc\n",
      "\twriting data for sample: CH310-3-1-May-nc\n",
      "\twriting data for sample: CH108-pl1-1-8-Oct-nc\n",
      "\twriting data for sample: CH610-pl1-7-2-Oct-nc\n",
      "\twriting data for sample: CH505-pl1-5-3-Oct-nc\n",
      "\twriting data for sample: CH117-pl1-3-1-Oct-nc\n",
      "\twriting data for sample: CH115-pl1-2-7-Oct-nc\n",
      "\twriting data for sample: CH508-pl1-5-6-Oct-nc\n",
      "\twriting data for sample: CH116-pl1-2-8-Oct-nc\n",
      "\twriting data for sample: CH501-pl1-4-7-Oct-nc\n",
      "\twriting data for sample: CH106-pl1-1-6-Oct-nc\n",
      "\twriting data for sample: CH307-pl1-4-2-Oct-nc\n",
      "\twriting data for sample: CH309-3-2-May-nc\n",
      "\twriting data for sample: CH305-pl1-3-8-Oct-nc\n",
      "\twriting data for sample: CH107-pl1-1-7-Oct-nc\n",
      "\twriting data for sample: CH509-7-8-May-nc\n",
      "\twriting data for sample: CH301-pl1-3-4-Oct-nc\n",
      "\twriting data for sample: CH110-pl1-2-2-Oct-nc\n",
      "\twriting data for sample: CH109-2-8-May-nc\n",
      "\twriting data for sample: CH105-pl1-1-5-Oct-nc\n",
      "\twriting data for sample: CH608-pl1-6-8-Oct-nc\n",
      "\twriting data for sample: CH503-pl1-5-1-Oct-nc\n",
      "\twriting data for sample: CH-BL1-pl1-7-3-Oct-nc\n",
      "\twriting data for sample: CH509-pl1-5-7-Oct-nc\n",
      "\twriting data for sample: neg-7-5-May-nc\n",
      "\twriting data for sample: CH605-pl1-6-5-Oct-nc\n",
      "\twriting data for sample: pcr-NEG-pl1-7-5-Oct-nc\n",
      "Crangonyx_pseudogracilis_0_OTU\n",
      "\twriting data for sample: CH113-pl1-2-5-Oct-nc\n",
      "\twriting data for sample: CH206-2-1-May-nc\n",
      "\twriting data for sample: CH304-pl1-3-7-Oct-nc\n",
      "\twriting data for sample: CH112-pl1-2-4-Oct-nc\n",
      "\twriting data for sample: CH602-pl1-6-2-Oct-nc\n",
      "\twriting data for sample: CH309-3-2-May-nc\n",
      "\twriting data for sample: CH204-2-3-May-nc\n",
      "\twriting data for sample: CH502-pl1-4-8-Oct-nc\n",
      "\twriting data for sample: CH308-pl1-4-3-Oct-nc\n",
      "\twriting data for sample: CH506-6-3-May-nc\n",
      "\twriting data for sample: CH205-2-2-May-nc\n",
      "\twriting data for sample: CH114-pl1-2-6-Oct-nc\n",
      "\twriting data for sample: CH310-pl1-4-5-Oct-nc\n",
      "\twriting data for sample: CH406-5-4-May-nc\n",
      "\twriting data for sample: CH106-1-3-May-nc\n",
      "\twriting data for sample: CH310-3-1-May-nc\n",
      "\twriting data for sample: CH607-pl1-6-7-Oct-nc\n",
      "\twriting data for sample: CH111-pl1-2-3-Oct-nc\n",
      "\twriting data for sample: CH309-pl1-4-4-Oct-nc\n",
      "\twriting data for sample: CH605-pl1-6-5-Oct-nc\n",
      "\twriting data for sample: CH302-pl1-3-5-Oct-nc\n",
      "\twriting data for sample: CH508-pl1-5-6-Oct-nc\n",
      "\twriting data for sample: CH601-pl1-6-1-Oct-nc\n",
      "\twriting data for sample: CH501-6-7-May-nc\n",
      "\twriting data for sample: CH507-pl1-5-5-Oct-nc\n",
      "\twriting data for sample: CH301-pl1-3-4-Oct-nc\n",
      "\twriting data for sample: CH606-pl1-6-6-Oct-nc\n",
      "\twriting data for sample: CH109-pl1-2-1-Oct-nc\n",
      "\twriting data for sample: CH313-4-6-May-nc\n",
      "\twriting data for sample: CH409-5-1-May-nc\n",
      "\twriting data for sample: CH105-1-4-May-nc\n",
      "\twriting data for sample: CH209-3-6-May-nc\n",
      "\twriting data for sample: CH404-5-6-May-nc\n",
      "\twriting data for sample: CH119-pl1-3-3-Oct-nc\n",
      "\twriting data for sample: CH403-5-7-May-nc\n",
      "\twriting data for sample: CH401-pl1-4-6-Oct-nc\n",
      "\twriting data for sample: CH118-pl1-3-2-Oct-nc\n",
      "\twriting data for sample: CH405-5-5-May-nc\n",
      "\twriting data for sample: CH505-6-4-May-nc\n",
      "\twriting data for sample: CH401-4-1-May-nc\n",
      "\twriting data for sample: CH604-pl1-6-4-Oct-nc\n",
      "\twriting data for sample: CH506-pl1-5-4-Oct-nc\n",
      "\twriting data for sample: CH408-5-2-May-nc\n",
      "\twriting data for sample: CH201-2-6-May-nc\n",
      "\twriting data for sample: CH316-4-3-May-nc\n",
      "\twriting data for sample: CH510-7-7-May-nc\n",
      "\twriting data for sample: CH308-3-3-May-nc\n",
      "\twriting data for sample: CH407-5-3-May-nc\n",
      "\twriting data for sample: CH311-4-8-May-nc\n",
      "\twriting data for sample: CH306-pl1-4-1-Oct-nc\n",
      "\twriting data for sample: CH501-pl1-4-7-Oct-nc\n",
      "\twriting data for sample: CH317-4-2-May-nc\n",
      "\twriting data for sample: CH312-4-7-May-nc\n",
      "\twriting data for sample: CH103-1-6-May-nc\n",
      "\twriting data for sample: CH503-pl1-5-1-Oct-nc\n",
      "\twriting data for sample: CH314-4-5-May-nc\n",
      "\twriting data for sample: CH305-pl1-3-8-Oct-nc\n",
      "\twriting data for sample: CH117-pl1-3-1-Oct-nc\n",
      "\twriting data for sample: CH115-pl1-2-7-Oct-nc\n",
      "\twriting data for sample: CH116-pl1-2-8-Oct-nc\n",
      "\twriting data for sample: CH510-pl1-5-8-Oct-nc\n",
      "\twriting data for sample: CH402-5-8-May-nc\n",
      "\twriting data for sample: CH307-pl1-4-2-Oct-nc\n",
      "\twriting data for sample: CH503-6-6-May-nc\n",
      "\twriting data for sample: CH603-pl1-6-3-Oct-nc\n",
      "\twriting data for sample: CH602-7-6-May-nc\n",
      "\twriting data for sample: CH315-4-4-May-nc\n",
      "\twriting data for sample: CH303-pl1-3-6-Oct-nc\n",
      "\twriting data for sample: CH608-pl1-6-8-Oct-nc\n",
      "\twriting data for sample: CH110-pl1-2-2-Oct-nc\n",
      "\twriting data for sample: CH203-2-4-May-nc\n",
      "\twriting data for sample: CH505-pl1-5-3-Oct-nc\n",
      "\twriting data for sample: CH509-pl1-5-7-Oct-nc\n",
      "\twriting data for sample: neg-7-5-May-nc\n",
      "\twriting data for sample: CH504-pl1-5-2-Oct-nc\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "for i in range(len(OTUs_as_list)):\n",
    "    otu=OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    print syn\n",
    "    outseqs = {}\n",
    "    \n",
    "    seqs = SeqIO.parse(open(syn+'.clipped.fasta','r'),'fasta')\n",
    "    for s in seqs:\n",
    "        sample = s.id.split(\"|\")[0]\n",
    "        \n",
    "        if not sample in outseqs:\n",
    "            outseqs[sample] = []\n",
    "        outseqs[sample].append(s)\n",
    "        \n",
    "    for sample in outseqs:\n",
    "        print \"\\twriting data for sample: %s\" %sample\n",
    "        out=open(sample+'.'+syn+'.clipped.fasta','w')\n",
    "        SeqIO.write(outseqs[sample],out,'fasta')\n",
    "        out.close()\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Identify most abundant OTU per sample."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex\n",
      "#CH206-2-1-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH206-2-1-May-nc|1_1114_27579_13621_1_ex']  -> CH206-2-1-May-nc|1_1114_27579_13621_1_ex  - OK\n",
      "#CH105-pl1-1-5-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH105-pl1-1-5-Oct-nc|1_1111_17590_9870_1_ex']  -> CH105-pl1-1-5-Oct-nc|1_1111_17590_9870_1_ex  - OK\n",
      "#CH504-pl1-5-2-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH504-pl1-5-2-Oct-nc|1_1103_6377_16927_1_ex']  -> CH504-pl1-5-2-Oct-nc|1_1103_6377_16927_1_ex  - OK\n",
      "#pcr-NEG-pl1-7-7-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['pcr-NEG-pl1-7-7-Oct-nc|1_1101_9829_3153_1_ex']  -> pcr-NEG-pl1-7-7-Oct-nc|1_1101_9829_3153_1_ex  - OK\n",
      "#CH208-3-7-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH208-3-7-May-nc|1_2107_13823_23739_1_ex', 'CH208-3-7-May-nc|1_1102_25191_15965_1_ex', 'CH208-3-7-May-nc|1_1107_24978_13102_1_ex', 'CH208-3-7-May-nc|1_2111_10616_10934_1_ex', 'CH208-3-7-May-nc|1_2113_13133_17253_1_ex']  -> CH208-3-7-May-nc|1_2113_13133_17253_1_ex  - OK\n",
      "#CH504-6-5-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH504-6-5-May-nc|1_2101_8721_18397_1_ex']  -> CH504-6-5-May-nc|1_2101_8721_18397_1_ex  - OK\n",
      "#CH108-1-1-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH108-1-1-May-nc|1_1105_22538_22905_1_ex']  -> CH108-1-1-May-nc|1_1105_22538_22905_1_ex  - OK\n",
      "#CH209-3-6-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH209-3-6-May-nc|1_1101_11049_12097_1_ex']  -> CH209-3-6-May-nc|1_1101_11049_12097_1_ex  - OK\n",
      "#CH109-2-8-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH109-2-8-May-nc|1_1104_4149_17971_1_ex', 'CH109-2-8-May-nc|1_1111_17882_15316_1_ex', 'CH109-2-8-May-nc|1_2103_19217_16098_1_ex', 'CH109-2-8-May-nc|1_2107_12255_24001_1_ex', 'CH109-2-8-May-nc|1_1105_9084_11993_1_ex']  -> CH109-2-8-May-nc|1_1105_9084_11993_1_ex  - OK\n",
      "#CH509-7-8-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH509-7-8-May-nc|1_2101_5882_5905_1_ex', 'CH509-7-8-May-nc|1_1105_24118_22790_1_ex', 'CH509-7-8-May-nc|1_1114_5815_14050_1_ex']  -> CH509-7-8-May-nc|1_1114_5815_14050_1_ex  - OK\n",
      "#CH210-3-5-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH210-3-5-May-nc|1_2102_7803_8624_1_ex']  -> CH210-3-5-May-nc|1_2102_7803_8624_1_ex  - OK\n",
      "#CH307-pl1-4-2-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH307-pl1-4-2-Oct-nc|1_2111_26938_16630_1_ex']  -> CH307-pl1-4-2-Oct-nc|1_2111_26938_16630_1_ex  - OK\n",
      "#CH119-pl1-3-3-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH119-pl1-3-3-Oct-nc|1_1114_17361_12540_1_ex', 'CH119-pl1-3-3-Oct-nc|1_1114_11909_28155_1_ex']  -> CH119-pl1-3-3-Oct-nc|1_1114_17361_12540_1_ex  - OK\n",
      "#CH115-pl1-2-7-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH115-pl1-2-7-Oct-nc|1_2110_24793_6820_1_ex']  -> CH115-pl1-2-7-Oct-nc|1_2110_24793_6820_1_ex  - OK\n",
      "#CH-BL2-pl1-7-4-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH-BL2-pl1-7-4-Oct-nc|1_1102_10818_23263_1_ex']  -> CH-BL2-pl1-7-4-Oct-nc|1_1102_10818_23263_1_ex  - OK\n",
      "#CH310-3-1-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH310-3-1-May-nc|1_2107_9200_6395_1_ex']  -> CH310-3-1-May-nc|1_2107_9200_6395_1_ex  - OK\n",
      "#CH116-pl1-2-8-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH116-pl1-2-8-Oct-nc|1_1101_15577_17803_1_ex']  -> CH116-pl1-2-8-Oct-nc|1_1101_15577_17803_1_ex  - OK\n",
      "#neg-7-5-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['neg-7-5-May-nc|1_1105_18185_18390_1_ex']  -> neg-7-5-May-nc|1_1105_18185_18390_1_ex  - OK\n",
      "#pcr-NEG-pl1-7-5-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['pcr-NEG-pl1-7-5-Oct-nc|1_2110_26480_22422_1_ex']  -> pcr-NEG-pl1-7-5-Oct-nc|1_2110_26480_22422_1_ex  - OK\n",
      "#CH117-pl1-3-1-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH117-pl1-3-1-Oct-nc|1_1112_21052_10874_1_ex', 'CH117-pl1-3-1-Oct-nc|1_1102_22697_21915_1_ex', 'CH117-pl1-3-1-Oct-nc|1_1102_4395_10505_1_ex']  -> CH117-pl1-3-1-Oct-nc|1_1112_21052_10874_1_ex  - OK\n",
      "#CH107-pl1-1-7-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH107-pl1-1-7-Oct-nc|1_1103_19227_9435_1_ex']  -> CH107-pl1-1-7-Oct-nc|1_1103_19227_9435_1_ex  - OK\n",
      "#CH507-pl1-5-5-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH507-pl1-5-5-Oct-nc|1_1105_7417_23503_1_ex']  -> CH507-pl1-5-5-Oct-nc|1_1105_7417_23503_1_ex  - OK\n",
      "#CH309-pl1-4-4-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH309-pl1-4-4-Oct-nc|1_2111_19020_11638_1_ex']  -> CH309-pl1-4-4-Oct-nc|1_2111_19020_11638_1_ex  - OK\n",
      "#CH304-pl1-3-7-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH304-pl1-3-7-Oct-nc|1_1105_25326_8473_1_ex']  -> CH304-pl1-3-7-Oct-nc|1_1105_25326_8473_1_ex  - OK\n",
      "#CH310-pl1-4-5-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH310-pl1-4-5-Oct-nc|1_1103_25241_20637_1_ex']  -> CH310-pl1-4-5-Oct-nc|1_1103_25241_20637_1_ex  - OK\n",
      "#CH113-pl1-2-5-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH113-pl1-2-5-Oct-nc|1_2101_27873_17936_1_ex']  -> CH113-pl1-2-5-Oct-nc|1_2101_27873_17936_1_ex  - OK\n",
      "#CH118-pl1-3-2-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH118-pl1-3-2-Oct-nc|1_2106_15982_25320_1_ex', 'CH118-pl1-3-2-Oct-nc|1_2108_16428_5700_1_ex', 'CH118-pl1-3-2-Oct-nc|1_2110_24095_23599_1_ex']  -> CH118-pl1-3-2-Oct-nc|1_2110_24095_23599_1_ex  - OK\n",
      "#CH602-pl1-6-2-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH602-pl1-6-2-Oct-nc|1_2112_17474_4212_1_ex']  -> CH602-pl1-6-2-Oct-nc|1_2112_17474_4212_1_ex  - OK\n",
      "#CH108-pl1-1-8-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH108-pl1-1-8-Oct-nc|1_1103_20873_20660_1_ex']  -> CH108-pl1-1-8-Oct-nc|1_1103_20873_20660_1_ex  - OK\n",
      "#CH102-pl1-1-2-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH102-pl1-1-2-Oct-nc|1_2112_22478_8174_1_ex']  -> CH102-pl1-1-2-Oct-nc|1_2112_22478_8174_1_ex  - OK\n",
      "#CH111-pl1-2-3-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH111-pl1-2-3-Oct-nc|1_1104_20289_25468_1_ex', 'CH111-pl1-2-3-Oct-nc|1_1108_10303_8437_1_ex']  -> CH111-pl1-2-3-Oct-nc|1_1108_10303_8437_1_ex  - OK\n",
      "#CH608-pl1-6-8-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH608-pl1-6-8-Oct-nc|1_2102_22468_24274_1_ex']  -> CH608-pl1-6-8-Oct-nc|1_2102_22468_24274_1_ex  - OK\n",
      "#CH112-pl1-2-4-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH112-pl1-2-4-Oct-nc|1_2109_10322_12814_1_ex']  -> CH112-pl1-2-4-Oct-nc|1_2109_10322_12814_1_ex  - OK\n",
      "#CH305-pl1-3-8-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH305-pl1-3-8-Oct-nc|1_1103_15287_26025_1_ex']  -> CH305-pl1-3-8-Oct-nc|1_1103_15287_26025_1_ex  - OK\n",
      "#CH106-pl1-1-6-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH106-pl1-1-6-Oct-nc|1_2110_15659_23451_1_ex']  -> CH106-pl1-1-6-Oct-nc|1_2110_15659_23451_1_ex  - OK\n",
      "#CH101-pl1-1-1-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH101-pl1-1-1-Oct-nc|1_2105_17136_11806_1_ex', 'CH101-pl1-1-1-Oct-nc|1_1101_19163_12271_1_ex', 'CH101-pl1-1-1-Oct-nc|1_1105_9774_10777_1_ex']  -> CH101-pl1-1-1-Oct-nc|1_1105_9774_10777_1_ex  - OK\n",
      "#CH610-pl1-7-2-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH610-pl1-7-2-Oct-nc|1_1105_18104_6500_1_ex']  -> CH610-pl1-7-2-Oct-nc|1_1105_18104_6500_1_ex  - OK\n",
      "#CH508-pl1-5-6-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH508-pl1-5-6-Oct-nc|1_1101_18482_26952_1_ex', 'CH508-pl1-5-6-Oct-nc|1_1113_12072_10057_1_ex', 'CH508-pl1-5-6-Oct-nc|1_1107_20724_7454_1_ex']  -> CH508-pl1-5-6-Oct-nc|1_1107_20724_7454_1_ex  - OK\n",
      "#CH107-1-2-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH107-1-2-May-nc|1_2107_7483_23780_1_ex']  -> CH107-1-2-May-nc|1_2107_7483_23780_1_ex  - OK\n",
      "#CH207-3-8-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH207-3-8-May-nc|1_1108_13776_14705_1_ex']  -> CH207-3-8-May-nc|1_1108_13776_14705_1_ex  - OK\n",
      "#CH606-pl1-6-6-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH606-pl1-6-6-Oct-nc|1_2114_18585_9443_1_ex']  -> CH606-pl1-6-6-Oct-nc|1_2114_18585_9443_1_ex  - OK\n",
      "#CH401-pl1-4-6-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH401-pl1-4-6-Oct-nc|1_1113_16057_10251_1_ex', 'CH401-pl1-4-6-Oct-nc|1_1103_18623_9709_1_ex', 'CH401-pl1-4-6-Oct-nc|1_2113_25554_12926_1_ex']  -> CH401-pl1-4-6-Oct-nc|1_2113_25554_12926_1_ex  - OK\n",
      "#CH202-2-5-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH202-2-5-May-nc|1_1111_14115_28583_1_ex', 'CH202-2-5-May-nc|1_2102_7674_11967_1_ex', 'CH202-2-5-May-nc|1_2111_13976_21014_1_ex', 'CH202-2-5-May-nc|1_2101_8683_13805_1_ex']  -> CH202-2-5-May-nc|1_2101_8683_13805_1_ex  - OK\n",
      "#CH306-pl1-4-1-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH306-pl1-4-1-Oct-nc|1_2113_14905_26849_1_ex']  -> CH306-pl1-4-1-Oct-nc|1_2113_14905_26849_1_ex  - OK\n",
      "#CH110-2-7-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH110-2-7-May-nc|1_1102_14199_27658_1_ex']  -> CH110-2-7-May-nc|1_1102_14199_27658_1_ex  - OK\n",
      "#CH509-pl1-5-7-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH509-pl1-5-7-Oct-nc|1_1105_12208_14070_1_ex', 'CH509-pl1-5-7-Oct-nc|1_1103_14803_12121_1_ex']  -> CH509-pl1-5-7-Oct-nc|1_1103_14803_12121_1_ex  - OK\n",
      "#CH605-pl1-6-5-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH605-pl1-6-5-Oct-nc|1_1102_7714_23811_1_ex', 'CH605-pl1-6-5-Oct-nc|1_2103_23579_23692_1_ex']  -> CH605-pl1-6-5-Oct-nc|1_1102_7714_23811_1_ex  - OK\n",
      "#CH204-2-3-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH204-2-3-May-nc|1_1101_15249_8454_1_ex']  -> CH204-2-3-May-nc|1_1101_15249_8454_1_ex  - OK\n",
      "#CH110-pl1-2-2-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH110-pl1-2-2-Oct-nc|1_1109_19936_9823_1_ex', 'CH110-pl1-2-2-Oct-nc|1_2102_18477_7604_1_ex', 'CH110-pl1-2-2-Oct-nc|1_2106_16297_7752_1_ex', 'CH110-pl1-2-2-Oct-nc|1_1110_10713_27863_1_ex', 'CH110-pl1-2-2-Oct-nc|1_2112_17573_21662_1_ex', 'CH110-pl1-2-2-Oct-nc|1_1111_11765_16896_1_ex']  -> CH110-pl1-2-2-Oct-nc|1_1111_11765_16896_1_ex  - OK\n",
      "#CH308-pl1-4-3-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH308-pl1-4-3-Oct-nc|1_1103_14822_12234_1_ex', 'CH308-pl1-4-3-Oct-nc|1_2101_24262_20530_1_ex']  -> CH308-pl1-4-3-Oct-nc|1_2101_24262_20530_1_ex  - OK\n",
      "#CH102-1-7-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH102-1-7-May-nc|1_1102_17379_27773_1_ex']  -> CH102-1-7-May-nc|1_1102_17379_27773_1_ex  - OK\n",
      "#CH505-pl1-5-3-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH505-pl1-5-3-Oct-nc|1_1111_27423_13701_1_ex']  -> CH505-pl1-5-3-Oct-nc|1_1111_27423_13701_1_ex  - OK\n",
      "#CH103-pl1-1-3-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH103-pl1-1-3-Oct-nc|1_1102_9913_11787_1_ex']  -> CH103-pl1-1-3-Oct-nc|1_1102_9913_11787_1_ex  - OK\n",
      "#CH104-pl1-1-4-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH104-pl1-1-4-Oct-nc|1_1107_12247_23331_1_ex', 'CH104-pl1-1-4-Oct-nc|1_1101_18512_16301_1_ex']  -> CH104-pl1-1-4-Oct-nc|1_1107_12247_23331_1_ex  - OK\n",
      "#CH607-pl1-6-7-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH607-pl1-6-7-Oct-nc|1_1110_22079_4171_1_ex']  -> CH607-pl1-6-7-Oct-nc|1_1110_22079_4171_1_ex  - OK\n",
      "#CH508-6-1-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH508-6-1-May-nc|1_1101_8646_6236_1_ex']  -> CH508-6-1-May-nc|1_1101_8646_6236_1_ex  - OK\n",
      "#CH502-pl1-4-8-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH502-pl1-4-8-Oct-nc|1_1101_24418_25110_1_ex']  -> CH502-pl1-4-8-Oct-nc|1_1101_24418_25110_1_ex  - OK\n",
      "#CH503-pl1-5-1-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH503-pl1-5-1-Oct-nc|1_2111_21188_16601_1_ex']  -> CH503-pl1-5-1-Oct-nc|1_2111_21188_16601_1_ex  - OK\n",
      "#CH301-pl1-3-4-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH301-pl1-3-4-Oct-nc|1_1111_12453_27617_1_ex']  -> CH301-pl1-3-4-Oct-nc|1_1111_12453_27617_1_ex  - OK\n",
      "#CH604-pl1-6-4-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH604-pl1-6-4-Oct-nc|1_1104_11193_6308_1_ex', 'CH604-pl1-6-4-Oct-nc|1_1101_9513_5482_1_ex', 'CH604-pl1-6-4-Oct-nc|1_2103_19820_7570_1_ex']  -> CH604-pl1-6-4-Oct-nc|1_2103_19820_7570_1_ex  - OK\n",
      "#pcr-POS-pl1-7-6-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['pcr-POS-pl1-7-6-Oct-nc|1_1108_6151_24160_1_ex']  -> pcr-POS-pl1-7-6-Oct-nc|1_1108_6151_24160_1_ex  - OK\n",
      "#CH601-pl1-6-1-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH601-pl1-6-1-Oct-nc|1_1102_12629_8189_1_ex', 'CH601-pl1-6-1-Oct-nc|1_2103_18237_7942_1_ex']  -> CH601-pl1-6-1-Oct-nc|1_2103_18237_7942_1_ex  - OK\n",
      "#CH510-pl1-5-8-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH510-pl1-5-8-Oct-nc|1_1111_8731_20032_1_ex', 'CH510-pl1-5-8-Oct-nc|1_1101_17841_4428_1_ex', 'CH510-pl1-5-8-Oct-nc|1_2104_27519_9634_1_ex', 'CH510-pl1-5-8-Oct-nc|1_1114_12008_15110_1_ex']  -> CH510-pl1-5-8-Oct-nc|1_1114_12008_15110_1_ex  - OK\n",
      "#CH114-pl1-2-6-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH114-pl1-2-6-Oct-nc|1_1105_15737_20636_1_ex']  -> CH114-pl1-2-6-Oct-nc|1_1105_15737_20636_1_ex  - OK\n",
      "#CH309-3-2-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH309-3-2-May-nc|1_1102_27653_12510_1_ex']  -> CH309-3-2-May-nc|1_1102_27653_12510_1_ex  - OK\n",
      "#CH-BL1-pl1-7-3-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH-BL1-pl1-7-3-Oct-nc|1_2111_14114_15025_1_ex']  -> CH-BL1-pl1-7-3-Oct-nc|1_2111_14114_15025_1_ex  - OK\n",
      "#CH507-6-2-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH507-6-2-May-nc|1_1103_16590_12766_1_ex']  -> CH507-6-2-May-nc|1_1103_16590_12766_1_ex  - OK\n",
      "#CH303-pl1-3-6-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH303-pl1-3-6-Oct-nc|1_2101_2773_19044_1_ex', 'CH303-pl1-3-6-Oct-nc|1_2101_15444_21743_1_ex', 'CH303-pl1-3-6-Oct-nc|1_1101_27561_20028_1_ex', 'CH303-pl1-3-6-Oct-nc|1_1101_21365_26861_1_ex', 'CH303-pl1-3-6-Oct-nc|1_1102_23591_13453_1_ex', 'CH303-pl1-3-6-Oct-nc|1_1109_21051_24535_1_ex', 'CH303-pl1-3-6-Oct-nc|1_1111_22301_13305_1_ex', 'CH303-pl1-3-6-Oct-nc|1_1104_21351_17798_1_ex', 'CH303-pl1-3-6-Oct-nc|1_1108_3489_19275_1_ex']  -> CH303-pl1-3-6-Oct-nc|1_1108_3489_19275_1_ex  - OK\n",
      "#CH101-1-8-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH101-1-8-May-nc|1_2114_22385_22080_1_ex']  -> CH101-1-8-May-nc|1_2114_22385_22080_1_ex  - OK\n",
      "#CH109-pl1-2-1-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH109-pl1-2-1-Oct-nc|1_1109_9384_21255_1_ex', 'CH109-pl1-2-1-Oct-nc|1_1103_21976_14388_1_ex']  -> CH109-pl1-2-1-Oct-nc|1_1103_21976_14388_1_ex  - OK\n",
      "#CH609-pl1-7-1-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH609-pl1-7-1-Oct-nc|1_1112_11242_24062_1_ex', 'CH609-pl1-7-1-Oct-nc|1_1102_6617_14688_1_ex']  -> CH609-pl1-7-1-Oct-nc|1_1102_6617_14688_1_ex  - OK\n",
      "#CH104-1-5-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH104-1-5-May-nc|1_2105_16181_17403_1_ex', 'CH104-1-5-May-nc|1_1103_23331_12069_1_ex', 'CH104-1-5-May-nc|1_1104_4936_17123_1_ex']  -> CH104-1-5-May-nc|1_1104_4936_17123_1_ex  - OK\n",
      "#CH501-pl1-4-7-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH501-pl1-4-7-Oct-nc|1_2110_14536_5793_1_ex']  -> CH501-pl1-4-7-Oct-nc|1_2110_14536_5793_1_ex  - OK\n",
      "#CH603-pl1-6-3-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH603-pl1-6-3-Oct-nc|1_2111_22687_15876_1_ex', 'CH603-pl1-6-3-Oct-nc|1_1102_3745_17711_1_ex', 'CH603-pl1-6-3-Oct-nc|1_2103_11992_17643_1_ex', 'CH603-pl1-6-3-Oct-nc|1_2111_14816_26851_1_ex']  -> CH603-pl1-6-3-Oct-nc|1_2111_14816_26851_1_ex  - OK\n",
      "#CH203-2-4-May-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH203-2-4-May-nc|1_1107_13081_11271_1_ex']  -> CH203-2-4-May-nc|1_1107_13081_11271_1_ex  - OK\n",
      "#CH302-pl1-3-5-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH302-pl1-3-5-Oct-nc|1_2112_23354_19511_1_ex', 'CH302-pl1-3-5-Oct-nc|1_1102_20186_15022_1_ex', 'CH302-pl1-3-5-Oct-nc|1_1108_3424_17463_1_ex', 'CH302-pl1-3-5-Oct-nc|1_1109_12193_6319_1_ex']  -> CH302-pl1-3-5-Oct-nc|1_1109_12193_6319_1_ex  - OK\n",
      "#CH506-pl1-5-4-Oct-nc.Crangonyx_floridanus_0_OTU.clipped  - ['CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex']  -> CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex  - OK\n",
      "extracting hts for Crangonyx_floridanus_0_OTU -> Crangonyx_floridanus_0_OTU_hts.fasta\n",
      "Crangonyx_pseudogracilis_0_OTU CH105-1-4-May-nc|1_2105_9715_15085_1_ex\n",
      "#CH501-pl1-4-7-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH501-pl1-4-7-Oct-nc|1_1114_21367_11968_1_ex']  -> CH501-pl1-4-7-Oct-nc|1_1114_21367_11968_1_ex  - OK\n",
      "#CH317-4-2-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH317-4-2-May-nc|1_1113_11527_10580_1_ex']  -> CH317-4-2-May-nc|1_1113_11527_10580_1_ex  - OK\n",
      "#CH402-5-8-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH402-5-8-May-nc|1_2114_5820_10860_1_ex']  -> CH402-5-8-May-nc|1_2114_5820_10860_1_ex  - OK\n",
      "#CH602-7-6-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH602-7-6-May-nc|1_1103_11222_3470_1_ex']  -> CH602-7-6-May-nc|1_1103_11222_3470_1_ex  - OK\n",
      "#CH605-pl1-6-5-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH605-pl1-6-5-Oct-nc|1_1110_16237_24188_1_ex', 'CH605-pl1-6-5-Oct-nc|1_1102_13824_27047_1_ex']  -> CH605-pl1-6-5-Oct-nc|1_1102_13824_27047_1_ex  - OK\n",
      "#CH409-5-1-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH409-5-1-May-nc|1_2113_12901_9622_1_ex']  -> CH409-5-1-May-nc|1_2113_12901_9622_1_ex  - OK\n",
      "#CH508-pl1-5-6-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH508-pl1-5-6-Oct-nc|1_2105_9297_23643_1_ex']  -> CH508-pl1-5-6-Oct-nc|1_2105_9297_23643_1_ex  - OK\n",
      "#CH314-4-5-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH314-4-5-May-nc|1_2113_14373_16842_1_ex']  -> CH314-4-5-May-nc|1_2113_14373_16842_1_ex  - OK\n",
      "#CH310-3-1-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH310-3-1-May-nc|1_1101_18604_17140_1_ex']  -> CH310-3-1-May-nc|1_1101_18604_17140_1_ex  - OK\n",
      "#CH507-pl1-5-5-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH507-pl1-5-5-Oct-nc|1_1110_17381_9508_1_ex', 'CH507-pl1-5-5-Oct-nc|1_1114_18451_14159_1_ex']  -> CH507-pl1-5-5-Oct-nc|1_1114_18451_14159_1_ex  - OK\n",
      "#CH116-pl1-2-8-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH116-pl1-2-8-Oct-nc|1_1103_14526_14749_1_ex']  -> CH116-pl1-2-8-Oct-nc|1_1103_14526_14749_1_ex  - OK\n",
      "#CH506-6-3-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH506-6-3-May-nc|1_2105_16158_22211_1_ex']  -> CH506-6-3-May-nc|1_2105_16158_22211_1_ex  - OK\n",
      "#CH105-1-4-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH105-1-4-May-nc|1_2105_9715_15085_1_ex']  -> CH105-1-4-May-nc|1_2105_9715_15085_1_ex  - OK\n",
      "#CH119-pl1-3-3-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH119-pl1-3-3-Oct-nc|1_1112_7975_13869_1_ex']  -> CH119-pl1-3-3-Oct-nc|1_1112_7975_13869_1_ex  - OK\n",
      "#CH313-4-6-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH313-4-6-May-nc|1_1101_17223_3813_1_ex']  -> CH313-4-6-May-nc|1_1101_17223_3813_1_ex  - OK\n",
      "#CH308-3-3-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH308-3-3-May-nc|1_1110_22091_26391_1_ex']  -> CH308-3-3-May-nc|1_1110_22091_26391_1_ex  - OK\n",
      "#CH506-pl1-5-4-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH506-pl1-5-4-Oct-nc|1_2108_19105_19739_1_ex']  -> CH506-pl1-5-4-Oct-nc|1_2108_19105_19739_1_ex  - OK\n",
      "#CH114-pl1-2-6-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH114-pl1-2-6-Oct-nc|1_1102_10014_10337_1_ex']  -> CH114-pl1-2-6-Oct-nc|1_1102_10014_10337_1_ex  - OK\n",
      "#CH502-pl1-4-8-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH502-pl1-4-8-Oct-nc|1_2111_10293_27692_1_ex', 'CH502-pl1-4-8-Oct-nc|1_1102_28237_13381_1_ex', 'CH502-pl1-4-8-Oct-nc|1_1107_21070_14741_1_ex', 'CH502-pl1-4-8-Oct-nc|1_2101_26438_18573_1_ex', 'CH502-pl1-4-8-Oct-nc|1_2102_19390_13080_1_ex', 'CH502-pl1-4-8-Oct-nc|1_1104_8157_16617_1_ex', 'CH502-pl1-4-8-Oct-nc|1_2103_11685_26793_1_ex', 'CH502-pl1-4-8-Oct-nc|1_2108_17273_27632_1_ex', 'CH502-pl1-4-8-Oct-nc|1_2109_21966_16778_1_ex']  -> CH502-pl1-4-8-Oct-nc|1_2109_21966_16778_1_ex  - OK\n",
      "#CH201-2-6-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH201-2-6-May-nc|1_2105_16108_25950_1_ex']  -> CH201-2-6-May-nc|1_2105_16108_25950_1_ex  - OK\n",
      "#CH310-pl1-4-5-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH310-pl1-4-5-Oct-nc|1_1102_9343_19183_1_ex', 'CH310-pl1-4-5-Oct-nc|1_1105_12388_26376_1_ex', 'CH310-pl1-4-5-Oct-nc|1_1106_16155_9070_1_ex', 'CH310-pl1-4-5-Oct-nc|1_1108_21815_6755_1_ex', 'CH310-pl1-4-5-Oct-nc|1_1109_3862_19237_1_ex', 'CH310-pl1-4-5-Oct-nc|1_1110_7811_17532_1_ex', 'CH310-pl1-4-5-Oct-nc|1_1113_22501_11697_1_ex', 'CH310-pl1-4-5-Oct-nc|1_2102_17322_23679_1_ex', 'CH310-pl1-4-5-Oct-nc|1_2105_11367_13744_1_ex', 'CH310-pl1-4-5-Oct-nc|1_2110_3817_15988_1_ex', 'CH310-pl1-4-5-Oct-nc|1_2114_15974_20148_1_ex']  -> CH310-pl1-4-5-Oct-nc|1_2114_15974_20148_1_ex  - OK\n",
      "#CH403-5-7-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH403-5-7-May-nc|1_2110_22767_17537_1_ex']  -> CH403-5-7-May-nc|1_2110_22767_17537_1_ex  - OK\n",
      "#CH109-pl1-2-1-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH109-pl1-2-1-Oct-nc|1_1111_17771_13255_1_ex']  -> CH109-pl1-2-1-Oct-nc|1_1111_17771_13255_1_ex  - OK\n",
      "#CH510-7-7-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH510-7-7-May-nc|1_1104_8944_8704_1_ex']  -> CH510-7-7-May-nc|1_1104_8944_8704_1_ex  - OK\n",
      "#CH408-5-2-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH408-5-2-May-nc|1_1104_18113_23871_1_ex']  -> CH408-5-2-May-nc|1_1104_18113_23871_1_ex  - OK\n",
      "#CH406-5-4-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH406-5-4-May-nc|1_2102_19675_24297_1_ex', 'CH406-5-4-May-nc|1_1101_6058_21446_1_ex']  -> CH406-5-4-May-nc|1_1101_6058_21446_1_ex  - OK\n",
      "#CH115-pl1-2-7-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH115-pl1-2-7-Oct-nc|1_1101_15281_24572_1_ex']  -> CH115-pl1-2-7-Oct-nc|1_1101_15281_24572_1_ex  - OK\n",
      "#CH312-4-7-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH312-4-7-May-nc|1_2114_10808_12998_1_ex', 'CH312-4-7-May-nc|1_1101_26865_8994_1_ex', 'CH312-4-7-May-nc|1_2111_20310_26284_1_ex', 'CH312-4-7-May-nc|1_1101_24633_10184_1_ex', 'CH312-4-7-May-nc|1_1111_15185_17021_1_ex', 'CH312-4-7-May-nc|1_2111_11547_5110_1_ex', 'CH312-4-7-May-nc|1_2112_4207_14922_1_ex', 'CH312-4-7-May-nc|1_1111_25351_20883_1_ex']  -> CH312-4-7-May-nc|1_1111_25351_20883_1_ex  - OK\n",
      "#CH405-5-5-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH405-5-5-May-nc|1_2114_7356_13964_1_ex']  -> CH405-5-5-May-nc|1_2114_7356_13964_1_ex  - OK\n",
      "#CH509-pl1-5-7-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH509-pl1-5-7-Oct-nc|1_1107_20609_7175_1_ex']  -> CH509-pl1-5-7-Oct-nc|1_1107_20609_7175_1_ex  - OK\n",
      "#CH103-1-6-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH103-1-6-May-nc|1_1108_12105_6796_1_ex', 'CH103-1-6-May-nc|1_1104_6632_7010_1_ex', 'CH103-1-6-May-nc|1_1102_10220_11006_1_ex', 'CH103-1-6-May-nc|1_1102_8729_18418_1_ex', 'CH103-1-6-May-nc|1_1103_17548_18310_1_ex', 'CH103-1-6-May-nc|1_1104_22429_18099_1_ex', 'CH103-1-6-May-nc|1_2102_8242_26920_1_ex', 'CH103-1-6-May-nc|1_2105_19984_26450_1_ex', 'CH103-1-6-May-nc|1_2105_6704_17543_1_ex', 'CH103-1-6-May-nc|1_2107_24186_20654_1_ex', 'CH103-1-6-May-nc|1_2111_2331_13823_1_ex', 'CH103-1-6-May-nc|1_1102_7800_20792_1_ex', 'CH103-1-6-May-nc|1_1103_11450_9537_1_ex', 'CH103-1-6-May-nc|1_1103_12563_12372_1_ex', 'CH103-1-6-May-nc|1_1103_28869_13963_1_ex', 'CH103-1-6-May-nc|1_1105_21661_12374_1_ex', 'CH103-1-6-May-nc|1_1106_11101_13741_1_ex', 'CH103-1-6-May-nc|1_1106_4461_12263_1_ex', 'CH103-1-6-May-nc|1_1108_15756_20165_1_ex', 'CH103-1-6-May-nc|1_1109_16945_20936_1_ex', 'CH103-1-6-May-nc|1_1110_15360_27795_1_ex', 'CH103-1-6-May-nc|1_1111_8704_12191_1_ex', 'CH103-1-6-May-nc|1_1112_10564_20603_1_ex', 'CH103-1-6-May-nc|1_1112_9067_12181_1_ex', 'CH103-1-6-May-nc|1_1113_18798_14438_1_ex', 'CH103-1-6-May-nc|1_1113_6543_25167_1_ex', 'CH103-1-6-May-nc|1_1114_17423_22807_1_ex', 'CH103-1-6-May-nc|1_1114_5811_12712_1_ex', 'CH103-1-6-May-nc|1_2101_13349_10607_1_ex', 'CH103-1-6-May-nc|1_2102_12847_12016_1_ex', 'CH103-1-6-May-nc|1_2103_23670_15957_1_ex', 'CH103-1-6-May-nc|1_2103_7276_18079_1_ex', 'CH103-1-6-May-nc|1_2103_8282_17074_1_ex', 'CH103-1-6-May-nc|1_2104_12159_15457_1_ex', 'CH103-1-6-May-nc|1_2104_13350_13191_1_ex', 'CH103-1-6-May-nc|1_2104_16146_20056_1_ex', 'CH103-1-6-May-nc|1_2105_8872_15343_1_ex', 'CH103-1-6-May-nc|1_2106_14015_28538_1_ex', 'CH103-1-6-May-nc|1_2106_29369_17802_1_ex', 'CH103-1-6-May-nc|1_2107_28718_16879_1_ex', 'CH103-1-6-May-nc|1_2107_7225_12330_1_ex', 'CH103-1-6-May-nc|1_2108_17060_9755_1_ex', 'CH103-1-6-May-nc|1_2110_17692_21574_1_ex', 'CH103-1-6-May-nc|1_2111_24870_18232_1_ex', 'CH103-1-6-May-nc|1_2112_10485_24664_1_ex', 'CH103-1-6-May-nc|1_2114_13502_14830_1_ex', 'CH103-1-6-May-nc|1_2114_8381_3963_1_ex', 'CH103-1-6-May-nc|1_1102_27565_22423_1_ex', 'CH103-1-6-May-nc|1_1104_29122_17093_1_ex', 'CH103-1-6-May-nc|1_1106_22165_8519_1_ex', 'CH103-1-6-May-nc|1_1107_8534_10537_1_ex', 'CH103-1-6-May-nc|1_1113_16637_21489_1_ex', 'CH103-1-6-May-nc|1_2105_4263_10599_1_ex', 'CH103-1-6-May-nc|1_2108_9477_20327_1_ex', 'CH103-1-6-May-nc|1_2113_18081_4955_1_ex', 'CH103-1-6-May-nc|1_2113_7436_14015_1_ex', 'CH103-1-6-May-nc|1_1105_25153_15980_1_ex', 'CH103-1-6-May-nc|1_2103_13183_23191_1_ex', 'CH103-1-6-May-nc|1_2110_8440_14312_1_ex', 'CH103-1-6-May-nc|1_2111_13493_24589_1_ex', 'CH103-1-6-May-nc|1_1101_24330_23874_1_ex', 'CH103-1-6-May-nc|1_2107_16462_6850_1_ex']  -> CH103-1-6-May-nc|1_2107_16462_6850_1_ex  - OK\n",
      "#CH603-pl1-6-3-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH603-pl1-6-3-Oct-nc|1_2106_19344_17133_1_ex']  -> CH603-pl1-6-3-Oct-nc|1_2106_19344_17133_1_ex  - OK\n",
      "#CH205-2-2-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH205-2-2-May-nc|1_1102_9440_27809_1_ex']  -> CH205-2-2-May-nc|1_1102_9440_27809_1_ex  - OK\n",
      "#CH309-pl1-4-4-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH309-pl1-4-4-Oct-nc|1_1101_20773_4368_1_ex']  -> CH309-pl1-4-4-Oct-nc|1_1101_20773_4368_1_ex  - OK\n",
      "#CH407-5-3-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH407-5-3-May-nc|1_1114_6447_21242_1_ex', 'CH407-5-3-May-nc|1_1106_2335_17338_1_ex', 'CH407-5-3-May-nc|1_2111_17745_21913_1_ex']  -> CH407-5-3-May-nc|1_2111_17745_21913_1_ex  - OK\n",
      "#CH305-pl1-3-8-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH305-pl1-3-8-Oct-nc|1_2110_10509_22653_1_ex']  -> CH305-pl1-3-8-Oct-nc|1_2110_10509_22653_1_ex  - OK\n",
      "#CH117-pl1-3-1-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH117-pl1-3-1-Oct-nc|1_1101_7880_21159_1_ex']  -> CH117-pl1-3-1-Oct-nc|1_1101_7880_21159_1_ex  - OK\n",
      "#CH608-pl1-6-8-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH608-pl1-6-8-Oct-nc|1_1107_5331_10619_1_ex']  -> CH608-pl1-6-8-Oct-nc|1_1107_5331_10619_1_ex  - OK\n",
      "#CH302-pl1-3-5-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH302-pl1-3-5-Oct-nc|1_1103_18851_7119_1_ex']  -> CH302-pl1-3-5-Oct-nc|1_1103_18851_7119_1_ex  - OK\n",
      "#CH315-4-4-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH315-4-4-May-nc|1_1101_10662_24904_1_ex', 'CH315-4-4-May-nc|1_2104_20609_4106_1_ex', 'CH315-4-4-May-nc|1_2113_14536_22073_1_ex', 'CH315-4-4-May-nc|1_1103_22687_12430_1_ex', 'CH315-4-4-May-nc|1_1104_15681_4019_1_ex', 'CH315-4-4-May-nc|1_1104_9004_10630_1_ex', 'CH315-4-4-May-nc|1_1106_6401_16448_1_ex', 'CH315-4-4-May-nc|1_1109_11432_26032_1_ex', 'CH315-4-4-May-nc|1_1114_20947_26158_1_ex', 'CH315-4-4-May-nc|1_2105_26332_17343_1_ex', 'CH315-4-4-May-nc|1_2107_23036_6643_1_ex', 'CH315-4-4-May-nc|1_2108_7435_24691_1_ex', 'CH315-4-4-May-nc|1_2109_10549_12380_1_ex', 'CH315-4-4-May-nc|1_2112_12159_9393_1_ex']  -> CH315-4-4-May-nc|1_2112_12159_9393_1_ex  - OK\n",
      "#CH604-pl1-6-4-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH604-pl1-6-4-Oct-nc|1_1113_14747_10301_1_ex']  -> CH604-pl1-6-4-Oct-nc|1_1113_14747_10301_1_ex  - OK\n",
      "#CH505-pl1-5-3-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH505-pl1-5-3-Oct-nc|1_1101_15663_9824_1_ex']  -> CH505-pl1-5-3-Oct-nc|1_1101_15663_9824_1_ex  - OK\n",
      "#CH307-pl1-4-2-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH307-pl1-4-2-Oct-nc|1_2102_5028_14498_1_ex']  -> CH307-pl1-4-2-Oct-nc|1_2102_5028_14498_1_ex  - OK\n",
      "#CH110-pl1-2-2-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH110-pl1-2-2-Oct-nc|1_1102_9949_5557_1_ex', 'CH110-pl1-2-2-Oct-nc|1_1103_25080_14554_1_ex']  -> CH110-pl1-2-2-Oct-nc|1_1103_25080_14554_1_ex  - OK\n",
      "#CH206-2-1-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH206-2-1-May-nc|1_1102_9367_9576_1_ex']  -> CH206-2-1-May-nc|1_1102_9367_9576_1_ex  - OK\n",
      "#neg-7-5-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['neg-7-5-May-nc|1_2104_14460_24998_1_ex']  -> neg-7-5-May-nc|1_2104_14460_24998_1_ex  - OK\n",
      "#CH505-6-4-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH505-6-4-May-nc|1_2105_8375_19090_1_ex']  -> CH505-6-4-May-nc|1_2105_8375_19090_1_ex  - OK\n",
      "#CH113-pl1-2-5-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH113-pl1-2-5-Oct-nc|1_1102_10926_16218_1_ex']  -> CH113-pl1-2-5-Oct-nc|1_1102_10926_16218_1_ex  - OK\n",
      "#CH209-3-6-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH209-3-6-May-nc|1_1101_23088_22908_1_ex']  -> CH209-3-6-May-nc|1_1101_23088_22908_1_ex  - OK\n",
      "#CH401-pl1-4-6-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH401-pl1-4-6-Oct-nc|1_1110_12109_12328_1_ex']  -> CH401-pl1-4-6-Oct-nc|1_1110_12109_12328_1_ex  - OK\n",
      "#CH601-pl1-6-1-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH601-pl1-6-1-Oct-nc|1_1102_14770_27980_1_ex']  -> CH601-pl1-6-1-Oct-nc|1_1102_14770_27980_1_ex  - OK\n",
      "#CH106-1-3-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH106-1-3-May-nc|1_1110_21439_22183_1_ex', 'CH106-1-3-May-nc|1_1111_5630_16700_1_ex', 'CH106-1-3-May-nc|1_1107_8315_18054_1_ex']  -> CH106-1-3-May-nc|1_1110_21439_22183_1_ex  - OK\n",
      "#CH303-pl1-3-6-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH303-pl1-3-6-Oct-nc|1_1102_15265_21686_1_ex']  -> CH303-pl1-3-6-Oct-nc|1_1102_15265_21686_1_ex  - OK\n",
      "#CH602-pl1-6-2-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH602-pl1-6-2-Oct-nc|1_1103_14601_27439_1_ex']  -> CH602-pl1-6-2-Oct-nc|1_1103_14601_27439_1_ex  - OK\n",
      "#CH118-pl1-3-2-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH118-pl1-3-2-Oct-nc|1_1107_19428_27614_1_ex']  -> CH118-pl1-3-2-Oct-nc|1_1107_19428_27614_1_ex  - OK\n",
      "#CH501-6-7-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH501-6-7-May-nc|1_2104_16742_20287_1_ex']  -> CH501-6-7-May-nc|1_2104_16742_20287_1_ex  - OK\n",
      "#CH112-pl1-2-4-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH112-pl1-2-4-Oct-nc|1_1102_12605_4879_1_ex']  -> CH112-pl1-2-4-Oct-nc|1_1102_12605_4879_1_ex  - OK\n",
      "#CH311-4-8-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH311-4-8-May-nc|1_2105_11929_22986_1_ex', 'CH311-4-8-May-nc|1_2111_16656_25801_1_ex', 'CH311-4-8-May-nc|1_1103_25211_6822_1_ex', 'CH311-4-8-May-nc|1_1106_9801_25151_1_ex', 'CH311-4-8-May-nc|1_1111_12946_17679_1_ex', 'CH311-4-8-May-nc|1_1111_5648_19871_1_ex', 'CH311-4-8-May-nc|1_1106_15509_26959_1_ex', 'CH311-4-8-May-nc|1_1109_24932_7432_1_ex']  -> CH311-4-8-May-nc|1_1109_24932_7432_1_ex  - OK\n",
      "#CH204-2-3-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH204-2-3-May-nc|1_1113_25654_15317_1_ex']  -> CH204-2-3-May-nc|1_1113_25654_15317_1_ex  - OK\n",
      "#CH401-4-1-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH401-4-1-May-nc|1_2102_8546_20040_1_ex']  -> CH401-4-1-May-nc|1_2102_8546_20040_1_ex  - OK\n",
      "#CH504-pl1-5-2-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH504-pl1-5-2-Oct-nc|1_1109_13608_27725_1_ex']  -> CH504-pl1-5-2-Oct-nc|1_1109_13608_27725_1_ex  - OK\n",
      "#CH308-pl1-4-3-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH308-pl1-4-3-Oct-nc|1_2110_9229_10986_1_ex', 'CH308-pl1-4-3-Oct-nc|1_1101_13768_18790_1_ex']  -> CH308-pl1-4-3-Oct-nc|1_1101_13768_18790_1_ex  - OK\n",
      "#CH316-4-3-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH316-4-3-May-nc|1_1101_17804_4020_1_ex']  -> CH316-4-3-May-nc|1_1101_17804_4020_1_ex  - OK\n",
      "#CH301-pl1-3-4-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH301-pl1-3-4-Oct-nc|1_1101_21762_24316_1_ex']  -> CH301-pl1-3-4-Oct-nc|1_1101_21762_24316_1_ex  - OK\n",
      "#CH309-3-2-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH309-3-2-May-nc|1_1107_21219_21838_1_ex']  -> CH309-3-2-May-nc|1_1107_21219_21838_1_ex  - OK\n",
      "#CH607-pl1-6-7-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH607-pl1-6-7-Oct-nc|1_1101_17852_23911_1_ex']  -> CH607-pl1-6-7-Oct-nc|1_1101_17852_23911_1_ex  - OK\n",
      "#CH404-5-6-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH404-5-6-May-nc|1_1109_23362_14780_1_ex']  -> CH404-5-6-May-nc|1_1109_23362_14780_1_ex  - OK\n",
      "#CH306-pl1-4-1-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH306-pl1-4-1-Oct-nc|1_1104_14965_4318_1_ex']  -> CH306-pl1-4-1-Oct-nc|1_1104_14965_4318_1_ex  - OK\n",
      "#CH503-pl1-5-1-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH503-pl1-5-1-Oct-nc|1_2102_12578_9443_1_ex']  -> CH503-pl1-5-1-Oct-nc|1_2102_12578_9443_1_ex  - OK\n",
      "#CH304-pl1-3-7-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH304-pl1-3-7-Oct-nc|1_1106_17586_25029_1_ex']  -> CH304-pl1-3-7-Oct-nc|1_1106_17586_25029_1_ex  - OK\n",
      "#CH503-6-6-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH503-6-6-May-nc|1_2112_13408_22985_1_ex']  -> CH503-6-6-May-nc|1_2112_13408_22985_1_ex  - OK\n",
      "#CH203-2-4-May-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH203-2-4-May-nc|1_1101_9790_13250_1_ex']  -> CH203-2-4-May-nc|1_1101_9790_13250_1_ex  - OK\n",
      "#CH606-pl1-6-6-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH606-pl1-6-6-Oct-nc|1_2109_10040_14462_1_ex']  -> CH606-pl1-6-6-Oct-nc|1_2109_10040_14462_1_ex  - OK\n",
      "#CH510-pl1-5-8-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH510-pl1-5-8-Oct-nc|1_1107_18552_12641_1_ex']  -> CH510-pl1-5-8-Oct-nc|1_1107_18552_12641_1_ex  - OK\n",
      "#CH111-pl1-2-3-Oct-nc.Crangonyx_pseudogracilis_0_OTU.clipped  - ['CH111-pl1-2-3-Oct-nc|1_1111_8862_24940_1_ex']  -> CH111-pl1-2-3-Oct-nc|1_1111_8862_24940_1_ex  - OK\n",
      "extracting hts for Crangonyx_pseudogracilis_0_OTU -> Crangonyx_pseudogracilis_0_OTU_hts.fasta\n"
     ]
    }
   ],
   "source": [
    "import glob\n",
    "import os\n",
    "import shlex, subprocess\n",
    "\n",
    "for i in range(len(OTUs_synonyms)):\n",
    "    print OTUs_synonyms[i],OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    hts_per_OTU=[]\n",
    "    seqs_per_OTU=[]\n",
    "    \n",
    "    for f in glob.glob('*.'+syn+'.clipped.fasta'):\n",
    "        prefix = f.replace('.fasta','')\n",
    "        print \"#\"+prefix,\n",
    "\n",
    "        #cluster at 100% similarity full length\n",
    "\n",
    "        mb.vsearch_cluster_full_length(infile=f, cluster_match=float(1), threads=3, sampleID=prefix)\n",
    "        \n",
    "        #output original centroid id as chosen by vsearch\n",
    "        cs=[]\n",
    "        for c in per_OTU_centroids[OTUs_as_list[i]]:\n",
    "            if f.split(\".\")[0] == c.split(\"|\")[0]:\n",
    "                cs.append(c)\n",
    "        print \" - %s\" %cs,\n",
    "                \n",
    "        hts_per_OTU.append(mb.find_most_abundant_seq_from_uc(uc=prefix+'.uc'))\n",
    "        print \" -> %s\" %hts_per_OTU[-1],\n",
    "        \n",
    "        if hts_per_OTU[-1] in cs:\n",
    "            print \" - OK\"\n",
    "        else:\n",
    "            print \" - adjust\"\n",
    "        \n",
    "\n",
    "#        os.remove(f+'.uc')\n",
    "#        os.remove(f+'_centroids.fasta')\n",
    "#        os.remove(f)\n",
    "\n",
    "    print \"extracting hts for %s -> %s\" %(syn,syn+'_hts.fasta')\n",
    "    for r in SeqIO.parse(syn+'.clipped.fasta', 'fasta'):\n",
    "        if r.id in hts_per_OTU:\n",
    "            seqs_per_OTU.append(r)\n",
    "                \n",
    "    fasta.close()\n",
    "#    print \"final cleanup .. \",\n",
    "#    os.remove(OTUs_synonyms[i]+'.fa')\n",
    "#    print \"DONE!\\n\"\n",
    "        \n",
    "    out=open(syn+'_hts.fasta', 'w')\n",
    "    SeqIO.write(seqs_per_OTU, out, 'fasta')\n",
    "    out.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each original OTU, cluster the chosen (most abundant) haplotypes for each sample at 100% to remove redundancy and identify the set of unique observed haplotypes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU_hts.fasta\n",
      "Crangonyx_pseudogracilis_0_OTU_hts.fasta\n"
     ]
    }
   ],
   "source": [
    "for f in glob.glob('*OTU_hts.fasta'):\n",
    "    print f\n",
    "    prefix = f.replace('.fasta','')\n",
    "    mb.vsearch_cluster_full_length(infile=f, cluster_match=float(1), threads=3, sampleID=prefix)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Align the non-redundant sequences with mafft."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "nseq =  64\n",
      "distance =  local\n",
      "iterate =  16\n",
      "cycle =  1\n",
      "nthread = 0\n",
      "lastonce = 0\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "    0 / 64\r",
      "    1 / 64\r",
      "    2 / 64\r",
      "    3 / 64\r",
      "    4 / 64\r",
      "    5 / 64\r",
      "    6 / 64\r",
      "    7 / 64\r",
      "    8 / 64\r",
      "    9 / 64\r",
      "   10 / 64\r",
      "   11 / 64\r",
      "   12 / 64\r",
      "   13 / 64\r",
      "   14 / 64\r",
      "   15 / 64\r",
      "   16 / 64\r",
      "   17 / 64\r",
      "   18 / 64\r",
      "   19 / 64\r",
      "   20 / 64\r",
      "   21 / 64\r",
      "   22 / 64\r",
      "   23 / 64\r",
      "   24 / 64\r",
      "   25 / 64\r",
      "   26 / 64\r",
      "   27 / 64\r",
      "   28 / 64\r",
      "   29 / 64\r",
      "   30 / 64\r",
      "   31 / 64\r",
      "   32 / 64\r",
      "   33 / 64\r",
      "   34 / 64\r",
      "   35 / 64\r",
      "   36 / 64\r",
      "   37 / 64\r",
      "   38 / 64\r",
      "   39 / 64\r",
      "   40 / 64\r",
      "   41 / 64\r",
      "   42 / 64\r",
      "   43 / 64\r",
      "   44 / 64\r",
      "   45 / 64\r",
      "   46 / 64\r",
      "   47 / 64\r",
      "   48 / 64\r",
      "   49 / 64\r",
      "   50 / 64\r",
      "   51 / 64\r",
      "   52 / 64\r",
      "   53 / 64\r",
      "   54 / 64\r",
      "   55 / 64\r",
      "   56 / 64\r",
      "   57 / 64\r",
      "   58 / 64\r",
      "   59 / 64\r",
      "   60 / 64\r",
      "   61 / 64\r",
      "   62 / 64\r\n",
      "\n",
      "##### writing hat3\n",
      "pairlocalalign (nuc) Version 7.123b alg=L, model=DNA200 (2),  2.000 ( 6.000), -0.099 (-0.297)\n",
      "0 thread(s)\n",
      "nthread = 0\n",
      "blosum 62 / kimura 200\n",
      "Loading 'hat3' ... \n",
      "done.\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "Loading 'hat2' ... done.\n",
      "Constructing a UPGMA tree ... \n",
      "\r",
      "    0 / 64\r",
      "   10 / 64\r",
      "   20 / 64\r",
      "   30 / 64\r",
      "   40 / 64\r",
      "   50 / 64\r",
      "   60 / 64\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "\r",
      "STEP     1 /63 c\r",
      "STEP     2 /63 c\r",
      "STEP     3 /63 c\r",
      "STEP     4 /63 c\r",
      "STEP     5 /63 c\r",
      "STEP     6 /63 c\r",
      "STEP     7 /63 c\r",
      "STEP     8 /63 c\r",
      "STEP     9 /63 c\r",
      "STEP    10 /63 c\r",
      "STEP    11 /63 c\r",
      "STEP    12 /63 c\r",
      "STEP    13 /63 c\r",
      "STEP    14 /63 c\r",
      "STEP    15 /63 c\r",
      "STEP    16 /63 c\r",
      "STEP    17 /63 c\r",
      "STEP    18 /63 c\r",
      "STEP    19 /63 c\r",
      "STEP    20 /63 c\r",
      "STEP    21 /63 c\r",
      "STEP    22 /63 c\r",
      "STEP    23 /63 c\r",
      "STEP    24 /63 c\r",
      "STEP    25 /63 c\r",
      "STEP    26 /63 c\r",
      "STEP    27 /63 c\r",
      "STEP    28 /63 c\r",
      "STEP    29 /63 c\r",
      "STEP    30 /63 c\r",
      "STEP    31 /63 c\r",
      "STEP    32 /63 c\r",
      "STEP    33 /63 c\r",
      "STEP    34 /63 c\r",
      "STEP    35 /63 c\r",
      "STEP    36 /63 c\r",
      "STEP    37 /63 c\r",
      "STEP    38 /63 c\r",
      "STEP    39 /63 c\r",
      "STEP    40 /63 c\r",
      "STEP    41 /63 c\r",
      "STEP    42 /63 c\r",
      "STEP    43 /63 c\r",
      "STEP    44 /63 c\r",
      "STEP    45 /63 c\r",
      "STEP    46 /63 c\r",
      "STEP    47 /63 c\r",
      "STEP    48 /63 c\r",
      "STEP    49 /63 c\r",
      "STEP    50 /63 c\r",
      "STEP    51 /63 c\r",
      "STEP    52 /63 c\r",
      "STEP    53 /63 c\n",
      "Reallocating..done. *alloclen = 1626\n",
      "\r",
      "STEP    54 /63 c\r",
      "STEP    55 /63 c\r",
      "STEP    56 /63 c\r",
      "STEP    57 /63 c\r",
      "STEP    58 /63 c\r",
      "STEP    59 /63 c\r",
      "STEP    60 /63 c\r",
      "STEP    61 /63 c\r",
      "STEP    62 /63 c\r",
      "STEP    63 /63 c\n",
      "done.\n",
      "tbfast (nuc) Version 7.123b alg=A, model=DNA200 (2),  1.530 ( 4.590), -0.000 (-0.000)\n",
      "0 thread(s)\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "\n",
      "\r",
      "    0 / 64\r",
      "   10 / 64\r",
      "   20 / 64\r",
      "   30 / 64\r",
      "   40 / 64\r",
      "   50 / 64\r",
      "   60 / 64\n",
      "Segment   1/  1    1- 320\n",
      "STEP 001-001-0  identical.   \r",
      "STEP 001-001-1  identical.   \r",
      "STEP 001-002-0  identical.   \r",
      "STEP 001-002-1  identical.   \r",
      "STEP 001-003-0  identical.   \r",
      "STEP 001-003-1  identical.   \r",
      "STEP 001-004-0  identical.   \r",
      "STEP 001-004-1  identical.   \r",
      "STEP 001-005-0  identical.   \r",
      "STEP 001-005-1  identical.   \r",
      "STEP 001-006-0  identical.   \r",
      "STEP 001-006-1  identical.   \r",
      "STEP 001-007-0  identical.   \r",
      "STEP 001-007-1  identical.   \r",
      "STEP 001-008-0  identical.   \r",
      "STEP 001-008-1  identical.   \r",
      "STEP 001-009-0  identical.   \r",
      "STEP 001-009-1  identical.   \r",
      "STEP 001-010-0  identical.   \r",
      "STEP 001-010-1  identical.   \r",
      "STEP 001-011-0  identical.   \r",
      "STEP 001-011-1  identical.   \r",
      "STEP 001-012-0  identical.   \r",
      "STEP 001-012-1  identical.   \r",
      "STEP 001-013-0  identical.   \r",
      "STEP 001-013-1  identical.   \r",
      "STEP 001-014-0  identical.   \r",
      "STEP 001-014-1  identical.   \r",
      "STEP 001-015-0  identical.   \r",
      "STEP 001-015-1  identical.   \r",
      "STEP 001-016-0  identical.   \r",
      "STEP 001-016-1  identical.   \r",
      "STEP 001-017-0  identical.   \r",
      "STEP 001-017-1  identical.   \r",
      "STEP 001-018-0  identical.   \r",
      "STEP 001-018-1  identical.   \r",
      "STEP 001-019-0  identical.   \r",
      "STEP 001-019-1  identical.   \r",
      "STEP 001-020-0  identical.   \r",
      "STEP 001-020-1  identical.   \r",
      "STEP 001-021-0  identical.   \r",
      "STEP 001-021-1  identical.   \r",
      "STEP 001-022-0  identical.   \r",
      "STEP 001-022-1  identical.   \r",
      "STEP 001-023-0  identical.   \r",
      "STEP 001-023-1  identical.   \r",
      "STEP 001-024-0  identical.   \r",
      "STEP 001-024-1  identical.   \r",
      "STEP 001-025-0  identical.   \r",
      "STEP 001-025-1  identical.   \r",
      "STEP 001-026-0  identical.   \r",
      "STEP 001-026-1  identical.   \r",
      "STEP 001-027-0  identical.   \r",
      "STEP 001-027-1  identical.   \r",
      "STEP 001-028-0  identical.   \r",
      "STEP 001-028-1  identical.   \r",
      "STEP 001-029-0  identical.   \r",
      "STEP 001-029-1  identical.   \r",
      "STEP 001-030-0  identical.   \r",
      "STEP 001-030-1  identical.   \r",
      "STEP 001-031-0  identical.   \r",
      "STEP 001-031-1  identical.   \r",
      "STEP 001-032-0  identical.   \r",
      "STEP 001-032-1  identical.   \r",
      "STEP 001-033-0  identical.   \r",
      "STEP 001-033-1  identical.   \r",
      "STEP 001-034-0  identical.   \r",
      "STEP 001-034-1  identical.   \r",
      "STEP 001-035-0  identical.   \r",
      "STEP 001-035-1  identical.   \r",
      "STEP 001-036-0  identical.   \r",
      "STEP 001-036-1  identical.   \r",
      "STEP 001-037-0  identical.   \r",
      "STEP 001-037-1  identical.   \r",
      "STEP 001-038-0  identical.   \r",
      "STEP 001-038-1  identical.   \r",
      "STEP 001-039-0  identical.   \r",
      "STEP 001-039-1  identical.   \r",
      "STEP 001-040-0  identical.   \r",
      "STEP 001-040-1  identical.   \r",
      "STEP 001-041-0  identical.   \r",
      "STEP 001-041-1  identical.   \r",
      "STEP 001-042-0  identical.   \r",
      "STEP 001-042-1  identical.   \r",
      "STEP 001-043-0  identical.   \r",
      "STEP 001-043-1  identical.   \r",
      "STEP 001-044-0  identical.   \r",
      "STEP 001-044-1  identical.   \r",
      "STEP 001-045-0  identical.   \r",
      "STEP 001-045-1  identical.   \r",
      "STEP 001-046-0  identical.   \r",
      "STEP 001-046-1  identical.   \r",
      "STEP 001-047-0  identical.   \r",
      "STEP 001-047-1  accepted.\r",
      "STEP 001-048-0  identical.   \r",
      "STEP 001-048-1  identical.   \r",
      "STEP 001-049-0  identical.   \r",
      "STEP 001-049-1  identical.   \r",
      "STEP 001-050-0  identical.   \r",
      "STEP 001-050-1  identical.   \r",
      "STEP 001-051-0  identical.   \r",
      "STEP 001-051-1  identical.   \r",
      "STEP 001-052-0  identical.   \r",
      "STEP 001-052-1  identical.   \r",
      "STEP 001-053-0  identical.   \r",
      "STEP 001-053-1  identical.   \r",
      "STEP 001-054-0  identical.   \r",
      "STEP 001-054-1  identical.   \r",
      "STEP 001-055-0  identical.   \r",
      "STEP 001-055-1  identical.   \r",
      "STEP 001-056-0  identical.   \r",
      "STEP 001-056-1  identical.   \r",
      "STEP 001-057-0  identical.   \r",
      "STEP 001-057-1  identical.   \r",
      "STEP 001-058-0  identical.   \r",
      "STEP 001-058-1  identical.   \r",
      "STEP 001-059-0  identical.   \r",
      "STEP 001-059-1  identical.   \r",
      "STEP 001-060-0  identical.   \r",
      "STEP 001-060-1  identical.   \r",
      "STEP 001-061-0  identical.   \r",
      "STEP 001-061-1  identical.   \r",
      "STEP 001-062-0  identical.   \r",
      "STEP 001-062-1  identical.   \r",
      "STEP 001-063-1  identical.   \r",
      "STEP 002-063-1  identical.   \r",
      "STEP 002-062-0  identical.   \r",
      "STEP 002-062-1  identical.   \r",
      "STEP 002-061-0  identical.   \r",
      "STEP 002-061-1  identical.   \r",
      "STEP 002-060-0  identical.   \r",
      "STEP 002-060-1  identical.   \r",
      "STEP 002-059-0  identical.   \r",
      "STEP 002-059-1  identical.   \r",
      "STEP 002-058-0  identical.   \r",
      "STEP 002-058-1  identical.   \r",
      "STEP 002-057-0  identical.   \r",
      "STEP 002-057-1  identical.   \r",
      "STEP 002-056-0  identical.   \r",
      "STEP 002-056-1  identical.   \r",
      "STEP 002-055-0  identical.   \r",
      "STEP 002-055-1  identical.   \r",
      "STEP 002-054-0  identical.   \r",
      "STEP 002-054-1  identical.   \r",
      "STEP 002-053-0  identical.   \r",
      "STEP 002-053-1  identical.   \r",
      "STEP 002-052-0  identical.   \r",
      "STEP 002-052-1  identical.   \r",
      "STEP 002-051-0  identical.   \r",
      "STEP 002-051-1  identical.   \r",
      "STEP 002-050-0  identical.   \r",
      "STEP 002-050-1  identical.   \r",
      "STEP 002-049-0  identical.   \r",
      "STEP 002-049-1  identical.   \r",
      "STEP 002-048-0  identical.   \r",
      "STEP 002-048-1  identical.   \r",
      "STEP 002-047-0  identical.   \r",
      "STEP 002-047-1  identical.   \r",
      "STEP 002-046-0  identical.   \r",
      "STEP 002-046-1  identical.   \r",
      "STEP 002-045-0  identical.   \r",
      "STEP 002-045-1  identical.   \r",
      "STEP 002-044-0  identical.   \r",
      "STEP 002-044-1  identical.   \r",
      "STEP 002-043-0  identical.   \r",
      "STEP 002-043-1  identical.   \r",
      "STEP 002-042-0  identical.   \r",
      "STEP 002-042-1  identical.   \r",
      "STEP 002-041-0  identical.   \r",
      "STEP 002-041-1  identical.   \r",
      "STEP 002-040-0  identical.   \r",
      "STEP 002-040-1  identical.   \r",
      "STEP 002-039-0  identical.   \r",
      "STEP 002-039-1  identical.   \r",
      "STEP 002-038-0  identical.   \r",
      "STEP 002-038-1  identical.   \r",
      "STEP 002-037-0  identical.   \r",
      "STEP 002-037-1  identical.   \r",
      "STEP 002-036-0  identical.   \r",
      "STEP 002-036-1  identical.   \r",
      "STEP 002-035-0  identical.   \r",
      "STEP 002-035-1  identical.   \r",
      "STEP 002-034-0  identical.   \r",
      "STEP 002-034-1  identical.   \r",
      "STEP 002-033-0  identical.   \r",
      "STEP 002-033-1  identical.   \r",
      "STEP 002-032-0  identical.   \r",
      "STEP 002-032-1  identical.   \r",
      "STEP 002-031-0  identical.   \r",
      "STEP 002-031-1  identical.   \r",
      "STEP 002-030-0  identical.   \r",
      "STEP 002-030-1  identical.   \r",
      "STEP 002-029-0  identical.   \r",
      "STEP 002-029-1  identical.   \r",
      "STEP 002-028-0  identical.   \r",
      "STEP 002-028-1  identical.   \r",
      "STEP 002-027-0  identical.   \r",
      "STEP 002-027-1  identical.   \r",
      "STEP 002-026-0  identical.   \r",
      "STEP 002-026-1  identical.   \r",
      "STEP 002-025-0  identical.   \r",
      "STEP 002-025-1  identical.   \r",
      "STEP 002-024-0  identical.   \r",
      "STEP 002-024-1  identical.   \r",
      "STEP 002-023-0  identical.   \r",
      "STEP 002-023-1  identical.   \r",
      "STEP 002-022-0  identical.   \r",
      "STEP 002-022-1  identical.   \r",
      "STEP 002-021-0  identical.   \r",
      "STEP 002-021-1  identical.   \r",
      "STEP 002-020-0  identical.   \r",
      "STEP 002-020-1  identical.   \r",
      "STEP 002-019-0  identical.   \r",
      "STEP 002-019-1  identical.   \r",
      "STEP 002-018-0  identical.   \r",
      "STEP 002-018-1  identical.   \r",
      "STEP 002-017-0  identical.   \r",
      "STEP 002-017-1  identical.   \r",
      "STEP 002-016-0  identical.   \r",
      "STEP 002-016-1  identical.   \r",
      "STEP 002-015-0  identical.   \r",
      "STEP 002-015-1  identical.   \r\n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.123b alg=A, model=DNA200 (2),  1.530 ( 4.590), -0.000 (-0.000)\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --legacygappenalty option.\n",
      "\n",
      "\n",
      "nseq =  57\n",
      "distance =  local\n",
      "iterate =  16\n",
      "cycle =  1\n",
      "nthread = 0\n",
      "lastonce = 0\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "    0 / 57\r",
      "    1 / 57\r",
      "    2 / 57\r",
      "    3 / 57\r",
      "    4 / 57\r",
      "    5 / 57\r",
      "    6 / 57\r",
      "    7 / 57\r",
      "    8 / 57\r",
      "    9 / 57\r",
      "   10 / 57\r",
      "   11 / 57\r",
      "   12 / 57\r",
      "   13 / 57\r",
      "   14 / 57\r",
      "   15 / 57\r",
      "   16 / 57\r",
      "   17 / 57\r",
      "   18 / 57\r",
      "   19 / 57\r",
      "   20 / 57\r",
      "   21 / 57\r",
      "   22 / 57\r",
      "   23 / 57\r",
      "   24 / 57\r",
      "   25 / 57\r",
      "   26 / 57\r",
      "   27 / 57\r",
      "   28 / 57\r",
      "   29 / 57\r",
      "   30 / 57\r",
      "   31 / 57\r",
      "   32 / 57\r",
      "   33 / 57\r",
      "   34 / 57\r",
      "   35 / 57\r",
      "   36 / 57\r",
      "   37 / 57\r",
      "   38 / 57\r",
      "   39 / 57\r",
      "   40 / 57\r",
      "   41 / 57\r",
      "   42 / 57\r",
      "   43 / 57\r",
      "   44 / 57\r",
      "   45 / 57\r",
      "   46 / 57\r",
      "   47 / 57\r",
      "   48 / 57\r",
      "   49 / 57\r",
      "   50 / 57\r",
      "   51 / 57\r",
      "   52 / 57\r",
      "   53 / 57\r",
      "   54 / 57\r",
      "   55 / 57\r\n",
      "\n",
      "##### writing hat3\n",
      "pairlocalalign (nuc) Version 7.123b alg=L, model=DNA200 (2),  2.000 ( 6.000), -0.099 (-0.297)\n",
      "0 thread(s)\n",
      "nthread = 0\n",
      "blosum 62 / kimura 200\n",
      "Loading 'hat3' ... \n",
      "done.\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "Loading 'hat2' ... done.\n",
      "Constructing a UPGMA tree ... \n",
      "\r",
      "    0 / 57\r",
      "   10 / 57\r",
      "   20 / 57\r",
      "   30 / 57\r",
      "   40 / 57\r",
      "   50 / 57\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "\r",
      "STEP     1 /56 c\r",
      "STEP     2 /56 c\r",
      "STEP     3 /56 c\r",
      "STEP     4 /56 c\r",
      "STEP     5 /56 c\r",
      "STEP     6 /56 c\r",
      "STEP     7 /56 c\r",
      "STEP     8 /56 c\r",
      "STEP     9 /56 c\r",
      "STEP    10 /56 c\r",
      "STEP    11 /56 c\r",
      "STEP    12 /56 c\r",
      "STEP    13 /56 c\r",
      "STEP    14 /56 c\r",
      "STEP    15 /56 c\r",
      "STEP    16 /56 c\r",
      "STEP    17 /56 c\r",
      "STEP    18 /56 c\r",
      "STEP    19 /56 c\r",
      "STEP    20 /56 c\r",
      "STEP    21 /56 c\r",
      "STEP    22 /56 c\r",
      "STEP    23 /56 c\r",
      "STEP    24 /56 c\r",
      "STEP    25 /56 c\r",
      "STEP    26 /56 c\r",
      "STEP    27 /56 c\r",
      "STEP    28 /56 c\r",
      "STEP    29 /56 c\r",
      "STEP    30 /56 c\r",
      "STEP    31 /56 c\r",
      "STEP    32 /56 c\r",
      "STEP    33 /56 c\r",
      "STEP    34 /56 c\r",
      "STEP    35 /56 c\r",
      "STEP    36 /56 c\r",
      "STEP    37 /56 c\r",
      "STEP    38 /56 c\r",
      "STEP    39 /56 c\r",
      "STEP    40 /56 c\r",
      "STEP    41 /56 c\r",
      "STEP    42 /56 c\r",
      "STEP    43 /56 c\r",
      "STEP    44 /56 c\r",
      "STEP    45 /56 c\r",
      "STEP    46 /56 c\r",
      "STEP    47 /56 c\r",
      "STEP    48 /56 c\r",
      "STEP    49 /56 c\r",
      "STEP    50 /56 c\r",
      "STEP    51 /56 c\r",
      "STEP    52 /56 c\r",
      "STEP    53 /56 c\r",
      "STEP    54 /56 c\r",
      "STEP    55 /56 c\r",
      "STEP    56 /56 c\n",
      "done.\n",
      "tbfast (nuc) Version 7.123b alg=A, model=DNA200 (2),  1.530 ( 4.590), -0.000 (-0.000)\n",
      "0 thread(s)\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating 200PAM scoring matrix for nucleotides ... done\n",
      "done\n",
      "done\n",
      "scoremtx = -1\n",
      "\n",
      "\r",
      "    0 / 57\r",
      "   10 / 57\r",
      "   20 / 57\r",
      "   30 / 57\r",
      "   40 / 57\r",
      "   50 / 57\n",
      "Segment   1/  1    1- 318\n",
      "STEP 001-001-0  identical.   \r",
      "STEP 001-001-1  identical.   \r",
      "STEP 001-002-0  identical.   \r",
      "STEP 001-002-1  identical.   \r",
      "STEP 001-003-0  identical.   \r",
      "STEP 001-003-1  identical.   \r",
      "STEP 001-004-0  identical.   \r",
      "STEP 001-004-1  identical.   \r",
      "STEP 001-005-0  identical.   \r",
      "STEP 001-005-1  identical.   \r",
      "STEP 001-006-0  identical.   \r",
      "STEP 001-006-1  identical.   \r",
      "STEP 001-007-0  identical.   \r",
      "STEP 001-007-1  identical.   \r",
      "STEP 001-008-0  identical.   \r",
      "STEP 001-008-1  identical.   \r",
      "STEP 001-009-0  identical.   \r",
      "STEP 001-009-1  identical.   \r",
      "STEP 001-010-0  identical.   \r",
      "STEP 001-010-1  identical.   \r",
      "STEP 001-011-0  identical.   \r",
      "STEP 001-011-1  identical.   \r",
      "STEP 001-012-0  identical.   \r",
      "STEP 001-012-1  identical.   \r",
      "STEP 001-013-0  identical.   \r",
      "STEP 001-013-1  identical.   \r",
      "STEP 001-014-0  identical.   \r",
      "STEP 001-014-1  identical.   \r",
      "STEP 001-015-0  identical.   \r",
      "STEP 001-015-1  identical.   \r",
      "STEP 001-016-0  identical.   \r",
      "STEP 001-016-1  identical.   \r",
      "STEP 001-017-0  identical.   \r",
      "STEP 001-017-1  identical.   \r",
      "STEP 001-018-0  identical.   \r",
      "STEP 001-018-1  identical.   \r",
      "STEP 001-019-0  identical.   \r",
      "STEP 001-019-1  identical.   \r",
      "STEP 001-020-0  identical.   \r",
      "STEP 001-020-1  identical.   \r",
      "STEP 001-021-0  identical.   \r",
      "STEP 001-021-1  identical.   \r",
      "STEP 001-022-0  identical.   \r",
      "STEP 001-022-1  identical.   \r",
      "STEP 001-023-0  identical.   \r",
      "STEP 001-023-1  identical.   \r",
      "STEP 001-024-0  identical.   \r",
      "STEP 001-024-1  identical.   \r",
      "STEP 001-025-0  identical.   \r",
      "STEP 001-025-1  identical.   \r",
      "STEP 001-026-0  identical.   \r",
      "STEP 001-026-1  identical.   \r",
      "STEP 001-027-0  identical.   \r",
      "STEP 001-027-1  identical.   \r",
      "STEP 001-028-0  identical.   \r",
      "STEP 001-028-1  identical.   \r",
      "STEP 001-029-0  identical.   \r",
      "STEP 001-029-1  identical.   \r",
      "STEP 001-030-0  identical.   \r",
      "STEP 001-030-1  identical.   \r",
      "STEP 001-031-0  identical.   \r",
      "STEP 001-031-1  identical.   \r",
      "STEP 001-032-0  identical.   \r",
      "STEP 001-032-1  identical.   \r",
      "STEP 001-033-0  identical.   \r",
      "STEP 001-033-1  identical.   \r",
      "STEP 001-034-0  identical.   \r",
      "STEP 001-034-1  identical.   \r",
      "STEP 001-035-0  identical.   \r",
      "STEP 001-035-1  identical.   \r",
      "STEP 001-036-0  identical.   \r",
      "STEP 001-036-1  identical.   \r",
      "STEP 001-037-0  identical.   \r",
      "STEP 001-037-1  identical.   \r",
      "STEP 001-038-0  identical.   \r",
      "STEP 001-038-1  identical.   \r",
      "STEP 001-039-0  identical.   \r",
      "STEP 001-039-1  identical.   \r",
      "STEP 001-040-0  identical.   \r",
      "STEP 001-040-1  identical.   \r",
      "STEP 001-041-0  identical.   \r",
      "STEP 001-041-1  identical.   \r",
      "STEP 001-042-0  identical.   \r",
      "STEP 001-042-1  identical.   \r",
      "STEP 001-043-0  identical.   \r",
      "STEP 001-043-1  identical.   \r",
      "STEP 001-044-0  identical.   \r",
      "STEP 001-044-1  identical.   \r",
      "STEP 001-045-0  identical.   \r",
      "STEP 001-045-1  identical.   \r",
      "STEP 001-046-0  identical.   \r",
      "STEP 001-046-1  identical.   \r",
      "STEP 001-047-0  identical.   \r",
      "STEP 001-047-1  identical.   \r",
      "STEP 001-048-0  identical.   \r",
      "STEP 001-048-1  identical.   \r",
      "STEP 001-049-0  identical.   \r",
      "STEP 001-049-1  identical.   \r",
      "STEP 001-050-0  identical.   \r",
      "STEP 001-050-1  identical.   \r",
      "STEP 001-051-0  identical.   \r",
      "STEP 001-051-1  identical.   \r",
      "STEP 001-052-0  identical.   \r",
      "STEP 001-052-1  identical.   \r",
      "STEP 001-053-0  identical.   \r",
      "STEP 001-053-1  identical.   \r",
      "STEP 001-054-0  identical.   \r",
      "STEP 001-054-1  identical.   \r",
      "STEP 001-055-0  identical.   \r",
      "STEP 001-055-1  identical.   \r",
      "STEP 001-056-1  identical.   \r",
      "STEP 002-056-1  identical.   \r",
      "STEP 002-055-0  identical.   \r",
      "STEP 002-055-1  identical.   \r\n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.123b alg=A, model=DNA200 (2),  1.530 ( 4.590), -0.000 (-0.000)\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --legacygappenalty option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for file in $(ls -1 *_hts_centroids*)\n",
    "do\n",
    "    out=$(echo -e \"$file\" | sed 's/fasta$/aln.fasta/')\n",
    "    mafft --localpair --maxiterate 1000 $file > $out\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Manually inspect alignments and remove dubious bases. Curated alignments were saved to directory `haplotype_alignments`.\n",
    "Cluster observed haplotypes again at 100% similarity to remove redundancy."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import glob\n",
    "\n",
    "for f in glob.glob('../haplotypes_alignments/*'):\n",
    "    print f\n",
    "    prefix = f.split(\"/\")[1].replace(\"_centroids.aln.fasta\",\"\")\n",
    "    print prefix\n",
    "    out = \"\"\n",
    "    seqs = SeqIO.parse(open(f,'r'), 'fasta')\n",
    "    for s in seqs:\n",
    "        out+=\">%s\\n%s\\n\" %(s.id,str(s.seq).replace(\"-\",\"\").upper())\n",
    "\n",
    "    fh = open(prefix+'.fasta','w')\n",
    "    fh.write(out)\n",
    "    fh.close()\n",
    "    \n",
    "    mb.vsearch_cluster_full_length(infile=prefix+'.fasta', cluster_match=float(1), threads=3, sampleID=prefix)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Give unique names to haplotypes and write to file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_pseudogracilis_0_OTU_hts_centroids.fasta\n",
      "Crangonyx_pseudogracilis_0_OTU_hts\n",
      "Crangonyx_floridanus_0_OTU_hts_centroids.fasta\n",
      "Crangonyx_floridanus_0_OTU_hts\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import glob\n",
    "\n",
    "for f in glob.glob('*_hts_centroids.fasta'):\n",
    "    print f\n",
    "    sp_prefix = f.split(\"_\")[0][0]+f.split(\"_\")[1][0]\n",
    "    prefix = f.replace(\"_centroids.fasta\",\"\")\n",
    "    print prefix\n",
    "    \n",
    "    count = 1\n",
    "\n",
    "    seqs = SeqIO.parse(open(f, 'r'), 'fasta')\n",
    "\n",
    "    fh=open(prefix+'.nr.fasta','w')\n",
    "    for s in seqs:\n",
    "        fh.write(\">\"+sp_prefix+'_UK_Mb-'+\"%02d\\n%s\\n\" %(count,s.seq))\n",
    "        count+=1\n",
    "    fh.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Concatenate all haplotypes into a single file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crangonyx_floridanus_0_OTU CH506-pl1-5-4-Oct-nc|1_1105_11317_10762_1_ex\n",
      "Crangonyx_floridanus\n",
      "Crangonyx_pseudogracilis_0_OTU CH105-1-4-May-nc|1_2105_9715_15085_1_ex\n",
      "Crangonyx_pseudogracilis\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import glob\n",
    "\n",
    "seqs = []\n",
    "\n",
    "for i in range(len(OTUs_synonyms)):\n",
    "    print OTUs_synonyms[i],OTUs_as_list[i]\n",
    "    syn=OTUs_synonyms[i]\n",
    "    sp=\"_\".join(syn.split(\"_\")[:2])\n",
    "    print sp\n",
    "#    prefix = sp.split(\"_\")[0][0]+sp.split(\"_\")[1][0]\n",
    "#    print prefix\n",
    "    for r in SeqIO.parse(open(syn+'_hts.nr.fasta','r'), 'fasta'):\n",
    "        r.description = sp+'|'+r.id\n",
    "        r.id = r.description\n",
    "        seqs.append(r)\n",
    "        \n",
    "out=open('Crangonyx_from_metaBEAT.fasta', 'w')\n",
    "SeqIO.write(seqs, out, 'fasta')\n",
    "out.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Compare the observed haplotypes to the full set of sequences to identify the samples which contain sequences that receive full length hits from the haplotypes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 0%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 1%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 2%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 2%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 3%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 4%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 5%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 5%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 6%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 7%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 8%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 8%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 9%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 10%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 10%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 11%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 12%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 13%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 13%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 14%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 15%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 16%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 16%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 17%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 18%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 19%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 19%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 20%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 21%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 22%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 22%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 23%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 24%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 25%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 25%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 26%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 27%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 28%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 28%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 29%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 30%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 30%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 31%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 32%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 33%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 33%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 34%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 35%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 36%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 36%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 37%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 38%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 39%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 39%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 40%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 41%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 42%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 42%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 43%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 44%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 45%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 45%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 46%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 47%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 48%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 48%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 49%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 50%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 50%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 51%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 52%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 53%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 53%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 54%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 55%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 56%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 56%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 57%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 58%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 59%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 59%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 60%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 61%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 62%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 62%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 63%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 64%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 65%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 65%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 66%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 67%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 67%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 68%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 69%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 70%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 70%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 71%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 72%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 73%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 73%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 74%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 75%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 76%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 76%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 77%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 78%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 79%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 79%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 80%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 81%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 82%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 82%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 83%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 84%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 85%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 85%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 86%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 87%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 87%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 88%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 89%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 90%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 90%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 91%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 92%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 93%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 93%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 94%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 95%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 96%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 96%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 97%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 98%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 99%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 99%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 100%  \r",
      "Reading file Crangonyx_floridanus_0_OTU.clipped.fasta 100%\n",
      "41872 nt in 135 seqs, min 306, max 315, avg 310\n",
      "Indexing sequences 0%  \r",
      "Indexing sequences 0%  \r",
      "Indexing sequences 1%  \r",
      "Indexing sequences 1%  \r",
      "Indexing sequences 2%  \r",
      "Indexing sequences 3%  \r",
      "Indexing sequences 4%  \r",
      "Indexing sequences 4%  \r",
      "Indexing sequences 5%  \r",
      "Indexing sequences 6%  \r",
      "Indexing sequences 7%  \r",
      "Indexing sequences 7%  \r",
      "Indexing sequences 8%  \r",
      "Indexing sequences 9%  \r",
      "Indexing sequences 10%  \r",
      "Indexing sequences 10%  \r",
      "Indexing sequences 11%  \r",
      "Indexing sequences 12%  \r",
      "Indexing sequences 13%  \r",
      "Indexing sequences 13%  \r",
      "Indexing sequences 14%  \r",
      "Indexing sequences 15%  \r",
      "Indexing sequences 16%  \r",
      "Indexing sequences 16%  \r",
      "Indexing sequences 17%  \r",
      "Indexing sequences 18%  \r",
      "Indexing sequences 19%  \r",
      "Indexing sequences 19%  \r",
      "Indexing sequences 20%  \r",
      "Indexing sequences 21%  \r",
      "Indexing sequences 21%  \r",
      "Indexing sequences 22%  \r",
      "Indexing sequences 23%  \r",
      "Indexing sequences 24%  \r",
      "Indexing sequences 24%  \r",
      "Indexing sequences 25%  \r",
      "Indexing sequences 26%  \r",
      "Indexing sequences 27%  \r",
      "Indexing sequences 27%  \r",
      "Indexing sequences 28%  \r",
      "Indexing sequences 29%  \r",
      "Indexing sequences 30%  \r",
      "Indexing sequences 30%  \r",
      "Indexing sequences 31%  \r",
      "Indexing sequences 32%  \r",
      "Indexing sequences 33%  \r",
      "Indexing sequences 33%  \r",
      "Indexing sequences 34%  \r",
      "Indexing sequences 35%  \r",
      "Indexing sequences 36%  \r",
      "Indexing sequences 36%  \r",
      "Indexing sequences 37%  \r",
      "Indexing sequences 38%  \r",
      "Indexing sequences 39%  \r",
      "Indexing sequences 39%  \r",
      "Indexing sequences 40%  \r",
      "Indexing sequences 41%  \r",
      "Indexing sequences 41%  \r",
      "Indexing sequences 42%  \r",
      "Indexing sequences 43%  \r",
      "Indexing sequences 44%  \r",
      "Indexing sequences 44%  \r",
      "Indexing sequences 45%  \r",
      "Indexing sequences 46%  \r",
      "Indexing sequences 47%  \r",
      "Indexing sequences 47%  \r",
      "Indexing sequences 48%  \r",
      "Indexing sequences 49%  \r",
      "Indexing sequences 50%  \r",
      "Indexing sequences 50%  \r",
      "Indexing sequences 51%  \r",
      "Indexing sequences 52%  \r",
      "Indexing sequences 53%  \r",
      "Indexing sequences 53%  \r",
      "Indexing sequences 54%  \r",
      "Indexing sequences 55%  \r",
      "Indexing sequences 56%  \r",
      "Indexing sequences 56%  \r",
      "Indexing sequences 57%  \r",
      "Indexing sequences 58%  \r",
      "Indexing sequences 59%  \r",
      "Indexing sequences 59%  \r",
      "Indexing sequences 60%  \r",
      "Indexing sequences 61%  \r",
      "Indexing sequences 61%  \r",
      "Indexing sequences 62%  \r",
      "Indexing sequences 63%  \r",
      "Indexing sequences 64%  \r",
      "Indexing sequences 64%  \r",
      "Indexing sequences 65%  \r",
      "Indexing sequences 66%  \r",
      "Indexing sequences 67%  \r",
      "Indexing sequences 67%  \r",
      "Indexing sequences 68%  \r",
      "Indexing sequences 69%  \r",
      "Indexing sequences 70%  \r",
      "Indexing sequences 70%  \r",
      "Indexing sequences 71%  \r",
      "Indexing sequences 72%  \r",
      "Indexing sequences 73%  \r",
      "Indexing sequences 73%  \r",
      "Indexing sequences 74%  \r",
      "Indexing sequences 75%  \r",
      "Indexing sequences 76%  \r",
      "Indexing sequences 76%  \r",
      "Indexing sequences 77%  \r",
      "Indexing sequences 78%  \r",
      "Indexing sequences 79%  \r",
      "Indexing sequences 79%  \r",
      "Indexing sequences 80%  \r",
      "Indexing sequences 81%  \r",
      "Indexing sequences 81%  \r",
      "Indexing sequences 82%  \r",
      "Indexing sequences 83%  \r",
      "Indexing sequences 84%  \r",
      "Indexing sequences 84%  \r",
      "Indexing sequences 85%  \r",
      "Indexing sequences 86%  \r",
      "Indexing sequences 87%  \r",
      "Indexing sequences 87%  \r",
      "Indexing sequences 88%  \r",
      "Indexing sequences 89%  \r",
      "Indexing sequences 90%  \r",
      "Indexing sequences 90%  \r",
      "Indexing sequences 91%  \r",
      "Indexing sequences 92%  \r",
      "Indexing sequences 93%  \r",
      "Indexing sequences 93%  \r",
      "Indexing sequences 94%  \r",
      "Indexing sequences 95%  \r",
      "Indexing sequences 96%  \r",
      "Indexing sequences 96%  \r",
      "Indexing sequences 97%  \r",
      "Indexing sequences 98%  \r",
      "Indexing sequences 99%  \r",
      "Indexing sequences 99%  \r",
      "Indexing sequences 100%\n",
      "Masking 0%  \r",
      "Masking 100%\n",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 1%  \r",
      "Counting unique k-mers 1%  \r",
      "Counting unique k-mers 2%  \r",
      "Counting unique k-mers 3%  \r",
      "Counting unique k-mers 4%  \r",
      "Counting unique k-mers 4%  \r",
      "Counting unique k-mers 5%  \r",
      "Counting unique k-mers 6%  \r",
      "Counting unique k-mers 7%  \r",
      "Counting unique k-mers 7%  \r",
      "Counting unique k-mers 8%  \r",
      "Counting unique k-mers 9%  \r",
      "Counting unique k-mers 10%  \r",
      "Counting unique k-mers 10%  \r",
      "Counting unique k-mers 11%  \r",
      "Counting unique k-mers 12%  \r",
      "Counting unique k-mers 13%  \r",
      "Counting unique k-mers 13%  \r",
      "Counting unique k-mers 14%  \r",
      "Counting unique k-mers 15%  \r",
      "Counting unique k-mers 16%  \r",
      "Counting unique k-mers 16%  \r",
      "Counting unique k-mers 17%  \r",
      "Counting unique k-mers 18%  \r",
      "Counting unique k-mers 19%  \r",
      "Counting unique k-mers 19%  \r",
      "Counting unique k-mers 20%  \r",
      "Counting unique k-mers 21%  \r",
      "Counting unique k-mers 21%  \r",
      "Counting unique k-mers 22%  \r",
      "Counting unique k-mers 23%  \r",
      "Counting unique k-mers 24%  \r",
      "Counting unique k-mers 24%  \r",
      "Counting unique k-mers 25%  \r",
      "Counting unique k-mers 26%  \r",
      "Counting unique k-mers 27%  \r",
      "Counting unique k-mers 27%  \r",
      "Counting unique k-mers 28%  \r",
      "Counting unique k-mers 29%  \r",
      "Counting unique k-mers 30%  \r",
      "Counting unique k-mers 30%  \r",
      "Counting unique k-mers 31%  \r",
      "Counting unique k-mers 32%  \r",
      "Counting unique k-mers 33%  \r",
      "Counting unique k-mers 33%  \r",
      "Counting unique k-mers 34%  \r",
      "Counting unique k-mers 35%  \r",
      "Counting unique k-mers 36%  \r",
      "Counting unique k-mers 36%  \r",
      "Counting unique k-mers 37%  \r",
      "Counting unique k-mers 38%  \r",
      "Counting unique k-mers 39%  \r",
      "Counting unique k-mers 39%  \r",
      "Counting unique k-mers 40%  \r",
      "Counting unique k-mers 41%  \r",
      "Counting unique k-mers 41%  \r",
      "Counting unique k-mers 42%  \r",
      "Counting unique k-mers 43%  \r",
      "Counting unique k-mers 44%  \r",
      "Counting unique k-mers 44%  \r",
      "Counting unique k-mers 45%  \r",
      "Counting unique k-mers 46%  \r",
      "Counting unique k-mers 47%  \r",
      "Counting unique k-mers 47%  \r",
      "Counting unique k-mers 48%  \r",
      "Counting unique k-mers 49%  \r",
      "Counting unique k-mers 50%  \r",
      "Counting unique k-mers 50%  \r",
      "Counting unique k-mers 51%  \r",
      "Counting unique k-mers 52%  \r",
      "Counting unique k-mers 53%  \r",
      "Counting unique k-mers 53%  \r",
      "Counting unique k-mers 54%  \r",
      "Counting unique k-mers 55%  \r",
      "Counting unique k-mers 56%  \r",
      "Counting unique k-mers 56%  \r",
      "Counting unique k-mers 57%  \r",
      "Counting unique k-mers 58%  \r",
      "Counting unique k-mers 59%  \r",
      "Counting unique k-mers 59%  \r",
      "Counting unique k-mers 60%  \r",
      "Counting unique k-mers 61%  \r",
      "Counting unique k-mers 61%  \r",
      "Counting unique k-mers 62%  \r",
      "Counting unique k-mers 63%  \r",
      "Counting unique k-mers 64%  \r",
      "Counting unique k-mers 64%  \r",
      "Counting unique k-mers 65%  \r",
      "Counting unique k-mers 66%  \r",
      "Counting unique k-mers 67%  \r",
      "Counting unique k-mers 67%  \r",
      "Counting unique k-mers 68%  \r",
      "Counting unique k-mers 69%  \r",
      "Counting unique k-mers 70%  \r",
      "Counting unique k-mers 70%  \r",
      "Counting unique k-mers 71%  \r",
      "Counting unique k-mers 72%  \r",
      "Counting unique k-mers 73%  \r",
      "Counting unique k-mers 73%  \r",
      "Counting unique k-mers 74%  \r",
      "Counting unique k-mers 75%  \r",
      "Counting unique k-mers 76%  \r",
      "Counting unique k-mers 76%  \r",
      "Counting unique k-mers 77%  \r",
      "Counting unique k-mers 78%  \r",
      "Counting unique k-mers 79%  \r",
      "Counting unique k-mers 79%  \r",
      "Counting unique k-mers 80%  \r",
      "Counting unique k-mers 81%  \r",
      "Counting unique k-mers 81%  \r",
      "Counting unique k-mers 82%  \r",
      "Counting unique k-mers 83%  \r",
      "Counting unique k-mers 84%  \r",
      "Counting unique k-mers 84%  \r",
      "Counting unique k-mers 85%  \r",
      "Counting unique k-mers 86%  \r",
      "Counting unique k-mers 87%  \r",
      "Counting unique k-mers 87%  \r",
      "Counting unique k-mers 88%  \r",
      "Counting unique k-mers 89%  \r",
      "Counting unique k-mers 90%  \r",
      "Counting unique k-mers 90%  \r",
      "Counting unique k-mers 91%  \r",
      "Counting unique k-mers 92%  \r",
      "Counting unique k-mers 93%  \r",
      "Counting unique k-mers 93%  \r",
      "Counting unique k-mers 94%  \r",
      "Counting unique k-mers 95%  \r",
      "Counting unique k-mers 96%  \r",
      "Counting unique k-mers 96%  \r",
      "Counting unique k-mers 97%  \r",
      "Counting unique k-mers 98%  \r",
      "Counting unique k-mers 99%  \r",
      "Counting unique k-mers 99%  \r",
      "Counting unique k-mers 100%\n",
      "Creating index of unique k-mers 0%  \r",
      "Creating index of unique k-mers 0%  \r",
      "Creating index of unique k-mers 1%  \r",
      "Creating index of unique k-mers 1%  \r",
      "Creating index of unique k-mers 2%  \r",
      "Creating index of unique k-mers 3%  \r",
      "Creating index of unique k-mers 4%  \r",
      "Creating index of unique k-mers 4%  \r",
      "Creating index of unique k-mers 5%  \r",
      "Creating index of unique k-mers 6%  \r",
      "Creating index of unique k-mers 7%  \r",
      "Creating index of unique k-mers 7%  \r",
      "Creating index of unique k-mers 8%  \r",
      "Creating index of unique k-mers 9%  \r",
      "Creating index of unique k-mers 10%  \r",
      "Creating index of unique k-mers 10%  \r",
      "Creating index of unique k-mers 11%  \r",
      "Creating index of unique k-mers 12%  \r",
      "Creating index of unique k-mers 13%  \r",
      "Creating index of unique k-mers 13%  \r",
      "Creating index of unique k-mers 14%  \r",
      "Creating index of unique k-mers 15%  \r",
      "Creating index of unique k-mers 16%  \r",
      "Creating index of unique k-mers 16%  \r",
      "Creating index of unique k-mers 17%  \r",
      "Creating index of unique k-mers 18%  \r",
      "Creating index of unique k-mers 19%  \r",
      "Creating index of unique k-mers 19%  \r",
      "Creating index of unique k-mers 20%  \r",
      "Creating index of unique k-mers 21%  \r",
      "Creating index of unique k-mers 21%  \r",
      "Creating index of unique k-mers 22%  \r",
      "Creating index of unique k-mers 23%  \r",
      "Creating index of unique k-mers 24%  \r",
      "Creating index of unique k-mers 24%  \r",
      "Creating index of unique k-mers 25%  \r",
      "Creating index of unique k-mers 26%  \r",
      "Creating index of unique k-mers 27%  \r",
      "Creating index of unique k-mers 27%  \r",
      "Creating index of unique k-mers 28%  \r",
      "Creating index of unique k-mers 29%  \r",
      "Creating index of unique k-mers 30%  \r",
      "Creating index of unique k-mers 30%  \r",
      "Creating index of unique k-mers 31%  \r",
      "Creating index of unique k-mers 32%  \r",
      "Creating index of unique k-mers 33%  \r",
      "Creating index of unique k-mers 33%  \r",
      "Creating index of unique k-mers 34%  \r",
      "Creating index of unique k-mers 35%  \r",
      "Creating index of unique k-mers 36%  \r",
      "Creating index of unique k-mers 36%  \r",
      "Creating index of unique k-mers 37%  \r",
      "Creating index of unique k-mers 38%  \r",
      "Creating index of unique k-mers 39%  \r",
      "Creating index of unique k-mers 39%  \r",
      "Creating index of unique k-mers 40%  \r",
      "Creating index of unique k-mers 41%  \r",
      "Creating index of unique k-mers 41%  \r",
      "Creating index of unique k-mers 42%  \r",
      "Creating index of unique k-mers 43%  \r",
      "Creating index of unique k-mers 44%  \r",
      "Creating index of unique k-mers 44%  \r",
      "Creating index of unique k-mers 45%  \r",
      "Creating index of unique k-mers 46%  \r",
      "Creating index of unique k-mers 47%  \r",
      "Creating index of unique k-mers 47%  \r",
      "Creating index of unique k-mers 48%  \r",
      "Creating index of unique k-mers 49%  \r",
      "Creating index of unique k-mers 50%  \r",
      "Creating index of unique k-mers 50%  \r",
      "Creating index of unique k-mers 51%  \r",
      "Creating index of unique k-mers 52%  \r",
      "Creating index of unique k-mers 53%  \r",
      "Creating index of unique k-mers 53%  \r",
      "Creating index of unique k-mers 54%  \r",
      "Creating index of unique k-mers 55%  \r",
      "Creating index of unique k-mers 56%  \r",
      "Creating index of unique k-mers 56%  \r",
      "Creating index of unique k-mers 57%  \r",
      "Creating index of unique k-mers 58%  \r",
      "Creating index of unique k-mers 59%  \r",
      "Creating index of unique k-mers 59%  \r",
      "Creating index of unique k-mers 60%  \r",
      "Creating index of unique k-mers 61%  \r",
      "Creating index of unique k-mers 61%  \r",
      "Creating index of unique k-mers 62%  \r",
      "Creating index of unique k-mers 63%  \r",
      "Creating index of unique k-mers 64%  \r",
      "Creating index of unique k-mers 64%  \r",
      "Creating index of unique k-mers 65%  \r",
      "Creating index of unique k-mers 66%  \r",
      "Creating index of unique k-mers 67%  \r",
      "Creating index of unique k-mers 67%  \r",
      "Creating index of unique k-mers 68%  \r",
      "Creating index of unique k-mers 69%  \r",
      "Creating index of unique k-mers 70%  \r",
      "Creating index of unique k-mers 70%  \r",
      "Creating index of unique k-mers 71%  \r",
      "Creating index of unique k-mers 72%  \r",
      "Creating index of unique k-mers 73%  \r",
      "Creating index of unique k-mers 73%  \r",
      "Creating index of unique k-mers 74%  \r",
      "Creating index of unique k-mers 75%  \r",
      "Creating index of unique k-mers 76%  \r",
      "Creating index of unique k-mers 76%  \r",
      "Creating index of unique k-mers 77%  \r",
      "Creating index of unique k-mers 78%  \r",
      "Creating index of unique k-mers 79%  \r",
      "Creating index of unique k-mers 79%  \r",
      "Creating index of unique k-mers 80%  \r",
      "Creating index of unique k-mers 81%  \r",
      "Creating index of unique k-mers 81%  \r",
      "Creating index of unique k-mers 82%  \r",
      "Creating index of unique k-mers 83%  \r",
      "Creating index of unique k-mers 84%  \r",
      "Creating index of unique k-mers 84%  \r",
      "Creating index of unique k-mers 85%  \r",
      "Creating index of unique k-mers 86%  \r",
      "Creating index of unique k-mers 87%  \r",
      "Creating index of unique k-mers 87%  \r",
      "Creating index of unique k-mers 88%  \r",
      "Creating index of unique k-mers 89%  \r",
      "Creating index of unique k-mers 90%  \r",
      "Creating index of unique k-mers 90%  \r",
      "Creating index of unique k-mers 91%  \r",
      "Creating index of unique k-mers 92%  \r",
      "Creating index of unique k-mers 93%  \r",
      "Creating index of unique k-mers 93%  \r",
      "Creating index of unique k-mers 94%  \r",
      "Creating index of unique k-mers 95%  \r",
      "Creating index of unique k-mers 96%  \r",
      "Creating index of unique k-mers 96%  \r",
      "Creating index of unique k-mers 97%  \r",
      "Creating index of unique k-mers 98%  \r",
      "Creating index of unique k-mers 99%  \r",
      "Creating index of unique k-mers 99%  \r",
      "Creating index of unique k-mers 100%\n",
      "Searching 0%  \r",
      "Searching 5%  \r",
      "Searching 8%  \r",
      "Searching 10%  \r",
      "Searching 16%  \r",
      "Searching 17%  \r",
      "Searching 19%  \r",
      "Searching 25%  \r",
      "Searching 28%  \r",
      "Searching 30%  \r",
      "Searching 31%  \r",
      "Searching 33%  \r",
      "Searching 36%  \r",
      "Searching 38%  \r",
      "Searching 41%  \r",
      "Searching 42%  \r",
      "Searching 46%  \r",
      "Searching 47%  \r",
      "Searching 49%  \r",
      "Searching 53%  \r",
      "Searching 56%  \r",
      "Searching 60%  \r",
      "Searching 64%  \r",
      "Searching 69%  \r",
      "Searching 72%  \r",
      "Searching 74%  \r",
      "Searching 75%  \r",
      "Searching 78%  \r",
      "Searching 80%  \r",
      "Searching 83%  \r",
      "Searching 86%  \r",
      "Searching 91%  \r",
      "Searching 92%  \r",
      "Searching 95%  \r",
      "Searching 97%  \r",
      "Searching 99%  \r",
      "Searching 100%  \r",
      "Searching 100%\n",
      "Matching query sequences: 64 of 64 (100.00%)\n",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 0%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 1%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 1%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 2%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 2%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 3%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 3%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 4%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 4%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 5%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 5%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 6%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 6%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 7%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 7%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 8%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 9%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 9%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 10%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 10%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 11%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 11%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 12%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 12%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 13%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 13%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 14%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 14%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 15%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 15%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 16%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 16%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 17%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 18%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 18%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 19%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 19%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 20%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 20%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 21%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 21%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 22%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 22%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 23%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 23%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 24%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 24%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 25%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 25%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 26%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 27%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 27%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 28%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 28%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 29%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 29%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 30%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 30%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 31%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 31%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 32%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 32%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 33%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 33%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 34%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 34%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 35%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 35%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 36%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 37%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 37%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 38%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 38%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 39%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 39%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 40%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 40%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 41%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 41%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 42%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 42%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 43%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 43%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 44%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 44%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 45%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 45%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 46%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 47%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 47%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 48%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 48%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 49%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 49%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 50%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 50%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 51%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 51%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 52%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 52%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 53%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 53%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 54%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 54%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 55%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 55%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 56%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 57%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 57%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 58%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 58%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 59%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 59%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 60%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 60%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 61%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 61%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 62%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 62%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 63%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 63%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 64%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 64%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 65%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 65%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 66%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 66%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 67%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 68%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 68%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 69%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 69%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 70%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 70%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 71%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 71%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 72%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 72%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 73%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 73%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 74%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 74%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 75%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 75%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 76%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 76%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 77%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 77%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 78%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 78%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 79%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 80%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 80%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 81%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 81%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 82%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 82%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 83%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 83%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 84%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 84%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 85%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 85%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 86%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 86%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 87%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 87%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 88%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 88%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 89%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 90%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 90%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 91%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 91%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 92%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 92%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 93%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 93%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 94%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 94%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 95%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 95%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 96%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 96%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 97%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 97%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 98%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 98%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 99%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 100%  \r",
      "Reading file Crangonyx_pseudogracilis_0_OTU.clipped.fasta 100%\n",
      "59167 nt in 190 seqs, min 305, max 314, avg 311\n",
      "Indexing sequences 0%  \r",
      "Indexing sequences 0%  \r",
      "Indexing sequences 1%  \r",
      "Indexing sequences 1%  \r",
      "Indexing sequences 2%  \r",
      "Indexing sequences 2%  \r",
      "Indexing sequences 3%  \r",
      "Indexing sequences 3%  \r",
      "Indexing sequences 4%  \r",
      "Indexing sequences 4%  \r",
      "Indexing sequences 5%  \r",
      "Indexing sequences 5%  \r",
      "Indexing sequences 6%  \r",
      "Indexing sequences 6%  \r",
      "Indexing sequences 7%  \r",
      "Indexing sequences 7%  \r",
      "Indexing sequences 8%  \r",
      "Indexing sequences 8%  \r",
      "Indexing sequences 9%  \r",
      "Indexing sequences 9%  \r",
      "Indexing sequences 10%  \r",
      "Indexing sequences 11%  \r",
      "Indexing sequences 11%  \r",
      "Indexing sequences 12%  \r",
      "Indexing sequences 12%  \r",
      "Indexing sequences 13%  \r",
      "Indexing sequences 13%  \r",
      "Indexing sequences 14%  \r",
      "Indexing sequences 14%  \r",
      "Indexing sequences 15%  \r",
      "Indexing sequences 15%  \r",
      "Indexing sequences 16%  \r",
      "Indexing sequences 16%  \r",
      "Indexing sequences 17%  \r",
      "Indexing sequences 17%  \r",
      "Indexing sequences 18%  \r",
      "Indexing sequences 18%  \r",
      "Indexing sequences 19%  \r",
      "Indexing sequences 19%  \r",
      "Indexing sequences 20%  \r",
      "Indexing sequences 21%  \r",
      "Indexing sequences 21%  \r",
      "Indexing sequences 22%  \r",
      "Indexing sequences 22%  \r",
      "Indexing sequences 23%  \r",
      "Indexing sequences 23%  \r",
      "Indexing sequences 24%  \r",
      "Indexing sequences 24%  \r",
      "Indexing sequences 25%  \r",
      "Indexing sequences 25%  \r",
      "Indexing sequences 26%  \r",
      "Indexing sequences 26%  \r",
      "Indexing sequences 27%  \r",
      "Indexing sequences 27%  \r",
      "Indexing sequences 28%  \r",
      "Indexing sequences 28%  \r",
      "Indexing sequences 29%  \r",
      "Indexing sequences 29%  \r",
      "Indexing sequences 30%  \r",
      "Indexing sequences 31%  \r",
      "Indexing sequences 31%  \r",
      "Indexing sequences 32%  \r",
      "Indexing sequences 32%  \r",
      "Indexing sequences 33%  \r",
      "Indexing sequences 33%  \r",
      "Indexing sequences 34%  \r",
      "Indexing sequences 34%  \r",
      "Indexing sequences 35%  \r",
      "Indexing sequences 35%  \r",
      "Indexing sequences 36%  \r",
      "Indexing sequences 36%  \r",
      "Indexing sequences 37%  \r",
      "Indexing sequences 37%  \r",
      "Indexing sequences 38%  \r",
      "Indexing sequences 38%  \r",
      "Indexing sequences 39%  \r",
      "Indexing sequences 39%  \r",
      "Indexing sequences 40%  \r",
      "Indexing sequences 41%  \r",
      "Indexing sequences 41%  \r",
      "Indexing sequences 42%  \r",
      "Indexing sequences 42%  \r",
      "Indexing sequences 43%  \r",
      "Indexing sequences 43%  \r",
      "Indexing sequences 44%  \r",
      "Indexing sequences 44%  \r",
      "Indexing sequences 45%  \r",
      "Indexing sequences 45%  \r",
      "Indexing sequences 46%  \r",
      "Indexing sequences 46%  \r",
      "Indexing sequences 47%  \r",
      "Indexing sequences 47%  \r",
      "Indexing sequences 48%  \r",
      "Indexing sequences 48%  \r",
      "Indexing sequences 49%  \r",
      "Indexing sequences 49%  \r",
      "Indexing sequences 50%  \r",
      "Indexing sequences 51%  \r",
      "Indexing sequences 51%  \r",
      "Indexing sequences 52%  \r",
      "Indexing sequences 52%  \r",
      "Indexing sequences 53%  \r",
      "Indexing sequences 53%  \r",
      "Indexing sequences 54%  \r",
      "Indexing sequences 54%  \r",
      "Indexing sequences 55%  \r",
      "Indexing sequences 55%  \r",
      "Indexing sequences 56%  \r",
      "Indexing sequences 56%  \r",
      "Indexing sequences 57%  \r",
      "Indexing sequences 57%  \r",
      "Indexing sequences 58%  \r",
      "Indexing sequences 58%  \r",
      "Indexing sequences 59%  \r",
      "Indexing sequences 59%  \r",
      "Indexing sequences 60%  \r",
      "Indexing sequences 61%  \r",
      "Indexing sequences 61%  \r",
      "Indexing sequences 62%  \r",
      "Indexing sequences 62%  \r",
      "Indexing sequences 63%  \r",
      "Indexing sequences 63%  \r",
      "Indexing sequences 64%  \r",
      "Indexing sequences 64%  \r",
      "Indexing sequences 65%  \r",
      "Indexing sequences 65%  \r",
      "Indexing sequences 66%  \r",
      "Indexing sequences 66%  \r",
      "Indexing sequences 67%  \r",
      "Indexing sequences 67%  \r",
      "Indexing sequences 68%  \r",
      "Indexing sequences 68%  \r",
      "Indexing sequences 69%  \r",
      "Indexing sequences 69%  \r",
      "Indexing sequences 70%  \r",
      "Indexing sequences 71%  \r",
      "Indexing sequences 71%  \r",
      "Indexing sequences 72%  \r",
      "Indexing sequences 72%  \r",
      "Indexing sequences 73%  \r",
      "Indexing sequences 73%  \r",
      "Indexing sequences 74%  \r",
      "Indexing sequences 74%  \r",
      "Indexing sequences 75%  \r",
      "Indexing sequences 75%  \r",
      "Indexing sequences 76%  \r",
      "Indexing sequences 76%  \r",
      "Indexing sequences 77%  \r",
      "Indexing sequences 77%  \r",
      "Indexing sequences 78%  \r",
      "Indexing sequences 78%  \r",
      "Indexing sequences 79%  \r",
      "Indexing sequences 79%  \r",
      "Indexing sequences 80%  \r",
      "Indexing sequences 81%  \r",
      "Indexing sequences 81%  \r",
      "Indexing sequences 82%  \r",
      "Indexing sequences 82%  \r",
      "Indexing sequences 83%  \r",
      "Indexing sequences 83%  \r",
      "Indexing sequences 84%  \r",
      "Indexing sequences 84%  \r",
      "Indexing sequences 85%  \r",
      "Indexing sequences 85%  \r",
      "Indexing sequences 86%  \r",
      "Indexing sequences 86%  \r",
      "Indexing sequences 87%  \r",
      "Indexing sequences 87%  \r",
      "Indexing sequences 88%  \r",
      "Indexing sequences 88%  \r",
      "Indexing sequences 89%  \r",
      "Indexing sequences 89%  \r",
      "Indexing sequences 90%  \r",
      "Indexing sequences 91%  \r",
      "Indexing sequences 91%  \r",
      "Indexing sequences 92%  \r",
      "Indexing sequences 92%  \r",
      "Indexing sequences 93%  \r",
      "Indexing sequences 93%  \r",
      "Indexing sequences 94%  \r",
      "Indexing sequences 94%  \r",
      "Indexing sequences 95%  \r",
      "Indexing sequences 95%  \r",
      "Indexing sequences 96%  \r",
      "Indexing sequences 96%  \r",
      "Indexing sequences 97%  \r",
      "Indexing sequences 97%  \r",
      "Indexing sequences 98%  \r",
      "Indexing sequences 98%  \r",
      "Indexing sequences 99%  \r",
      "Indexing sequences 99%  \r",
      "Indexing sequences 100%\n",
      "Masking 0%  \r",
      "Masking 100%\n",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 1%  \r",
      "Counting unique k-mers 1%  \r",
      "Counting unique k-mers 2%  \r",
      "Counting unique k-mers 2%  \r",
      "Counting unique k-mers 3%  \r",
      "Counting unique k-mers 3%  \r",
      "Counting unique k-mers 4%  \r",
      "Counting unique k-mers 4%  \r",
      "Counting unique k-mers 5%  \r",
      "Counting unique k-mers 5%  \r",
      "Counting unique k-mers 6%  \r",
      "Counting unique k-mers 6%  \r",
      "Counting unique k-mers 7%  \r",
      "Counting unique k-mers 7%  \r",
      "Counting unique k-mers 8%  \r",
      "Counting unique k-mers 8%  \r",
      "Counting unique k-mers 9%  \r",
      "Counting unique k-mers 9%  \r",
      "Counting unique k-mers 10%  \r",
      "Counting unique k-mers 11%  \r",
      "Counting unique k-mers 11%  \r",
      "Counting unique k-mers 12%  \r",
      "Counting unique k-mers 12%  \r",
      "Counting unique k-mers 13%  \r",
      "Counting unique k-mers 13%  \r",
      "Counting unique k-mers 14%  \r",
      "Counting unique k-mers 14%  \r",
      "Counting unique k-mers 15%  \r",
      "Counting unique k-mers 15%  \r",
      "Counting unique k-mers 16%  \r",
      "Counting unique k-mers 16%  \r",
      "Counting unique k-mers 17%  \r",
      "Counting unique k-mers 17%  \r",
      "Counting unique k-mers 18%  \r",
      "Counting unique k-mers 18%  \r",
      "Counting unique k-mers 19%  \r",
      "Counting unique k-mers 19%  \r",
      "Counting unique k-mers 20%  \r",
      "Counting unique k-mers 21%  \r",
      "Counting unique k-mers 21%  \r",
      "Counting unique k-mers 22%  \r",
      "Counting unique k-mers 22%  \r",
      "Counting unique k-mers 23%  \r",
      "Counting unique k-mers 23%  \r",
      "Counting unique k-mers 24%  \r",
      "Counting unique k-mers 24%  \r",
      "Counting unique k-mers 25%  \r",
      "Counting unique k-mers 25%  \r",
      "Counting unique k-mers 26%  \r",
      "Counting unique k-mers 26%  \r",
      "Counting unique k-mers 27%  \r",
      "Counting unique k-mers 27%  \r",
      "Counting unique k-mers 28%  \r",
      "Counting unique k-mers 28%  \r",
      "Counting unique k-mers 29%  \r",
      "Counting unique k-mers 29%  \r",
      "Counting unique k-mers 30%  \r",
      "Counting unique k-mers 31%  \r",
      "Counting unique k-mers 31%  \r",
      "Counting unique k-mers 32%  \r",
      "Counting unique k-mers 32%  \r",
      "Counting unique k-mers 33%  \r",
      "Counting unique k-mers 33%  \r",
      "Counting unique k-mers 34%  \r",
      "Counting unique k-mers 34%  \r",
      "Counting unique k-mers 35%  \r",
      "Counting unique k-mers 35%  \r",
      "Counting unique k-mers 36%  \r",
      "Counting unique k-mers 36%  \r",
      "Counting unique k-mers 37%  \r",
      "Counting unique k-mers 37%  \r",
      "Counting unique k-mers 38%  \r",
      "Counting unique k-mers 38%  \r",
      "Counting unique k-mers 39%  \r",
      "Counting unique k-mers 39%  \r",
      "Counting unique k-mers 40%  \r",
      "Counting unique k-mers 41%  \r",
      "Counting unique k-mers 41%  \r",
      "Counting unique k-mers 42%  \r",
      "Counting unique k-mers 42%  \r",
      "Counting unique k-mers 43%  \r",
      "Counting unique k-mers 43%  \r",
      "Counting unique k-mers 44%  \r",
      "Counting unique k-mers 44%  \r",
      "Counting unique k-mers 45%  \r",
      "Counting unique k-mers 45%  \r",
      "Counting unique k-mers 46%  \r",
      "Counting unique k-mers 46%  \r",
      "Counting unique k-mers 47%  \r",
      "Counting unique k-mers 47%  \r",
      "Counting unique k-mers 48%  \r",
      "Counting unique k-mers 48%  \r",
      "Counting unique k-mers 49%  \r",
      "Counting unique k-mers 49%  \r",
      "Counting unique k-mers 50%  \r",
      "Counting unique k-mers 51%  \r",
      "Counting unique k-mers 51%  \r",
      "Counting unique k-mers 52%  \r",
      "Counting unique k-mers 52%  \r",
      "Counting unique k-mers 53%  \r",
      "Counting unique k-mers 53%  \r",
      "Counting unique k-mers 54%  \r",
      "Counting unique k-mers 54%  \r",
      "Counting unique k-mers 55%  \r",
      "Counting unique k-mers 55%  \r",
      "Counting unique k-mers 56%  \r",
      "Counting unique k-mers 56%  \r",
      "Counting unique k-mers 57%  \r",
      "Counting unique k-mers 57%  \r",
      "Counting unique k-mers 58%  \r",
      "Counting unique k-mers 58%  \r",
      "Counting unique k-mers 59%  \r",
      "Counting unique k-mers 59%  \r",
      "Counting unique k-mers 60%  \r",
      "Counting unique k-mers 61%  \r",
      "Counting unique k-mers 61%  \r",
      "Counting unique k-mers 62%  \r",
      "Counting unique k-mers 62%  \r",
      "Counting unique k-mers 63%  \r",
      "Counting unique k-mers 63%  \r",
      "Counting unique k-mers 64%  \r",
      "Counting unique k-mers 64%  \r",
      "Counting unique k-mers 65%  \r",
      "Counting unique k-mers 65%  \r",
      "Counting unique k-mers 66%  \r",
      "Counting unique k-mers 66%  \r",
      "Counting unique k-mers 67%  \r",
      "Counting unique k-mers 67%  \r",
      "Counting unique k-mers 68%  \r",
      "Counting unique k-mers 68%  \r",
      "Counting unique k-mers 69%  \r",
      "Counting unique k-mers 69%  \r",
      "Counting unique k-mers 70%  \r",
      "Counting unique k-mers 71%  \r",
      "Counting unique k-mers 71%  \r",
      "Counting unique k-mers 72%  \r",
      "Counting unique k-mers 72%  \r",
      "Counting unique k-mers 73%  \r",
      "Counting unique k-mers 73%  \r",
      "Counting unique k-mers 74%  \r",
      "Counting unique k-mers 74%  \r",
      "Counting unique k-mers 75%  \r",
      "Counting unique k-mers 75%  \r",
      "Counting unique k-mers 76%  \r",
      "Counting unique k-mers 76%  \r",
      "Counting unique k-mers 77%  \r",
      "Counting unique k-mers 77%  \r",
      "Counting unique k-mers 78%  \r",
      "Counting unique k-mers 78%  \r",
      "Counting unique k-mers 79%  \r",
      "Counting unique k-mers 79%  \r",
      "Counting unique k-mers 80%  \r",
      "Counting unique k-mers 81%  \r",
      "Counting unique k-mers 81%  \r",
      "Counting unique k-mers 82%  \r",
      "Counting unique k-mers 82%  \r",
      "Counting unique k-mers 83%  \r",
      "Counting unique k-mers 83%  \r",
      "Counting unique k-mers 84%  \r",
      "Counting unique k-mers 84%  \r",
      "Counting unique k-mers 85%  \r",
      "Counting unique k-mers 85%  \r",
      "Counting unique k-mers 86%  \r",
      "Counting unique k-mers 86%  \r",
      "Counting unique k-mers 87%  \r",
      "Counting unique k-mers 87%  \r",
      "Counting unique k-mers 88%  \r",
      "Counting unique k-mers 88%  \r",
      "Counting unique k-mers 89%  \r",
      "Counting unique k-mers 89%  \r",
      "Counting unique k-mers 90%  \r",
      "Counting unique k-mers 91%  \r",
      "Counting unique k-mers 91%  \r",
      "Counting unique k-mers 92%  \r",
      "Counting unique k-mers 92%  \r",
      "Counting unique k-mers 93%  \r",
      "Counting unique k-mers 93%  \r",
      "Counting unique k-mers 94%  \r",
      "Counting unique k-mers 94%  \r",
      "Counting unique k-mers 95%  \r",
      "Counting unique k-mers 95%  \r",
      "Counting unique k-mers 96%  \r",
      "Counting unique k-mers 96%  \r",
      "Counting unique k-mers 97%  \r",
      "Counting unique k-mers 97%  \r",
      "Counting unique k-mers 98%  \r",
      "Counting unique k-mers 98%  \r",
      "Counting unique k-mers 99%  \r",
      "Counting unique k-mers 99%  \r",
      "Counting unique k-mers 100%\n",
      "Creating index of unique k-mers 0%  \r",
      "Creating index of unique k-mers 0%  \r",
      "Creating index of unique k-mers 1%  \r",
      "Creating index of unique k-mers 1%  \r",
      "Creating index of unique k-mers 2%  \r",
      "Creating index of unique k-mers 2%  \r",
      "Creating index of unique k-mers 3%  \r",
      "Creating index of unique k-mers 3%  \r",
      "Creating index of unique k-mers 4%  \r",
      "Creating index of unique k-mers 4%  \r",
      "Creating index of unique k-mers 5%  \r",
      "Creating index of unique k-mers 5%  \r",
      "Creating index of unique k-mers 6%  \r",
      "Creating index of unique k-mers 6%  \r",
      "Creating index of unique k-mers 7%  \r",
      "Creating index of unique k-mers 7%  \r",
      "Creating index of unique k-mers 8%  \r",
      "Creating index of unique k-mers 8%  \r",
      "Creating index of unique k-mers 9%  \r",
      "Creating index of unique k-mers 9%  \r",
      "Creating index of unique k-mers 10%  \r",
      "Creating index of unique k-mers 11%  \r",
      "Creating index of unique k-mers 11%  \r",
      "Creating index of unique k-mers 12%  \r",
      "Creating index of unique k-mers 12%  \r",
      "Creating index of unique k-mers 13%  \r",
      "Creating index of unique k-mers 13%  \r",
      "Creating index of unique k-mers 14%  \r",
      "Creating index of unique k-mers 14%  \r",
      "Creating index of unique k-mers 15%  \r",
      "Creating index of unique k-mers 15%  \r",
      "Creating index of unique k-mers 16%  \r",
      "Creating index of unique k-mers 16%  \r",
      "Creating index of unique k-mers 17%  \r",
      "Creating index of unique k-mers 17%  \r",
      "Creating index of unique k-mers 18%  \r",
      "Creating index of unique k-mers 18%  \r",
      "Creating index of unique k-mers 19%  \r",
      "Creating index of unique k-mers 19%  \r",
      "Creating index of unique k-mers 20%  \r",
      "Creating index of unique k-mers 21%  \r",
      "Creating index of unique k-mers 21%  \r",
      "Creating index of unique k-mers 22%  \r",
      "Creating index of unique k-mers 22%  \r",
      "Creating index of unique k-mers 23%  \r",
      "Creating index of unique k-mers 23%  \r",
      "Creating index of unique k-mers 24%  \r",
      "Creating index of unique k-mers 24%  \r",
      "Creating index of unique k-mers 25%  \r",
      "Creating index of unique k-mers 25%  \r",
      "Creating index of unique k-mers 26%  \r",
      "Creating index of unique k-mers 26%  \r",
      "Creating index of unique k-mers 27%  \r",
      "Creating index of unique k-mers 27%  \r",
      "Creating index of unique k-mers 28%  \r",
      "Creating index of unique k-mers 28%  \r",
      "Creating index of unique k-mers 29%  \r",
      "Creating index of unique k-mers 29%  \r",
      "Creating index of unique k-mers 30%  \r",
      "Creating index of unique k-mers 31%  \r",
      "Creating index of unique k-mers 31%  \r",
      "Creating index of unique k-mers 32%  \r",
      "Creating index of unique k-mers 32%  \r",
      "Creating index of unique k-mers 33%  \r",
      "Creating index of unique k-mers 33%  \r",
      "Creating index of unique k-mers 34%  \r",
      "Creating index of unique k-mers 34%  \r",
      "Creating index of unique k-mers 35%  \r",
      "Creating index of unique k-mers 35%  \r",
      "Creating index of unique k-mers 36%  \r",
      "Creating index of unique k-mers 36%  \r",
      "Creating index of unique k-mers 37%  \r",
      "Creating index of unique k-mers 37%  \r",
      "Creating index of unique k-mers 38%  \r",
      "Creating index of unique k-mers 38%  \r",
      "Creating index of unique k-mers 39%  \r",
      "Creating index of unique k-mers 39%  \r",
      "Creating index of unique k-mers 40%  \r",
      "Creating index of unique k-mers 41%  \r",
      "Creating index of unique k-mers 41%  \r",
      "Creating index of unique k-mers 42%  \r",
      "Creating index of unique k-mers 42%  \r",
      "Creating index of unique k-mers 43%  \r",
      "Creating index of unique k-mers 43%  \r",
      "Creating index of unique k-mers 44%  \r",
      "Creating index of unique k-mers 44%  \r",
      "Creating index of unique k-mers 45%  \r",
      "Creating index of unique k-mers 45%  \r",
      "Creating index of unique k-mers 46%  \r",
      "Creating index of unique k-mers 46%  \r",
      "Creating index of unique k-mers 47%  \r",
      "Creating index of unique k-mers 47%  \r",
      "Creating index of unique k-mers 48%  \r",
      "Creating index of unique k-mers 48%  \r",
      "Creating index of unique k-mers 49%  \r",
      "Creating index of unique k-mers 49%  \r",
      "Creating index of unique k-mers 50%  \r",
      "Creating index of unique k-mers 51%  \r",
      "Creating index of unique k-mers 51%  \r",
      "Creating index of unique k-mers 52%  \r",
      "Creating index of unique k-mers 52%  \r",
      "Creating index of unique k-mers 53%  \r",
      "Creating index of unique k-mers 53%  \r",
      "Creating index of unique k-mers 54%  \r",
      "Creating index of unique k-mers 54%  \r",
      "Creating index of unique k-mers 55%  \r",
      "Creating index of unique k-mers 55%  \r",
      "Creating index of unique k-mers 56%  \r",
      "Creating index of unique k-mers 56%  \r",
      "Creating index of unique k-mers 57%  \r",
      "Creating index of unique k-mers 57%  \r",
      "Creating index of unique k-mers 58%  \r",
      "Creating index of unique k-mers 58%  \r",
      "Creating index of unique k-mers 59%  \r",
      "Creating index of unique k-mers 59%  \r",
      "Creating index of unique k-mers 60%  \r",
      "Creating index of unique k-mers 61%  \r",
      "Creating index of unique k-mers 61%  \r",
      "Creating index of unique k-mers 62%  \r",
      "Creating index of unique k-mers 62%  \r",
      "Creating index of unique k-mers 63%  \r",
      "Creating index of unique k-mers 63%  \r",
      "Creating index of unique k-mers 64%  \r",
      "Creating index of unique k-mers 64%  \r",
      "Creating index of unique k-mers 65%  \r",
      "Creating index of unique k-mers 65%  \r",
      "Creating index of unique k-mers 66%  \r",
      "Creating index of unique k-mers 66%  \r",
      "Creating index of unique k-mers 67%  \r",
      "Creating index of unique k-mers 67%  \r",
      "Creating index of unique k-mers 68%  \r",
      "Creating index of unique k-mers 68%  \r",
      "Creating index of unique k-mers 69%  \r",
      "Creating index of unique k-mers 69%  \r",
      "Creating index of unique k-mers 70%  \r",
      "Creating index of unique k-mers 71%  \r",
      "Creating index of unique k-mers 71%  \r",
      "Creating index of unique k-mers 72%  \r",
      "Creating index of unique k-mers 72%  \r",
      "Creating index of unique k-mers 73%  \r",
      "Creating index of unique k-mers 73%  \r",
      "Creating index of unique k-mers 74%  \r",
      "Creating index of unique k-mers 74%  \r",
      "Creating index of unique k-mers 75%  \r",
      "Creating index of unique k-mers 75%  \r",
      "Creating index of unique k-mers 76%  \r",
      "Creating index of unique k-mers 76%  \r",
      "Creating index of unique k-mers 77%  \r",
      "Creating index of unique k-mers 77%  \r",
      "Creating index of unique k-mers 78%  \r",
      "Creating index of unique k-mers 78%  \r",
      "Creating index of unique k-mers 79%  \r",
      "Creating index of unique k-mers 79%  \r",
      "Creating index of unique k-mers 80%  \r",
      "Creating index of unique k-mers 81%  \r",
      "Creating index of unique k-mers 81%  \r",
      "Creating index of unique k-mers 82%  \r",
      "Creating index of unique k-mers 82%  \r",
      "Creating index of unique k-mers 83%  \r",
      "Creating index of unique k-mers 83%  \r",
      "Creating index of unique k-mers 84%  \r",
      "Creating index of unique k-mers 84%  \r",
      "Creating index of unique k-mers 85%  \r",
      "Creating index of unique k-mers 85%  \r",
      "Creating index of unique k-mers 86%  \r",
      "Creating index of unique k-mers 86%  \r",
      "Creating index of unique k-mers 87%  \r",
      "Creating index of unique k-mers 87%  \r",
      "Creating index of unique k-mers 88%  \r",
      "Creating index of unique k-mers 88%  \r",
      "Creating index of unique k-mers 89%  \r",
      "Creating index of unique k-mers 89%  \r",
      "Creating index of unique k-mers 90%  \r",
      "Creating index of unique k-mers 91%  \r",
      "Creating index of unique k-mers 91%  \r",
      "Creating index of unique k-mers 92%  \r",
      "Creating index of unique k-mers 92%  \r",
      "Creating index of unique k-mers 93%  \r",
      "Creating index of unique k-mers 93%  \r",
      "Creating index of unique k-mers 94%  \r",
      "Creating index of unique k-mers 94%  \r",
      "Creating index of unique k-mers 95%  \r",
      "Creating index of unique k-mers 95%  \r",
      "Creating index of unique k-mers 96%  \r",
      "Creating index of unique k-mers 96%  \r",
      "Creating index of unique k-mers 97%  \r",
      "Creating index of unique k-mers 97%  \r",
      "Creating index of unique k-mers 98%  \r",
      "Creating index of unique k-mers 98%  \r",
      "Creating index of unique k-mers 99%  \r",
      "Creating index of unique k-mers 99%  \r",
      "Creating index of unique k-mers 100%\n",
      "Searching 0%  \r",
      "Searching 11%  \r",
      "Searching 16%  \r",
      "Searching 19%  \r",
      "Searching 21%  \r",
      "Searching 25%  \r",
      "Searching 28%  \r",
      "Searching 30%  \r",
      "Searching 35%  \r",
      "Searching 41%  \r",
      "Searching 44%  \r",
      "Searching 46%  \r",
      "Searching 53%  \r",
      "Searching 55%  \r",
      "Searching 58%  \r",
      "Searching 60%  \r",
      "Searching 70%  \r",
      "Searching 77%  \r",
      "Searching 81%  \r",
      "Searching 91%  \r",
      "Searching 93%  \r",
      "Searching 100%  \r",
      "Searching 100%\n",
      "Matching query sequences: 57 of 57 (100.00%)\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for s in $(ls -1 *_hts.nr.fasta)\n",
    "do\n",
    "    prefix=$(echo -e \"$s\" | cut -d \".\" -f 1 | sed 's/_hts//')\n",
    "    vsearch --usearch_global $s  \\\n",
    "    --strand both \\\n",
    "    --db $prefix.clipped.fasta \\\n",
    "    --id 0.97 --query_cov 1 --maxaccepts 100000 \\\n",
    "    --blast6out $prefix.vs.full.blast.out \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove redundancy from Sanger sequences produced with Folmer primers."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 0%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 8%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 17%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 25%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 33%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 40%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 48%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 56%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 64%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 71%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 79%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 87%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 95%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 100%  \r",
      "Reading file ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta 100%\n",
      "8081 nt in 13 seqs, min 406, max 658, avg 622\n",
      "Indexing sequences 0%  \r",
      "Indexing sequences 0%  \r",
      "Indexing sequences 8%  \r",
      "Indexing sequences 15%  \r",
      "Indexing sequences 23%  \r",
      "Indexing sequences 31%  \r",
      "Indexing sequences 38%  \r",
      "Indexing sequences 46%  \r",
      "Indexing sequences 54%  \r",
      "Indexing sequences 62%  \r",
      "Indexing sequences 69%  \r",
      "Indexing sequences 77%  \r",
      "Indexing sequences 85%  \r",
      "Indexing sequences 92%  \r",
      "Indexing sequences 100%\n",
      "Masking 0%  \r",
      "Masking 100%\n",
      "Sorting by length 0%  \r",
      "Sorting by length 100%\n",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 0%  \r",
      "Counting unique k-mers 8%  \r",
      "Counting unique k-mers 15%  \r",
      "Counting unique k-mers 23%  \r",
      "Counting unique k-mers 31%  \r",
      "Counting unique k-mers 38%  \r",
      "Counting unique k-mers 46%  \r",
      "Counting unique k-mers 54%  \r",
      "Counting unique k-mers 62%  \r",
      "Counting unique k-mers 69%  \r",
      "Counting unique k-mers 77%  \r",
      "Counting unique k-mers 85%  \r",
      "Counting unique k-mers 92%  \r",
      "Counting unique k-mers 100%\n",
      "Clustering 0%  \r",
      "Clustering 24%  \r",
      "Clustering 49%  \r",
      "Clustering 73%  \r",
      "Clustering 95%  \r",
      "Clustering 100%  \r",
      "Clustering 100%\n",
      "Writing clusters 0%  \r",
      "Writing clusters 0%  \r",
      "Writing clusters 8%  \r",
      "Writing clusters 15%  \r",
      "Writing clusters 23%  \r",
      "Writing clusters 31%  \r",
      "Writing clusters 38%  \r",
      "Writing clusters 46%  \r",
      "Writing clusters 54%  \r",
      "Writing clusters 62%  \r",
      "Writing clusters 69%  \r",
      "Writing clusters 77%  \r",
      "Writing clusters 85%  \r",
      "Writing clusters 92%  \r",
      "Writing clusters 100%\n",
      "Clusters: 6 Size min 1, max 4, avg 2.2\n",
      "Singletons: 3, 23.1% of seqs, 50.0% of clusters\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "vsearch --cluster_fast ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER_full.fasta --id 1.0 --strand both --threads 3 \\\n",
    "--centroids ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER.nr.fasta \\\n",
    "--uc ../../4-infer_phylogeny/sequences/c.flor_c.pse_SANGER.nr.uc --query_cov 1"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
