{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Illumina read processing and taxonomic classification of query sequences ###"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are using our custom pipeline [metaBEAT](https://github.com/HullUni-bioinformatics/metaBEAT) to process the Illumina data and taxonomically identify query sequences. \n",
    "\n",
    "For full reproducibility metaBEAT was run inside a docker container - [here](https://hub.docker.com/r/chrishah/metabeat/).\n",
    "\n",
    "After initial read quality trimming, merging and clustering, query sequences are filtered to remove potential chimeras against a reference database composed of COI sequences of _Craongonyx pseudogracilis_, _Crangonyx floridanus_, _Crangonyx islandicus_, and the positive control taxa _Osmia bicornis_ (all downloaded from Genbank as described [here](https://localhost:8888/notebooks/C_floridanus/1-NCBI_references/C_floridanus_NCBI_reference.ipynb)). Taxonomic assignment was performed using BLAST and a lowest common ancestor (LCA) approach, as described in the paper.\n",
    "\n",
    "The file `Querymap.txt` contains the sample IDs and the location of the Illumina read files, plus the barcodes and instructions to clip off the first number of bases of both the forward and reverse reads, in order to remove any primers and heterogeneity spacers.\n",
    "\n",
    "The file `REFlist.txt` points towards the sequences included in the reference database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "neg-7-5-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tGTAGAGAG\tGTAAGGAG\t32\t30\r\n",
      "CH101-1-8-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tTCGCCTTA\tCTAAGCCT\t26\t33\r\n",
      "CH102-1-7-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tTCGCCTTA\tAAGGAGTA\t26\t32\r\n",
      "CH103-1-6-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tTCGCCTTA\tACTGCATA\t26\t31\r\n",
      "CH104-1-5-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tTCGCCTTA\tGTAAGGAG\t26\t30\r\n",
      "CH105-1-4-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tTCGCCTTA\tAGAGTAGA\t26\t29\r\n",
      "CH106-1-3-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tTCGCCTTA\tTATCCTCT\t26\t28\r\n",
      "CH107-1-2-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tTCGCCTTA\tCTCTCTAT\t26\t27\r\n",
      "CH108-1-1-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tTCGCCTTA\tTAGATCGC\t26\t26\r\n",
      "CH109-2-8-May\tfastq\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz\t../../Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\tCTAGTACG\tCTAAGCCT\t27\t33\r\n"
     ]
    }
   ],
   "source": [
    "!head Querymap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We prepare the list for the reference database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for gb in $(ls -1 ../1-NCBI_references/*.gb | grep \"refseqs\" -v)\n",
    "do\n",
    "    echo -e \"$gb\\tgb\"\n",
    "done > REFlist.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../1-NCBI_references/C.islandicus_NCBI_2018.gb\tgb\r\n",
      "../1-NCBI_references/Crangonyx_NCBI_2018.gb\tgb\r\n",
      "../1-NCBI_references/O.bicornis_NCBI_2018.gb\tgb\r\n",
      "../1-NCBI_references/Synurella_NCBI_2018.gb\tgb\r\n"
     ]
    }
   ],
   "source": [
    "!cat REFlist.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Trimming, merging and clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.97.11-global\r\n"
     ]
    }
   ],
   "source": [
    "!metaBEAT_global.py --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: metaBEAT.py [-h] [-Q <FILE>] [-B <FILE>] [--g_queries <FILE>] [-v] [-s]\r\n",
      "                   [-f] [-p] [-k] [-t] [-b] [-m <string>] [-n <INT>] [-E] [-e]\r\n",
      "                   [--read_stats_off] [--PCR_primer <FILE>] [--bc_dist <INT>]\r\n",
      "                   [--trim_adapter <FILE>] [--trim_qual <INT>] [--phred <INT>]\r\n",
      "                   [--trim_window <INT>] [--read_crop <INT>]\r\n",
      "                   [--trim_minlength <INT>] [--merge] [--product_length <INT>]\r\n",
      "                   [--merged_only] [--forward_only] [--length_filter <INT>]\r\n",
      "                   [--length_deviation <FLOAT>] [-R <FILE>] [--gb_out <FILE>]\r\n",
      "                   [--rec_check] [--gb_to_taxid <FILE>] [--cluster]\r\n",
      "                   [--clust_match <FLOAT>] [--clust_cov <INT>]\r\n",
      "                   [--blast_db <PATH>] [--blast_xml <PATH>]\r\n",
      "                   [--update_taxonomy] [--taxonomy_db <FILE>]\r\n",
      "                   [--min_ident <FLOAT>] [--min_ali_length <FLOAT>]\r\n",
      "                   [--bitscore_skim_LCA <FLOAT>] [--bitscore_skim_adjust_off]\r\n",
      "                   [--min_bit <INT>] [--refpkg <DIR>] [--jplace <FILE>]\r\n",
      "                   [--jellyfish_hash_size <STR>] [--kraken_db <DIR>]\r\n",
      "                   [--kraken_score_threshold <FLOAT>] [--rm_kraken_db]\r\n",
      "                   [-o OUTPUT_PREFIX] [--metadata METADATA]\r\n",
      "                   [-@ <email-address>] [--version]\r\n",
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool version:\r\n",
      "v.0.97.11-global\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -Q <FILE>, --querylist <FILE>\r\n",
      "                        file containing a list of query files\r\n",
      "  -B <FILE>, --BIOM_input <FILE>\r\n",
      "                        OTU table in BIOM format\r\n",
      "  --g_queries <FILE>    fasta file containing query sequences (in combination\r\n",
      "                        with '-B' sequence headers are expected to match)\r\n",
      "  -v, --verbose         turn verbose output on\r\n",
      "  -s, --seqinfo         write out seq_info.csv file\r\n",
      "  -f, --fasta           write out ref.fasta file\r\n",
      "  -p, --pplace          perform phylogenetic placement\r\n",
      "  -k, --kraken          perform phylogenetic placement\r\n",
      "  -t, --taxids          write out taxid.txt file\r\n",
      "  -b, --blast           compile local blast db and blast queries\r\n",
      "  -m <string>, --marker <string>\r\n",
      "                        marker ID (default: marker)\r\n",
      "  -n <INT>, --n_threads <INT>\r\n",
      "                        Number of threads (default: 1)\r\n",
      "  -E, --extract_centroid_reads\r\n",
      "                        extract centroid reads to files\r\n",
      "  -e, --extract_all_reads\r\n",
      "                        extract reads to files\r\n",
      "  --read_stats_off      ommit writing read stats to file\r\n",
      "  --version             show program's version number and exit\r\n",
      "\r\n",
      "Query preprocessing:\r\n",
      "  The parameters in this group affect how the query sequences are processed\r\n",
      "\r\n",
      "  --PCR_primer <FILE>   PCR primers (provided in fasta file) to be clipped\r\n",
      "                        from reads\r\n",
      "  --bc_dist <INT>       Number of mismatches allowed in barcode sequences\r\n",
      "  --trim_adapter <FILE>\r\n",
      "                        trim adapters provided in file\r\n",
      "  --trim_qual <INT>     minimum phred quality score (default: 30)\r\n",
      "  --phred <INT>         phred quality score offset - 33 or 64 (default: 33)\r\n",
      "  --trim_window <INT>   sliding window size (default: 5) for trimming; if\r\n",
      "                        average quality drops below the specified minimum\r\n",
      "                        quality all subsequent bases are removed from the\r\n",
      "                        reads\r\n",
      "  --read_crop <INT>     Crop reads to this length if they are longer than that\r\n",
      "                        (default: off)\r\n",
      "  --trim_minlength <INT>\r\n",
      "                        minimum length of reads to be retained after trimming\r\n",
      "                        (default: 50)\r\n",
      "  --merge               attempt to merge paired-end reads\r\n",
      "  --product_length <INT>\r\n",
      "                        estimated length of PCR product (specifying this\r\n",
      "                        option increases merging efficiency)\r\n",
      "  --merged_only         only process successfully merged read-pairs\r\n",
      "  --forward_only        only process sequences that contain forward reads\r\n",
      "                        (i.e. unmerged forward reads and merged reads)\r\n",
      "  --length_filter <INT>\r\n",
      "                        only process reads, which are within +/- 10 percent of\r\n",
      "                        this length\r\n",
      "  --length_deviation <FLOAT>\r\n",
      "                        allowed deviation (in percent) from length specified\r\n",
      "                        by --length_filter (default=0.1)\r\n",
      "\r\n",
      "Reference:\r\n",
      "  The parameters in this group affect the reference to be used in the\r\n",
      "  analyses\r\n",
      "\r\n",
      "  -R <FILE>, --REFlist <FILE>\r\n",
      "                        file containing a list of files to be used as\r\n",
      "                        reference sequences\r\n",
      "  --gb_out <FILE>       output the corrected gb file\r\n",
      "  --rec_check           check records to be used as reference\r\n",
      "  --gb_to_taxid <FILE>  comma delimited file containing 'gb accession,taxid'\r\n",
      "                        for a list of taxa\r\n",
      "\r\n",
      "Query clustering options:\r\n",
      "  The parameters in this group affect read clustering\r\n",
      "\r\n",
      "  --cluster             perform clustering of query sequences using vsearch\r\n",
      "  --clust_match <FLOAT>\r\n",
      "                        identity threshold for clustering in percent (default:\r\n",
      "                        1)\r\n",
      "  --clust_cov <INT>     minimum number of records in cluster (default: 1)\r\n",
      "\r\n",
      "BLAST search:\r\n",
      "  The parameters in this group affect BLAST search and BLAST based taxonomic\r\n",
      "  assignment\r\n",
      "\r\n",
      "  --blast_db <PATH>     path to precompiled blast database\r\n",
      "  --blast_xml <PATH>    path to Blast result in xml format\r\n",
      "  --update_taxonomy     Download/update taxonomy database. Database will be\r\n",
      "                        called 'taxonomy.db' and will be compiled in the same\r\n",
      "                        location as the metaBEAT.py script.\r\n",
      "  --taxonomy_db <FILE>  taxonomy database file location. In case it's not the\r\n",
      "                        default, which is 'taxonomy.db' in the same directory\r\n",
      "                        as the metaBEAT.py script.\r\n",
      "  --min_ident <FLOAT>   minimum identity threshold in percent (default: 0.80)\r\n",
      "  --min_ali_length <FLOAT>\r\n",
      "                        minimum alignment length in percent of total query\r\n",
      "                        length (default: 0.95)\r\n",
      "  --bitscore_skim_LCA <FLOAT>\r\n",
      "                        Only BLAST hits with bitscores differing by less than\r\n",
      "                        this factor from the top hit (bitscore skim window)\r\n",
      "                        will be considered for LCA (0-1; default: 0.1)\r\n",
      "  --bitscore_skim_adjust_off\r\n",
      "                        Per default a 100% identity BLAST top hit across the\r\n",
      "                        minimum alignment length triggers an adjustment of the\r\n",
      "                        bitscore skim window to '0', i.e. only hits with\r\n",
      "                        bitscores as good as the top hit are considered for\r\n",
      "                        LCA. This flag switches this behaviour off.\r\n",
      "  --min_bit <INT>       minimum bitscore (default: 80)\r\n",
      "\r\n",
      "Phylogenetic placement:\r\n",
      "  The parameters in this group affect phylogenetic placement\r\n",
      "\r\n",
      "  --refpkg <DIR>        PATH to refpkg for pplacer\r\n",
      "  --jplace <FILE>       phylogenetic placement result from prefious pplacer\r\n",
      "                        run in *.jplace format\r\n",
      "\r\n",
      "Kraken:\r\n",
      "  The parameters in this group affect taxonomic assignment using Kraken\r\n",
      "\r\n",
      "  --jellyfish_hash_size <STR>\r\n",
      "                        jellyfish hash size to control memory usage during\r\n",
      "                        kraken database building. A table size of '6400M' will\r\n",
      "                        require ~44G of RAM. '2700M' -> 20G RAM.\r\n",
      "  --kraken_db <DIR>     PATH to a Kraken database\r\n",
      "  --kraken_score_threshold <FLOAT>\r\n",
      "                        minimum proportion of k-mers to support assignment\r\n",
      "                        (0-1; default: 0)\r\n",
      "  --rm_kraken_db        Remove Kraken database after successful completion\r\n",
      "\r\n",
      "BIOM OUTPUT:\r\n",
      "  The arguments in this groups affect the output in BIOM format\r\n",
      "\r\n",
      "  -o OUTPUT_PREFIX, --output_prefix OUTPUT_PREFIX\r\n",
      "                        prefix for BIOM output files (default='metaBEAT')\r\n",
      "  --metadata METADATA   comma delimited file containing metadata (optional)\r\n",
      "\r\n",
      "Entrez identification:\r\n",
      "  metaBEAT is querying the NCBI Entrez databases, please provide an email\r\n",
      "  address for identification\r\n",
      "\r\n",
      "  -@ <email-address>, --email <email-address>\r\n",
      "                        provide your email address for identification to NCBI\r\n"
     ]
    }
   ],
   "source": [
    "!metaBEAT_global.py --help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We run metaBEAT to cluster and trim the sequences from the metabarcoding data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!metaBEAT_global.py \\\n",
    "-Q Querymap.txt \\\n",
    "--trim_qual 30 \\\n",
    "--length_filter 313 \\\n",
    "--product_length 313 \\\n",
    "--merge --merged_only \\\n",
    "-m COI -n 5 -v -@ m.benucci@2015.hull.ac.uk > metaBEAT_trim.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool\n",
      "version: v.0.97.11-global\n",
      "\n",
      "\n",
      "Wed Jul 24 08:49:52 2019\n",
      "\n",
      "/usr/bin/metaBEAT_global.py -Q Querymap.txt --trim_qual 30 --length_filter 313 --product_length 313 --merge --merged_only -m COI -n 5 -v -@ m.benucci@2015.hull.ac.uk\n",
      "\n",
      "\n",
      "metaBEAT may be querying NCBI's Entrez databases to fetch/verify taxonomic ids. Entrez User requirements state that you need to identify yourself by providing an email address so that NCBI can contact you in case there is a problem.\n",
      "\n",
      "You have specified: 'm.benucci@2015.hull.ac.uk'\n",
      "\n",
      "\n",
      "Parsing querylist file\n",
      "\n",
      "Number of samples to process: 127\n",
      "Sequence input format: defaultdict(<type 'int'>, {'fastq': 127})\n",
      "Barcodes for demultiplexing provided for 127 samples\n",
      "Cropping instructions provided for 127 samples\n",
      "\n",
      "\n",
      "Wed Jul 24 08:49:52 2019\n",
      "\n",
      "\n",
      "### DEMULTIPLEXING ###\n",
      "\n",
      "assessing basic characteristics\n",
      "data comes as paired end - ok\n",
      "process_shortreads -1 /home/working/Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz -2 /home/working/Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz -i gzfastq -o . -b barcodes --inline_inline -y fastq\n",
      "\n",
      "Using Phred+33 encoding for quality scores.\n",
      "Reads trimmed shorter than 31 nucleotides will be discarded.\n",
      "Found 1 paired input file(s).\n",
      "Searching for single and paired-end, inlined barcodes.\n",
      "Loaded 52 barcodes (8bp / 8bp).\n",
      "Processing file 1 of 1 [Gp-gutcont3_S3_L001_R1_001.fastq.gz]\n",
      "  Reading data from\n",
      "    /home/working/Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R1_001.fastq.gz and\n",
      "    /home/working/Seqrun_COI_Aug17/data_files/Gp-gutcont3_S3_L001_R2_001.fastq.gz\n",
      "  5989798 total reads; -1593600 ambiguous barcodes; +0 recovered; -0 low quality reads; 4396198 retained reads.\n",
      "    0 trimmed reads; 0 orphaned paired-ends.\n",
      "Closing files, flushing buffers...\n",
      "Outputing details to log: './process_shortreads.log'\n",
      "\n",
      "5989798 total sequences;\n",
      "  1593600 ambiguous barcode drops;\n",
      "  0 low quality read drops;\n",
      "  0 trimmed reads;\n",
      "  0 orphaned paired-end reads;\n",
      "4396198 retained reads.\n",
      "\n",
      "assessing basic characteristics\n",
      "data comes as paired end - ok\n",
      "process_shortreads -1 /home/working/Seqrun_COI_Aug18_1/source_files/GW-WB-gutcont-eDNA_S3_L001_R1_001.fastq.gz -2 /home/working/Seqrun_COI_Aug18_1/source_files/GW-WB-gutcont-eDNA_S3_L001_R2_001.fastq.gz -i gzfastq -o . -b barcodes --inline_inline -y fastq\n",
      "\n",
      "Using Phred+33 encoding for quality scores.\n",
      "Reads trimmed shorter than 31 nucleotides will be discarded.\n",
      "Found 1 paired input file(s).\n",
      "Searching for single and paired-end, inlined barcodes.\n",
      "Loaded 10 barcodes (8bp / 8bp).\n",
      "Processing file 1 of 1 [GW-WB-gutcont-eDNA_S3_L001_R1_001.fastq.gz]\n",
      "  Reading data from\n",
      "    /home/working/Seqrun_COI_Aug18_1/source_files/GW-WB-gutcont-eDNA_S3_L001_R1_001.fastq.gz and\n",
      "    /home/working/Seqrun_COI_Aug18_1/source_files/GW-WB-gutcont-eDNA_S3_L001_R2_001.fastq.gz\n",
      "  6111486 total reads; -5741892 ambiguous barcodes; +0 recovered; -0 low quality reads; 369594 retained reads.\n",
      "    0 trimmed reads; 0 orphaned paired-ends.\n",
      "Closing files, flushing buffers...\n",
      "Outputing details to log: './process_shortreads.log'\n",
      "\n",
      "6111486 total sequences;\n",
      "  5741892 ambiguous barcode drops;\n",
      "  0 low quality read drops;\n",
      "  0 trimmed reads;\n",
      "  0 orphaned paired-end reads;\n",
      "369594 retained reads.\n",
      "\n",
      "assessing basic characteristics\n",
      "data comes as paired end - ok\n",
      "process_shortreads -1 /home/working/Seqrun_COI_Aug17/data_files/Dv-gutcont2-eDNA_S2_L001_R1_001.fastq.gz -2 /home/working/Seqrun_COI_Aug17/data_files/Dv-gutcont2-eDNA_S2_L001_R2_001.fastq.gz -i gzfastq -o . -b barcodes --inline_inline -y fastq\n",
      "\n",
      "Using Phred+33 encoding for quality scores.\n",
      "Reads trimmed shorter than 31 nucleotides will be discarded.\n",
      "Found 1 paired input file(s).\n",
      "Searching for single and paired-end, inlined barcodes.\n",
      "Loaded 10 barcodes (8bp / 8bp).\n",
      "Processing file 1 of 1 [Dv-gutcont2-eDNA_S2_L001_R1_001.fastq.gz]\n",
      "  Reading data from\n",
      "    /home/working/Seqrun_COI_Aug17/data_files/Dv-gutcont2-eDNA_S2_L001_R1_001.fastq.gz and\n",
      "    /home/working/Seqrun_COI_Aug17/data_files/Dv-gutcont2-eDNA_S2_L001_R2_001.fastq.gz\n",
      "  2203096 total reads; -1915238 ambiguous barcodes; +0 recovered; -0 low quality reads; 287858 retained reads.\n",
      "    0 trimmed reads; 0 orphaned paired-ends.\n",
      "Closing files, flushing buffers...\n",
      "Outputing details to log: './process_shortreads.log'\n",
      "\n",
      "2203096 total sequences;\n",
      "  1915238 ambiguous barcode drops;\n",
      "  0 low quality read drops;\n",
      "  0 trimmed reads;\n"
     ]
    }
   ],
   "source": [
    "!head -n 100 metaBEAT_trim.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "### READ CLIPPING ###\r\n",
      "\r\n",
      "Cropping the first 26 bases off reads in file: pos-9-7-May_forward.paired.fastq.gz\r\n",
      "Cropping the first 26 bases off reads in file: pos-9-7-May_forward.singletons.fastq.gz\r\n",
      "Cropping the first 32 bases off reads in file: pos-9-7-May_reverse.paired.fastq.gz\r\n",
      "Cropping the first 32 bases off reads in file: pos-9-7-May_reverse.singletons.fastq.gz\r\n",
      "\r\n",
      "### MERGING READ PAIRS ###\r\n",
      "\r\n",
      "merging paired-end reads with flash\r\n",
      "\r\n",
      "flash pos-9-7-May_forward.paired.headclipped.fastq.gz pos-9-7-May_reverse.paired.headclipped.fastq.gz -M 313 -t 5 -p 33 -o pos-9-7-May -z\r\n",
      "[FLASH] Starting FLASH v1.2.11\r\n",
      "[FLASH] Fast Length Adjustment of SHort reads\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Input files:\r\n",
      "[FLASH]     pos-9-7-May_forward.paired.headclipped.fastq.gz\r\n",
      "[FLASH]     pos-9-7-May_reverse.paired.headclipped.fastq.gz\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Output files:\r\n",
      "[FLASH]     ./pos-9-7-May.extendedFrags.fastq.gz\r\n",
      "[FLASH]     ./pos-9-7-May.notCombined_1.fastq.gz\r\n",
      "[FLASH]     ./pos-9-7-May.notCombined_2.fastq.gz\r\n",
      "[FLASH]     ./pos-9-7-May.hist\r\n",
      "[FLASH]     ./pos-9-7-May.histogram\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Parameters:\r\n",
      "[FLASH]     Min overlap:           10\r\n",
      "[FLASH]     Max overlap:           313\r\n",
      "[FLASH]     Max mismatch density:  0.250000\r\n",
      "[FLASH]     Allow \"outie\" pairs:   false\r\n",
      "[FLASH]     Cap mismatch quals:    false\r\n",
      "[FLASH]     Combiner threads:      5\r\n",
      "[FLASH]     Input format:          FASTQ, phred_offset=33\r\n",
      "[FLASH]     Output format:         FASTQ, phred_offset=33, gzip\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Starting reader and writer threads\r\n",
      "[FLASH] Starting 5 combiner threads\r\n",
      "[FLASH] Processed 9 read pairs\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Read combination statistics:\r\n",
      "[FLASH]     Total pairs:      9\r\n",
      "[FLASH]     Combined pairs:   9\r\n",
      "[FLASH]     Uncombined pairs: 0\r\n",
      "[FLASH]     Percent combined: 100.00%\r\n",
      "[FLASH]  \r\n",
      "[FLASH] Writing histogram files.\r\n",
      "[FLASH]  \r\n",
      "[FLASH] FLASH v1.2.11 complete!\r\n",
      "[FLASH] 0.002 seconds elapsed\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "### Fastq to fasta (plus filtering for length, merged-only, etc. if applicable) ###\r\n",
      "\r\n",
      "\r\n",
      "keeping only merged reads for subsequent analyses\r\n",
      "\r\n",
      "1 sequences removed by length filter\r\n",
      "\r\n",
      "\r\n",
      "8 sequences survived filtering\r\n",
      "\r\n",
      "The queryfile contains 8 query sequences.\r\n",
      "\r\n",
      "WRITING BASIC READ STATS TO FILE\r\n",
      "\r\n",
      "\r\n",
      "### GLOBAL CLUSTERING ###\r\n",
      "\r\n",
      "\r\n",
      "Concatenating for global clustering\r\n",
      "\r\n",
      "\r\n",
      "Performing global clustering\r\n",
      "\r\n",
      "vsearch --cluster_fast /home/working/C_floridanus/2-metaBEAT/GLOBAL/global_queries.fasta --id 1.00 --strand both --threads 5 --centroids global_centroids.fasta --uc global.uc\r\n",
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\r\n",
      "https://github.com/torognes/vsearch\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "parse vsearch uc file\r\n",
      "\r\n",
      "\r\n",
      "writing denovo OTU table\r\n",
      "\r\n",
      "\r\n",
      "Please select a method for taxonomic assignment:\r\n",
      "--blast\r\n",
      "--pplace\r\n",
      "--kraken\r\n",
      "\r\n",
      "\r\n",
      "##### DONE! #####\r\n",
      "\r\n",
      "\r\n",
      "Wed Jul 24 13:23:59 2019\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!tail -n 100 metaBEAT_trim.log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Chimera detection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We then run the chimera detection step using the `.gb` files we generated before."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "u'/home/working/C_floridanus/2-metaBEAT'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘chimera_detection’: File exists\r\n"
     ]
    }
   ],
   "source": [
    "!mkdir chimera_detection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/C_floridanus/2-metaBEAT/chimera_detection\n"
     ]
    }
   ],
   "source": [
    "cd chimera_detection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C_floridanus_NCBI_reference.ipynb  O.bicornis_NCBI_2018.gb\r\n",
      "c-islandicus_accession.txt\t   pos-taxa_accession.txt\r\n",
      "C.islandicus_NCBI_2018.gb\t   synurella_accession.txt\r\n",
      "Crangonyx_accession.txt\t\t   Synurella_NCBI_2018.gb\r\n",
      "Crangonyx_NCBI_2018.gb\r\n"
     ]
    }
   ],
   "source": [
    "!ls ../../1-NCBI_references/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Write REFmap\n",
    "for file in $(ls ../../1-NCBI_references/* | grep \"NCBI_2018.gb$\")\n",
    "do\n",
    "    echo -e \"$file\\tgb\"\n",
    "done > chimera_REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../../1-NCBI_references/C.islandicus_NCBI_2018.gb\tgb\r\n",
      "../../1-NCBI_references/Crangonyx_NCBI_2018.gb\tgb\r\n",
      "../../1-NCBI_references/O.bicornis_NCBI_2018.gb\tgb\r\n",
      "../../1-NCBI_references/Synurella_NCBI_2018.gb\tgb\r\n"
     ]
    }
   ],
   "source": [
    "!cat chimera_REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool\n",
      "version: v.0.97.11-global\n",
      "\n",
      "\n",
      "Wed Jul 24 13:25:05 2019\n",
      "\n",
      "/usr/bin/metaBEAT_global.py -R chimera_REFmap.txt -f -@ M.Benucci@2015.hull.ac.uk\n",
      "\n",
      "\n",
      "metaBEAT may be querying NCBI's Entrez databases to fetch/verify taxonomic ids. Entrez User requirements state that you need to identify yourself by providing an email address so that NCBI can contact you in case there is a problem.\n",
      "\n",
      "You have specified: 'M.Benucci@2015.hull.ac.uk'\n",
      "\n",
      "\n",
      "######## PROCESSING REFERENCE DATA ########\n",
      "\n",
      "\n",
      "processing ../../1-NCBI_references/Crangonyx_NCBI_2018.gb (containing 62 records)\n",
      "\n",
      "total number of valid records: 62\n",
      "\n",
      "\n",
      "processing ../../1-NCBI_references/Synurella_NCBI_2018.gb (containing 3 records)\n",
      "\n",
      "total number of valid records: 65\n",
      "\n",
      "\n",
      "processing ../../1-NCBI_references/C.islandicus_NCBI_2018.gb (containing 4 records)\n",
      "\n",
      "total number of valid records: 69\n",
      "\n",
      "\n",
      "processing ../../1-NCBI_references/O.bicornis_NCBI_2018.gb (containing 2 records)\n",
      "\n",
      "total number of valid records: 71\n",
      "\n",
      "write out reference sequences to refs.fasta\n",
      "\n",
      "\n",
      "Wed Jul 24 13:25:05 2019\n",
      "\n",
      "\n",
      "### GLOBAL CLUSTERING ###\n",
      "\n",
      "\n",
      "Concatenating for global clustering\n",
      "\n",
      "\n",
      "Performing global clustering\n",
      "\n",
      "vsearch --cluster_fast /home/working/C_floridanus/2-metaBEAT/chimera_detection/GLOBAL/global_queries.fasta --id 1.00 --strand both --threads 1 --centroids global_centroids.fasta --uc global.uc\n",
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "\n",
      "\n",
      "parse vsearch uc file\n",
      "\n",
      "\n",
      "writing denovo OTU table\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/usr/bin/metaBEAT_global.py\", line 2575, in <module>\n",
      "    out.write(BIOM_tables_per_method['OTU_denovo'].to_tsv()) #to_json('generaged by test', direct_io=out)\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/biom/table.py\", line 4027, in to_tsv\n",
      "    observation_column_name)\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/biom/table.py\", line 1268, in delimited_self\n",
      "    raise TableException(\"Cannot delimit self if I don't have data...\")\n",
      "biom.exception.TableException: Cannot delimit self if I don't have data...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-R chimera_REFmap.txt \\\n",
    "-f \\\n",
    "-@ M.Benucci@2015.hull.ac.uk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">AB513800|326578|Crangonyx floridanus\r\n",
      "CCAGAGCTGTTGGTACAGCCTTAAGCATAATTATCCGAATCGAACTGGCAACACCAGGAAATATTATTGAAGACGATCAAATTTACAATGTTATAGTAACAGCCCATGCCTTTGTTATAATTTTTTTTATGGTTATACCCATCATGATTGGAGGGTTTGGTAACTGACTAGTGCCTCTTATATTAGGCAGCCCTGATATAGCATTTCCTCGAATAAATAATATAAGATTTTGATTATTACCTCCCTCGTTATGTCTTCTGCTTATAAGAAGTTTAATCGAAAGAGGGGTAGGAACAGGATGAACTGTCTACCCGCCTTTAGCATCTACAGCTGCTCATAGAGGTGCTTCTGTAGACTTAGCTATTTTCTCTCTTCACCTAGCAGGTGCCTCCTCTATTTTAGGTTCAATTAACTTTATTTCCACAGTAATAAATATACGAGTAAAAAATATATTAATAGACCAAATCCCTTTATTTGTTTGAGCTATTTTCTTCACTACTATTCTTCTTCTTCTTTCTTTACCTGTTCTAGCAGGAGCTATCACAATACTTTTAACAGACCGTAATCTCAATACATCATTCTTTGACCCTTCTGGGGGGGGTGACCCTATCTTGTAC\r\n",
      ">AB513801|326578|Crangonyx floridanus\r\n",
      "GCCTGAGCCAGAGCTGTTGGTACAGCCTTAAGCATAATTATCCGAATCGAACTGGCAACACCAGGAAATATTATTGAAGACGATCAAATTTACAATGTTATAGTAACAGCCCATGCCTTTGTTATAATTTTTTTTATGGTTATACCCATCATGATTGGAGGGTTTGGTAACTGACTAGTGCCTCTTATATTAGGCAGCCCTGATATAGCATTTCCTCGAATAAATAATATAAGATTTTGATTATTACCTCCCTCGTTATGTCTTCTGCTTATAAGAAGTTTAATCGAAAGAGGGGTAGGAACAGGATGAACTGTCTACCCGCCTTTAGCATCTACAGCTGCTCATAGAGGTGCTTCTGTAGACTTAGCTATTTTCTCTCTTCACCTAGCAGGTGCCTCCTCTATTTTAGGTTCAATTAACTTTATTTCCACAGTAATAAATATACGAGTAAAAAATATATTAATAGACCAAATCCCTTTATTTGTTTGAGCTATTTTCTTCACTACTATTCTTCTTCTTCTTTCTTTACCTGTTCTAGCAGGAGCTATCACAATACTTTTAACAGACCGTAATCTCAATACATCATTCTTTGACCCTTCTGGGGGGGGTGACCCTATCTTGTAC\r\n",
      ">AB513802|326578|Crangonyx floridanus\r\n",
      "GCCTGAGCCAGAGCTGTTGGTACAGCCTTAAGCATAATTATCCGAATCGAACTGGCAACACCAGGAAATATTATTGAAGACGATCAAATTTACAATGTTATAGTAACAGCCCATGCCTTTGTTATAATTTTTTTTATGGTTATACCCATCATGATTGGAGGGTTTGGTAACTGACTAGTGCCTCTTATATTAGGCAGCCCTGATATAGCATTTCCTCGAATAAATAATATAAGATTTTGATTATTACCTCCCTCGTTATGTCTTCTGCTTATAAGAAGTTTAATCGAAAGAGGGGTAGGAACAGGATGAACTGTCTACCCGCCTTTAGCATCTACAGCTGCTCATAGAGGTGCTTCTGTAGACTTAGCTATTTTCTCTCTTCACCTAGCAGGTGCCTCCTCTATTTTAGGTTCAATTAACTTTATTTCCACAGTAATAAATATACGAGTAAAAAATATATTAATAGACCAAATCCCTTTATTTGTTTGAGCTATTTTCTTCACTACTATTCTTCTTCTTCTTTCTTTACCTGTTCTAGCAGGAGCTATCACAATACTTTTAACAGACCGTAATCTCAATACATCATTCTTTGACCCTTCTGGGGGGGGTGACCCTATCTTGTAC\r\n",
      ">AB513803|326578|Crangonyx floridanus\r\n",
      "TGAGCCAGAGCTGTTGGTACAGCCTTAAGCATAATTATCCGAATCGAACTGGCAACACCAGGAAATATTATTGAAGACGATCAAATTTACAATGTTATAGTAACAGCCCATGCCTTTGTTATAATTTTTTTTATGGTTATACCCATCATGATTGGAGGGTTTGGTAACTGACTAGTGCCTCTTATATTAGGCAGCCCTGATATAGCATTTCCTCGAATAAATAATATAAGATTTTGATTATTACCTCCCTCGTTATGTCTTCTGCTTATAAGAAGTTTAATCGAAAGAGGGGTAGGAACAGGATGAACTGTCTACCCGCCTTTAGCATCTACAGCTGCTCATAGAGGTGCTTCTGTAGACTTAGCTATTTTCTCTCTTCACCTAGCAGGTGCCTCCTCTATTTTAGGTTCAATTAACTTTATTTCCACAGTAATAAATATACGAGTAAAAAATATATTAATAGACCAAATCCCTTTATTTGTTTGAGCTATTTTCTTCACTACTATTCTTCTTCTTCTTTCTTTACCTGTTCTAGCAGGAGCTATCACAATACTTTTAACAGACCGTAATCTCAATACATCATTCTTTGACCCTTCTGGGGGGGGTGACCCTATCTTGTAC\r\n",
      ">AB513804|326578|Crangonyx floridanus\r\n",
      "GCCTGAGCCAGAGCTGTTGGTACAGCCTTAAGCATAATTATCCGAATCGAACTGGCAACACCAGGAAATATTATTGAAGACGATCAAATTTACAATGTTATAGTAACAGCCCATGCCTTTGTTATAATTTTTTTTATGGTTATACCCATCATGATTGGAGGGTTTGGTAACTGACTAGTGCCTCTTATATTAGGCAGCCCTGATATAGCATTTCCTCGAATAAATAATATAAGATTTTGATTATTACCTCCCTCGTTATGTCTTCTGCTTATAAGAAGTTTAATCGAAAGAGGGGTAGGAACAGGATGAACTGTCTACCCGCCTTTAGCATCTACAGCTGCTCATAGAGGTGCTTCTGTAGACTTAGCTATTTTCTCTCTTCACCTAGCAGGTGCCTCCTCTATTTTAGGTTCAATTAACTTTATTTCCACAGTAATAAATATACGAGTAAAAAATATATTAATAGACCAAATCCCTTTATTTGTTTGAGCTATTTTCTTCACTACTATTCTTCTTCTTCTTTCTTTACCTGTTCTAGCAGGAGCTATCACAATACTTTTAACAGACCGTAATCTCAATACATCATTCTTTGACCCTTCTGGGGGGGGTGACCCTATCTTGTAC\r\n"
     ]
    }
   ],
   "source": [
    "!head refs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "### Detecting chimeras in neg-7-5-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH101-1-8-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH102-1-7-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH103-1-6-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH104-1-5-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH105-1-4-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH106-1-3-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH107-1-2-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH108-1-1-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH109-2-8-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH110-2-7-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH201-2-6-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH202-2-5-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH203-2-4-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH204-2-3-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH205-2-2-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH206-2-1-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH207-3-8-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH208-3-7-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH209-3-6-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH210-3-5-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH211-3-4-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH308-3-3-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH309-3-2-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH310-3-1-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH311-4-8-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH312-4-7-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH313-4-6-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH314-4-5-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH315-4-4-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH316-4-3-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH317-4-2-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH401-4-1-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH402-5-8-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH403-5-7-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH404-5-6-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH405-5-5-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH406-5-4-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH407-5-3-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH408-5-2-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH409-5-1-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH410-6-8-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH501-6-7-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH503-6-6-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH504-6-5-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH505-6-4-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH506-6-3-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH507-6-2-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH508-6-1-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH509-7-8-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH510-7-7-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH602-7-6-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in neg-6-2-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH-BL-7-2-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH6-7-3-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH5-7-4-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH4-7-5-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH3-7-6-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH2-7-7-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH1-7-8-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pos-9-7-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in neg-9-8-May ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH101-pl1-1-1-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH102-pl1-1-2-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH103-pl1-1-3-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH104-pl1-1-4-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH105-pl1-1-5-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH106-pl1-1-6-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH107-pl1-1-7-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH108-pl1-1-8-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH109-pl1-2-1-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH110-pl1-2-2-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH111-pl1-2-3-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH112-pl1-2-4-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH113-pl1-2-5-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH114-pl1-2-6-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH115-pl1-2-7-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH116-pl1-2-8-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH117-pl1-3-1-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH118-pl1-3-2-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH119-pl1-3-3-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH301-pl1-3-4-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH302-pl1-3-5-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH303-pl1-3-6-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH304-pl1-3-7-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH305-pl1-3-8-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH306-pl1-4-1-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH307-pl1-4-2-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH308-pl1-4-3-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH309-pl1-4-4-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH310-pl1-4-5-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH401-pl1-4-6-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH501-pl1-4-7-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH502-pl1-4-8-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH503-pl1-5-1-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH504-pl1-5-2-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH505-pl1-5-3-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH506-pl1-5-4-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH507-pl1-5-5-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH508-pl1-5-6-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH509-pl1-5-7-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH510-pl1-5-8-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH601-pl1-6-1-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH602-pl1-6-2-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH603-pl1-6-3-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH604-pl1-6-4-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH605-pl1-6-5-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH606-pl1-6-6-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH607-pl1-6-7-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH608-pl1-6-8-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH609-pl1-7-1-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH610-pl1-7-2-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH-BL1-pl1-7-3-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in CH-BL2-pl1-7-4-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pcr-NEG-pl1-7-5-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pcr-POS-pl1-7-6-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pcr-NEG-pl1-7-7-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pcr-NEG-pl3-6-5-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pcr-POS-pl3-6-6-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in pcr-NEG-pl3-6-7-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH-BL-pl3-6-8-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH1-pl3-7-1-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH2-pl3-7-2-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH3-pl3-7-3-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH4-pl3-7-4-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH5-pl3-7-5-Oct ###\n",
      "\n",
      "\n",
      "### Detecting chimeras in eCH6-pl3-7-6-Oct ###\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘neg-7-5-May’: File exists\n",
      "mkdir: cannot create directory ‘CH101-1-8-May’: File exists\n",
      "mkdir: cannot create directory ‘CH102-1-7-May’: File exists\n",
      "mkdir: cannot create directory ‘CH103-1-6-May’: File exists\n",
      "mkdir: cannot create directory ‘CH104-1-5-May’: File exists\n",
      "mkdir: cannot create directory ‘CH105-1-4-May’: File exists\n",
      "mkdir: cannot create directory ‘CH106-1-3-May’: File exists\n",
      "mkdir: cannot create directory ‘CH107-1-2-May’: File exists\n",
      "mkdir: cannot create directory ‘CH108-1-1-May’: File exists\n",
      "mkdir: cannot create directory ‘CH109-2-8-May’: File exists\n",
      "mkdir: cannot create directory ‘CH110-2-7-May’: File exists\n",
      "mkdir: cannot create directory ‘CH201-2-6-May’: File exists\n",
      "mkdir: cannot create directory ‘CH202-2-5-May’: File exists\n",
      "mkdir: cannot create directory ‘CH203-2-4-May’: File exists\n",
      "mkdir: cannot create directory ‘CH204-2-3-May’: File exists\n",
      "mkdir: cannot create directory ‘CH205-2-2-May’: File exists\n",
      "mkdir: cannot create directory ‘CH206-2-1-May’: File exists\n",
      "mkdir: cannot create directory ‘CH207-3-8-May’: File exists\n",
      "mkdir: cannot create directory ‘CH208-3-7-May’: File exists\n",
      "mkdir: cannot create directory ‘CH209-3-6-May’: File exists\n",
      "mkdir: cannot create directory ‘CH210-3-5-May’: File exists\n",
      "mkdir: cannot create directory ‘CH211-3-4-May’: File exists\n",
      "mkdir: cannot create directory ‘CH308-3-3-May’: File exists\n",
      "mkdir: cannot create directory ‘CH309-3-2-May’: File exists\n",
      "mkdir: cannot create directory ‘CH310-3-1-May’: File exists\n",
      "mkdir: cannot create directory ‘CH311-4-8-May’: File exists\n",
      "mkdir: cannot create directory ‘CH312-4-7-May’: File exists\n",
      "mkdir: cannot create directory ‘CH313-4-6-May’: File exists\n",
      "mkdir: cannot create directory ‘CH314-4-5-May’: File exists\n",
      "mkdir: cannot create directory ‘CH315-4-4-May’: File exists\n",
      "mkdir: cannot create directory ‘CH316-4-3-May’: File exists\n",
      "mkdir: cannot create directory ‘CH317-4-2-May’: File exists\n",
      "mkdir: cannot create directory ‘CH401-4-1-May’: File exists\n",
      "mkdir: cannot create directory ‘CH402-5-8-May’: File exists\n",
      "mkdir: cannot create directory ‘CH403-5-7-May’: File exists\n",
      "mkdir: cannot create directory ‘CH404-5-6-May’: File exists\n",
      "mkdir: cannot create directory ‘CH405-5-5-May’: File exists\n",
      "mkdir: cannot create directory ‘CH406-5-4-May’: File exists\n",
      "mkdir: cannot create directory ‘CH407-5-3-May’: File exists\n",
      "mkdir: cannot create directory ‘CH408-5-2-May’: File exists\n",
      "mkdir: cannot create directory ‘CH409-5-1-May’: File exists\n",
      "mkdir: cannot create directory ‘CH410-6-8-May’: File exists\n",
      "mkdir: cannot create directory ‘CH501-6-7-May’: File exists\n",
      "mkdir: cannot create directory ‘CH503-6-6-May’: File exists\n",
      "mkdir: cannot create directory ‘CH504-6-5-May’: File exists\n",
      "mkdir: cannot create directory ‘CH505-6-4-May’: File exists\n",
      "mkdir: cannot create directory ‘CH506-6-3-May’: File exists\n",
      "mkdir: cannot create directory ‘CH507-6-2-May’: File exists\n",
      "mkdir: cannot create directory ‘CH508-6-1-May’: File exists\n",
      "mkdir: cannot create directory ‘CH509-7-8-May’: File exists\n",
      "mkdir: cannot create directory ‘CH510-7-7-May’: File exists\n",
      "mkdir: cannot create directory ‘CH602-7-6-May’: File exists\n",
      "mkdir: cannot create directory ‘neg-6-2-May’: File exists\n",
      "mkdir: cannot create directory ‘eCH-BL-7-2-May’: File exists\n",
      "mkdir: cannot create directory ‘eCH6-7-3-May’: File exists\n",
      "mkdir: cannot create directory ‘eCH5-7-4-May’: File exists\n",
      "mkdir: cannot create directory ‘eCH4-7-5-May’: File exists\n",
      "mkdir: cannot create directory ‘eCH3-7-6-May’: File exists\n",
      "mkdir: cannot create directory ‘eCH2-7-7-May’: File exists\n",
      "mkdir: cannot create directory ‘eCH1-7-8-May’: File exists\n",
      "mkdir: cannot create directory ‘pos-9-7-May’: File exists\n",
      "mkdir: cannot create directory ‘neg-9-8-May’: File exists\n",
      "mkdir: cannot create directory ‘CH101-pl1-1-1-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH102-pl1-1-2-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH103-pl1-1-3-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH104-pl1-1-4-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH105-pl1-1-5-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH106-pl1-1-6-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH107-pl1-1-7-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH108-pl1-1-8-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH109-pl1-2-1-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH110-pl1-2-2-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH111-pl1-2-3-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH112-pl1-2-4-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH113-pl1-2-5-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH114-pl1-2-6-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH115-pl1-2-7-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH116-pl1-2-8-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH117-pl1-3-1-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH118-pl1-3-2-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH119-pl1-3-3-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH301-pl1-3-4-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH302-pl1-3-5-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH303-pl1-3-6-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH304-pl1-3-7-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH305-pl1-3-8-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH306-pl1-4-1-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH307-pl1-4-2-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH308-pl1-4-3-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH309-pl1-4-4-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH310-pl1-4-5-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH401-pl1-4-6-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH501-pl1-4-7-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH502-pl1-4-8-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH503-pl1-5-1-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH504-pl1-5-2-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH505-pl1-5-3-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH506-pl1-5-4-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH507-pl1-5-5-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH508-pl1-5-6-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH509-pl1-5-7-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH510-pl1-5-8-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH601-pl1-6-1-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH602-pl1-6-2-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH603-pl1-6-3-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH604-pl1-6-4-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH605-pl1-6-5-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH606-pl1-6-6-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH607-pl1-6-7-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH608-pl1-6-8-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH609-pl1-7-1-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH610-pl1-7-2-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH-BL1-pl1-7-3-Oct’: File exists\n",
      "mkdir: cannot create directory ‘CH-BL2-pl1-7-4-Oct’: File exists\n",
      "mkdir: cannot create directory ‘pcr-NEG-pl1-7-5-Oct’: File exists\n",
      "mkdir: cannot create directory ‘pcr-POS-pl1-7-6-Oct’: File exists\n",
      "mkdir: cannot create directory ‘pcr-NEG-pl1-7-7-Oct’: File exists\n",
      "mkdir: cannot create directory ‘pcr-NEG-pl3-6-5-Oct’: File exists\n",
      "mkdir: cannot create directory ‘pcr-POS-pl3-6-6-Oct’: File exists\n",
      "mkdir: cannot create directory ‘pcr-NEG-pl3-6-7-Oct’: File exists\n",
      "mkdir: cannot create directory ‘eCH-BL-pl3-6-8-Oct’: File exists\n",
      "mkdir: cannot create directory ‘eCH1-pl3-7-1-Oct’: File exists\n",
      "mkdir: cannot create directory ‘eCH2-pl3-7-2-Oct’: File exists\n",
      "mkdir: cannot create directory ‘eCH3-pl3-7-3-Oct’: File exists\n",
      "mkdir: cannot create directory ‘eCH4-pl3-7-4-Oct’: File exists\n",
      "mkdir: cannot create directory ‘eCH5-pl3-7-5-Oct’: File exists\n",
      "mkdir: cannot create directory ‘eCH6-pl3-7-6-Oct’: File exists\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "\n",
    "for a in $(cut -f 1 ../Querymap.txt)\n",
    "do\n",
    "    if [ -s ../$a/$a\\_trimmed.fasta ]\n",
    "    then\n",
    "        echo -e \"\\n### Detecting chimeras in $a ###\\n\"\n",
    "        mkdir $a\n",
    "        cd $a\n",
    "        vsearch --uchime_ref ../../$a/$a\\_trimmed.fasta --db ../refs.fasta \\\n",
    "        --nonchimeras $a-nonchimeras.fasta --chimeras $a-chimeras.fasta &> log \n",
    "        cd ..\n",
    "\n",
    "    else\n",
    "        echo -e \"$a is empty\"\n",
    "    fi\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Blast of non-chimera sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/C_floridanus/2-metaBEAT\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Querymap\n",
    "for a in $(ls -l chimera_detection/ | grep \"^d\" | perl -ne 'chomp; @a=split(\" \"); print \"$a[-1]\\n\"')\n",
    "do\n",
    "    echo -e \"$a-nc\\tfasta\\tchimera_detection/$a/$a-nonchimeras.fasta\"\n",
    "done > nonchimera_Querymap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CH101-1-8-May-nc\tfasta\tchimera_detection/CH101-1-8-May/CH101-1-8-May-nonchimeras.fasta\r\n",
      "CH101-pl1-1-1-Oct-nc\tfasta\tchimera_detection/CH101-pl1-1-1-Oct/CH101-pl1-1-1-Oct-nonchimeras.fasta\r\n",
      "CH102-1-7-May-nc\tfasta\tchimera_detection/CH102-1-7-May/CH102-1-7-May-nonchimeras.fasta\r\n",
      "CH102-pl1-1-2-Oct-nc\tfasta\tchimera_detection/CH102-pl1-1-2-Oct/CH102-pl1-1-2-Oct-nonchimeras.fasta\r\n",
      "CH103-1-6-May-nc\tfasta\tchimera_detection/CH103-1-6-May/CH103-1-6-May-nonchimeras.fasta\r\n",
      "CH103-pl1-1-3-Oct-nc\tfasta\tchimera_detection/CH103-pl1-1-3-Oct/CH103-pl1-1-3-Oct-nonchimeras.fasta\r\n",
      "CH104-1-5-May-nc\tfasta\tchimera_detection/CH104-1-5-May/CH104-1-5-May-nonchimeras.fasta\r\n",
      "CH104-pl1-1-4-Oct-nc\tfasta\tchimera_detection/CH104-pl1-1-4-Oct/CH104-pl1-1-4-Oct-nonchimeras.fasta\r\n",
      "CH105-1-4-May-nc\tfasta\tchimera_detection/CH105-1-4-May/CH105-1-4-May-nonchimeras.fasta\r\n",
      "CH105-pl1-1-5-Oct-nc\tfasta\tchimera_detection/CH105-pl1-1-5-Oct/CH105-pl1-1-5-Oct-nonchimeras.fasta\r\n",
      "CH106-1-3-May-nc\tfasta\tchimera_detection/CH106-1-3-May/CH106-1-3-May-nonchimeras.fasta\r\n",
      "CH106-pl1-1-6-Oct-nc\tfasta\tchimera_detection/CH106-pl1-1-6-Oct/CH106-pl1-1-6-Oct-nonchimeras.fasta\r\n",
      "CH107-1-2-May-nc\tfasta\tchimera_detection/CH107-1-2-May/CH107-1-2-May-nonchimeras.fasta\r\n",
      "CH107-pl1-1-7-Oct-nc\tfasta\tchimera_detection/CH107-pl1-1-7-Oct/CH107-pl1-1-7-Oct-nonchimeras.fasta\r\n",
      "CH108-1-1-May-nc\tfasta\tchimera_detection/CH108-1-1-May/CH108-1-1-May-nonchimeras.fasta\r\n",
      "CH108-pl1-1-8-Oct-nc\tfasta\tchimera_detection/CH108-pl1-1-8-Oct/CH108-pl1-1-8-Oct-nonchimeras.fasta\r\n",
      "CH109-2-8-May-nc\tfasta\tchimera_detection/CH109-2-8-May/CH109-2-8-May-nonchimeras.fasta\r\n",
      "CH109-pl1-2-1-Oct-nc\tfasta\tchimera_detection/CH109-pl1-2-1-Oct/CH109-pl1-2-1-Oct-nonchimeras.fasta\r\n",
      "CH110-2-7-May-nc\tfasta\tchimera_detection/CH110-2-7-May/CH110-2-7-May-nonchimeras.fasta\r\n",
      "CH110-pl1-2-2-Oct-nc\tfasta\tchimera_detection/CH110-pl1-2-2-Oct/CH110-pl1-2-2-Oct-nonchimeras.fasta\r\n",
      "CH111-pl1-2-3-Oct-nc\tfasta\tchimera_detection/CH111-pl1-2-3-Oct/CH111-pl1-2-3-Oct-nonchimeras.fasta\r\n",
      "CH112-pl1-2-4-Oct-nc\tfasta\tchimera_detection/CH112-pl1-2-4-Oct/CH112-pl1-2-4-Oct-nonchimeras.fasta\r\n",
      "CH113-pl1-2-5-Oct-nc\tfasta\tchimera_detection/CH113-pl1-2-5-Oct/CH113-pl1-2-5-Oct-nonchimeras.fasta\r\n",
      "CH114-pl1-2-6-Oct-nc\tfasta\tchimera_detection/CH114-pl1-2-6-Oct/CH114-pl1-2-6-Oct-nonchimeras.fasta\r\n",
      "CH115-pl1-2-7-Oct-nc\tfasta\tchimera_detection/CH115-pl1-2-7-Oct/CH115-pl1-2-7-Oct-nonchimeras.fasta\r\n",
      "CH116-pl1-2-8-Oct-nc\tfasta\tchimera_detection/CH116-pl1-2-8-Oct/CH116-pl1-2-8-Oct-nonchimeras.fasta\r\n",
      "CH117-pl1-3-1-Oct-nc\tfasta\tchimera_detection/CH117-pl1-3-1-Oct/CH117-pl1-3-1-Oct-nonchimeras.fasta\r\n",
      "CH118-pl1-3-2-Oct-nc\tfasta\tchimera_detection/CH118-pl1-3-2-Oct/CH118-pl1-3-2-Oct-nonchimeras.fasta\r\n",
      "CH119-pl1-3-3-Oct-nc\tfasta\tchimera_detection/CH119-pl1-3-3-Oct/CH119-pl1-3-3-Oct-nonchimeras.fasta\r\n",
      "CH201-2-6-May-nc\tfasta\tchimera_detection/CH201-2-6-May/CH201-2-6-May-nonchimeras.fasta\r\n",
      "CH202-2-5-May-nc\tfasta\tchimera_detection/CH202-2-5-May/CH202-2-5-May-nonchimeras.fasta\r\n",
      "CH203-2-4-May-nc\tfasta\tchimera_detection/CH203-2-4-May/CH203-2-4-May-nonchimeras.fasta\r\n",
      "CH204-2-3-May-nc\tfasta\tchimera_detection/CH204-2-3-May/CH204-2-3-May-nonchimeras.fasta\r\n",
      "CH205-2-2-May-nc\tfasta\tchimera_detection/CH205-2-2-May/CH205-2-2-May-nonchimeras.fasta\r\n",
      "CH206-2-1-May-nc\tfasta\tchimera_detection/CH206-2-1-May/CH206-2-1-May-nonchimeras.fasta\r\n",
      "CH207-3-8-May-nc\tfasta\tchimera_detection/CH207-3-8-May/CH207-3-8-May-nonchimeras.fasta\r\n",
      "CH208-3-7-May-nc\tfasta\tchimera_detection/CH208-3-7-May/CH208-3-7-May-nonchimeras.fasta\r\n",
      "CH209-3-6-May-nc\tfasta\tchimera_detection/CH209-3-6-May/CH209-3-6-May-nonchimeras.fasta\r\n",
      "CH210-3-5-May-nc\tfasta\tchimera_detection/CH210-3-5-May/CH210-3-5-May-nonchimeras.fasta\r\n",
      "CH211-3-4-May-nc\tfasta\tchimera_detection/CH211-3-4-May/CH211-3-4-May-nonchimeras.fasta\r\n",
      "CH301-pl1-3-4-Oct-nc\tfasta\tchimera_detection/CH301-pl1-3-4-Oct/CH301-pl1-3-4-Oct-nonchimeras.fasta\r\n",
      "CH302-pl1-3-5-Oct-nc\tfasta\tchimera_detection/CH302-pl1-3-5-Oct/CH302-pl1-3-5-Oct-nonchimeras.fasta\r\n",
      "CH303-pl1-3-6-Oct-nc\tfasta\tchimera_detection/CH303-pl1-3-6-Oct/CH303-pl1-3-6-Oct-nonchimeras.fasta\r\n",
      "CH304-pl1-3-7-Oct-nc\tfasta\tchimera_detection/CH304-pl1-3-7-Oct/CH304-pl1-3-7-Oct-nonchimeras.fasta\r\n",
      "CH305-pl1-3-8-Oct-nc\tfasta\tchimera_detection/CH305-pl1-3-8-Oct/CH305-pl1-3-8-Oct-nonchimeras.fasta\r\n",
      "CH306-pl1-4-1-Oct-nc\tfasta\tchimera_detection/CH306-pl1-4-1-Oct/CH306-pl1-4-1-Oct-nonchimeras.fasta\r\n",
      "CH307-pl1-4-2-Oct-nc\tfasta\tchimera_detection/CH307-pl1-4-2-Oct/CH307-pl1-4-2-Oct-nonchimeras.fasta\r\n",
      "CH308-3-3-May-nc\tfasta\tchimera_detection/CH308-3-3-May/CH308-3-3-May-nonchimeras.fasta\r\n",
      "CH308-pl1-4-3-Oct-nc\tfasta\tchimera_detection/CH308-pl1-4-3-Oct/CH308-pl1-4-3-Oct-nonchimeras.fasta\r\n",
      "CH309-3-2-May-nc\tfasta\tchimera_detection/CH309-3-2-May/CH309-3-2-May-nonchimeras.fasta\r\n",
      "CH309-pl1-4-4-Oct-nc\tfasta\tchimera_detection/CH309-pl1-4-4-Oct/CH309-pl1-4-4-Oct-nonchimeras.fasta\r\n",
      "CH310-3-1-May-nc\tfasta\tchimera_detection/CH310-3-1-May/CH310-3-1-May-nonchimeras.fasta\r\n",
      "CH310-pl1-4-5-Oct-nc\tfasta\tchimera_detection/CH310-pl1-4-5-Oct/CH310-pl1-4-5-Oct-nonchimeras.fasta\r\n",
      "CH311-4-8-May-nc\tfasta\tchimera_detection/CH311-4-8-May/CH311-4-8-May-nonchimeras.fasta\r\n",
      "CH312-4-7-May-nc\tfasta\tchimera_detection/CH312-4-7-May/CH312-4-7-May-nonchimeras.fasta\r\n",
      "CH313-4-6-May-nc\tfasta\tchimera_detection/CH313-4-6-May/CH313-4-6-May-nonchimeras.fasta\r\n",
      "CH314-4-5-May-nc\tfasta\tchimera_detection/CH314-4-5-May/CH314-4-5-May-nonchimeras.fasta\r\n",
      "CH315-4-4-May-nc\tfasta\tchimera_detection/CH315-4-4-May/CH315-4-4-May-nonchimeras.fasta\r\n",
      "CH316-4-3-May-nc\tfasta\tchimera_detection/CH316-4-3-May/CH316-4-3-May-nonchimeras.fasta\r\n",
      "CH317-4-2-May-nc\tfasta\tchimera_detection/CH317-4-2-May/CH317-4-2-May-nonchimeras.fasta\r\n",
      "CH401-4-1-May-nc\tfasta\tchimera_detection/CH401-4-1-May/CH401-4-1-May-nonchimeras.fasta\r\n",
      "CH401-pl1-4-6-Oct-nc\tfasta\tchimera_detection/CH401-pl1-4-6-Oct/CH401-pl1-4-6-Oct-nonchimeras.fasta\r\n",
      "CH402-5-8-May-nc\tfasta\tchimera_detection/CH402-5-8-May/CH402-5-8-May-nonchimeras.fasta\r\n",
      "CH403-5-7-May-nc\tfasta\tchimera_detection/CH403-5-7-May/CH403-5-7-May-nonchimeras.fasta\r\n",
      "CH404-5-6-May-nc\tfasta\tchimera_detection/CH404-5-6-May/CH404-5-6-May-nonchimeras.fasta\r\n",
      "CH405-5-5-May-nc\tfasta\tchimera_detection/CH405-5-5-May/CH405-5-5-May-nonchimeras.fasta\r\n",
      "CH406-5-4-May-nc\tfasta\tchimera_detection/CH406-5-4-May/CH406-5-4-May-nonchimeras.fasta\r\n",
      "CH407-5-3-May-nc\tfasta\tchimera_detection/CH407-5-3-May/CH407-5-3-May-nonchimeras.fasta\r\n",
      "CH408-5-2-May-nc\tfasta\tchimera_detection/CH408-5-2-May/CH408-5-2-May-nonchimeras.fasta\r\n",
      "CH409-5-1-May-nc\tfasta\tchimera_detection/CH409-5-1-May/CH409-5-1-May-nonchimeras.fasta\r\n",
      "CH410-6-8-May-nc\tfasta\tchimera_detection/CH410-6-8-May/CH410-6-8-May-nonchimeras.fasta\r\n",
      "CH501-6-7-May-nc\tfasta\tchimera_detection/CH501-6-7-May/CH501-6-7-May-nonchimeras.fasta\r\n",
      "CH501-pl1-4-7-Oct-nc\tfasta\tchimera_detection/CH501-pl1-4-7-Oct/CH501-pl1-4-7-Oct-nonchimeras.fasta\r\n",
      "CH502-pl1-4-8-Oct-nc\tfasta\tchimera_detection/CH502-pl1-4-8-Oct/CH502-pl1-4-8-Oct-nonchimeras.fasta\r\n",
      "CH503-6-6-May-nc\tfasta\tchimera_detection/CH503-6-6-May/CH503-6-6-May-nonchimeras.fasta\r\n",
      "CH503-pl1-5-1-Oct-nc\tfasta\tchimera_detection/CH503-pl1-5-1-Oct/CH503-pl1-5-1-Oct-nonchimeras.fasta\r\n",
      "CH504-6-5-May-nc\tfasta\tchimera_detection/CH504-6-5-May/CH504-6-5-May-nonchimeras.fasta\r\n",
      "CH504-pl1-5-2-Oct-nc\tfasta\tchimera_detection/CH504-pl1-5-2-Oct/CH504-pl1-5-2-Oct-nonchimeras.fasta\r\n",
      "CH505-6-4-May-nc\tfasta\tchimera_detection/CH505-6-4-May/CH505-6-4-May-nonchimeras.fasta\r\n",
      "CH505-pl1-5-3-Oct-nc\tfasta\tchimera_detection/CH505-pl1-5-3-Oct/CH505-pl1-5-3-Oct-nonchimeras.fasta\r\n",
      "CH506-6-3-May-nc\tfasta\tchimera_detection/CH506-6-3-May/CH506-6-3-May-nonchimeras.fasta\r\n",
      "CH506-pl1-5-4-Oct-nc\tfasta\tchimera_detection/CH506-pl1-5-4-Oct/CH506-pl1-5-4-Oct-nonchimeras.fasta\r\n",
      "CH507-6-2-May-nc\tfasta\tchimera_detection/CH507-6-2-May/CH507-6-2-May-nonchimeras.fasta\r\n",
      "CH507-pl1-5-5-Oct-nc\tfasta\tchimera_detection/CH507-pl1-5-5-Oct/CH507-pl1-5-5-Oct-nonchimeras.fasta\r\n",
      "CH508-6-1-May-nc\tfasta\tchimera_detection/CH508-6-1-May/CH508-6-1-May-nonchimeras.fasta\r\n",
      "CH508-pl1-5-6-Oct-nc\tfasta\tchimera_detection/CH508-pl1-5-6-Oct/CH508-pl1-5-6-Oct-nonchimeras.fasta\r\n",
      "CH509-7-8-May-nc\tfasta\tchimera_detection/CH509-7-8-May/CH509-7-8-May-nonchimeras.fasta\r\n",
      "CH509-pl1-5-7-Oct-nc\tfasta\tchimera_detection/CH509-pl1-5-7-Oct/CH509-pl1-5-7-Oct-nonchimeras.fasta\r\n",
      "CH510-7-7-May-nc\tfasta\tchimera_detection/CH510-7-7-May/CH510-7-7-May-nonchimeras.fasta\r\n",
      "CH510-pl1-5-8-Oct-nc\tfasta\tchimera_detection/CH510-pl1-5-8-Oct/CH510-pl1-5-8-Oct-nonchimeras.fasta\r\n",
      "CH601-pl1-6-1-Oct-nc\tfasta\tchimera_detection/CH601-pl1-6-1-Oct/CH601-pl1-6-1-Oct-nonchimeras.fasta\r\n",
      "CH602-7-6-May-nc\tfasta\tchimera_detection/CH602-7-6-May/CH602-7-6-May-nonchimeras.fasta\r\n",
      "CH602-pl1-6-2-Oct-nc\tfasta\tchimera_detection/CH602-pl1-6-2-Oct/CH602-pl1-6-2-Oct-nonchimeras.fasta\r\n",
      "CH603-pl1-6-3-Oct-nc\tfasta\tchimera_detection/CH603-pl1-6-3-Oct/CH603-pl1-6-3-Oct-nonchimeras.fasta\r\n",
      "CH604-pl1-6-4-Oct-nc\tfasta\tchimera_detection/CH604-pl1-6-4-Oct/CH604-pl1-6-4-Oct-nonchimeras.fasta\r\n",
      "CH605-pl1-6-5-Oct-nc\tfasta\tchimera_detection/CH605-pl1-6-5-Oct/CH605-pl1-6-5-Oct-nonchimeras.fasta\r\n",
      "CH606-pl1-6-6-Oct-nc\tfasta\tchimera_detection/CH606-pl1-6-6-Oct/CH606-pl1-6-6-Oct-nonchimeras.fasta\r\n",
      "CH607-pl1-6-7-Oct-nc\tfasta\tchimera_detection/CH607-pl1-6-7-Oct/CH607-pl1-6-7-Oct-nonchimeras.fasta\r\n",
      "CH608-pl1-6-8-Oct-nc\tfasta\tchimera_detection/CH608-pl1-6-8-Oct/CH608-pl1-6-8-Oct-nonchimeras.fasta\r\n",
      "CH609-pl1-7-1-Oct-nc\tfasta\tchimera_detection/CH609-pl1-7-1-Oct/CH609-pl1-7-1-Oct-nonchimeras.fasta\r\n",
      "CH610-pl1-7-2-Oct-nc\tfasta\tchimera_detection/CH610-pl1-7-2-Oct/CH610-pl1-7-2-Oct-nonchimeras.fasta\r\n",
      "CH-BL1-pl1-7-3-Oct-nc\tfasta\tchimera_detection/CH-BL1-pl1-7-3-Oct/CH-BL1-pl1-7-3-Oct-nonchimeras.fasta\r\n",
      "CH-BL2-pl1-7-4-Oct-nc\tfasta\tchimera_detection/CH-BL2-pl1-7-4-Oct/CH-BL2-pl1-7-4-Oct-nonchimeras.fasta\r\n",
      "eCH1-7-8-May-nc\tfasta\tchimera_detection/eCH1-7-8-May/eCH1-7-8-May-nonchimeras.fasta\r\n",
      "eCH1-pl3-7-1-Oct-nc\tfasta\tchimera_detection/eCH1-pl3-7-1-Oct/eCH1-pl3-7-1-Oct-nonchimeras.fasta\r\n",
      "eCH2-7-7-May-nc\tfasta\tchimera_detection/eCH2-7-7-May/eCH2-7-7-May-nonchimeras.fasta\r\n",
      "eCH2-pl3-7-2-Oct-nc\tfasta\tchimera_detection/eCH2-pl3-7-2-Oct/eCH2-pl3-7-2-Oct-nonchimeras.fasta\r\n",
      "eCH3-7-6-May-nc\tfasta\tchimera_detection/eCH3-7-6-May/eCH3-7-6-May-nonchimeras.fasta\r\n",
      "eCH3-pl3-7-3-Oct-nc\tfasta\tchimera_detection/eCH3-pl3-7-3-Oct/eCH3-pl3-7-3-Oct-nonchimeras.fasta\r\n",
      "eCH4-7-5-May-nc\tfasta\tchimera_detection/eCH4-7-5-May/eCH4-7-5-May-nonchimeras.fasta\r\n",
      "eCH4-pl3-7-4-Oct-nc\tfasta\tchimera_detection/eCH4-pl3-7-4-Oct/eCH4-pl3-7-4-Oct-nonchimeras.fasta\r\n",
      "eCH5-7-4-May-nc\tfasta\tchimera_detection/eCH5-7-4-May/eCH5-7-4-May-nonchimeras.fasta\r\n",
      "eCH5-pl3-7-5-Oct-nc\tfasta\tchimera_detection/eCH5-pl3-7-5-Oct/eCH5-pl3-7-5-Oct-nonchimeras.fasta\r\n",
      "eCH6-7-3-May-nc\tfasta\tchimera_detection/eCH6-7-3-May/eCH6-7-3-May-nonchimeras.fasta\r\n",
      "eCH6-pl3-7-6-Oct-nc\tfasta\tchimera_detection/eCH6-pl3-7-6-Oct/eCH6-pl3-7-6-Oct-nonchimeras.fasta\r\n",
      "eCH-BL-7-2-May-nc\tfasta\tchimera_detection/eCH-BL-7-2-May/eCH-BL-7-2-May-nonchimeras.fasta\r\n",
      "eCH-BL-pl3-6-8-Oct-nc\tfasta\tchimera_detection/eCH-BL-pl3-6-8-Oct/eCH-BL-pl3-6-8-Oct-nonchimeras.fasta\r\n",
      "GLOBAL-nc\tfasta\tchimera_detection/GLOBAL/GLOBAL-nonchimeras.fasta\r\n",
      "neg-6-2-May-nc\tfasta\tchimera_detection/neg-6-2-May/neg-6-2-May-nonchimeras.fasta\r\n",
      "neg-7-5-May-nc\tfasta\tchimera_detection/neg-7-5-May/neg-7-5-May-nonchimeras.fasta\r\n",
      "neg-9-8-May-nc\tfasta\tchimera_detection/neg-9-8-May/neg-9-8-May-nonchimeras.fasta\r\n",
      "pcr-NEG-pl1-7-5-Oct-nc\tfasta\tchimera_detection/pcr-NEG-pl1-7-5-Oct/pcr-NEG-pl1-7-5-Oct-nonchimeras.fasta\r\n",
      "pcr-NEG-pl1-7-7-Oct-nc\tfasta\tchimera_detection/pcr-NEG-pl1-7-7-Oct/pcr-NEG-pl1-7-7-Oct-nonchimeras.fasta\r\n",
      "pcr-NEG-pl3-6-5-Oct-nc\tfasta\tchimera_detection/pcr-NEG-pl3-6-5-Oct/pcr-NEG-pl3-6-5-Oct-nonchimeras.fasta\r\n",
      "pcr-NEG-pl3-6-7-Oct-nc\tfasta\tchimera_detection/pcr-NEG-pl3-6-7-Oct/pcr-NEG-pl3-6-7-Oct-nonchimeras.fasta\r\n",
      "pcr-POS-pl1-7-6-Oct-nc\tfasta\tchimera_detection/pcr-POS-pl1-7-6-Oct/pcr-POS-pl1-7-6-Oct-nonchimeras.fasta\r\n",
      "pcr-POS-pl3-6-6-Oct-nc\tfasta\tchimera_detection/pcr-POS-pl3-6-6-Oct/pcr-POS-pl3-6-6-Oct-nonchimeras.fasta\r\n",
      "pos-9-7-May-nc\tfasta\tchimera_detection/pos-9-7-May/pos-9-7-May-nonchimeras.fasta\r\n"
     ]
    }
   ],
   "source": [
    "!cat nonchimera_Querymap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!sed '/GLOBAL/d' nonchimera_Querymap.txt > nonchimera_Querymap_final.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CH101-1-8-May-nc\tfasta\tchimera_detection/CH101-1-8-May/CH101-1-8-May-nonchimeras.fasta\r\n",
      "CH101-pl1-1-1-Oct-nc\tfasta\tchimera_detection/CH101-pl1-1-1-Oct/CH101-pl1-1-1-Oct-nonchimeras.fasta\r\n",
      "CH102-1-7-May-nc\tfasta\tchimera_detection/CH102-1-7-May/CH102-1-7-May-nonchimeras.fasta\r\n",
      "CH102-pl1-1-2-Oct-nc\tfasta\tchimera_detection/CH102-pl1-1-2-Oct/CH102-pl1-1-2-Oct-nonchimeras.fasta\r\n",
      "CH103-1-6-May-nc\tfasta\tchimera_detection/CH103-1-6-May/CH103-1-6-May-nonchimeras.fasta\r\n",
      "CH103-pl1-1-3-Oct-nc\tfasta\tchimera_detection/CH103-pl1-1-3-Oct/CH103-pl1-1-3-Oct-nonchimeras.fasta\r\n",
      "CH104-1-5-May-nc\tfasta\tchimera_detection/CH104-1-5-May/CH104-1-5-May-nonchimeras.fasta\r\n",
      "CH104-pl1-1-4-Oct-nc\tfasta\tchimera_detection/CH104-pl1-1-4-Oct/CH104-pl1-1-4-Oct-nonchimeras.fasta\r\n",
      "CH105-1-4-May-nc\tfasta\tchimera_detection/CH105-1-4-May/CH105-1-4-May-nonchimeras.fasta\r\n",
      "CH105-pl1-1-5-Oct-nc\tfasta\tchimera_detection/CH105-pl1-1-5-Oct/CH105-pl1-1-5-Oct-nonchimeras.fasta\r\n",
      "CH106-1-3-May-nc\tfasta\tchimera_detection/CH106-1-3-May/CH106-1-3-May-nonchimeras.fasta\r\n",
      "CH106-pl1-1-6-Oct-nc\tfasta\tchimera_detection/CH106-pl1-1-6-Oct/CH106-pl1-1-6-Oct-nonchimeras.fasta\r\n",
      "CH107-1-2-May-nc\tfasta\tchimera_detection/CH107-1-2-May/CH107-1-2-May-nonchimeras.fasta\r\n",
      "CH107-pl1-1-7-Oct-nc\tfasta\tchimera_detection/CH107-pl1-1-7-Oct/CH107-pl1-1-7-Oct-nonchimeras.fasta\r\n",
      "CH108-1-1-May-nc\tfasta\tchimera_detection/CH108-1-1-May/CH108-1-1-May-nonchimeras.fasta\r\n",
      "CH108-pl1-1-8-Oct-nc\tfasta\tchimera_detection/CH108-pl1-1-8-Oct/CH108-pl1-1-8-Oct-nonchimeras.fasta\r\n",
      "CH109-2-8-May-nc\tfasta\tchimera_detection/CH109-2-8-May/CH109-2-8-May-nonchimeras.fasta\r\n",
      "CH109-pl1-2-1-Oct-nc\tfasta\tchimera_detection/CH109-pl1-2-1-Oct/CH109-pl1-2-1-Oct-nonchimeras.fasta\r\n",
      "CH110-2-7-May-nc\tfasta\tchimera_detection/CH110-2-7-May/CH110-2-7-May-nonchimeras.fasta\r\n",
      "CH110-pl1-2-2-Oct-nc\tfasta\tchimera_detection/CH110-pl1-2-2-Oct/CH110-pl1-2-2-Oct-nonchimeras.fasta\r\n",
      "CH111-pl1-2-3-Oct-nc\tfasta\tchimera_detection/CH111-pl1-2-3-Oct/CH111-pl1-2-3-Oct-nonchimeras.fasta\r\n",
      "CH112-pl1-2-4-Oct-nc\tfasta\tchimera_detection/CH112-pl1-2-4-Oct/CH112-pl1-2-4-Oct-nonchimeras.fasta\r\n",
      "CH113-pl1-2-5-Oct-nc\tfasta\tchimera_detection/CH113-pl1-2-5-Oct/CH113-pl1-2-5-Oct-nonchimeras.fasta\r\n",
      "CH114-pl1-2-6-Oct-nc\tfasta\tchimera_detection/CH114-pl1-2-6-Oct/CH114-pl1-2-6-Oct-nonchimeras.fasta\r\n",
      "CH115-pl1-2-7-Oct-nc\tfasta\tchimera_detection/CH115-pl1-2-7-Oct/CH115-pl1-2-7-Oct-nonchimeras.fasta\r\n",
      "CH116-pl1-2-8-Oct-nc\tfasta\tchimera_detection/CH116-pl1-2-8-Oct/CH116-pl1-2-8-Oct-nonchimeras.fasta\r\n",
      "CH117-pl1-3-1-Oct-nc\tfasta\tchimera_detection/CH117-pl1-3-1-Oct/CH117-pl1-3-1-Oct-nonchimeras.fasta\r\n",
      "CH118-pl1-3-2-Oct-nc\tfasta\tchimera_detection/CH118-pl1-3-2-Oct/CH118-pl1-3-2-Oct-nonchimeras.fasta\r\n",
      "CH119-pl1-3-3-Oct-nc\tfasta\tchimera_detection/CH119-pl1-3-3-Oct/CH119-pl1-3-3-Oct-nonchimeras.fasta\r\n",
      "CH201-2-6-May-nc\tfasta\tchimera_detection/CH201-2-6-May/CH201-2-6-May-nonchimeras.fasta\r\n",
      "CH202-2-5-May-nc\tfasta\tchimera_detection/CH202-2-5-May/CH202-2-5-May-nonchimeras.fasta\r\n",
      "CH203-2-4-May-nc\tfasta\tchimera_detection/CH203-2-4-May/CH203-2-4-May-nonchimeras.fasta\r\n",
      "CH204-2-3-May-nc\tfasta\tchimera_detection/CH204-2-3-May/CH204-2-3-May-nonchimeras.fasta\r\n",
      "CH205-2-2-May-nc\tfasta\tchimera_detection/CH205-2-2-May/CH205-2-2-May-nonchimeras.fasta\r\n",
      "CH206-2-1-May-nc\tfasta\tchimera_detection/CH206-2-1-May/CH206-2-1-May-nonchimeras.fasta\r\n",
      "CH207-3-8-May-nc\tfasta\tchimera_detection/CH207-3-8-May/CH207-3-8-May-nonchimeras.fasta\r\n",
      "CH208-3-7-May-nc\tfasta\tchimera_detection/CH208-3-7-May/CH208-3-7-May-nonchimeras.fasta\r\n",
      "CH209-3-6-May-nc\tfasta\tchimera_detection/CH209-3-6-May/CH209-3-6-May-nonchimeras.fasta\r\n",
      "CH210-3-5-May-nc\tfasta\tchimera_detection/CH210-3-5-May/CH210-3-5-May-nonchimeras.fasta\r\n",
      "CH211-3-4-May-nc\tfasta\tchimera_detection/CH211-3-4-May/CH211-3-4-May-nonchimeras.fasta\r\n",
      "CH301-pl1-3-4-Oct-nc\tfasta\tchimera_detection/CH301-pl1-3-4-Oct/CH301-pl1-3-4-Oct-nonchimeras.fasta\r\n",
      "CH302-pl1-3-5-Oct-nc\tfasta\tchimera_detection/CH302-pl1-3-5-Oct/CH302-pl1-3-5-Oct-nonchimeras.fasta\r\n",
      "CH303-pl1-3-6-Oct-nc\tfasta\tchimera_detection/CH303-pl1-3-6-Oct/CH303-pl1-3-6-Oct-nonchimeras.fasta\r\n",
      "CH304-pl1-3-7-Oct-nc\tfasta\tchimera_detection/CH304-pl1-3-7-Oct/CH304-pl1-3-7-Oct-nonchimeras.fasta\r\n",
      "CH305-pl1-3-8-Oct-nc\tfasta\tchimera_detection/CH305-pl1-3-8-Oct/CH305-pl1-3-8-Oct-nonchimeras.fasta\r\n",
      "CH306-pl1-4-1-Oct-nc\tfasta\tchimera_detection/CH306-pl1-4-1-Oct/CH306-pl1-4-1-Oct-nonchimeras.fasta\r\n",
      "CH307-pl1-4-2-Oct-nc\tfasta\tchimera_detection/CH307-pl1-4-2-Oct/CH307-pl1-4-2-Oct-nonchimeras.fasta\r\n",
      "CH308-3-3-May-nc\tfasta\tchimera_detection/CH308-3-3-May/CH308-3-3-May-nonchimeras.fasta\r\n",
      "CH308-pl1-4-3-Oct-nc\tfasta\tchimera_detection/CH308-pl1-4-3-Oct/CH308-pl1-4-3-Oct-nonchimeras.fasta\r\n",
      "CH309-3-2-May-nc\tfasta\tchimera_detection/CH309-3-2-May/CH309-3-2-May-nonchimeras.fasta\r\n",
      "CH309-pl1-4-4-Oct-nc\tfasta\tchimera_detection/CH309-pl1-4-4-Oct/CH309-pl1-4-4-Oct-nonchimeras.fasta\r\n",
      "CH310-3-1-May-nc\tfasta\tchimera_detection/CH310-3-1-May/CH310-3-1-May-nonchimeras.fasta\r\n",
      "CH310-pl1-4-5-Oct-nc\tfasta\tchimera_detection/CH310-pl1-4-5-Oct/CH310-pl1-4-5-Oct-nonchimeras.fasta\r\n",
      "CH311-4-8-May-nc\tfasta\tchimera_detection/CH311-4-8-May/CH311-4-8-May-nonchimeras.fasta\r\n",
      "CH312-4-7-May-nc\tfasta\tchimera_detection/CH312-4-7-May/CH312-4-7-May-nonchimeras.fasta\r\n",
      "CH313-4-6-May-nc\tfasta\tchimera_detection/CH313-4-6-May/CH313-4-6-May-nonchimeras.fasta\r\n",
      "CH314-4-5-May-nc\tfasta\tchimera_detection/CH314-4-5-May/CH314-4-5-May-nonchimeras.fasta\r\n",
      "CH315-4-4-May-nc\tfasta\tchimera_detection/CH315-4-4-May/CH315-4-4-May-nonchimeras.fasta\r\n",
      "CH316-4-3-May-nc\tfasta\tchimera_detection/CH316-4-3-May/CH316-4-3-May-nonchimeras.fasta\r\n",
      "CH317-4-2-May-nc\tfasta\tchimera_detection/CH317-4-2-May/CH317-4-2-May-nonchimeras.fasta\r\n",
      "CH401-4-1-May-nc\tfasta\tchimera_detection/CH401-4-1-May/CH401-4-1-May-nonchimeras.fasta\r\n",
      "CH401-pl1-4-6-Oct-nc\tfasta\tchimera_detection/CH401-pl1-4-6-Oct/CH401-pl1-4-6-Oct-nonchimeras.fasta\r\n",
      "CH402-5-8-May-nc\tfasta\tchimera_detection/CH402-5-8-May/CH402-5-8-May-nonchimeras.fasta\r\n",
      "CH403-5-7-May-nc\tfasta\tchimera_detection/CH403-5-7-May/CH403-5-7-May-nonchimeras.fasta\r\n",
      "CH404-5-6-May-nc\tfasta\tchimera_detection/CH404-5-6-May/CH404-5-6-May-nonchimeras.fasta\r\n",
      "CH405-5-5-May-nc\tfasta\tchimera_detection/CH405-5-5-May/CH405-5-5-May-nonchimeras.fasta\r\n",
      "CH406-5-4-May-nc\tfasta\tchimera_detection/CH406-5-4-May/CH406-5-4-May-nonchimeras.fasta\r\n",
      "CH407-5-3-May-nc\tfasta\tchimera_detection/CH407-5-3-May/CH407-5-3-May-nonchimeras.fasta\r\n",
      "CH408-5-2-May-nc\tfasta\tchimera_detection/CH408-5-2-May/CH408-5-2-May-nonchimeras.fasta\r\n",
      "CH409-5-1-May-nc\tfasta\tchimera_detection/CH409-5-1-May/CH409-5-1-May-nonchimeras.fasta\r\n",
      "CH410-6-8-May-nc\tfasta\tchimera_detection/CH410-6-8-May/CH410-6-8-May-nonchimeras.fasta\r\n",
      "CH501-6-7-May-nc\tfasta\tchimera_detection/CH501-6-7-May/CH501-6-7-May-nonchimeras.fasta\r\n",
      "CH501-pl1-4-7-Oct-nc\tfasta\tchimera_detection/CH501-pl1-4-7-Oct/CH501-pl1-4-7-Oct-nonchimeras.fasta\r\n",
      "CH502-pl1-4-8-Oct-nc\tfasta\tchimera_detection/CH502-pl1-4-8-Oct/CH502-pl1-4-8-Oct-nonchimeras.fasta\r\n",
      "CH503-6-6-May-nc\tfasta\tchimera_detection/CH503-6-6-May/CH503-6-6-May-nonchimeras.fasta\r\n",
      "CH503-pl1-5-1-Oct-nc\tfasta\tchimera_detection/CH503-pl1-5-1-Oct/CH503-pl1-5-1-Oct-nonchimeras.fasta\r\n",
      "CH504-6-5-May-nc\tfasta\tchimera_detection/CH504-6-5-May/CH504-6-5-May-nonchimeras.fasta\r\n",
      "CH504-pl1-5-2-Oct-nc\tfasta\tchimera_detection/CH504-pl1-5-2-Oct/CH504-pl1-5-2-Oct-nonchimeras.fasta\r\n",
      "CH505-6-4-May-nc\tfasta\tchimera_detection/CH505-6-4-May/CH505-6-4-May-nonchimeras.fasta\r\n",
      "CH505-pl1-5-3-Oct-nc\tfasta\tchimera_detection/CH505-pl1-5-3-Oct/CH505-pl1-5-3-Oct-nonchimeras.fasta\r\n",
      "CH506-6-3-May-nc\tfasta\tchimera_detection/CH506-6-3-May/CH506-6-3-May-nonchimeras.fasta\r\n",
      "CH506-pl1-5-4-Oct-nc\tfasta\tchimera_detection/CH506-pl1-5-4-Oct/CH506-pl1-5-4-Oct-nonchimeras.fasta\r\n",
      "CH507-6-2-May-nc\tfasta\tchimera_detection/CH507-6-2-May/CH507-6-2-May-nonchimeras.fasta\r\n",
      "CH507-pl1-5-5-Oct-nc\tfasta\tchimera_detection/CH507-pl1-5-5-Oct/CH507-pl1-5-5-Oct-nonchimeras.fasta\r\n",
      "CH508-6-1-May-nc\tfasta\tchimera_detection/CH508-6-1-May/CH508-6-1-May-nonchimeras.fasta\r\n",
      "CH508-pl1-5-6-Oct-nc\tfasta\tchimera_detection/CH508-pl1-5-6-Oct/CH508-pl1-5-6-Oct-nonchimeras.fasta\r\n",
      "CH509-7-8-May-nc\tfasta\tchimera_detection/CH509-7-8-May/CH509-7-8-May-nonchimeras.fasta\r\n",
      "CH509-pl1-5-7-Oct-nc\tfasta\tchimera_detection/CH509-pl1-5-7-Oct/CH509-pl1-5-7-Oct-nonchimeras.fasta\r\n",
      "CH510-7-7-May-nc\tfasta\tchimera_detection/CH510-7-7-May/CH510-7-7-May-nonchimeras.fasta\r\n",
      "CH510-pl1-5-8-Oct-nc\tfasta\tchimera_detection/CH510-pl1-5-8-Oct/CH510-pl1-5-8-Oct-nonchimeras.fasta\r\n",
      "CH601-pl1-6-1-Oct-nc\tfasta\tchimera_detection/CH601-pl1-6-1-Oct/CH601-pl1-6-1-Oct-nonchimeras.fasta\r\n",
      "CH602-7-6-May-nc\tfasta\tchimera_detection/CH602-7-6-May/CH602-7-6-May-nonchimeras.fasta\r\n",
      "CH602-pl1-6-2-Oct-nc\tfasta\tchimera_detection/CH602-pl1-6-2-Oct/CH602-pl1-6-2-Oct-nonchimeras.fasta\r\n",
      "CH603-pl1-6-3-Oct-nc\tfasta\tchimera_detection/CH603-pl1-6-3-Oct/CH603-pl1-6-3-Oct-nonchimeras.fasta\r\n",
      "CH604-pl1-6-4-Oct-nc\tfasta\tchimera_detection/CH604-pl1-6-4-Oct/CH604-pl1-6-4-Oct-nonchimeras.fasta\r\n",
      "CH605-pl1-6-5-Oct-nc\tfasta\tchimera_detection/CH605-pl1-6-5-Oct/CH605-pl1-6-5-Oct-nonchimeras.fasta\r\n",
      "CH606-pl1-6-6-Oct-nc\tfasta\tchimera_detection/CH606-pl1-6-6-Oct/CH606-pl1-6-6-Oct-nonchimeras.fasta\r\n",
      "CH607-pl1-6-7-Oct-nc\tfasta\tchimera_detection/CH607-pl1-6-7-Oct/CH607-pl1-6-7-Oct-nonchimeras.fasta\r\n",
      "CH608-pl1-6-8-Oct-nc\tfasta\tchimera_detection/CH608-pl1-6-8-Oct/CH608-pl1-6-8-Oct-nonchimeras.fasta\r\n",
      "CH609-pl1-7-1-Oct-nc\tfasta\tchimera_detection/CH609-pl1-7-1-Oct/CH609-pl1-7-1-Oct-nonchimeras.fasta\r\n",
      "CH610-pl1-7-2-Oct-nc\tfasta\tchimera_detection/CH610-pl1-7-2-Oct/CH610-pl1-7-2-Oct-nonchimeras.fasta\r\n",
      "CH-BL1-pl1-7-3-Oct-nc\tfasta\tchimera_detection/CH-BL1-pl1-7-3-Oct/CH-BL1-pl1-7-3-Oct-nonchimeras.fasta\r\n",
      "CH-BL2-pl1-7-4-Oct-nc\tfasta\tchimera_detection/CH-BL2-pl1-7-4-Oct/CH-BL2-pl1-7-4-Oct-nonchimeras.fasta\r\n",
      "eCH1-7-8-May-nc\tfasta\tchimera_detection/eCH1-7-8-May/eCH1-7-8-May-nonchimeras.fasta\r\n",
      "eCH1-pl3-7-1-Oct-nc\tfasta\tchimera_detection/eCH1-pl3-7-1-Oct/eCH1-pl3-7-1-Oct-nonchimeras.fasta\r\n",
      "eCH2-7-7-May-nc\tfasta\tchimera_detection/eCH2-7-7-May/eCH2-7-7-May-nonchimeras.fasta\r\n",
      "eCH2-pl3-7-2-Oct-nc\tfasta\tchimera_detection/eCH2-pl3-7-2-Oct/eCH2-pl3-7-2-Oct-nonchimeras.fasta\r\n",
      "eCH3-7-6-May-nc\tfasta\tchimera_detection/eCH3-7-6-May/eCH3-7-6-May-nonchimeras.fasta\r\n",
      "eCH3-pl3-7-3-Oct-nc\tfasta\tchimera_detection/eCH3-pl3-7-3-Oct/eCH3-pl3-7-3-Oct-nonchimeras.fasta\r\n",
      "eCH4-7-5-May-nc\tfasta\tchimera_detection/eCH4-7-5-May/eCH4-7-5-May-nonchimeras.fasta\r\n",
      "eCH4-pl3-7-4-Oct-nc\tfasta\tchimera_detection/eCH4-pl3-7-4-Oct/eCH4-pl3-7-4-Oct-nonchimeras.fasta\r\n",
      "eCH5-7-4-May-nc\tfasta\tchimera_detection/eCH5-7-4-May/eCH5-7-4-May-nonchimeras.fasta\r\n",
      "eCH5-pl3-7-5-Oct-nc\tfasta\tchimera_detection/eCH5-pl3-7-5-Oct/eCH5-pl3-7-5-Oct-nonchimeras.fasta\r\n",
      "eCH6-7-3-May-nc\tfasta\tchimera_detection/eCH6-7-3-May/eCH6-7-3-May-nonchimeras.fasta\r\n",
      "eCH6-pl3-7-6-Oct-nc\tfasta\tchimera_detection/eCH6-pl3-7-6-Oct/eCH6-pl3-7-6-Oct-nonchimeras.fasta\r\n",
      "eCH-BL-7-2-May-nc\tfasta\tchimera_detection/eCH-BL-7-2-May/eCH-BL-7-2-May-nonchimeras.fasta\r\n",
      "eCH-BL-pl3-6-8-Oct-nc\tfasta\tchimera_detection/eCH-BL-pl3-6-8-Oct/eCH-BL-pl3-6-8-Oct-nonchimeras.fasta\r\n",
      "neg-6-2-May-nc\tfasta\tchimera_detection/neg-6-2-May/neg-6-2-May-nonchimeras.fasta\r\n",
      "neg-7-5-May-nc\tfasta\tchimera_detection/neg-7-5-May/neg-7-5-May-nonchimeras.fasta\r\n",
      "neg-9-8-May-nc\tfasta\tchimera_detection/neg-9-8-May/neg-9-8-May-nonchimeras.fasta\r\n",
      "pcr-NEG-pl1-7-5-Oct-nc\tfasta\tchimera_detection/pcr-NEG-pl1-7-5-Oct/pcr-NEG-pl1-7-5-Oct-nonchimeras.fasta\r\n",
      "pcr-NEG-pl1-7-7-Oct-nc\tfasta\tchimera_detection/pcr-NEG-pl1-7-7-Oct/pcr-NEG-pl1-7-7-Oct-nonchimeras.fasta\r\n",
      "pcr-NEG-pl3-6-5-Oct-nc\tfasta\tchimera_detection/pcr-NEG-pl3-6-5-Oct/pcr-NEG-pl3-6-5-Oct-nonchimeras.fasta\r\n",
      "pcr-NEG-pl3-6-7-Oct-nc\tfasta\tchimera_detection/pcr-NEG-pl3-6-7-Oct/pcr-NEG-pl3-6-7-Oct-nonchimeras.fasta\r\n",
      "pcr-POS-pl1-7-6-Oct-nc\tfasta\tchimera_detection/pcr-POS-pl1-7-6-Oct/pcr-POS-pl1-7-6-Oct-nonchimeras.fasta\r\n",
      "pcr-POS-pl3-6-6-Oct-nc\tfasta\tchimera_detection/pcr-POS-pl3-6-6-Oct/pcr-POS-pl3-6-6-Oct-nonchimeras.fasta\r\n",
      "pos-9-7-May-nc\tfasta\tchimera_detection/pos-9-7-May/pos-9-7-May-nonchimeras.fasta\r\n"
     ]
    }
   ],
   "source": [
    "!cat nonchimera_Querymap_final.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We now run the BLAST search of the non-chimera sequences against the reference sequences we downloaded from literature and that were saved as `REFlist.txt`.\n",
    "We clustered the sequences with 97% match, retaining clusters that have a minimum of 6 sequences in each. For the identity assignment we used a minimum identity of 97% with a minimum alignment lenght of 85%."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!metaBEAT_global.py \\\n",
    "-Q nonchimera_Querymap_final.txt \\\n",
    "-R REFlist.txt \\\n",
    "--cluster --clust_match 0.97 --clust_cov 6 \\\n",
    "--blast --min_ident 0.97 --min_ali_length 0.85 \\\n",
    "-m COI -n 5 -v -@ m.benucci@2015.hull.ac.uk \\\n",
    "-o COI_28062019_merged-only_nonchimera_cl97cov6_blast_min97_ali0.85_ref > metaBEAT.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "metaBEAT - metaBarcoding and Environmental DNA Analyses tool\r\n",
      "version: v.0.97.11-global\r\n",
      "\r\n",
      "\r\n",
      "Wed Jul 24 17:45:42 2019\r\n",
      "\r\n",
      "/usr/bin/metaBEAT_global.py -Q nonchimera_Querymap_final.txt -R REFlist.txt --cluster --clust_match 0.97 --clust_cov 6 --blast --min_ident 0.97 --min_ali_length 0.85 -m COI -n 5 -v -@ m.benucci@2015.hull.ac.uk -o COI_28062019_merged-only_nonchimera_cl97cov6_blast_min97_ali0.85_ref\r\n",
      "\r\n",
      "\r\n",
      "metaBEAT may be querying NCBI's Entrez databases to fetch/verify taxonomic ids. Entrez User requirements state that you need to identify yourself by providing an email address so that NCBI can contact you in case there is a problem.\r\n",
      "\r\n",
      "You have specified: 'm.benucci@2015.hull.ac.uk'\r\n",
      "\r\n",
      "taxonomy.db found at /usr/bin/taxonomy.db\r\n",
      "Taxonomy database is 0 days old - good!\r\n",
      "\r\n",
      "Parsing querylist file\r\n",
      "\r\n",
      "Number of samples to process: 127\r\n",
      "Sequence input format: defaultdict(<type 'int'>, {'fasta': 127})\r\n",
      "Barcodes for demultiplexing provided for 0 samples\r\n",
      "Cropping instructions provided for 0 samples\r\n",
      "\r\n",
      "\r\n",
      "######## PROCESSING REFERENCE DATA ########\r\n",
      "\r\n",
      "\r\n",
      "processing ../1-NCBI_references/C.islandicus_NCBI_2018.gb (containing 4 records)\r\n",
      "\r\n",
      "total number of valid records: 4\r\n",
      "\r\n",
      "\r\n",
      "processing ../1-NCBI_references/Synurella_NCBI_2018.gb (containing 3 records)\r\n",
      "\r\n",
      "total number of valid records: 7\r\n",
      "\r\n",
      "\r\n",
      "processing ../1-NCBI_references/O.bicornis_NCBI_2018.gb (containing 2 records)\r\n",
      "\r\n",
      "total number of valid records: 9\r\n",
      "\r\n",
      "\r\n",
      "processing ../1-NCBI_references/Crangonyx_NCBI_2018.gb (containing 62 records)\r\n",
      "\r\n",
      "total number of valid records: 71\r\n",
      "\r\n",
      "\r\n",
      "Wed Jul 24 17:45:42 2019\r\n",
      "\r\n",
      "\r\n",
      "##### processing query ID: CH-BL1-pl1-7-3-Oct-nc #####\r\n",
      "\r\n",
      "\r\n",
      "WARNING - EXPECTING input data to come in SINGLE FASTA FILE\r\n",
      "\r\n",
      "\r\n",
      "### CLUSTERING ###\r\n",
      "\r\n",
      "\r\n",
      "clustering using vsearch\r\n",
      "vsearch --cluster_fast /home/working/C_floridanus/2-metaBEAT/chimera_detection/CH-BL1-pl1-7-3-Oct/CH-BL1-pl1-7-3-Oct-nonchimeras.fasta --id 0.97 --strand both --threads 5 --centroids CH-BL1-pl1-7-3-Oct-nc_centroids.fasta --uc CH-BL1-pl1-7-3-Oct-nc.uc\r\n",
      "vsearch v1.1.0_linux_x86_64, 15.6GB RAM, 8 cores\r\n",
      "https://github.com/torognes/vsearch\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "parse vsearch uc file\r\n",
      "\r\n",
      "\r\n",
      "reduce cluster files - minimum coverage: 6\r\n",
      "\r\n",
      "1_2111_14114_15025_1_ex - ok\r\n",
      "1_1109_18398_5634_1_ex - ok\r\n",
      "excluding 1_2114_25062_15405_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "excluding 1_1101_13542_15692_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "excluding 1_1101_4975_14261_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "1_1102_13795_3394_1_ex - ok\r\n",
      "excluding 1_1104_18123_16328_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "1_1105_5808_23415_1_ex - ok\r\n",
      "excluding 1_1112_18787_22456_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "excluding 1_2102_24781_13477_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "excluding 1_2107_11982_15436_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "excluding 1_2108_7825_11596_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "excluding 1_2110_25044_11899_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "excluding 1_2112_24824_25633_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "excluding 1_2112_4429_17086_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n",
      "excluding 1_2113_22505_11687_1_ex from further analyses - cluster_size filter\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 100 metaBEAT.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "assigned LCA Crangonyx floridanus (taxid 326578) at level species\r\n",
      "\r\n",
      "attempting LCA assignment for CH103-pl1-1-3-Oct-nc|1_1109_8180_18246_1_ex\r\n",
      "found LCA 326578 at level species\r\n",
      "assigned LCA Crangonyx floridanus (taxid 326578) at level species\r\n",
      "\r\n",
      "attempting LCA assignment for CH119-pl1-3-3-Oct-nc|1_1103_9815_11448_1_ex\r\n",
      "found LCA 326578 at level species\r\n",
      "assigned LCA Crangonyx floridanus (taxid 326578) at level species\r\n",
      "\r\n",
      "attempting LCA assignment for CH505-pl1-5-3-Oct-nc|1_2111_20094_14700_1_ex\r\n",
      "found LCA 326578 at level species\r\n",
      "assigned LCA Crangonyx floridanus (taxid 326578) at level species\r\n",
      "\r\n",
      "attempting LCA assignment for CH105-1-4-May-nc|1_2105_9715_15085_1_ex\r\n",
      "found LCA 326568 at level species\r\n",
      "assigned LCA Crangonyx pseudogracilis (taxid 326568) at level species\r\n",
      "\r\n",
      "attempting LCA assignment for CH103-1-6-May-nc|1_2106_20229_21512_1_ex\r\n",
      "found LCA 326568 at level species\r\n",
      "assigned LCA Crangonyx pseudogracilis (taxid 326568) at level species\r\n",
      "\r\n",
      "attempting LCA assignment for pcr-POS-pl3-6-6-Oct-nc|1_2105_26198_12675_1_ex\r\n",
      "found LCA 1437190 at level species\r\n",
      "assigned LCA Osmia bicornis (taxid 1437190) at level species\r\n",
      "\r\n",
      "attempting LCA assignment for CH303-pl1-3-6-Oct-nc|1_2111_11656_11528_1_ex\r\n",
      "found LCA 326578 at level species\r\n",
      "assigned LCA Crangonyx floridanus (taxid 326578) at level species\r\n",
      "\r\n",
      "all queries have been successfully assigned to a taxonomy\r\n",
      "\r\n",
      "\r\n",
      "##### FORMATTING AND WRITING BIOM OUTPUT #####\r\n",
      "\r\n",
      "\r\n",
      "##### BLAST ANALYSIS DONE #####\r\n",
      "\r\n",
      "Find result tables in '/home/working/C_floridanus/2-metaBEAT/GLOBAL/BLAST_0.97'\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "Wed Jul 24 17:57:13 2019\r\n",
      "\r\n",
      "\r\n",
      "##### DONE! #####\r\n",
      "\r\n",
      "\r\n",
      "Wed Jul 24 17:57:13 2019\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!tail -n 50 metaBEAT.log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "The final OTU table can be found in the file: \n",
    "\n",
    "`GLOBAL/BLAST_0.97/COI_28062019_merged-only_nonchimera_cl97cov6_blast_min97_ali0.85_ref-by-taxonomy-readcounts.blast.tsv`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extracting now the sequences we are interested in."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import metaBEAT_global_misc_functions as mb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Identify samples which contained reads assigned to _C. floridanus_ before filtering."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Specified BIOM input format 'json' - ok!\n",
      "CH-BL1-pl1-7-3-Oct-nc.blast\t(90.0383 %)\n",
      "CH-BL2-pl1-7-4-Oct-nc.blast\t(88.2554 %)\n",
      "CH101-1-8-May-nc.blast\t(99.5666 %)\n",
      "CH101-pl1-1-1-Oct-nc.blast\t(99.7500 %)\n",
      "CH102-1-7-May-nc.blast\t(99.4167 %)\n",
      "CH102-pl1-1-2-Oct-nc.blast\t(99.6956 %)\n",
      "CH103-1-6-May-nc.blast\t(0.0644 %)\n",
      "CH103-pl1-1-3-Oct-nc.blast\t(98.9692 %)\n",
      "CH104-1-5-May-nc.blast\t(99.6886 %)\n",
      "CH104-pl1-1-4-Oct-nc.blast\t(95.2475 %)\n",
      "CH105-1-4-May-nc.blast\t(0.1071 %)\n",
      "CH105-pl1-1-5-Oct-nc.blast\t(99.6454 %)\n",
      "CH106-1-3-May-nc.blast\t(0.0627 %)\n",
      "CH106-pl1-1-6-Oct-nc.blast\t(98.1265 %)\n",
      "CH107-1-2-May-nc.blast\t(98.9977 %)\n",
      "CH107-pl1-1-7-Oct-nc.blast\t(99.8794 %)\n",
      "CH108-1-1-May-nc.blast\t(99.8094 %)\n",
      "CH108-pl1-1-8-Oct-nc.blast\t(99.6482 %)\n",
      "CH109-2-8-May-nc.blast\t(99.6020 %)\n",
      "CH109-pl1-2-1-Oct-nc.blast\t(94.7490 %)\n",
      "CH110-2-7-May-nc.blast\t(99.8381 %)\n",
      "CH110-pl1-2-2-Oct-nc.blast\t(94.2847 %)\n",
      "CH111-pl1-2-3-Oct-nc.blast\t(21.1116 %)\n",
      "CH112-pl1-2-4-Oct-nc.blast\t(97.5969 %)\n",
      "CH113-pl1-2-5-Oct-nc.blast\t(89.8442 %)\n",
      "CH114-pl1-2-6-Oct-nc.blast\t(77.9786 %)\n",
      "CH115-pl1-2-7-Oct-nc.blast\t(95.0008 %)\n",
      "CH116-pl1-2-8-Oct-nc.blast\t(89.6136 %)\n",
      "CH117-pl1-3-1-Oct-nc.blast\t(95.2146 %)\n",
      "CH118-pl1-3-2-Oct-nc.blast\t(93.2841 %)\n",
      "CH119-pl1-3-3-Oct-nc.blast\t(86.7163 %)\n",
      "CH201-2-6-May-nc.blast\t(0.3123 %)\n",
      "CH202-2-5-May-nc.blast\t(99.8612 %)\n",
      "CH203-2-4-May-nc.blast\t(19.0871 %)\n",
      "CH204-2-3-May-nc.blast\t(4.7790 %)\n",
      "CH205-2-2-May-nc.blast\t(0.7772 %)\n",
      "CH206-2-1-May-nc.blast\t(92.6752 %)\n",
      "CH207-3-8-May-nc.blast\t(100.0000 %)\n",
      "CH208-3-7-May-nc.blast\t(99.8739 %)\n",
      "CH209-3-6-May-nc.blast\t(97.3219 %)\n",
      "CH210-3-5-May-nc.blast\t(94.2414 %)\n",
      "CH211-3-4-May-nc.blast\t(0.0168 %)\n",
      "CH301-pl1-3-4-Oct-nc.blast\t(93.7817 %)\n",
      "CH302-pl1-3-5-Oct-nc.blast\t(26.0771 %)\n",
      "CH303-pl1-3-6-Oct-nc.blast\t(83.2712 %)\n",
      "CH304-pl1-3-7-Oct-nc.blast\t(94.5074 %)\n",
      "CH305-pl1-3-8-Oct-nc.blast\t(67.9007 %)\n",
      "CH306-pl1-4-1-Oct-nc.blast\t(90.2895 %)\n",
      "CH307-pl1-4-2-Oct-nc.blast\t(85.3880 %)\n",
      "CH308-pl1-4-3-Oct-nc.blast\t(11.0938 %)\n",
      "CH309-3-2-May-nc.blast\t(2.0602 %)\n",
      "CH309-pl1-4-4-Oct-nc.blast\t(86.7605 %)\n",
      "CH310-3-1-May-nc.blast\t(8.2596 %)\n",
      "CH310-pl1-4-5-Oct-nc.blast\t(78.2097 %)\n",
      "CH401-pl1-4-6-Oct-nc.blast\t(12.7197 %)\n",
      "CH403-5-7-May-nc.blast\t(0.0362 %)\n",
      "CH405-5-5-May-nc.blast\t(0.0325 %)\n",
      "CH501-6-7-May-nc.blast\t(1.1431 %)\n",
      "CH501-pl1-4-7-Oct-nc.blast\t(56.7290 %)\n",
      "CH502-pl1-4-8-Oct-nc.blast\t(20.9107 %)\n",
      "CH503-pl1-5-1-Oct-nc.blast\t(95.4929 %)\n",
      "CH504-6-5-May-nc.blast\t(92.8809 %)\n",
      "CH504-pl1-5-2-Oct-nc.blast\t(94.6672 %)\n",
      "CH505-pl1-5-3-Oct-nc.blast\t(85.8449 %)\n",
      "CH506-pl1-5-4-Oct-nc.blast\t(72.0645 %)\n",
      "CH507-6-2-May-nc.blast\t(100.0000 %)\n",
      "CH507-pl1-5-5-Oct-nc.blast\t(85.1517 %)\n",
      "CH508-6-1-May-nc.blast\t(100.0000 %)\n",
      "CH508-pl1-5-6-Oct-nc.blast\t(16.1260 %)\n",
      "CH509-7-8-May-nc.blast\t(100.0000 %)\n",
      "CH509-pl1-5-7-Oct-nc.blast\t(93.8691 %)\n",
      "CH510-7-7-May-nc.blast\t(0.6258 %)\n",
      "CH510-pl1-5-8-Oct-nc.blast\t(89.2541 %)\n",
      "CH601-pl1-6-1-Oct-nc.blast\t(94.5790 %)\n",
      "CH602-pl1-6-2-Oct-nc.blast\t(94.6124 %)\n",
      "CH603-pl1-6-3-Oct-nc.blast\t(17.4641 %)\n",
      "CH604-pl1-6-4-Oct-nc.blast\t(95.1357 %)\n",
      "CH605-pl1-6-5-Oct-nc.blast\t(88.2136 %)\n",
      "CH606-pl1-6-6-Oct-nc.blast\t(84.8564 %)\n",
      "CH607-pl1-6-7-Oct-nc.blast\t(91.8847 %)\n",
      "CH608-pl1-6-8-Oct-nc.blast\t(88.1507 %)\n",
      "CH609-pl1-7-1-Oct-nc.blast\t(98.5925 %)\n",
      "CH610-pl1-7-2-Oct-nc.blast\t(98.3307 %)\n",
      "neg-7-5-May-nc.blast\t(18.1818 %)\n",
      "pcr-NEG-pl1-7-5-Oct-nc.blast\t(86.9013 %)\n",
      "pcr-NEG-pl1-7-7-Oct-nc.blast\t(89.3382 %)\n",
      "pcr-POS-pl1-7-6-Oct-nc.blast\t(41.0218 %)\n"
     ]
    }
   ],
   "source": [
    "mb.find_target(BIOM=mb.load_BIOM('GLOBAL/BLAST_0.97/COI_28062019_merged-only_nonchimera_cl97cov6_blast_min97_ali0.85_ref-by-taxonomy-readcounts.blast.biom'), \n",
    "               target='Crangonyx_floridanus')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The OTU by taxonomy `GLOBAL/BLAST_0.97/COI_28062019_merged-only_nonchimera_cl97cov6_blast_min97_ali0.85_ref-OTU-taxonomy.blast.tsv` shows _Crangonyx floridanus_ (_Cf_) and _Crangonyx pseudogracilis_ (_Cp_) OTUs contain 87.9% of total N of reads. 2 OTUs in particular contain 87.5% of the total N reads; respectively 1 OTU with 51.6% reads belonging to _Cf_, and 1 OTU with 35.9% belonging to _Cp_.\n",
    "\n",
    "We filter the raw OTU table, and in a given sample we remove OTUs that were not supported by at least 2% of the reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Specified BIOM input format 'json' - ok!\n",
      "\n",
      "Filtering at level: 2.0 %\n",
      "\n",
      "Removing 610 OTUs for lack of support\n",
      "\n",
      "Writing 'filtered.biom'\n",
      "Writing 'filtered.tsv'\n",
      "Writing 'filtered-collapsed.biom'\n",
      "Writing 'filtered-collapsed.tsv'\n"
     ]
    }
   ],
   "source": [
    "#load raw OTU table\n",
    "to_filter = mb.load_BIOM(table='GLOBAL/BLAST_0.97/COI_28062019_merged-only_nonchimera_cl97cov6_blast_min97_ali0.85_ref-OTU-taxonomy.blast.biom')\n",
    "\n",
    "#filter at 1%\n",
    "filtered = mb.filter_BIOM_by_per_sample_read_prop(BIOM=to_filter, min_prop=0.02)\n",
    "\n",
    "#write to file\n",
    "mb.write_BIOM(filtered, target_prefix='filtered' )\n",
    "\n",
    "#collapse OTUs by taxonomy\n",
    "filtered_collapsed = mb.collapse_biom_by_taxonomy(in_table=filtered)\n",
    "\n",
    "#write to file\n",
    "mb.write_BIOM(filtered_collapsed, target_prefix='filtered-collapsed')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "we identify samples containing sequences assigned to _C. floridanus_."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CH-BL1-pl1-7-3-Oct-nc.blast\t(90.7552 %)\n",
      "CH-BL2-pl1-7-4-Oct-nc.blast\t(88.2554 %)\n",
      "CH101-1-8-May-nc.blast\t(100.0000 %)\n",
      "CH101-pl1-1-1-Oct-nc.blast\t(100.0000 %)\n",
      "CH102-1-7-May-nc.blast\t(100.0000 %)\n",
      "CH102-pl1-1-2-Oct-nc.blast\t(100.0000 %)\n",
      "CH103-pl1-1-3-Oct-nc.blast\t(100.0000 %)\n",
      "CH104-1-5-May-nc.blast\t(100.0000 %)\n",
      "CH104-pl1-1-4-Oct-nc.blast\t(100.0000 %)\n",
      "CH105-pl1-1-5-Oct-nc.blast\t(100.0000 %)\n",
      "CH106-pl1-1-6-Oct-nc.blast\t(100.0000 %)\n",
      "CH107-1-2-May-nc.blast\t(100.0000 %)\n",
      "CH107-pl1-1-7-Oct-nc.blast\t(100.0000 %)\n",
      "CH108-1-1-May-nc.blast\t(100.0000 %)\n",
      "CH108-pl1-1-8-Oct-nc.blast\t(100.0000 %)\n",
      "CH109-2-8-May-nc.blast\t(100.0000 %)\n",
      "CH109-pl1-2-1-Oct-nc.blast\t(95.6085 %)\n",
      "CH110-2-7-May-nc.blast\t(100.0000 %)\n",
      "CH110-pl1-2-2-Oct-nc.blast\t(94.3920 %)\n",
      "CH111-pl1-2-3-Oct-nc.blast\t(21.4096 %)\n",
      "CH112-pl1-2-4-Oct-nc.blast\t(97.7373 %)\n",
      "CH113-pl1-2-5-Oct-nc.blast\t(90.3129 %)\n",
      "CH114-pl1-2-6-Oct-nc.blast\t(80.7732 %)\n",
      "CH115-pl1-2-7-Oct-nc.blast\t(95.3055 %)\n",
      "CH116-pl1-2-8-Oct-nc.blast\t(89.5879 %)\n",
      "CH117-pl1-3-1-Oct-nc.blast\t(96.1352 %)\n",
      "CH118-pl1-3-2-Oct-nc.blast\t(95.4572 %)\n",
      "CH119-pl1-3-3-Oct-nc.blast\t(87.4251 %)\n",
      "CH202-2-5-May-nc.blast\t(100.0000 %)\n",
      "CH203-2-4-May-nc.blast\t(19.0871 %)\n",
      "CH204-2-3-May-nc.blast\t(4.8662 %)\n",
      "CH206-2-1-May-nc.blast\t(92.6752 %)\n",
      "CH207-3-8-May-nc.blast\t(100.0000 %)\n",
      "CH208-3-7-May-nc.blast\t(100.0000 %)\n",
      "CH209-3-6-May-nc.blast\t(97.7895 %)\n",
      "CH210-3-5-May-nc.blast\t(94.2414 %)\n",
      "CH301-pl1-3-4-Oct-nc.blast\t(95.4589 %)\n",
      "CH302-pl1-3-5-Oct-nc.blast\t(26.7896 %)\n",
      "CH303-pl1-3-6-Oct-nc.blast\t(83.5330 %)\n",
      "CH304-pl1-3-7-Oct-nc.blast\t(94.7740 %)\n",
      "CH305-pl1-3-8-Oct-nc.blast\t(68.1530 %)\n",
      "CH306-pl1-4-1-Oct-nc.blast\t(90.7080 %)\n",
      "CH307-pl1-4-2-Oct-nc.blast\t(85.9195 %)\n",
      "CH308-pl1-4-3-Oct-nc.blast\t(11.2833 %)\n",
      "CH309-3-2-May-nc.blast\t(2.0844 %)\n",
      "CH309-pl1-4-4-Oct-nc.blast\t(87.8240 %)\n",
      "CH310-3-1-May-nc.blast\t(8.3958 %)\n",
      "CH310-pl1-4-5-Oct-nc.blast\t(78.6368 %)\n",
      "CH401-pl1-4-6-Oct-nc.blast\t(12.8118 %)\n",
      "CH501-pl1-4-7-Oct-nc.blast\t(56.9524 %)\n",
      "CH502-pl1-4-8-Oct-nc.blast\t(21.0821 %)\n",
      "CH503-pl1-5-1-Oct-nc.blast\t(95.7363 %)\n",
      "CH504-6-5-May-nc.blast\t(92.8809 %)\n",
      "CH504-pl1-5-2-Oct-nc.blast\t(95.1211 %)\n",
      "CH505-pl1-5-3-Oct-nc.blast\t(86.3547 %)\n",
      "CH506-pl1-5-4-Oct-nc.blast\t(72.4321 %)\n",
      "CH507-6-2-May-nc.blast\t(100.0000 %)\n",
      "CH507-pl1-5-5-Oct-nc.blast\t(86.0071 %)\n",
      "CH508-6-1-May-nc.blast\t(100.0000 %)\n",
      "CH508-pl1-5-6-Oct-nc.blast\t(15.5060 %)\n",
      "CH509-7-8-May-nc.blast\t(100.0000 %)\n",
      "CH509-pl1-5-7-Oct-nc.blast\t(93.9991 %)\n",
      "CH510-pl1-5-8-Oct-nc.blast\t(89.3020 %)\n",
      "CH601-pl1-6-1-Oct-nc.blast\t(95.1703 %)\n",
      "CH602-pl1-6-2-Oct-nc.blast\t(94.8962 %)\n",
      "CH603-pl1-6-3-Oct-nc.blast\t(17.5988 %)\n",
      "CH604-pl1-6-4-Oct-nc.blast\t(95.3248 %)\n",
      "CH605-pl1-6-5-Oct-nc.blast\t(88.6710 %)\n",
      "CH606-pl1-6-6-Oct-nc.blast\t(84.9662 %)\n",
      "CH607-pl1-6-7-Oct-nc.blast\t(92.3518 %)\n",
      "CH608-pl1-6-8-Oct-nc.blast\t(88.2148 %)\n",
      "CH609-pl1-7-1-Oct-nc.blast\t(100.0000 %)\n",
      "CH610-pl1-7-2-Oct-nc.blast\t(100.0000 %)\n",
      "neg-7-5-May-nc.blast\t(18.1818 %)\n",
      "pcr-NEG-pl1-7-5-Oct-nc.blast\t(87.5000 %)\n",
      "pcr-NEG-pl1-7-7-Oct-nc.blast\t(89.3382 %)\n",
      "pcr-POS-pl1-7-6-Oct-nc.blast\t(41.7431 %)\n"
     ]
    }
   ],
   "source": [
    "mb.find_target(filtered_collapsed, target='Crangonyx_floridanus')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And we identify the samples containing sequences assigned to _C. pseudogracilis_."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CH103-1-6-May-nc.blast\t(100.0000 %)\n",
      "CH105-1-4-May-nc.blast\t(100.0000 %)\n",
      "CH106-1-3-May-nc.blast\t(100.0000 %)\n",
      "CH109-pl1-2-1-Oct-nc.blast\t(4.3915 %)\n",
      "CH110-pl1-2-2-Oct-nc.blast\t(5.6080 %)\n",
      "CH111-pl1-2-3-Oct-nc.blast\t(78.5904 %)\n",
      "CH112-pl1-2-4-Oct-nc.blast\t(2.2627 %)\n",
      "CH113-pl1-2-5-Oct-nc.blast\t(9.6871 %)\n",
      "CH114-pl1-2-6-Oct-nc.blast\t(16.3531 %)\n",
      "CH115-pl1-2-7-Oct-nc.blast\t(4.6945 %)\n",
      "CH116-pl1-2-8-Oct-nc.blast\t(10.4121 %)\n",
      "CH117-pl1-3-1-Oct-nc.blast\t(3.8648 %)\n",
      "CH118-pl1-3-2-Oct-nc.blast\t(4.5428 %)\n",
      "CH119-pl1-3-3-Oct-nc.blast\t(12.5749 %)\n",
      "CH201-2-6-May-nc.blast\t(100.0000 %)\n",
      "CH203-2-4-May-nc.blast\t(77.1784 %)\n",
      "CH204-2-3-May-nc.blast\t(93.0657 %)\n",
      "CH205-2-2-May-nc.blast\t(100.0000 %)\n",
      "CH206-2-1-May-nc.blast\t(7.3248 %)\n",
      "CH209-3-6-May-nc.blast\t(2.2105 %)\n",
      "CH301-pl1-3-4-Oct-nc.blast\t(4.5411 %)\n",
      "CH302-pl1-3-5-Oct-nc.blast\t(70.9771 %)\n",
      "CH303-pl1-3-6-Oct-nc.blast\t(16.4670 %)\n",
      "CH304-pl1-3-7-Oct-nc.blast\t(5.2260 %)\n",
      "CH305-pl1-3-8-Oct-nc.blast\t(5.2702 %)\n",
      "CH306-pl1-4-1-Oct-nc.blast\t(9.2920 %)\n",
      "CH307-pl1-4-2-Oct-nc.blast\t(14.0805 %)\n",
      "CH308-3-3-May-nc.blast\t(100.0000 %)\n",
      "CH308-pl1-4-3-Oct-nc.blast\t(84.0204 %)\n",
      "CH309-3-2-May-nc.blast\t(97.9156 %)\n",
      "CH309-pl1-4-4-Oct-nc.blast\t(12.1760 %)\n",
      "CH310-3-1-May-nc.blast\t(91.6042 %)\n",
      "CH310-pl1-4-5-Oct-nc.blast\t(21.3632 %)\n",
      "CH311-4-8-May-nc.blast\t(100.0000 %)\n",
      "CH312-4-7-May-nc.blast\t(100.0000 %)\n",
      "CH313-4-6-May-nc.blast\t(100.0000 %)\n",
      "CH314-4-5-May-nc.blast\t(100.0000 %)\n",
      "CH315-4-4-May-nc.blast\t(100.0000 %)\n",
      "CH316-4-3-May-nc.blast\t(100.0000 %)\n",
      "CH317-4-2-May-nc.blast\t(100.0000 %)\n",
      "CH401-4-1-May-nc.blast\t(100.0000 %)\n",
      "CH401-pl1-4-6-Oct-nc.blast\t(85.0651 %)\n",
      "CH402-5-8-May-nc.blast\t(100.0000 %)\n",
      "CH403-5-7-May-nc.blast\t(100.0000 %)\n",
      "CH404-5-6-May-nc.blast\t(100.0000 %)\n",
      "CH405-5-5-May-nc.blast\t(100.0000 %)\n",
      "CH406-5-4-May-nc.blast\t(100.0000 %)\n",
      "CH407-5-3-May-nc.blast\t(100.0000 %)\n",
      "CH408-5-2-May-nc.blast\t(100.0000 %)\n",
      "CH409-5-1-May-nc.blast\t(100.0000 %)\n",
      "CH501-6-7-May-nc.blast\t(100.0000 %)\n",
      "CH501-pl1-4-7-Oct-nc.blast\t(11.0918 %)\n",
      "CH502-pl1-4-8-Oct-nc.blast\t(76.4128 %)\n",
      "CH503-6-6-May-nc.blast\t(100.0000 %)\n",
      "CH503-pl1-5-1-Oct-nc.blast\t(4.2637 %)\n",
      "CH504-pl1-5-2-Oct-nc.blast\t(4.8789 %)\n",
      "CH505-6-4-May-nc.blast\t(100.0000 %)\n",
      "CH505-pl1-5-3-Oct-nc.blast\t(13.6453 %)\n",
      "CH506-6-3-May-nc.blast\t(100.0000 %)\n",
      "CH506-pl1-5-4-Oct-nc.blast\t(27.5679 %)\n",
      "CH507-pl1-5-5-Oct-nc.blast\t(9.5633 %)\n",
      "CH508-pl1-5-6-Oct-nc.blast\t(84.4940 %)\n",
      "CH509-pl1-5-7-Oct-nc.blast\t(6.0009 %)\n",
      "CH510-7-7-May-nc.blast\t(100.0000 %)\n",
      "CH510-pl1-5-8-Oct-nc.blast\t(10.6980 %)\n",
      "CH601-pl1-6-1-Oct-nc.blast\t(4.8297 %)\n",
      "CH602-7-6-May-nc.blast\t(36.5854 %)\n",
      "CH602-pl1-6-2-Oct-nc.blast\t(5.1038 %)\n",
      "CH603-pl1-6-3-Oct-nc.blast\t(82.4012 %)\n",
      "CH604-pl1-6-4-Oct-nc.blast\t(4.6752 %)\n",
      "CH605-pl1-6-5-Oct-nc.blast\t(11.3290 %)\n",
      "CH606-pl1-6-6-Oct-nc.blast\t(15.0338 %)\n",
      "CH607-pl1-6-7-Oct-nc.blast\t(7.6482 %)\n",
      "CH608-pl1-6-8-Oct-nc.blast\t(8.8616 %)\n",
      "neg-7-5-May-nc.blast\t(15.9091 %)\n"
     ]
    }
   ],
   "source": [
    "mb.find_target(filtered_collapsed, target='Crangonyx_pseudogracilis')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
